<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2024-02-27" />

<title>Quality control of C133_Neeland data</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-inflammation-CITEseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Quality control of C133_Neeland data</h1>
<h3 class="subtitle">Batches 0-6</h3>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">February 27, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-27
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>paed-inflammation-CITEseq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong335d800">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 335d800
</a>
</p>
</div>
<div id="strongRepositoryversionstrong335d800"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 335d800.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  analysis/03.0_identify_doublets.Rmd
    Untracked:  analysis/04.0_filter_doublets.Rmd
    Untracked:  analysis/05.0_remove_ambient.R
    Untracked:  analysis/06.0_azimuth_annotation.Rmd
    Untracked:  analysis/06.1_azimuth_annotation_decontx.Rmd
    Untracked:  code/dropletutils.R
    Untracked:  code/utility.R
    Untracked:  data/.DS_Store
    Untracked:  data/C133_Neeland_batch0/
    Untracked:  data/C133_Neeland_batch1/
    Untracked:  data/C133_Neeland_batch2/
    Untracked:  data/C133_Neeland_batch3/
    Untracked:  data/C133_Neeland_batch4/
    Untracked:  data/C133_Neeland_batch5/
    Untracked:  data/C133_Neeland_batch6/
    Untracked:  renv.lock
    Untracked:  renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitignore
    Modified:   analysis/01.0_preprocess_batch0.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/02.0_quality_control.Rmd</code>)
and HTML (<code>docs/02.0_quality_control.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
335d800
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-27
</td>
<td>
wflow_publish("analysis/02.0_quality_control.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>suppressPackageStartupMessages({
  library(BiocStyle)
  library(tidyverse)
  library(here)
  library(glue)
  library(patchwork)
  library(scran)
  library(scater)
  library(scuttle)
  library(cowplot)
})

source(here(&quot;code&quot;,&quot;utility.R&quot;))</code></pre>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<pre class="r"><code>files &lt;- list.files(here(&quot;data&quot;,
                         paste0(&quot;C133_Neeland_batch&quot;, 0:6),
                         &quot;data&quot;,
                         &quot;SCEs&quot;),
                    pattern = &quot;preprocessed&quot;,
                    full.names = TRUE)
               
sceLst &lt;- sapply(files, function(fn){
  readRDS(file = fn)
})

sceLst</code></pre>
<pre><code>$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch0/data/SCEs/C133_Neeland_batch0.preprocessed.SCE.rds`
class: SingleCellExperiment 
dim: 33538 34583 
metadata(1): Samples
assays(1): counts
rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475
  ENSG00000268674
rowData names(3): ID Symbol Type
colnames(34583): 1_AAACCCAAGCTAGTTC-1 1_AAACCCACAAGATTGA-1 ...
  4_TTTGTTGTCTAGTACG-1 4_TTTGTTGTCTCGAACA-1
colData names(5): Barcode Capture sum detected total
reducedDimNames(0):
mainExpName: Gene Expression
altExpNames(0):

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch1/data/SCEs/C133_Neeland_batch1.preprocessed.SCE.rds`
class: SingleCellExperiment 
dim: 36601 24823 
metadata(1): Samples
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(3): ID Symbol Type
colnames(24823): 1_AAACCCACACTTCCTG-1 1_AAACCCACAGACAAAT-1 ...
  2_TTTGTTGTCATTGGTG-1 2_TTTGTTGTCGATGGAG-1
colData names(11): Barcode Capture ... GeneticDonor vireo
reducedDimNames(0):
mainExpName: Gene Expression
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch2/data/SCEs/C133_Neeland_batch2.preprocessed.SCE.rds`
class: SingleCellExperiment 
dim: 36601 53160 
metadata(1): Samples
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(3): ID Symbol Type
colnames(53160): 1_AAACCCAAGACCTGGA-1 1_AAACCCAAGACTGTTC-1 ...
  2_TTTGTTGTCTCATGGA-1 2_TTTGTTGTCTCCAAGA-1
colData names(11): Barcode Capture ... GeneticDonor vireo
reducedDimNames(0):
mainExpName: Gene Expression
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch3/data/SCEs/C133_Neeland_batch3.preprocessed.SCE.rds`
class: SingleCellExperiment 
dim: 36601 64842 
metadata(1): Samples
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(3): ID Symbol Type
colnames(64842): 1_AAACCCAAGCAGCACA-1 1_AAACCCAAGCATCTTG-1 ...
  2_TTTGTTGTCTAGGCCG-1 2_TTTGTTGTCTCGGCTT-1
colData names(11): Barcode Capture ... GeneticDonor vireo
reducedDimNames(0):
mainExpName: Gene Expression
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch4/data/SCEs/C133_Neeland_batch4.preprocessed.SCE.rds`
class: SingleCellExperiment 
dim: 36601 50208 
metadata(1): Samples
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(3): ID Symbol Type
colnames(50208): 1_AAACCCAAGCGTTAGG-1 1_AAACCCAAGGATTTGA-1 ...
  2_TTTGTTGTCGACGATT-1 2_TTTGTTGTCTAGGCCG-1
colData names(11): Barcode Capture ... GeneticDonor vireo
reducedDimNames(0):
mainExpName: Gene Expression
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch5/data/SCEs/C133_Neeland_batch5.preprocessed.SCE.rds`
class: SingleCellExperiment 
dim: 36601 50668 
metadata(1): Samples
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(3): ID Symbol Type
colnames(50668): 1_AAACCCAAGAAGATCT-1 1_AAACCCAAGATGCAGC-1 ...
  2_TTTGTTGTCGGATTAC-1 2_TTTGTTGTCTGAGAGG-1
colData names(11): Barcode Capture ... GeneticDonor vireo
reducedDimNames(0):
mainExpName: Gene Expression
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch6/data/SCEs/C133_Neeland_batch6.preprocessed.SCE.rds`
class: SingleCellExperiment 
dim: 36601 51119 
metadata(1): Samples
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(3): ID Symbol Type
colnames(51119): 1_AAACCCAAGAAGCGCT-1 1_AAACCCAAGACTCATC-1 ...
  2_TTTGTTGTCGAGAATA-1 2_TTTGTTGTCTACTGAG-1
colData names(11): Barcode Capture ... GeneticDonor vireo
reducedDimNames(0):
mainExpName: Gene Expression
altExpNames(2): HTO ADT</code></pre>
<div id="incorporating-gene-based-annotation" class="section level2">
<h2>Incorporating gene-based annotation</h2>
<p>Having quantified gene expression against the Ensembl gene
annotation, we have Ensembl-style identifiers for the genes. These
identifiers are used as they are unambiguous and highly stable. However,
they are difficult to interpret compared to the gene symbols which are
more commonly used in the literature. Given the Ensembl identifiers, we
obtain the corresponding gene symbols using annotation packages
available through Bioconductor. Henceforth, we will use gene symbols
(where available) to refer to genes in our analysis and otherwise use
the Ensembl-style gene identifiers<a href="#fn1" class="footnote-ref"
id="fnref1"><sup>1</sup></a>.</p>
<pre class="r"><code>sceLst &lt;- sapply(sceLst, function(sce){
  add_gene_information(sce)
})</code></pre>
</div>
</div>
<div id="quality-control" class="section level1">
<h1>Quality control</h1>
<div id="define-the-quality-control-metrics" class="section level2">
<h2>Define the quality control metrics</h2>
<p>Low-quality cells need to be removed to ensure that technical effects
do not distort downstream analysis results. We use several quality
control (QC) metrics to measure the quality of the cells:</p>
<ul>
<li><code>sum</code>: This measures the library size of the cells, which
is the total sum of counts across both genes and spike-in transcripts.
We want cells to have high library sizes as this means more RNA has been
successfully captured during library preparation.</li>
<li><code>detected</code>: This is the number of expressed features<a
href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a> in each
cell. Cells with few expressed features are likely to be of poor
quality, as the diverse transcript population has not been successful
captured.</li>
<li><code>subsets_Mito_percent</code>: This measures the proportion of
UMIs which are mapped to mitochondrial RNA. If there is a higher than
expected proportion of mitochondrial RNA this is often symptomatic of a
cell which is under stress and is therefore of low quality and will not
be used for the analysis.</li>
<li><code>subsets_Ribo_percent</code>: This measures the proportion of
UMIs which are mapped to ribosomal protein genes. If there is a higher
than expected proportion of ribosomal protein gene expression this is
often symptomatic of a cell which is of compromised quality and we may
want to exclude it from the analysis.</li>
</ul>
<p>In summary, we aim to identify cells with low library sizes, few
expressed genes, and very high percentages of mitochondrial and
ribosomal protein gene expression.</p>
<pre class="r"><code>sceLst &lt;- sapply(sceLst, function(sce){
  
  colData(sce) &lt;- colData(sce)[, !str_detect(colnames(colData(sce)), 
                                             &quot;sum|detected|percent|total&quot;)]
  sce &lt;- addPerCellQC(sce, 
                      subsets = list(Mito = which(rowData(sce)$is_mito), 
                                     Ribo = which(rowData(sce)$is_ribo)))
  
  sce
})</code></pre>
</div>
<div id="visualise-the-qc-metrics" class="section level2">
<h2>Visualise the QC metrics</h2>
<p>Figure @ref(fig:qcplot-by-genetic-donor) shows that the vast majority
of samples are good-quality:</p>
<p>As we would expect, the <code>doublet</code> droplets have larger
library sizes and more genes detected. The <code>unassigned</code>
droplets generally have smaller library sizes and fewer genes
detected.</p>
<pre class="r"><code># for batch 0 each capture is from a different donor
sceLst[[1]]$GeneticDonor &lt;- sceLst[[1]]$Capture

p &lt;- vector(&quot;list&quot;, length(sceLst))
for(i in 1:length(sceLst)){
  sce &lt;- sceLst[[i]]
  
  
  p1 &lt;- plotColData(
    sce,
    &quot;sum&quot;,
    x = &quot;GeneticDonor&quot;,
    other_fields = c(&quot;Capture&quot;),
    colour_by = &quot;GeneticDonor&quot;,
    point_size = 1) +
    scale_y_log10() +
    theme(axis.text.x = element_blank()) +
    geom_hline(yintercept = 500,
               linetype = &quot;dotted&quot;) +
    annotation_logticks(
      sides = &quot;l&quot;,
      short = unit(0.03, &quot;cm&quot;),
      mid = unit(0.06, &quot;cm&quot;),
      long = unit(0.09, &quot;cm&quot;))
  p2 &lt;- plotColData(
    sce,
    &quot;detected&quot;,
    x = &quot;GeneticDonor&quot;,
    other_fields = c(&quot;Capture&quot;),
    colour_by = &quot;GeneticDonor&quot;,
    point_size = 1) +
    theme(axis.text.x = element_blank())
  p3 &lt;- plotColData(
    sce,
    &quot;subsets_Mito_percent&quot;,
    x = &quot;GeneticDonor&quot;,
    other_fields = c(&quot;Capture&quot;),
    colour_by = &quot;GeneticDonor&quot;,
    point_size = 1) +
    theme(axis.text.x = element_blank())
  p4 &lt;- plotColData(
    sce,
    &quot;subsets_Ribo_percent&quot;,
    x = &quot;GeneticDonor&quot;,
    other_fields = c(&quot;Capture&quot;),
    colour_by = &quot;GeneticDonor&quot;,
    point_size = 1) +
    theme(axis.text.x = element_blank())
  
  p[[i]] &lt;- p1 + p2 + p3 + p4 + 
    plot_layout(guides = &quot;collect&quot;, ncol = 2) +
    plot_annotation(title = glue(&quot;Batch {i-1}&quot;))
}

p</code></pre>
<pre><code>[[1]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-by-genetic-donor-1.png" alt="Distributions of various QC metrics for all cells in the dataset. This includes the library sizes, number of genes detected, and percentage of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset. This
includes the library sizes, number of genes detected, and percentage of
reads mapped to mitochondrial genes.
</p>
</div>
<pre><code>
[[2]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-by-genetic-donor-2.png" alt="Distributions of various QC metrics for all cells in the dataset. This includes the library sizes, number of genes detected, and percentage of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset. This
includes the library sizes, number of genes detected, and percentage of
reads mapped to mitochondrial genes.
</p>
</div>
<pre><code>
[[3]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-by-genetic-donor-3.png" alt="Distributions of various QC metrics for all cells in the dataset. This includes the library sizes, number of genes detected, and percentage of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset. This
includes the library sizes, number of genes detected, and percentage of
reads mapped to mitochondrial genes.
</p>
</div>
<pre><code>
[[4]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-by-genetic-donor-4.png" alt="Distributions of various QC metrics for all cells in the dataset. This includes the library sizes, number of genes detected, and percentage of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset. This
includes the library sizes, number of genes detected, and percentage of
reads mapped to mitochondrial genes.
</p>
</div>
<pre><code>
[[5]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-by-genetic-donor-5.png" alt="Distributions of various QC metrics for all cells in the dataset. This includes the library sizes, number of genes detected, and percentage of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset. This
includes the library sizes, number of genes detected, and percentage of
reads mapped to mitochondrial genes.
</p>
</div>
<pre><code>
[[6]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-by-genetic-donor-6.png" alt="Distributions of various QC metrics for all cells in the dataset. This includes the library sizes, number of genes detected, and percentage of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset. This
includes the library sizes, number of genes detected, and percentage of
reads mapped to mitochondrial genes.
</p>
</div>
<pre><code>
[[7]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-by-genetic-donor-7.png" alt="Distributions of various QC metrics for all cells in the dataset. This includes the library sizes, number of genes detected, and percentage of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset. This
includes the library sizes, number of genes detected, and percentage of
reads mapped to mitochondrial genes.
</p>
</div>
</div>
<div id="identify-outliers-by-each-metric" class="section level2">
<h2>Identify outliers by each metric</h2>
<p>Filtering on the mitochondrial proportion can identify
stressed/damaged cells and so we seek to identify droplets with
unusually large mitochondrial proportions (i.e. outliers). Outlier
thresholds are defined based on the median absolute deviation (MADs)
from the median value of the metric across all cells. Here, we opt to
use <code>donor</code>-specific thresholds to account for
<code>donor</code>-specific differences<a href="#fn3"
class="footnote-ref" id="fnref3"><sup>3</sup></a>.</p>
<p>The following table summarises the QC cutoffs:</p>
<pre class="r"><code># for batch 0, remove droplets with library size &lt; 500 for consistency with other batches
sceLst[[1]] &lt;- sceLst[[1]][, sceLst[[1]]$sum &gt;= 500]

# identify % mito outliers
sceLst &lt;- sapply(sceLst, function(sce){
  sce$mito_drop &lt;- isOutlier(
    metric = sce$subsets_Mito_percent, 
    nmads = 3, 
    type = &quot;higher&quot;,
    batch = sce$GeneticDonor,
    subset = !grepl(&quot;Unknown&quot;, sce$GeneticDonor))
  
  data.frame(
    sample = factor(
      colnames(attributes(sce$mito_drop)$thresholds),
      levels(sce$GeneticDonor)),
    lower = attributes(sce$mito_drop)$thresholds[&quot;higher&quot;, ]) %&gt;%
    arrange(sample) %&gt;%
    knitr::kable(caption = &quot;Sample-specific %mito cutoffs&quot;, digits = 1) %&gt;%
    print()
  
  sce
})</code></pre>
<table>
<caption>Sample-specific %mito cutoffs</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">sample</th>
<th align="right">lower</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">A</td>
<td align="right">19.3</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="left">B</td>
<td align="right">20.1</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="left">C</td>
<td align="right">15.0</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="left">D</td>
<td align="right">14.8</td>
</tr>
</tbody>
</table>
<table>
<caption>Sample-specific %mito cutoffs</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">sample</th>
<th align="right">lower</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">A</td>
<td align="right">6.9</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="left">B</td>
<td align="right">14.5</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="left">C</td>
<td align="right">11.4</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="left">D</td>
<td align="right">12.5</td>
</tr>
<tr class="odd">
<td align="left">E</td>
<td align="left">E</td>
<td align="right">12.9</td>
</tr>
<tr class="even">
<td align="left">F</td>
<td align="left">F</td>
<td align="right">14.5</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">G</td>
<td align="right">11.1</td>
</tr>
<tr class="even">
<td align="left">H</td>
<td align="left">H</td>
<td align="right">13.1</td>
</tr>
<tr class="odd">
<td align="left">Doublet</td>
<td align="left">Doublet</td>
<td align="right">11.7</td>
</tr>
<tr class="even">
<td align="left">Unknown</td>
<td align="left">Unknown</td>
<td align="right">11.1</td>
</tr>
</tbody>
</table>
<table>
<caption>Sample-specific %mito cutoffs</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">sample</th>
<th align="right">lower</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">A</td>
<td align="right">12.8</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="left">B</td>
<td align="right">15.3</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="left">C</td>
<td align="right">15.9</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="left">D</td>
<td align="right">15.7</td>
</tr>
<tr class="odd">
<td align="left">Doublet</td>
<td align="left">Doublet</td>
<td align="right">14.4</td>
</tr>
<tr class="even">
<td align="left">Unknown</td>
<td align="left">Unknown</td>
<td align="right">14.4</td>
</tr>
</tbody>
</table>
<table>
<caption>Sample-specific %mito cutoffs</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">sample</th>
<th align="right">lower</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">A</td>
<td align="right">13.1</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="left">B</td>
<td align="right">9.2</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="left">C</td>
<td align="right">9.6</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="left">D</td>
<td align="right">9.7</td>
</tr>
<tr class="odd">
<td align="left">E</td>
<td align="left">E</td>
<td align="right">9.1</td>
</tr>
<tr class="even">
<td align="left">F</td>
<td align="left">F</td>
<td align="right">9.2</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">G</td>
<td align="right">12.6</td>
</tr>
<tr class="even">
<td align="left">H</td>
<td align="left">H</td>
<td align="right">9.9</td>
</tr>
<tr class="odd">
<td align="left">Doublet</td>
<td align="left">Doublet</td>
<td align="right">9.3</td>
</tr>
<tr class="even">
<td align="left">Unknown</td>
<td align="left">Unknown</td>
<td align="right">9.7</td>
</tr>
</tbody>
</table>
<table>
<caption>Sample-specific %mito cutoffs</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">sample</th>
<th align="right">lower</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">A</td>
<td align="right">11.4</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="left">B</td>
<td align="right">10.8</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="left">C</td>
<td align="right">8.0</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="left">D</td>
<td align="right">9.1</td>
</tr>
<tr class="odd">
<td align="left">E</td>
<td align="left">E</td>
<td align="right">10.5</td>
</tr>
<tr class="even">
<td align="left">F</td>
<td align="left">F</td>
<td align="right">9.0</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">G</td>
<td align="right">11.8</td>
</tr>
<tr class="even">
<td align="left">Doublet</td>
<td align="left">Doublet</td>
<td align="right">9.6</td>
</tr>
<tr class="odd">
<td align="left">Unknown</td>
<td align="left">Unknown</td>
<td align="right">9.6</td>
</tr>
</tbody>
</table>
<table>
<caption>Sample-specific %mito cutoffs</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">sample</th>
<th align="right">lower</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">A</td>
<td align="right">13.1</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="left">B</td>
<td align="right">14.7</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="left">C</td>
<td align="right">16.2</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="left">D</td>
<td align="right">11.4</td>
</tr>
<tr class="odd">
<td align="left">E</td>
<td align="left">E</td>
<td align="right">12.2</td>
</tr>
<tr class="even">
<td align="left">F</td>
<td align="left">F</td>
<td align="right">15.1</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="left">G</td>
<td align="right">11.7</td>
</tr>
<tr class="even">
<td align="left">H</td>
<td align="left">H</td>
<td align="right">11.9</td>
</tr>
<tr class="odd">
<td align="left">Doublet</td>
<td align="left">Doublet</td>
<td align="right">13.0</td>
</tr>
<tr class="even">
<td align="left">Unknown</td>
<td align="left">Unknown</td>
<td align="right">12.9</td>
</tr>
</tbody>
</table>
<table>
<caption>Sample-specific %mito cutoffs</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">sample</th>
<th align="right">lower</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left">A</td>
<td align="right">9.2</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="left">B</td>
<td align="right">11.3</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="left">C</td>
<td align="right">11.2</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="left">D</td>
<td align="right">10.1</td>
</tr>
<tr class="odd">
<td align="left">Doublet</td>
<td align="left">Doublet</td>
<td align="right">10.5</td>
</tr>
<tr class="even">
<td align="left">Unknown</td>
<td align="left">Unknown</td>
<td align="right">10.3</td>
</tr>
</tbody>
</table>
<p>The vast majority of cells are retained for all samples.</p>
<pre class="r"><code>sceFlt &lt;- sapply(sceLst, function(sce){
  scePre &lt;- sce
  keep &lt;- !sce$mito_drop
  scePre$keep &lt;- keep
  sce &lt;- sce[, keep]
  
  data.frame(
    ByMito = tapply(
      scePre$mito_drop, 
      scePre$GeneticDonor, 
      sum,
      na.rm = TRUE),
    Remaining = as.vector(unname(table(sce$GeneticDonor))),
    PercRemaining = round(
      100 * as.vector(unname(table(sce$GeneticDonor))) /
        as.vector(
          unname(
            table(scePre$GeneticDonor))), 1)) %&gt;%
    tibble::rownames_to_column(&quot;GeneticDonor&quot;) %&gt;%
    dplyr::arrange(dplyr::desc(PercRemaining)) %&gt;%
    knitr::kable(
      caption = &quot;Number of droplets removed by each QC step and the number of droplets remaining.&quot;) %&gt;%
    print()
  
  sce
})</code></pre>
<table>
<caption>Number of droplets removed by each QC step and the number of
droplets remaining.</caption>
<thead>
<tr class="header">
<th align="left">GeneticDonor</th>
<th align="right">ByMito</th>
<th align="right">Remaining</th>
<th align="right">PercRemaining</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">D</td>
<td align="right">994</td>
<td align="right">9820</td>
<td align="right">90.8</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="right">946</td>
<td align="right">9129</td>
<td align="right">90.6</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="right">462</td>
<td align="right">3620</td>
<td align="right">88.7</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="right">588</td>
<td align="right">4370</td>
<td align="right">88.1</td>
</tr>
</tbody>
</table>
<table>
<caption>Number of droplets removed by each QC step and the number of
droplets remaining.</caption>
<thead>
<tr class="header">
<th align="left">GeneticDonor</th>
<th align="right">ByMito</th>
<th align="right">Remaining</th>
<th align="right">PercRemaining</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">G</td>
<td align="right">174</td>
<td align="right">3093</td>
<td align="right">94.7</td>
</tr>
<tr class="even">
<td align="left">H</td>
<td align="right">164</td>
<td align="right">2554</td>
<td align="right">94.0</td>
</tr>
<tr class="odd">
<td align="left">Doublet</td>
<td align="right">137</td>
<td align="right">1904</td>
<td align="right">93.3</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="right">211</td>
<td align="right">2846</td>
<td align="right">93.1</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="right">172</td>
<td align="right">2219</td>
<td align="right">92.8</td>
</tr>
<tr class="even">
<td align="left">F</td>
<td align="right">198</td>
<td align="right">2082</td>
<td align="right">91.3</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="right">348</td>
<td align="right">3450</td>
<td align="right">90.8</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="right">218</td>
<td align="right">1615</td>
<td align="right">88.1</td>
</tr>
<tr class="odd">
<td align="left">E</td>
<td align="right">300</td>
<td align="right">2207</td>
<td align="right">88.0</td>
</tr>
<tr class="even">
<td align="left">Unknown</td>
<td align="right">276</td>
<td align="right">655</td>
<td align="right">70.4</td>
</tr>
</tbody>
</table>
<table>
<caption>Number of droplets removed by each QC step and the number of
droplets remaining.</caption>
<thead>
<tr class="header">
<th align="left">GeneticDonor</th>
<th align="right">ByMito</th>
<th align="right">Remaining</th>
<th align="right">PercRemaining</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Doublet</td>
<td align="right">368</td>
<td align="right">8723</td>
<td align="right">96.0</td>
</tr>
<tr class="even">
<td align="left">A</td>
<td align="right">1039</td>
<td align="right">15039</td>
<td align="right">93.5</td>
</tr>
<tr class="odd">
<td align="left">B</td>
<td align="right">165</td>
<td align="right">1962</td>
<td align="right">92.2</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="right">422</td>
<td align="right">4289</td>
<td align="right">91.0</td>
</tr>
<tr class="odd">
<td align="left">D</td>
<td align="right">1665</td>
<td align="right">15698</td>
<td align="right">90.4</td>
</tr>
<tr class="even">
<td align="left">Unknown</td>
<td align="right">496</td>
<td align="right">3294</td>
<td align="right">86.9</td>
</tr>
</tbody>
</table>
<table>
<caption>Number of droplets removed by each QC step and the number of
droplets remaining.</caption>
<thead>
<tr class="header">
<th align="left">GeneticDonor</th>
<th align="right">ByMito</th>
<th align="right">Remaining</th>
<th align="right">PercRemaining</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Doublet</td>
<td align="right">377</td>
<td align="right">10627</td>
<td align="right">96.6</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="right">515</td>
<td align="right">11404</td>
<td align="right">95.7</td>
</tr>
<tr class="odd">
<td align="left">E</td>
<td align="right">518</td>
<td align="right">10324</td>
<td align="right">95.2</td>
</tr>
<tr class="even">
<td align="left">H</td>
<td align="right">319</td>
<td align="right">5815</td>
<td align="right">94.8</td>
</tr>
<tr class="odd">
<td align="left">B</td>
<td align="right">488</td>
<td align="right">7233</td>
<td align="right">93.7</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="right">290</td>
<td align="right">3854</td>
<td align="right">93.0</td>
</tr>
<tr class="odd">
<td align="left">F</td>
<td align="right">187</td>
<td align="right">2365</td>
<td align="right">92.7</td>
</tr>
<tr class="even">
<td align="left">G</td>
<td align="right">398</td>
<td align="right">3887</td>
<td align="right">90.7</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="right">370</td>
<td align="right">2529</td>
<td align="right">87.2</td>
</tr>
<tr class="even">
<td align="left">Unknown</td>
<td align="right">1053</td>
<td align="right">2289</td>
<td align="right">68.5</td>
</tr>
</tbody>
</table>
<table>
<caption>Number of droplets removed by each QC step and the number of
droplets remaining.</caption>
<thead>
<tr class="header">
<th align="left">GeneticDonor</th>
<th align="right">ByMito</th>
<th align="right">Remaining</th>
<th align="right">PercRemaining</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">D</td>
<td align="right">460</td>
<td align="right">11833</td>
<td align="right">96.3</td>
</tr>
<tr class="even">
<td align="left">Doublet</td>
<td align="right">265</td>
<td align="right">6289</td>
<td align="right">96.0</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="right">188</td>
<td align="right">3911</td>
<td align="right">95.4</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="right">325</td>
<td align="right">6631</td>
<td align="right">95.3</td>
</tr>
<tr class="odd">
<td align="left">E</td>
<td align="right">662</td>
<td align="right">11919</td>
<td align="right">94.7</td>
</tr>
<tr class="even">
<td align="left">A</td>
<td align="right">5</td>
<td align="right">66</td>
<td align="right">93.0</td>
</tr>
<tr class="odd">
<td align="left">F</td>
<td align="right">258</td>
<td align="right">2844</td>
<td align="right">91.7</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="right">250</td>
<td align="right">2267</td>
<td align="right">90.1</td>
</tr>
<tr class="odd">
<td align="left">Unknown</td>
<td align="right">619</td>
<td align="right">1416</td>
<td align="right">69.6</td>
</tr>
</tbody>
</table>
<table>
<caption>Number of droplets removed by each QC step and the number of
droplets remaining.</caption>
<thead>
<tr class="header">
<th align="left">GeneticDonor</th>
<th align="right">ByMito</th>
<th align="right">Remaining</th>
<th align="right">PercRemaining</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Doublet</td>
<td align="right">294</td>
<td align="right">6169</td>
<td align="right">95.5</td>
</tr>
<tr class="even">
<td align="left">G</td>
<td align="right">549</td>
<td align="right">9728</td>
<td align="right">94.7</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="right">441</td>
<td align="right">7479</td>
<td align="right">94.4</td>
</tr>
<tr class="even">
<td align="left">H</td>
<td align="right">510</td>
<td align="right">8392</td>
<td align="right">94.3</td>
</tr>
<tr class="odd">
<td align="left">D</td>
<td align="right">303</td>
<td align="right">2729</td>
<td align="right">90.0</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="right">159</td>
<td align="right">1124</td>
<td align="right">87.6</td>
</tr>
<tr class="odd">
<td align="left">E</td>
<td align="right">700</td>
<td align="right">4937</td>
<td align="right">87.6</td>
</tr>
<tr class="even">
<td align="left">F</td>
<td align="right">386</td>
<td align="right">2679</td>
<td align="right">87.4</td>
</tr>
<tr class="odd">
<td align="left">A</td>
<td align="right">222</td>
<td align="right">1478</td>
<td align="right">86.9</td>
</tr>
<tr class="even">
<td align="left">Unknown</td>
<td align="right">613</td>
<td align="right">1776</td>
<td align="right">74.3</td>
</tr>
</tbody>
</table>
<table>
<caption>Number of droplets removed by each QC step and the number of
droplets remaining.</caption>
<thead>
<tr class="header">
<th align="left">GeneticDonor</th>
<th align="right">ByMito</th>
<th align="right">Remaining</th>
<th align="right">PercRemaining</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Doublet</td>
<td align="right">228</td>
<td align="right">6388</td>
<td align="right">96.6</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="right">924</td>
<td align="right">14327</td>
<td align="right">93.9</td>
</tr>
<tr class="odd">
<td align="left">B</td>
<td align="right">576</td>
<td align="right">8652</td>
<td align="right">93.8</td>
</tr>
<tr class="even">
<td align="left">A</td>
<td align="right">736</td>
<td align="right">9377</td>
<td align="right">92.7</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="right">817</td>
<td align="right">7741</td>
<td align="right">90.5</td>
</tr>
<tr class="even">
<td align="left">Unknown</td>
<td align="right">452</td>
<td align="right">901</td>
<td align="right">66.6</td>
</tr>
</tbody>
</table>
<p>Of concern is whether the cells removed during QC preferentially
derive from particular experimental groups. Reassuringly, Figure
@ref(fig:barplot-highlighting-outliers) shows that this is not the
case.</p>
<pre class="r"><code>p &lt;- lapply(1:length(sceLst), function(i){
  sce &lt;- sceLst[[i]]
  flt &lt;- sceFlt[[i]]
  
  sce$keep &lt;- colnames(sce) %in% colnames(flt)
  ggcells(sce) +
    geom_bar(aes(x = GeneticDonor, fill = keep)) + 
    ylab(&quot;Number of droplets&quot;) + 
    theme_cowplot(font_size = 7) + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    facet_grid(GeneticDonor ~ ., scales = &quot;free_y&quot;)
})

p</code></pre>
<pre><code>[[1]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/barplot-highlighting-outliers-1.png" alt="Droplets removed during QC, stratified by `Sample`. " width="672" />
<p class="caption">
Droplets removed during QC, stratified by <code>Sample</code>.
</p>
</div>
<pre><code>
[[2]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/barplot-highlighting-outliers-2.png" alt="Droplets removed during QC, stratified by `Sample`. " width="672" />
<p class="caption">
Droplets removed during QC, stratified by <code>Sample</code>.
</p>
</div>
<pre><code>
[[3]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/barplot-highlighting-outliers-3.png" alt="Droplets removed during QC, stratified by `Sample`. " width="672" />
<p class="caption">
Droplets removed during QC, stratified by <code>Sample</code>.
</p>
</div>
<pre><code>
[[4]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/barplot-highlighting-outliers-4.png" alt="Droplets removed during QC, stratified by `Sample`. " width="672" />
<p class="caption">
Droplets removed during QC, stratified by <code>Sample</code>.
</p>
</div>
<pre><code>
[[5]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/barplot-highlighting-outliers-5.png" alt="Droplets removed during QC, stratified by `Sample`. " width="672" />
<p class="caption">
Droplets removed during QC, stratified by <code>Sample</code>.
</p>
</div>
<pre><code>
[[6]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/barplot-highlighting-outliers-6.png" alt="Droplets removed during QC, stratified by `Sample`. " width="672" />
<p class="caption">
Droplets removed during QC, stratified by <code>Sample</code>.
</p>
</div>
<pre><code>
[[7]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/barplot-highlighting-outliers-7.png" alt="Droplets removed during QC, stratified by `Sample`. " width="672" />
<p class="caption">
Droplets removed during QC, stratified by <code>Sample</code>.
</p>
</div>
<p>Finally, Figure @ref(fig:qcplot-highlighting-outliers) compares the
QC metrics of the discarded and retained droplets.</p>
<pre class="r"><code>p &lt;- lapply(1:length(sceLst), function(i){
  sce &lt;- sceLst[[i]]
  flt &lt;- sceFlt[[i]]
  
  sce$keep &lt;- colnames(sce) %in% colnames(flt)
  
  p1 &lt;- plotColData(
    sce,
    &quot;sum&quot;,
    x = &quot;GeneticDonor&quot;,
    colour_by = &quot;keep&quot;,
    point_size = 0.5) +
    scale_y_log10() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    annotation_logticks(
      sides = &quot;l&quot;,
      short = unit(0.03, &quot;cm&quot;),
      mid = unit(0.06, &quot;cm&quot;),
      long = unit(0.09, &quot;cm&quot;))
  p2 &lt;- plotColData(
    sce,
    &quot;detected&quot;,
    x = &quot;GeneticDonor&quot;,
    colour_by = &quot;keep&quot;,
    point_size = 0.5) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))
  p3 &lt;- plotColData(
    sce,
    &quot;subsets_Mito_percent&quot;,
    x = &quot;GeneticDonor&quot;,
    colour_by = &quot;keep&quot;,
    point_size = 0.5) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))
  p4 &lt;- plotColData(
    sce,
    &quot;subsets_Ribo_percent&quot;,
    x = &quot;GeneticDonor&quot;,
    colour_by = &quot;keep&quot;,
    point_size = 0.5) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))
  p1 + p2 + p3 + p4 + plot_layout(guides = &quot;collect&quot;)
})

p</code></pre>
<pre><code>[[1]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-highlighting-outliers-1.png" alt="Distribution of QC metrics for each plate in the dataset. Each point represents a cell and is colored according to whether it was discarded during the QC process. Note that a cell will only be kept if it passes the relevant threshold for all QC metrics." width="672" />
<p class="caption">
Distribution of QC metrics for each plate in the dataset. Each point
represents a cell and is colored according to whether it was discarded
during the QC process. Note that a cell will only be kept if it passes
the relevant threshold for all QC metrics.
</p>
</div>
<pre><code>
[[2]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-highlighting-outliers-2.png" alt="Distribution of QC metrics for each plate in the dataset. Each point represents a cell and is colored according to whether it was discarded during the QC process. Note that a cell will only be kept if it passes the relevant threshold for all QC metrics." width="672" />
<p class="caption">
Distribution of QC metrics for each plate in the dataset. Each point
represents a cell and is colored according to whether it was discarded
during the QC process. Note that a cell will only be kept if it passes
the relevant threshold for all QC metrics.
</p>
</div>
<pre><code>
[[3]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-highlighting-outliers-3.png" alt="Distribution of QC metrics for each plate in the dataset. Each point represents a cell and is colored according to whether it was discarded during the QC process. Note that a cell will only be kept if it passes the relevant threshold for all QC metrics." width="672" />
<p class="caption">
Distribution of QC metrics for each plate in the dataset. Each point
represents a cell and is colored according to whether it was discarded
during the QC process. Note that a cell will only be kept if it passes
the relevant threshold for all QC metrics.
</p>
</div>
<pre><code>
[[4]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-highlighting-outliers-4.png" alt="Distribution of QC metrics for each plate in the dataset. Each point represents a cell and is colored according to whether it was discarded during the QC process. Note that a cell will only be kept if it passes the relevant threshold for all QC metrics." width="672" />
<p class="caption">
Distribution of QC metrics for each plate in the dataset. Each point
represents a cell and is colored according to whether it was discarded
during the QC process. Note that a cell will only be kept if it passes
the relevant threshold for all QC metrics.
</p>
</div>
<pre><code>
[[5]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-highlighting-outliers-5.png" alt="Distribution of QC metrics for each plate in the dataset. Each point represents a cell and is colored according to whether it was discarded during the QC process. Note that a cell will only be kept if it passes the relevant threshold for all QC metrics." width="672" />
<p class="caption">
Distribution of QC metrics for each plate in the dataset. Each point
represents a cell and is colored according to whether it was discarded
during the QC process. Note that a cell will only be kept if it passes
the relevant threshold for all QC metrics.
</p>
</div>
<pre><code>
[[6]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-highlighting-outliers-6.png" alt="Distribution of QC metrics for each plate in the dataset. Each point represents a cell and is colored according to whether it was discarded during the QC process. Note that a cell will only be kept if it passes the relevant threshold for all QC metrics." width="672" />
<p class="caption">
Distribution of QC metrics for each plate in the dataset. Each point
represents a cell and is colored according to whether it was discarded
during the QC process. Note that a cell will only be kept if it passes
the relevant threshold for all QC metrics.
</p>
</div>
<pre><code>
[[7]]</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-highlighting-outliers-7.png" alt="Distribution of QC metrics for each plate in the dataset. Each point represents a cell and is colored according to whether it was discarded during the QC process. Note that a cell will only be kept if it passes the relevant threshold for all QC metrics." width="672" />
<p class="caption">
Distribution of QC metrics for each plate in the dataset. Each point
represents a cell and is colored according to whether it was discarded
during the QC process. Note that a cell will only be kept if it passes
the relevant threshold for all QC metrics.
</p>
</div>
</div>
<div id="filter-out-unassigned-droplets" class="section level2">
<h2>Filter out unassigned droplets</h2>
<p>Remove droplets that could not be assigned using genetics.</p>
<pre class="r"><code>sceFlt &lt;- sapply(sceFlt, function(sce){
  
  sce &lt;- sce[, sce$GeneticDonor != &quot;Unknown&quot;]
  sce
})</code></pre>
</div>
</div>
<div id="qc-summary" class="section level1">
<h1>QC summary</h1>
<p>We had already removed droplets that have unusually small library
sizes or number of genes detected by the process of identifying empty
droplets. We have now further removed droplets whose mitochondrial
proportions we deem to be an outlier.</p>
<p>To conclude, Figure @ref(fig:qcplot-post-outlier-removal) shows that
following QC that most samples have similar QC metrics, as is to be
expected, and Figure@ref(fig:experiment-by-donor-postqc) summarises the
experimental design following QC.</p>
<pre class="r"><code>p &lt;- lapply(sceFlt, function(sce){
  p1 &lt;- plotColData(
    sce,
    &quot;sum&quot;,
    x = &quot;GeneticDonor&quot;,
    other_fields = c(&quot;Capture&quot;, &quot;GeneticDonor&quot;),
    colour_by = &quot;GeneticDonor&quot;,
    point_size = 0.5) +
    scale_y_log10() +
    theme(axis.text.x = element_blank()) +
    annotation_logticks(
      sides = &quot;l&quot;,
      short = unit(0.03, &quot;cm&quot;),
      mid = unit(0.06, &quot;cm&quot;),
      long = unit(0.09, &quot;cm&quot;))
  p2 &lt;- plotColData(
    sce,
    &quot;detected&quot;,
    x = &quot;GeneticDonor&quot;,
    other_fields = c(&quot;Capture&quot;, &quot;GeneticDonor&quot;),
    colour_by = &quot;GeneticDonor&quot;,
    point_size = 0.5) +
    theme(axis.text.x = element_blank())
  p3 &lt;- plotColData(
    sce,
    &quot;subsets_Mito_percent&quot;,
    x = &quot;GeneticDonor&quot;,
    other_fields = c(&quot;Capture&quot;, &quot;GeneticDonor&quot;),
    colour_by = &quot;GeneticDonor&quot;,
    point_size = 0.5) +
    theme(axis.text.x = element_blank())
  p4 &lt;- plotColData(
    sce,
    &quot;subsets_Ribo_percent&quot;,
    x = &quot;GeneticDonor&quot;,
    other_fields = c(&quot;Capture&quot;, &quot;GeneticDonor&quot;),
    colour_by = &quot;GeneticDonor&quot;,
    point_size = 0.5) +
    theme(axis.text.x = element_blank())
  p1 + p2 + p3 + p4 + plot_layout(guides = &quot;collect&quot;, ncol = 2)
})

p</code></pre>
<pre><code>$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch0/data/SCEs/C133_Neeland_batch0.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-post-outlier-removal-1.png" alt="Distributions of various QC metrics for all cells in the dataset passing QC. This includes the library sizes and proportion of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset passing
QC. This includes the library sizes and proportion of reads mapped to
mitochondrial genes.
</p>
</div>
<pre><code>
$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch1/data/SCEs/C133_Neeland_batch1.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-post-outlier-removal-2.png" alt="Distributions of various QC metrics for all cells in the dataset passing QC. This includes the library sizes and proportion of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset passing
QC. This includes the library sizes and proportion of reads mapped to
mitochondrial genes.
</p>
</div>
<pre><code>
$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch2/data/SCEs/C133_Neeland_batch2.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-post-outlier-removal-3.png" alt="Distributions of various QC metrics for all cells in the dataset passing QC. This includes the library sizes and proportion of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset passing
QC. This includes the library sizes and proportion of reads mapped to
mitochondrial genes.
</p>
</div>
<pre><code>
$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch3/data/SCEs/C133_Neeland_batch3.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-post-outlier-removal-4.png" alt="Distributions of various QC metrics for all cells in the dataset passing QC. This includes the library sizes and proportion of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset passing
QC. This includes the library sizes and proportion of reads mapped to
mitochondrial genes.
</p>
</div>
<pre><code>
$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch4/data/SCEs/C133_Neeland_batch4.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-post-outlier-removal-5.png" alt="Distributions of various QC metrics for all cells in the dataset passing QC. This includes the library sizes and proportion of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset passing
QC. This includes the library sizes and proportion of reads mapped to
mitochondrial genes.
</p>
</div>
<pre><code>
$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch5/data/SCEs/C133_Neeland_batch5.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-post-outlier-removal-6.png" alt="Distributions of various QC metrics for all cells in the dataset passing QC. This includes the library sizes and proportion of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset passing
QC. This includes the library sizes and proportion of reads mapped to
mitochondrial genes.
</p>
</div>
<pre><code>
$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch6/data/SCEs/C133_Neeland_batch6.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/qcplot-post-outlier-removal-7.png" alt="Distributions of various QC metrics for all cells in the dataset passing QC. This includes the library sizes and proportion of reads mapped to mitochondrial genes." width="672" />
<p class="caption">
Distributions of various QC metrics for all cells in the dataset passing
QC. This includes the library sizes and proportion of reads mapped to
mitochondrial genes.
</p>
</div>
<p>Update <code>batch0</code> object to include <code>dmmHTO</code>
column to align with other batches.</p>
<pre class="r"><code>batch &lt;- grepl(&quot;batch0&quot;, names(sceFlt))
sceFlt[batch][[1]]$dmmHTO &lt;- sceFlt[batch][[1]]$Capture</code></pre>
<pre class="r"><code>p &lt;- lapply(sceFlt, function(sce){
  p1 &lt;- ggcells(sce) + 
    geom_bar(
      aes(x = GeneticDonor, fill = dmmHTO),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab(&quot;Frequency&quot;) +
    theme_cowplot(font_size = 10) 
  p2 &lt;- ggcells(sce) + 
    geom_bar(
      aes(x = GeneticDonor, fill = Capture),
      position = position_fill(reverse = TRUE)) +
    coord_flip() +
    ylab(&quot;Frequency&quot;) +
    theme_cowplot(font_size = 10)
  p3 &lt;- ggcells(sce) + 
    geom_bar(aes(x = GeneticDonor, fill = GeneticDonor)) + 
    coord_flip() + 
    ylab(&quot;Number of droplets&quot;) + 
    theme_cowplot(font_size = 10) + 
    geom_text(stat=&#39;count&#39;, aes(x = GeneticDonor, label=..count..), hjust=1.5, size=2) +
    guides(fill = FALSE)
  p1 / p2 / p3 + plot_layout(guides = &quot;collect&quot;)
})

p</code></pre>
<pre><code>$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch0/data/SCEs/C133_Neeland_batch0.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/experiment-by-donor-postqc-1.png" alt="Breakdown of the samples following QC." width="672" />
<p class="caption">
Breakdown of the samples following QC.
</p>
</div>
<pre><code>
$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch1/data/SCEs/C133_Neeland_batch1.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/experiment-by-donor-postqc-2.png" alt="Breakdown of the samples following QC." width="672" />
<p class="caption">
Breakdown of the samples following QC.
</p>
</div>
<pre><code>
$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch2/data/SCEs/C133_Neeland_batch2.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/experiment-by-donor-postqc-3.png" alt="Breakdown of the samples following QC." width="672" />
<p class="caption">
Breakdown of the samples following QC.
</p>
</div>
<pre><code>
$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch3/data/SCEs/C133_Neeland_batch3.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/experiment-by-donor-postqc-4.png" alt="Breakdown of the samples following QC." width="672" />
<p class="caption">
Breakdown of the samples following QC.
</p>
</div>
<pre><code>
$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch4/data/SCEs/C133_Neeland_batch4.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/experiment-by-donor-postqc-5.png" alt="Breakdown of the samples following QC." width="672" />
<p class="caption">
Breakdown of the samples following QC.
</p>
</div>
<pre><code>
$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch5/data/SCEs/C133_Neeland_batch5.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/experiment-by-donor-postqc-6.png" alt="Breakdown of the samples following QC." width="672" />
<p class="caption">
Breakdown of the samples following QC.
</p>
</div>
<pre><code>
$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch6/data/SCEs/C133_Neeland_batch6.preprocessed.SCE.rds`</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/02.0_quality_control.Rmd/experiment-by-donor-postqc-7.png" alt="Breakdown of the samples following QC." width="672" />
<p class="caption">
Breakdown of the samples following QC.
</p>
</div>
</div>
<div id="save-data" class="section level1">
<h1>Save data</h1>
<pre class="r"><code>batches &lt;- str_extract(names(sceFlt), &quot;batch[0-6]&quot;)

sapply(1:length(sceFlt), function(i){
  out &lt;- here(&quot;data&quot;,
              paste0(&quot;C133_Neeland_&quot;, batches[i]),
              &quot;data&quot;, 
              &quot;SCEs&quot;, 
              glue(&quot;C133_Neeland_{batches[i]}.quality_filtered.SCE.rds&quot;))
  if(!file.exists(out)) saveRDS(sceFlt[[i]], out)
  fs::file_chmod(out, &quot;664&quot;)
  if(any(str_detect(fs::group_ids()$group_name, 
                    &quot;oshlack_lab&quot;))) fs::file_chown(out, 
                                                    group_id = &quot;oshlack_lab&quot;)
})</code></pre>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL

[[4]]
NULL

[[5]]
NULL

[[6]]
NULL

[[7]]
NULL</code></pre>
</div>
<div id="session-info" class="section level1 appendix">
<h1 class="appendix">Session info</h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Melbourne
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] msigdbr_7.5.1                          
 [2] Homo.sapiens_1.3.1                     
 [3] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
 [4] org.Hs.eg.db_3.18.0                    
 [5] GO.db_3.18.0                           
 [6] OrganismDbi_1.44.0                     
 [7] EnsDb.Hsapiens.v86_2.99.0              
 [8] ensembldb_2.26.0                       
 [9] AnnotationFilter_1.26.0                
[10] GenomicFeatures_1.54.3                 
[11] AnnotationDbi_1.64.1                   
[12] cowplot_1.1.3                          
[13] scater_1.30.1                          
[14] scran_1.30.2                           
[15] scuttle_1.12.0                         
[16] SingleCellExperiment_1.24.0            
[17] SummarizedExperiment_1.32.0            
[18] Biobase_2.62.0                         
[19] GenomicRanges_1.54.1                   
[20] GenomeInfoDb_1.38.6                    
[21] IRanges_2.36.0                         
[22] S4Vectors_0.40.2                       
[23] BiocGenerics_0.48.1                    
[24] MatrixGenerics_1.14.0                  
[25] matrixStats_1.2.0                      
[26] patchwork_1.2.0                        
[27] glue_1.7.0                             
[28] here_1.0.1                             
[29] lubridate_1.9.3                        
[30] forcats_1.0.0                          
[31] stringr_1.5.1                          
[32] dplyr_1.1.4                            
[33] purrr_1.0.2                            
[34] readr_2.1.5                            
[35] tidyr_1.3.1                            
[36] tibble_3.2.1                           
[37] ggplot2_3.4.4                          
[38] tidyverse_2.0.0                        
[39] BiocStyle_2.30.0                       
[40] workflowr_1.7.1                        

loaded via a namespace (and not attached):
  [1] later_1.3.2               BiocIO_1.12.0            
  [3] bitops_1.0-7              filelock_1.0.3           
  [5] graph_1.80.0              XML_3.99-0.16.1          
  [7] lifecycle_1.0.4           edgeR_4.0.15             
  [9] rprojroot_2.0.4           processx_3.8.3           
 [11] lattice_0.22-5            magrittr_2.0.3           
 [13] limma_3.58.1              sass_0.4.8               
 [15] rmarkdown_2.25            jquerylib_0.1.4          
 [17] yaml_2.3.8                metapod_1.10.1           
 [19] httpuv_1.6.14             DBI_1.2.1                
 [21] abind_1.4-5               zlibbioc_1.48.0          
 [23] RCurl_1.98-1.14           rappdirs_0.3.3           
 [25] git2r_0.33.0              GenomeInfoDbData_1.2.11  
 [27] ggrepel_0.9.5             irlba_2.3.5.1            
 [29] dqrng_0.3.2               DelayedMatrixStats_1.24.0
 [31] codetools_0.2-19          DelayedArray_0.28.0      
 [33] xml2_1.3.6                tidyselect_1.2.0         
 [35] farver_2.1.1              ScaledMatrix_1.10.0      
 [37] viridis_0.6.5             BiocFileCache_2.10.1     
 [39] GenomicAlignments_1.38.2  jsonlite_1.8.8           
 [41] BiocNeighbors_1.20.2      tools_4.3.2              
 [43] progress_1.2.3            Rcpp_1.0.12              
 [45] gridExtra_2.3             SparseArray_1.2.4        
 [47] xfun_0.42                 withr_3.0.0              
 [49] BiocManager_1.30.22       fastmap_1.1.1            
 [51] bluster_1.12.0            fansi_1.0.6              
 [53] callr_3.7.3               digest_0.6.34            
 [55] rsvd_1.0.5                timechange_0.3.0         
 [57] R6_2.5.1                  colorspace_2.1-0         
 [59] biomaRt_2.58.2            RSQLite_2.3.5            
 [61] utf8_1.2.4                generics_0.1.3           
 [63] renv_1.0.3                rtracklayer_1.62.0       
 [65] prettyunits_1.2.0         httr_1.4.7               
 [67] S4Arrays_1.2.0            whisker_0.4.1            
 [69] pkgconfig_2.0.3           gtable_0.3.4             
 [71] blob_1.2.4                XVector_0.42.0           
 [73] htmltools_0.5.7           RBGL_1.78.0              
 [75] ProtGenerics_1.34.0       scales_1.3.0             
 [77] png_0.1-8                 knitr_1.45               
 [79] rstudioapi_0.15.0         tzdb_0.4.0               
 [81] rjson_0.2.21              curl_5.2.0               
 [83] cachem_1.0.8              parallel_4.3.2           
 [85] vipor_0.4.7               restfulr_0.0.15          
 [87] pillar_1.9.0              grid_4.3.2               
 [89] vctrs_0.6.5               promises_1.2.1           
 [91] BiocSingular_1.18.0       dbplyr_2.4.0             
 [93] beachmat_2.18.1           cluster_2.1.6            
 [95] beeswarm_0.4.0            evaluate_0.23            
 [97] cli_3.6.2                 locfit_1.5-9.8           
 [99] compiler_4.3.2            Rsamtools_2.18.0         
[101] rlang_1.1.3               crayon_1.5.2             
[103] labeling_0.4.3            ps_1.7.6                 
[105] getPass_0.2-4             fs_1.6.3                 
[107] ggbeeswarm_0.7.2          stringi_1.8.3            
[109] viridisLite_0.4.2         BiocParallel_1.36.0      
[111] babelgene_22.9            munsell_0.5.0            
[113] Biostrings_2.70.2         lazyeval_0.2.2           
[115] Matrix_1.6-5              hms_1.1.3                
[117] sparseMatrixStats_1.14.0  bit64_4.0.5              
[119] KEGGREST_1.42.0           statmod_1.5.0            
[121] highr_0.10                igraph_2.0.1.1           
[123] memoise_2.0.1             bslib_0.6.1              
[125] bit_4.0.5                </code></pre>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr />
<ol>
<li id="fn1"><p>Some care is taken to account for missing and duplicate
gene symbols; missing symbols are replaced with the Ensembl identifier
and duplicated symbols are concatenated with the (unique) Ensembl
identifiers.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>The number of expressed features refers to the number of
genes which have non-zero counts (i.e. they have been identified in the
cell at least once)<a href="#fnref2"
class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>It is important to note that we only using droplets
assigned to a sample (i.e. we ignore <code>unassigned</code> droplets)
for the calculation of these thresholds.<a href="#fnref3"
class="footnote-back">↩︎</a></p></li>
</ol>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
