<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2024-02-27" />

<title>Filter doublets and add sample information to C133_Neeland data</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-inflammation-CITEseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Filter doublets and add sample information
to C133_Neeland data</h1>
<h3 class="subtitle">Batches 0-6</h3>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">February 27, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-27
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>paed-inflammation-CITEseq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong2176c1e">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 2176c1e
</a>
</p>
</div>
<div id="strongRepositoryversionstrong2176c1e"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 2176c1e.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  analysis/05.0_remove_ambient.Rmd
    Untracked:  analysis/06.0_azimuth_annotation.Rmd
    Untracked:  analysis/06.1_azimuth_annotation_decontx.Rmd
    Untracked:  code/dropletutils.R
    Untracked:  code/utility.R
    Untracked:  data/.DS_Store
    Untracked:  data/C133_Neeland_batch0/
    Untracked:  data/C133_Neeland_batch1/
    Untracked:  data/C133_Neeland_batch2/
    Untracked:  data/C133_Neeland_batch3/
    Untracked:  data/C133_Neeland_batch4/
    Untracked:  data/C133_Neeland_batch5/
    Untracked:  data/C133_Neeland_batch6/
    Untracked:  data/CZI_samples_design_with_micro.xlsx
    Untracked:  renv.lock
    Untracked:  renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitignore
    Modified:   analysis/01.0_preprocess_batch0.Rmd
    Modified:   analysis/01.1_preprocess_batch1.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/04.0_filter_doublets.Rmd</code>)
and HTML (<code>docs/04.0_filter_doublets.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
c2985bb
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
59f1b28
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-27
</td>
<td>
wflow_publish("analysis/04.0_filter_doublets.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>suppressPackageStartupMessages({
  library(BiocStyle)
  library(tidyverse)
  library(here)
  library(glue)
  library(patchwork)
  library(paletteer)
  library(readxl)
  library(ggupset)
  library(stringr)
})</code></pre>
</div>
<div id="load-data" class="section level1">
<h1>Load Data</h1>
<pre class="r"><code>files &lt;- list.files(here(&quot;data&quot;,
                         paste0(&quot;C133_Neeland_batch&quot;, 0:6),
                         &quot;data&quot;, 
                         &quot;SCEs&quot;), 
                      pattern = &quot;doublets_called&quot;,
                      full.names = TRUE)
  
sceLst &lt;- sapply(files, function(fn){
    readRDS(file = fn)
})

sceLst</code></pre>
<pre><code>$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch0/data/SCEs/C133_Neeland_batch0.doublets_called.SCE.rds`
class: SingleCellExperiment 
dim: 33538 26939 
metadata(0):
assays(1): counts
rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475
  ENSG00000268674
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(26939): 1_AAACCCAAGCTAGTTC-1 1_AAACCCACAGTCGCTG-1 ...
  4_TTTGTTGTCTAGTACG-1 4_TTTGTTGTCTCGAACA-1
colData names(25): Barcode Capture ... scDblFinder.cxds_score batch
reducedDimNames(0):
mainExpName: NULL
altExpNames(0):

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch1/data/SCEs/C133_Neeland_batch1.doublets_called.SCE.rds`
class: SingleCellExperiment 
dim: 36601 21970 
metadata(0):
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(21970): 1_AAACCCACACTTCCTG-1 1_AAACCCACAGACAAAT-1 ...
  2_TTTGTTGTCATTGGTG-1 2_TTTGTTGTCGATGGAG-1
colData names(32): Barcode Capture ... scDblFinder.cxds_score batch
reducedDimNames(0):
mainExpName: NULL
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch2/data/SCEs/C133_Neeland_batch2.doublets_called.SCE.rds`
class: SingleCellExperiment 
dim: 36601 45711 
metadata(0):
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(45711): 1_AAACCCAAGACCTGGA-1 1_AAACCCAAGACTGTTC-1 ...
  2_TTTGTTGTCTCATGGA-1 2_TTTGTTGTCTCCAAGA-1
colData names(32): Barcode Capture ... scDblFinder.cxds_score batch
reducedDimNames(0):
mainExpName: NULL
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch3/data/SCEs/C133_Neeland_batch3.doublets_called.SCE.rds`
class: SingleCellExperiment 
dim: 36601 58038 
metadata(0):
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(58038): 1_AAACCCAAGCAGCACA-1 1_AAACCCAAGCATCTTG-1 ...
  2_TTTGTTGTCTAGGCCG-1 2_TTTGTTGTCTCGGCTT-1
colData names(32): Barcode Capture ... scDblFinder.cxds_score batch
reducedDimNames(0):
mainExpName: NULL
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch4/data/SCEs/C133_Neeland_batch4.doublets_called.SCE.rds`
class: SingleCellExperiment 
dim: 36601 45760 
metadata(0):
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(45760): 1_AAACCCAAGCGTTAGG-1 1_AAACCCAAGGATTTGA-1 ...
  2_TTTGTTGTCGACGATT-1 2_TTTGTTGTCTAGGCCG-1
colData names(32): Barcode Capture ... scDblFinder.cxds_score batch
reducedDimNames(0):
mainExpName: NULL
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch5/data/SCEs/C133_Neeland_batch5.doublets_called.SCE.rds`
class: SingleCellExperiment 
dim: 36601 44715 
metadata(0):
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(44715): 1_AAACCCAAGAAGATCT-1 1_AAACCCAAGATGCAGC-1 ...
  2_TTTGTTGTCGGATTAC-1 2_TTTGTTGTCTGAGAGG-1
colData names(32): Barcode Capture ... scDblFinder.cxds_score batch
reducedDimNames(0):
mainExpName: NULL
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch6/data/SCEs/C133_Neeland_batch6.doublets_called.SCE.rds`
class: SingleCellExperiment 
dim: 36601 46485 
metadata(0):
assays(1): counts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(46485): 1_AAACCCAAGAAGCGCT-1 1_AAACCCAAGACTCATC-1 ...
  2_TTTGTTGTCGAGAATA-1 2_TTTGTTGTCTACTGAG-1
colData names(32): Barcode Capture ... scDblFinder.cxds_score batch
reducedDimNames(0):
mainExpName: NULL
altExpNames(2): HTO ADT</code></pre>
</div>
<div id="load-sample-information" class="section level1">
<h1>Load sample information</h1>
<pre class="r"><code>ext_metadata_df &lt;- read_excel(
  here(&quot;data/CZI_samples_design_with_micro.xlsx&quot;),
  col_types = 
    c(&quot;text&quot;, &quot;numeric&quot;, &quot;text&quot;, &quot;text&quot;, &quot;text&quot;, &quot;numeric&quot;, &quot;numeric&quot;, &quot;text&quot;,
      &quot;numeric&quot;, &quot;text&quot;, &quot;numeric&quot;, &quot;numeric&quot;, &quot;numeric&quot;, &quot;numeric&quot;,
      &quot;text&quot;, &quot;text&quot;, &quot;text&quot;)) %&gt;%
  mutate(HASHTAG = paste0(&quot;Human_HTO_&quot;, HASHTAG)) %&gt;%
  mutate(sample.id = ave(Donor_ID, Donor_ID, 
                         FUN = function(i) paste0(&quot;sample_&quot;, i, &#39;.&#39;, seq_along(i)))) %&gt;%
  dplyr::rename(Batch = BATCH, 
                Donor = Donor_ID, 
                Hashtag = HASHTAG,
                arest.id = AREST_ID) %&gt;%
  mutate(Donor = as.factor(Donor))
knitr::kable(ext_metadata_df)</code></pre>
<table>
<colgroup>
<col width="2%" />
<col width="1%" />
<col width="2%" />
<col width="6%" />
<col width="2%" />
<col width="3%" />
<col width="3%" />
<col width="1%" />
<col width="1%" />
<col width="3%" />
<col width="5%" />
<col width="5%" />
<col width="4%" />
<col width="4%" />
<col width="30%" />
<col width="10%" />
<col width="6%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">arest.id</th>
<th align="left">Donor</th>
<th align="left">Disease</th>
<th align="left">Treatment</th>
<th align="left">Severity</th>
<th align="right">Experiment</th>
<th align="right">Age</th>
<th align="left">Sex</th>
<th align="right">Batch</th>
<th align="left">Hashtag</th>
<th align="right">Micobiology_done</th>
<th align="right">Bacteria_detected</th>
<th align="right">Virus_detected</th>
<th align="right">Fungi_detected</th>
<th align="left">Bacteria_type</th>
<th align="left">Virus_type</th>
<th align="left">Fungi_type</th>
<th align="left">sample.id</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">M1C141</td>
<td align="left">25</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">Severe</td>
<td align="right">2</td>
<td align="right">4.9100000</td>
<td align="left">F</td>
<td align="right">1</td>
<td align="left">Human_HTO_3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">Haemophilus parainfluenzae</td>
<td align="left">NA</td>
<td align="left">Aspergillus fumigatus</td>
<td align="left">sample_25.1</td>
</tr>
<tr class="even">
<td align="left">M1C156</td>
<td align="left">26</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">2</td>
<td align="right">5.0493151</td>
<td align="left">M</td>
<td align="right">1</td>
<td align="left">Human_HTO_6</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">S. aureus</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_26.1</td>
</tr>
<tr class="odd">
<td align="left">M1C130</td>
<td align="left">27</td>
<td align="left">CF</td>
<td align="left">treated (ivacaftor)</td>
<td align="left">mild</td>
<td align="right">2</td>
<td align="right">4.9178082</td>
<td align="left">M</td>
<td align="right">1</td>
<td align="left">Human_HTO_7</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_27.1</td>
</tr>
<tr class="even">
<td align="left">M1C122</td>
<td align="left">29</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">severe</td>
<td align="right">2</td>
<td align="right">5.9890411</td>
<td align="left">F</td>
<td align="right">1</td>
<td align="left">Human_HTO_1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">S. aureus, Stenotrophomonas, E.coli, Haemophilus
influenzae</td>
<td align="left">Parainfluenza virus 2</td>
<td align="left">NA</td>
<td align="left">sample_29.1</td>
</tr>
<tr class="odd">
<td align="left">M1C118</td>
<td align="left">30</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">severe</td>
<td align="right">2</td>
<td align="right">6.0164384</td>
<td align="left">F</td>
<td align="right">1</td>
<td align="left">Human_HTO_2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Mycobacterium Avium, Haemophilus parainfluenzae</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_30.1</td>
</tr>
<tr class="even">
<td align="left">M1C125</td>
<td align="left">31</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">2</td>
<td align="right">5.8904110</td>
<td align="left">F</td>
<td align="right">1</td>
<td align="left">Human_HTO_4</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_31.1</td>
</tr>
<tr class="odd">
<td align="left">M1C1265</td>
<td align="left">32</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">2</td>
<td align="right">5.9260274</td>
<td align="left">F</td>
<td align="right">1</td>
<td align="left">Human_HTO_5</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_32.1</td>
</tr>
<tr class="even">
<td align="left">M1C129</td>
<td align="left">33</td>
<td align="left">CF</td>
<td align="left">treated (ivacaftor)</td>
<td align="left">severe</td>
<td align="right">2</td>
<td align="right">5.9506849</td>
<td align="left">M</td>
<td align="right">1</td>
<td align="left">Human_HTO_8</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">S. aureus, Haemophilus parainfluenzae,
Stenotrophomonas</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_33.1</td>
</tr>
<tr class="odd">
<td align="left">M1N050</td>
<td align="left">5</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="right">3</td>
<td align="right">8.3972603</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="left">Human_HTO_7</td>
<td align="right">0</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_5.1</td>
</tr>
<tr class="even">
<td align="left">M1N057</td>
<td align="left">10</td>
<td align="left">Wheeze</td>
<td align="left">Wheeze</td>
<td align="left">Wheeze</td>
<td align="right">3</td>
<td align="right">3.6657534</td>
<td align="left">M</td>
<td align="right">2</td>
<td align="left">Human_HTO_6</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Haemophilus influenzae</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_10.1</td>
</tr>
<tr class="odd">
<td align="left">M1C161.1</td>
<td align="left">37</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">2.9479452</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="left">Human_HTO_9</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Haemophilus influenzae</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_37.1</td>
</tr>
<tr class="even">
<td align="left">M1C161.2</td>
<td align="left">37</td>
<td align="left">CF</td>
<td align="left">treated (orkambi)</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">4.5726027</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="left">Human_HTO_10</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Haemphilus parainfluenzae, Haemophilus
Parahemolyticus</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_37.2</td>
</tr>
<tr class="odd">
<td align="left">M1C161.3</td>
<td align="left">37</td>
<td align="left">CF</td>
<td align="left">treated (orkambi)</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">4.9178082</td>
<td align="left">F</td>
<td align="right">2</td>
<td align="left">Human_HTO_12</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">S. aureus, Haemophilus parainfluenza</td>
<td align="left">Rhinovirus, Enterovirus</td>
<td align="left">NA</td>
<td align="left">sample_37.3</td>
</tr>
<tr class="even">
<td align="left">M1C159.1</td>
<td align="left">38</td>
<td align="left">CF</td>
<td align="left">treated (ivacaftor)</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">2.9698630</td>
<td align="left">M</td>
<td align="right">2</td>
<td align="left">Human_HTO_13</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Haemophilus parainfluenzae, Haemophilus
parahaemolyticus, Haemophilus haemolyticus</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_38.1</td>
</tr>
<tr class="odd">
<td align="left">M1C159.2</td>
<td align="left">38</td>
<td align="left">CF</td>
<td align="left">treated (ivacaftor)</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">4.0575342</td>
<td align="left">M</td>
<td align="right">2</td>
<td align="left">Human_HTO_14</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">Haemophilus influenzae</td>
<td align="left">Enterovirus, Human metapneumovirus</td>
<td align="left">NA</td>
<td align="left">sample_38.2</td>
</tr>
<tr class="even">
<td align="left">M1C159.3</td>
<td align="left">38</td>
<td align="left">CF</td>
<td align="left">treated (ivacaftor)</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">5.0931507</td>
<td align="left">M</td>
<td align="right">2</td>
<td align="left">Human_HTO_15</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Haemophilus Haemolyticus, Haemophilus
parainfluenza</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_38.3</td>
</tr>
<tr class="odd">
<td align="left">M1N047</td>
<td align="left">1</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="right">3</td>
<td align="right">0.8356164</td>
<td align="left">M</td>
<td align="right">3</td>
<td align="left">Human_HTO_6</td>
<td align="right">0</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_1.1</td>
</tr>
<tr class="even">
<td align="left">M1N063</td>
<td align="left">2</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="right">3</td>
<td align="right">1.1068493</td>
<td align="left">M</td>
<td align="right">3</td>
<td align="left">Human_HTO_7</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">moraxella catarrhalis</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_2.1</td>
</tr>
<tr class="odd">
<td align="left">M1N048</td>
<td align="left">6</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="right">3</td>
<td align="right">5.3534247</td>
<td align="left">M</td>
<td align="right">3</td>
<td align="left">Human_HTO_14</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">S. aureus</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_6.1</td>
</tr>
<tr class="even">
<td align="left">M1N062</td>
<td align="left">7</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="right">3</td>
<td align="right">4.7095890</td>
<td align="left">F</td>
<td align="right">3</td>
<td align="left">Human_HTO_15</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_7.1</td>
</tr>
<tr class="odd">
<td align="left">M1C178</td>
<td align="left">15</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">0.9369863</td>
<td align="left">M</td>
<td align="right">3</td>
<td align="left">Human_HTO_12</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">Enterobacter cloacae, Haemophilus parainfluenzae</td>
<td align="left">NA</td>
<td align="left">Candida</td>
<td align="left">sample_15.1</td>
</tr>
<tr class="even">
<td align="left">M1C179</td>
<td align="left">16</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">0.9232877</td>
<td align="left">F</td>
<td align="right">3</td>
<td align="left">Human_HTO_13</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">Aspergillus fumigatus</td>
<td align="left">sample_16.1</td>
</tr>
<tr class="odd">
<td align="left">M1C168</td>
<td align="left">21</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">severe</td>
<td align="right">3</td>
<td align="right">2.9780822</td>
<td align="left">F</td>
<td align="right">3</td>
<td align="left">Human_HTO_9</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Haemophilus influenzae, MRSA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_21.1</td>
</tr>
<tr class="even">
<td align="left">M1C164</td>
<td align="left">22</td>
<td align="left">CF</td>
<td align="left">treated (ivacaftor)</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">3.0301370</td>
<td align="left">M</td>
<td align="right">3</td>
<td align="left">Human_HTO_10</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">S. aureus, Haemophilus haemolyticus</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_22.1</td>
</tr>
<tr class="odd">
<td align="left">M1N053</td>
<td align="left">8</td>
<td align="left">Wheeze</td>
<td align="left">Wheeze</td>
<td align="left">Wheeze</td>
<td align="right">3</td>
<td align="right">1.0438356</td>
<td align="left">M</td>
<td align="right">4</td>
<td align="left">Human_HTO_7</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">Haemophilus influenzae, Haemophilus parainfluenzae</td>
<td align="left">RSV</td>
<td align="left">NA</td>
<td align="left">sample_8.1</td>
</tr>
<tr class="even">
<td align="left">M1N061</td>
<td align="left">9</td>
<td align="left">Wheeze</td>
<td align="left">Wheeze</td>
<td align="left">Wheeze</td>
<td align="right">3</td>
<td align="right">0.9698630</td>
<td align="left">M</td>
<td align="right">4</td>
<td align="left">Human_HTO_9</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="left">Rhinovirus, Adenovirus</td>
<td align="left">NA</td>
<td align="left">sample_9.1</td>
</tr>
<tr class="odd">
<td align="left">M1N059</td>
<td align="left">12</td>
<td align="left">CSLD</td>
<td align="left">CSLD</td>
<td align="left">CSLD</td>
<td align="right">3</td>
<td align="right">3.8739726</td>
<td align="left">F</td>
<td align="right">4</td>
<td align="left">Human_HTO_6</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_12.1</td>
</tr>
<tr class="even">
<td align="left">M1N056</td>
<td align="left">13</td>
<td align="left">CSLD</td>
<td align="left">CSLD</td>
<td align="left">CSLD</td>
<td align="right">3</td>
<td align="right">5.9287671</td>
<td align="left">F</td>
<td align="right">4</td>
<td align="left">Human_HTO_10</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">S. aureus, Haemophilus parainfluenzae</td>
<td align="left">NA</td>
<td align="left">Aspergillus fumigatus</td>
<td align="left">sample_13.1</td>
</tr>
<tr class="odd">
<td align="left">M1C181</td>
<td align="left">17</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">0.9698630</td>
<td align="left">F</td>
<td align="right">4</td>
<td align="left">Human_HTO_12</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">Burkholderia cenocepacia</td>
<td align="left">Parainfluenza virus 3, RSV</td>
<td align="left">NA</td>
<td align="left">sample_17.1</td>
</tr>
<tr class="even">
<td align="left">M1C193</td>
<td align="left">18</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">1.1041096</td>
<td align="left">M</td>
<td align="right">4</td>
<td align="left">Human_HTO_13</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">Klebsiella Aerogens, Haemophilus parainfluenzae</td>
<td align="left">Rhnovirus, Enterovirus</td>
<td align="left">NA</td>
<td align="left">sample_18.1</td>
</tr>
<tr class="odd">
<td align="left">M1C152.1</td>
<td align="left">39</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">3.9260274</td>
<td align="left">F</td>
<td align="right">4</td>
<td align="left">Human_HTO_14</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Streptococcus pneumoniae, Moraxella catarrhalis ,
Haemophilus parainfluenzae, Haemophilus influenzae</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_39.1</td>
</tr>
<tr class="even">
<td align="left">M1C152.2</td>
<td align="left">39</td>
<td align="left">CF</td>
<td align="left">treated (orkambi)</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">6.1890411</td>
<td align="left">F</td>
<td align="right">4</td>
<td align="left">Human_HTO_15</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">Streptococcus pneumoniae, Haemophilus influenzae</td>
<td align="left">Rhinovirus</td>
<td align="left">NA</td>
<td align="left">sample_39.2</td>
</tr>
<tr class="odd">
<td align="left">M1N055</td>
<td align="left">3</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="right">3</td>
<td align="right">1.0958904</td>
<td align="left">M</td>
<td align="right">5</td>
<td align="left">Human_HTO_6</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">Haemophilus influenzae, Moraxella catarrhalis</td>
<td align="left">Rhinovirus</td>
<td align="left">NA</td>
<td align="left">sample_3.1</td>
</tr>
<tr class="even">
<td align="left">M1N065</td>
<td align="left">11</td>
<td align="left">Wheeze</td>
<td align="left">Wheeze</td>
<td align="left">Wheeze</td>
<td align="right">3</td>
<td align="right">5.9013699</td>
<td align="left">M</td>
<td align="right">5</td>
<td align="left">Human_HTO_12</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_11.1</td>
</tr>
<tr class="odd">
<td align="left">M1N054</td>
<td align="left">14</td>
<td align="left">CSLD</td>
<td align="left">CSLD</td>
<td align="left">CSLD</td>
<td align="right">3</td>
<td align="right">7.4602740</td>
<td align="left">F</td>
<td align="right">5</td>
<td align="left">Human_HTO_7</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Haemophilus influenzae</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_14.1</td>
</tr>
<tr class="even">
<td align="left">M1C177</td>
<td align="left">19</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">0.9506849</td>
<td align="left">F</td>
<td align="right">5</td>
<td align="left">Human_HTO_9</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_19.1</td>
</tr>
<tr class="odd">
<td align="left">M1C188</td>
<td align="left">20</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">severe</td>
<td align="right">3</td>
<td align="right">1.4739726</td>
<td align="left">M</td>
<td align="right">5</td>
<td align="left">Human_HTO_10</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Klebsiella Oxytoca, Staph Aureus (Methicillin
Resistant)</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_20.1</td>
</tr>
<tr class="even">
<td align="left">M1C158</td>
<td align="left">23</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">severe</td>
<td align="right">3</td>
<td align="right">3.9232877</td>
<td align="left">F</td>
<td align="right">5</td>
<td align="left">Human_HTO_13</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">MRSA</td>
<td align="left">Influenza A</td>
<td align="left">NA</td>
<td align="left">sample_23.1</td>
</tr>
<tr class="odd">
<td align="left">M1C155</td>
<td align="left">24</td>
<td align="left">CF</td>
<td align="left">treated (ivacaftor)</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">5.2356164</td>
<td align="left">M</td>
<td align="right">5</td>
<td align="left">Human_HTO_14</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_24.1</td>
</tr>
<tr class="even">
<td align="left">M1C123</td>
<td align="left">28</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">5.9753425</td>
<td align="left">F</td>
<td align="right">5</td>
<td align="left">Human_HTO_15</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="left">Pseudomonas aeruginosa</td>
<td align="left">Influenza A</td>
<td align="left">NA</td>
<td align="left">sample_28.1</td>
</tr>
<tr class="odd">
<td align="left">M1N051</td>
<td align="left">4</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="right">3</td>
<td align="right">1.2246575</td>
<td align="left">M</td>
<td align="right">6</td>
<td align="left">Human_HTO_6</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">Haemophilus influenzae</td>
<td align="left">RSV</td>
<td align="left">NA</td>
<td align="left">sample_4.1</td>
</tr>
<tr class="even">
<td align="left">M1C184.1</td>
<td align="left">34</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">0.5232877</td>
<td align="left">F</td>
<td align="right">6</td>
<td align="left">Human_HTO_7</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="left">Influenza A, RSV</td>
<td align="left">NA</td>
<td align="left">sample_34.1</td>
</tr>
<tr class="odd">
<td align="left">M1C184.2</td>
<td align="left">34</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">1.0657534</td>
<td align="left">F</td>
<td align="right">6</td>
<td align="left">Human_HTO_9</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Haemophilus parainfluenzae</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_34.2</td>
</tr>
<tr class="even">
<td align="left">M1C184.3</td>
<td align="left">34</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">2.0575342</td>
<td align="left">F</td>
<td align="right">6</td>
<td align="left">Human_HTO_10</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Psuedomonas aeuroginosa, Haemophilus parainfluenzae,
haemophilus parahaemolyticus</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_34.3</td>
</tr>
<tr class="odd">
<td align="left">M1C183.1</td>
<td align="left">35</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">1.0575342</td>
<td align="left">M</td>
<td align="right">6</td>
<td align="left">Human_HTO_12</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">Serratia marcescens, Haemophilus parainfluenzae,
Bordetella pertussis</td>
<td align="left">Rhinovirus, Adenovirus</td>
<td align="left">NA</td>
<td align="left">sample_35.1</td>
</tr>
<tr class="even">
<td align="left">M1C183.2</td>
<td align="left">35</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">3</td>
<td align="right">1.9917808</td>
<td align="left">M</td>
<td align="right">6</td>
<td align="left">Human_HTO_13</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">Serratia Marcesens, Pseudomonas Aeruginosa, Haemophilus
parainfluenzae</td>
<td align="left">Rhinovirus, Enterovirus</td>
<td align="left">NA</td>
<td align="left">sample_35.2</td>
</tr>
<tr class="odd">
<td align="left">M1C185.1</td>
<td align="left">36</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">severe</td>
<td align="right">3</td>
<td align="right">0.9616438</td>
<td align="left">F</td>
<td align="right">6</td>
<td align="left">Human_HTO_14</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="left">RSV</td>
<td align="left">NA</td>
<td align="left">sample_36.1</td>
</tr>
<tr class="even">
<td align="left">M1C185.2</td>
<td align="left">36</td>
<td align="left">CF</td>
<td align="left">treated (orkambi)</td>
<td align="left">severe</td>
<td align="right">3</td>
<td align="right">2.2602740</td>
<td align="left">F</td>
<td align="right">6</td>
<td align="left">Human_HTO_15</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">S. aureus, Haemophilus parainfluenzae</td>
<td align="left">NA</td>
<td align="left">Candida</td>
<td align="left">sample_36.2</td>
</tr>
<tr class="odd">
<td align="left">M1C160 </td>
<td align="left">40</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">severe</td>
<td align="right">1</td>
<td align="right">2.9900000</td>
<td align="left">M</td>
<td align="right">0</td>
<td align="left">Human_HTO_A</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">Stenotrophomonas maltophilia, Enterobacter cloacae,
Burkholderia gladioli.</td>
<td align="left">RSV</td>
<td align="left">Cadida, Aspergillus</td>
<td align="left">sample_40.1</td>
</tr>
<tr class="even">
<td align="left">M1C162 </td>
<td align="left">41</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="right">1</td>
<td align="right">2.9900000</td>
<td align="left">M</td>
<td align="right">0</td>
<td align="left">Human_HTO_B</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Bortadella Pertussis</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_41.1</td>
</tr>
<tr class="odd">
<td align="left">M1C163 </td>
<td align="left">42</td>
<td align="left">CF</td>
<td align="left">treated (ivacaftor)</td>
<td align="left">severe</td>
<td align="right">1</td>
<td align="right">3.0300000</td>
<td align="left">M</td>
<td align="right">0</td>
<td align="left">Human_HTO_D</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_42.1</td>
</tr>
<tr class="even">
<td align="left">M1N052</td>
<td align="left">43</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="left">Healthy</td>
<td align="right">1</td>
<td align="right">3.0000000</td>
<td align="left">M</td>
<td align="right">0</td>
<td align="left">Human_HTO_C</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">S. aureus, Haemophilus parainfluenzae, Haemophilus
Hemolyticus, Haemophilus Parahemolyticus</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">sample_43.1</td>
</tr>
</tbody>
</table>
</div>
<div id="assign-sample-ids-to-cells" class="section level1">
<h1>Assign sample IDs to cells</h1>
<p>For genetically distinct samples, cells are assigned to a sample
using the genetic assignment; the sample information is then assigned
using the corresponding HTO. For longitudinal samples <em>only</em>,
both cells and sample information are assigned to the individual samples
using HTOs.</p>
<pre class="r"><code>sceLst &lt;- sapply(1:length(sceLst), function(i){
  message(i)
  sce &lt;- sceLst[[i]]
  
  # convert genetic and HTO labels to character
  sce$GeneticDonor &lt;- as.character(sce$GeneticDonor)
  sce$dmmHTO &lt;- ifelse(!stringr::str_detect(sce$dmmHTO, &quot;Human_HTO_&quot;) &amp; 
                         !(stringr::str_detect(sce$dmmHTO, &quot;Doublet|Negative&quot;)),
                       paste0(&quot;Human_HTO_&quot;, sce$dmmHTO),
                       as.character(sce$dmmHTO))
  
  # match genetics and HTOs
  tab &lt;- table(sce$GeneticDonor, sce$dmmHTO)
  tab &lt;- tab[!rownames(tab) %in% c(&quot;Doublet&quot;,&quot;Unknown&quot;),
             !colnames(tab) %in% c(&quot;Doublet&quot;,&quot;Negative&quot;)]
  # extract all genetically unique samples in this batch
  sub &lt;- tab[rowSums(tab/rowSums(tab) &gt; 0.2) == 1, , drop = FALSE]
  # create dictionary to translate genetics to HTO
  dict &lt;- apply(sub, 1, function(x){
    colnames(tab)[which(x == max(x))]
  })
  dict &lt;- c(dict, &quot;Doublet&quot; = &quot;Doublet&quot;, &quot;Unknown&quot; = &quot;Negative&quot;)
 
  sce$ProcessedHTO &lt;- NA
  # assign HTO based on genetics
  genetics &lt;- sce$GeneticDonor %in% names(dict)
  sce$ProcessedHTO[genetics] &lt;- unname(dict[sce$GeneticDonor[genetics]])
  
  # if there are longitudinal samples also assign HTO based on hashtags
  if(dim(sub)[1] != dim(sub)[2]){
    HTO &lt;- !sce$GeneticDonor %in% names(dict)
    sce$ProcessedHTO[HTO] &lt;- sce$dmmHTO[HTO]
    
  }
  
  # add sample metadata based on processed HTO
  colData(sce) %&gt;%
    data.frame() %&gt;%
    rownames_to_column(&quot;cell&quot;) %&gt;%
    left_join(ext_metadata_df %&gt;% 
                dplyr::filter(Batch == (i-1)) %&gt;%
                data.frame, 
              by = c(&quot;ProcessedHTO&quot; = &quot;Hashtag&quot;)) %&gt;%
    mutate(sample.id = ifelse(is.na(sample.id), 
                              ProcessedHTO,
                              sample.id)) %&gt;%
    column_to_rownames(&quot;cell&quot;) %&gt;% 
    DataFrame() -&gt; colData(sce)
  
  sce
})</code></pre>
<div id="number-of-cells-per-sample" class="section level2">
<h2>Number of cells per sample</h2>
<pre class="r"><code>p &lt;- lapply(1:length(sceLst), function(i){
  sce &lt;- sceLst[[i]]
  colData(sce) %&gt;%
    data.frame() %&gt;%
    ggplot(aes(x = sample.id, fill = Donor)) +
    geom_bar(position = position_dodge()) + 
    geom_text(aes(label = ..count..),
              stat = &quot;count&quot;, vjust = 1.5, 
              colour = &quot;black&quot;, size = 2) +
    theme(legend.position = &quot;none&quot;) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
    ggtitle(glue(&quot;batch {i-1}&quot;))
})

wrap_plots(p, ncol = 2)</code></pre>
<p><img src="figure/04.0_filter_doublets.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
c2985bb
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="no.-of-doublets-called-per-batch" class="section level2">
<h2>No. of doublets called per batch</h2>
<pre class="r"><code>p &lt;- lapply(1:length(sceLst), function(i){
  sce &lt;- sceLst[[i]]
  colData(sce) %&gt;%
    data.frame() %&gt;%
    mutate(scds = ifelse(hybrid_call, &quot;Doublet&quot;, &quot;Singlet&quot;),
           scdf = ifelse(scDblFinder.class == &quot;doublet&quot;, &quot;Doublet&quot;, &quot;Singlet&quot;)) %&gt;%
    dplyr::select(ProcessedHTO, scds, scdf, Capture) %&gt;% 
    rownames_to_column(var = &quot;cell&quot;) %&gt;%
    mutate(vireo_dmm_dbl = (ProcessedHTO == &quot;Doublet&quot;),
           scds_dbl = (scds == &quot;Doublet&quot;),
           scdf_dbl = (scdf == &quot;Doublet&quot;)) %&gt;%
    dplyr::select(cell, vireo_dmm_dbl, scds_dbl, scdf_dbl, Capture) %&gt;% 
    pivot_longer(cols = c(vireo_dmm_dbl, scds_dbl, scdf_dbl), names_to = &quot;method&quot;) %&gt;%
    dplyr::filter(value == TRUE) %&gt;%
    group_by(cell) %&gt;%
    summarise(data = list(method), Capture) %&gt;%
    rowwise() %&gt;%
    ggplot(aes(x = data)) +
    geom_bar() +
    scale_x_upset(n_intersections = 20) +
    geom_text(stat = &#39;count&#39;, aes(label = after_stat(count)), 
              vjust = -0.5, size = 2) +
    ggtitle(glue(&quot;batch {i-1}&quot;))
})

wrap_plots(p, ncol = 2)</code></pre>
<p><img src="figure/04.0_filter_doublets.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
c2985bb
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="doublet-filtering" class="section level1">
<h1>Doublet filtering</h1>
<p>We HAVE to filter out cells that are called doublets by either
<code>vireo</code> or <code>demuxmix</code> (for relevant samples only)
because we have no singlet assignment to a sample for them. We also HAVE
to filter out negative cells as they also have no singlet assignment. We
will also filter out cells that are called doublets by BOTH
<code>scds</code> and <code>scDblFinder</code> (both methods were
suggested in the demuxafy paper) that are not already in the
<code>vireo/demuxmix doublet pool</code>. The percentages for those
doublets calls per capture are shown below.</p>
<pre class="r"><code>lapply(sceLst, function(sce){
  scds_scdf &lt;- table(sce$Capture, sce$hybrid_call &amp; sce$scDblFinder.class == &quot;doublet&quot; &amp;
                       !sce$ProcessedHTO == &quot;Doublet&quot;)
  vireo_dmm &lt;- table(sce$Capture, sce$ProcessedHTO == &quot;Doublet&quot;)
  
  if(length(levels(sce$Capture)) &lt; 4) {
    data.frame(scds_scdf_dbl = scds_scdf[,2]/scds_scdf[,1]*100,
               vireo_dmm_dbl = vireo_dmm[,2]/vireo_dmm[,1]*100) -&gt; tab
  } else {
    data.frame(scds_scdf_dbl = scds_scdf[,2]/scds_scdf[,1]*100) -&gt; tab
  }
  
  tab %&gt;% knitr::kable()
})</code></pre>
<p>[[1]]</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">scds_scdf_dbl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="right">3.104529</td>
</tr>
<tr class="even">
<td align="left">B</td>
<td align="right">3.236475</td>
</tr>
<tr class="odd">
<td align="left">C</td>
<td align="right">6.003251</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="right">5.830370</td>
</tr>
</tbody>
</table>
<p>[[2]]</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">scds_scdf_dbl</th>
<th align="right">vireo_dmm_dbl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C133_batch1_1</td>
<td align="right">2.545172</td>
<td align="right">7.79128</td>
</tr>
<tr class="even">
<td align="left">C133_batch1_2</td>
<td align="right">2.533237</td>
<td align="right">11.10678</td>
</tr>
</tbody>
</table>
<p>[[3]]</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">scds_scdf_dbl</th>
<th align="right">vireo_dmm_dbl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C133_batch2_1</td>
<td align="right">1.635894</td>
<td align="right">29.69542</td>
</tr>
<tr class="even">
<td align="left">C133_batch2_2</td>
<td align="right">1.523074</td>
<td align="right">57.17682</td>
</tr>
</tbody>
</table>
<p>[[4]]</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">scds_scdf_dbl</th>
<th align="right">vireo_dmm_dbl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C133_batch3_1</td>
<td align="right">2.370001</td>
<td align="right">24.63025</td>
</tr>
<tr class="even">
<td align="left">C133_batch3_2</td>
<td align="right">2.425704</td>
<td align="right">20.25649</td>
</tr>
</tbody>
</table>
<p>[[5]]</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">scds_scdf_dbl</th>
<th align="right">vireo_dmm_dbl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C133_batch4_1</td>
<td align="right">3.619517</td>
<td align="right">18.65984</td>
</tr>
<tr class="even">
<td align="left">C133_batch4_2</td>
<td align="right">2.475446</td>
<td align="right">22.84497</td>
</tr>
</tbody>
</table>
<p>[[6]]</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">scds_scdf_dbl</th>
<th align="right">vireo_dmm_dbl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C133_batch5_1</td>
<td align="right">3.343934</td>
<td align="right">13.83751</td>
</tr>
<tr class="even">
<td align="left">C133_batch5_2</td>
<td align="right">3.217268</td>
<td align="right">18.20225</td>
</tr>
</tbody>
</table>
<p>[[7]]</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">scds_scdf_dbl</th>
<th align="right">vireo_dmm_dbl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C133_batch6_1</td>
<td align="right">1.518363</td>
<td align="right">42.77545</td>
</tr>
<tr class="even">
<td align="left">C133_batch6_2</td>
<td align="right">1.508269</td>
<td align="right">31.80439</td>
</tr>
</tbody>
</table>
<p>First, filter out <em>wheeze</em> and <em>CSLD</em> samples as we
will focus only on <em>CF</em>.</p>
<pre class="r"><code>sceLst &lt;- lapply(sceLst, function(sce){
  sce[, !sce$Disease %in% c(&quot;Wheeze&quot;, &quot;CSLD&quot;)]
})

p &lt;- lapply(1:length(sceLst), function(i){
  sce &lt;- sceLst[[i]]
  colData(sce) %&gt;%
    data.frame() %&gt;%
    ggplot(aes(x = sample.id, fill = Donor)) +
    geom_bar(position = position_dodge()) + 
    geom_text(aes(label = ..count..),
              stat = &quot;count&quot;, vjust = 1.5, 
              colour = &quot;black&quot;, size = 2) +
    theme(legend.position = &quot;none&quot;) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
    ggtitle(glue(&quot;batch {i-1}&quot;))
})

wrap_plots(p, ncol = 2)</code></pre>
<p><img src="figure/04.0_filter_doublets.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
c2985bb
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Then, filter out doublets.</p>
<pre class="r"><code>sceLst &lt;- lapply(sceLst, function(sce){
  sce[, !(sce$ProcessedHTO %in% c(&quot;Doublet&quot;) | 
                                  (sce$hybrid_call &amp; sce$scDblFinder.class == &quot;doublet&quot;))]
})

p &lt;- lapply(1:length(sceLst), function(i){
  sce &lt;- sceLst[[i]]
  colData(sce) %&gt;%
    data.frame() %&gt;%
    ggplot(aes(x = sample.id, fill = Donor)) +
    geom_bar(position = position_dodge()) + 
    geom_text(aes(label = ..count..),
              stat = &quot;count&quot;, vjust = 1.5, 
              colour = &quot;black&quot;, size = 2) +
    theme(legend.position = &quot;none&quot;) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
    ggtitle(glue(&quot;batch {i-1}&quot;))
})

wrap_plots(p, ncol = 2)</code></pre>
<p><img src="figure/04.0_filter_doublets.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
c2985bb
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Finally, filter out negatives cells.</p>
<pre class="r"><code>sceLst &lt;- lapply(sceLst, function(sce){
  sce[, !(sce$sample.id %in% &quot;Negative&quot;)]
})

p &lt;- lapply(1:length(sceLst), function(i){
  sce &lt;- sceLst[[i]]
  colData(sce) %&gt;%
    data.frame() %&gt;%
    ggplot(aes(x = sample.id, fill = Donor)) +
    geom_bar(position = position_dodge()) + 
    geom_text(aes(label = ..count..),
              stat = &quot;count&quot;, vjust = 1.5, 
              colour = &quot;black&quot;, size = 2) +
    theme(legend.position = &quot;none&quot;) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
    ggtitle(glue(&quot;batch {i-1}&quot;))
})

wrap_plots(p, ncol = 2)</code></pre>
<p><img src="figure/04.0_filter_doublets.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
c2985bb
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cell-counts-per-sample-after-filtering" class="section level1">
<h1>Cell counts per sample after filtering</h1>
<pre class="r"><code>p &lt;- lapply(sceLst, function(sce){
  colData(sce) %&gt;%
    data.frame() %&gt;%
    mutate(donor.fct = factor(Donor, labels = 1:length(unique(Donor)))) %&gt;%
    ggplot(aes(x = sample.id, fill = donor.fct)) +
    geom_bar(position = position_dodge()) + 
    geom_text(aes(label = ..count..),
              stat = &quot;count&quot;, vjust = 1.5, 
              colour = &quot;black&quot;, size = 2) +
    theme(legend.position = &quot;none&quot;) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))
})

wrap_plots(p, ncol = 2)</code></pre>
<p><img src="figure/04.0_filter_doublets.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
c2985bb
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="qc-per-sample-after-filtering" class="section level1">
<h1>QC per sample after filtering</h1>
<pre class="r"><code>p &lt;- lapply(sceLst, function(sce){
  colData(sce) %&gt;%
    data.frame() %&gt;% 
    mutate(donor.fct = factor(Donor, labels = 1:length(unique(Donor)))) %&gt;%
    ggplot(aes(x = sample.id, y = sum,
               fill = donor.fct)) +
    geom_violin() +
    scale_y_log10() +
    theme(legend.position = &quot;none&quot;) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
    labs(x = &quot;Sample&quot;,
         y = &quot;Library size (log10 scale)&quot;) -&gt; p1
  
  colData(sce) %&gt;%
    data.frame() %&gt;% 
    mutate(donor.fct = factor(Donor, labels = 1:length(unique(Donor)))) %&gt;%
    ggplot(aes(x = sample.id, y = detected,
               fill = donor.fct)) +
    geom_violin() +
    scale_y_log10() +
    theme(legend.position = &quot;none&quot;) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
    labs(x = &quot;Sample&quot;,
         y = &quot;No. genes detected (log10 scale)&quot;) -&gt; p2
  
  p1 + p2
})

wrap_plots(p, ncol = 1) &amp;
  theme(axis.title.y = element_text(size = 9))</code></pre>
<p><img src="figure/04.0_filter_doublets.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
c2985bb
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-data" class="section level1">
<h1>Save data</h1>
<pre class="r"><code>batches &lt;- str_extract(files, &quot;batch[0-6]&quot;)

sapply(1:length(sceLst), function(i){
  out &lt;- here(&quot;data&quot;,
              paste0(&quot;C133_Neeland_&quot;, batches[i]),
              &quot;data&quot;, 
              &quot;SCEs&quot;, 
              glue(&quot;C133_Neeland_{batches[i]}.doublets_filtered.SCE.rds&quot;))
  if(!file.exists(out)) saveRDS(sceLst[[i]], out)
  fs::file_chmod(out, &quot;664&quot;)
  if(any(str_detect(fs::group_ids()$group_name, 
                    &quot;oshlack_lab&quot;))) fs::file_chown(out, 
                                                    group_id = &quot;oshlack_lab&quot;)
})</code></pre>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL

[[4]]
NULL

[[5]]
NULL

[[6]]
NULL

[[7]]
NULL</code></pre>
</div>
<div id="session-info" class="section level1 appendix">
<h1 class="appendix">Session info</h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Melbourne
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] SingleCellExperiment_1.24.0 SummarizedExperiment_1.32.0
 [3] Biobase_2.62.0              GenomicRanges_1.54.1       
 [5] GenomeInfoDb_1.38.6         IRanges_2.36.0             
 [7] S4Vectors_0.40.2            BiocGenerics_0.48.1        
 [9] MatrixGenerics_1.14.0       matrixStats_1.2.0          
[11] ggupset_0.3.0               readxl_1.4.3               
[13] paletteer_1.6.0             patchwork_1.2.0            
[15] glue_1.7.0                  here_1.0.1                 
[17] lubridate_1.9.3             forcats_1.0.0              
[19] stringr_1.5.1               dplyr_1.1.4                
[21] purrr_1.0.2                 readr_2.1.5                
[23] tidyr_1.3.1                 tibble_3.2.1               
[25] ggplot2_3.4.4               tidyverse_2.0.0            
[27] BiocStyle_2.30.0            workflowr_1.7.1            

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0        farver_2.1.1            bitops_1.0-7           
 [4] fastmap_1.1.1           RCurl_1.98-1.14         promises_1.2.1         
 [7] digest_0.6.34           timechange_0.3.0        lifecycle_1.0.4        
[10] processx_3.8.3          magrittr_2.0.3          compiler_4.3.2         
[13] rlang_1.1.3             sass_0.4.8              tools_4.3.2            
[16] utf8_1.2.4              yaml_2.3.8              knitr_1.45             
[19] labeling_0.4.3          S4Arrays_1.2.0          DelayedArray_0.28.0    
[22] abind_1.4-5             withr_3.0.0             grid_4.3.2             
[25] fansi_1.0.6             git2r_0.33.0            colorspace_2.1-0       
[28] scales_1.3.0            cli_3.6.2               rmarkdown_2.25         
[31] crayon_1.5.2            generics_0.1.3          rstudioapi_0.15.0      
[34] httr_1.4.7              tzdb_0.4.0              cachem_1.0.8           
[37] zlibbioc_1.48.0         BiocManager_1.30.22     cellranger_1.1.0       
[40] XVector_0.42.0          vctrs_0.6.5             Matrix_1.6-5           
[43] jsonlite_1.8.8          callr_3.7.3             hms_1.1.3              
[46] jquerylib_0.1.4         rematch2_2.1.2          ps_1.7.6               
[49] stringi_1.8.3           gtable_0.3.4            later_1.3.2            
[52] munsell_0.5.0           pillar_1.9.0            htmltools_0.5.7        
[55] GenomeInfoDbData_1.2.11 R6_2.5.1                rprojroot_2.0.4        
[58] evaluate_0.23           lattice_0.22-5          highr_0.10             
[61] renv_1.0.3              httpuv_1.6.14           bslib_0.6.1            
[64] Rcpp_1.0.12             SparseArray_1.2.4       whisker_0.4.1          
[67] xfun_0.42               fs_1.6.3                getPass_0.2-4          
[70] pkgconfig_2.0.3        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
