<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2024-03-04" />

<title>Azimuth annotation of C133_Neeland data</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-inflammation-CITEseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Azimuth annotation of C133_Neeland
data</h1>
<h3 class="subtitle">Batches 0-6 (No ambient removal)</h3>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">March 04, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>paed-inflammation-CITEseq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong23ddb68">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 23ddb68
</a>
</p>
</div>
<div id="strongRepositoryversionstrong23ddb68"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 23ddb68.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  analysis/07.0_integrate_cluster_t_cells.Rmd
    Untracked:  code/dropletutils.R
    Untracked:  code/utility.R
    Untracked:  data/.DS_Store
    Untracked:  data/C133_Neeland_batch0/
    Untracked:  data/C133_Neeland_batch1/
    Untracked:  data/C133_Neeland_batch2/
    Untracked:  data/C133_Neeland_batch3/
    Untracked:  data/C133_Neeland_batch4/
    Untracked:  data/C133_Neeland_batch5/
    Untracked:  data/C133_Neeland_batch6/
    Untracked:  data/CZI_samples_design_with_micro.xlsx
    Untracked:  renv.lock
    Untracked:  renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitignore
    Modified:   analysis/01.0_preprocess_batch0.Rmd
    Modified:   analysis/01.1_preprocess_batch1.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/06.0_azimuth_annotation.Rmd</code>) and HTML
(<code>docs/06.0_azimuth_annotation.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
23ddb68
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-03-04
</td>
<td>
wflow_publish(c("analysis/06.0_azimuth_annotation.Rmd",
"analysis/06.1_azimuth_annotation_decontx.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
e16353b
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
ea5d591
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-29
</td>
<td>
wflow_publish("analysis/06.0_azimuth_annotation.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
7bf7f42
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
3f94bd1
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-29
</td>
<td>
wflow_publish("analysis/06.0_azimuth_annotation.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>suppressPackageStartupMessages({
  library(BiocStyle)
  library(tidyverse)
  library(here)
  library(glue)
  library(patchwork)
  library(scran)
  library(scater)
  library(scuttle)
  library(scMerge)
  library(ggupset)
  library(Seurat)
  library(SeuratData)
  library(Azimuth)
})</code></pre>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<pre class="r"><code>files &lt;- list.files(here(&quot;data&quot;,
                         paste0(&quot;C133_Neeland_batch&quot;, 0:6),
                         &quot;data&quot;, 
                         &quot;SCEs&quot;), 
                    pattern = &quot;ambient_removed&quot;,
                    full.names = TRUE)
  
sceLst &lt;- sapply(files, function(fn){
    readRDS(file = fn)
})

sceLst</code></pre>
<pre><code>$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch0/data/SCEs/C133_Neeland_batch0.ambient_removed.SCE.rds`
class: SingleCellExperiment 
dim: 33538 25635 
metadata(1): decontX
assays(2): counts decontXcounts
rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475
  ENSG00000268674
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(25635): 1_AAACCCAAGCTAGTTC-1 1_AAACCCACAGTCGCTG-1 ...
  4_TTTGTTGTCTAGTACG-1 4_TTTGTTGTCTCGAACA-1
colData names(45): Barcode Capture ... decontX_contamination
  decontX_clusters
reducedDimNames(4): decontX_A_UMAP decontX_B_UMAP decontX_C_UMAP
  decontX_D_UMAP
mainExpName: NULL
altExpNames(0):

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch1/data/SCEs/C133_Neeland_batch1.ambient_removed.SCE.rds`
class: SingleCellExperiment 
dim: 36601 19522 
metadata(1): decontX
assays(2): counts decontXcounts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(19522): 1_AAACCCACACTTCCTG-1 1_AAACCCACAGACAAAT-1 ...
  2_TTTGTTGTCATTGGTG-1 2_TTTGTTGTCGATGGAG-1
colData names(58): Barcode Capture ... decontX_contamination
  decontX_clusters
reducedDimNames(1): decontX_UMAP
mainExpName: NULL
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch2/data/SCEs/C133_Neeland_batch2.ambient_removed.SCE.rds`
class: SingleCellExperiment 
dim: 36601 22386 
metadata(1): decontX
assays(2): counts decontXcounts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(22386): 1_AAACCCAAGACTGTTC-1 1_AAACCCAAGATGATTG-1 ...
  2_TTTGTTGTCCAAGGGA-1 2_TTTGTTGTCCTTCTAA-1
colData names(58): Barcode Capture ... decontX_contamination
  decontX_clusters
reducedDimNames(1): decontX_UMAP
mainExpName: NULL
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch3/data/SCEs/C133_Neeland_batch3.ambient_removed.SCE.rds`
class: SingleCellExperiment 
dim: 36601 46052 
metadata(1): decontX
assays(2): counts decontXcounts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(46052): 1_AAACCCAAGCAGCACA-1 1_AAACCCAAGCATCTTG-1 ...
  2_TTTGTTGTCTAGGCCG-1 2_TTTGTTGTCTCGGCTT-1
colData names(58): Barcode Capture ... decontX_contamination
  decontX_clusters
reducedDimNames(1): decontX_UMAP
mainExpName: NULL
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch4/data/SCEs/C133_Neeland_batch4.ambient_removed.SCE.rds`
class: SingleCellExperiment 
dim: 36601 18858 
metadata(1): decontX
assays(2): counts decontXcounts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(18858): 1_AAACCCAAGGATTTGA-1 1_AAACCCAAGTCTCTGA-1 ...
  2_TTTGTTGCATGTGGCC-1 2_TTTGTTGGTCAACATC-1
colData names(58): Barcode Capture ... decontX_contamination
  decontX_clusters
reducedDimNames(1): decontX_UMAP
mainExpName: NULL
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch5/data/SCEs/C133_Neeland_batch5.ambient_removed.SCE.rds`
class: SingleCellExperiment 
dim: 36601 32959 
metadata(1): decontX
assays(2): counts decontXcounts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(32959): 1_AAACCCAAGAAGATCT-1 1_AAACCCAAGGAGAGGC-1 ...
  2_TTTGTTGTCGGATTAC-1 2_TTTGTTGTCTGAGAGG-1
colData names(58): Barcode Capture ... decontX_contamination
  decontX_clusters
reducedDimNames(1): decontX_UMAP
mainExpName: NULL
altExpNames(2): HTO ADT

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch6/data/SCEs/C133_Neeland_batch6.ambient_removed.SCE.rds`
class: SingleCellExperiment 
dim: 36601 31275 
metadata(1): decontX
assays(2): counts decontXcounts
rownames(36601): ENSG00000243485 ENSG00000237613 ... ENSG00000278817
  ENSG00000277196
rowData names(20): ID Symbol ... is_mito is_pseudogene
colnames(31275): 1_AAACCCAAGAAGCGCT-1 1_AAACCCAAGACTCATC-1 ...
  2_TTTGTTGTCCCGAGTG-1 2_TTTGTTGTCGAGAATA-1
colData names(58): Barcode Capture ... decontX_contamination
  decontX_clusters
reducedDimNames(1): decontX_UMAP
mainExpName: NULL
altExpNames(2): HTO ADT</code></pre>
<div id="convert-to-seurat-objects" class="section level2">
<h2>Convert to <em>Seurat</em> objects</h2>
<pre class="r"><code>seuLst &lt;- sapply(sceLst, function(sce){
  # Azimuth relies on gene symbols for annotation to add these as row names
  rownames(sce) &lt;- rowData(sce)$Symbol
  
  # sum any rows that have the same gene symbol
  CAGEfightR::utilsAggregateRows(counts(sce), 
                                 factor(rownames(sce)), 
                                 sparse = TRUE) -&gt; count_matrix 
  
  seu &lt;- Seurat::CreateSeuratObject(counts = count_matrix,
                                    meta.data = data.frame(colData(sce)))
  seu
})

seuLst</code></pre>
<pre><code>$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch0/data/SCEs/C133_Neeland_batch0.ambient_removed.SCE.rds`
An object of class Seurat 
33514 features across 25635 samples within 1 assay 
Active assay: RNA (33514 features, 0 variable features)

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch1/data/SCEs/C133_Neeland_batch1.ambient_removed.SCE.rds`
An object of class Seurat 
36591 features across 19522 samples within 1 assay 
Active assay: RNA (36591 features, 0 variable features)

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch2/data/SCEs/C133_Neeland_batch2.ambient_removed.SCE.rds`
An object of class Seurat 
36591 features across 22386 samples within 1 assay 
Active assay: RNA (36591 features, 0 variable features)

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch3/data/SCEs/C133_Neeland_batch3.ambient_removed.SCE.rds`
An object of class Seurat 
36591 features across 46052 samples within 1 assay 
Active assay: RNA (36591 features, 0 variable features)

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch4/data/SCEs/C133_Neeland_batch4.ambient_removed.SCE.rds`
An object of class Seurat 
36591 features across 18858 samples within 1 assay 
Active assay: RNA (36591 features, 0 variable features)

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch5/data/SCEs/C133_Neeland_batch5.ambient_removed.SCE.rds`
An object of class Seurat 
36591 features across 32959 samples within 1 assay 
Active assay: RNA (36591 features, 0 variable features)

$`/Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/data/C133_Neeland_batch6/data/SCEs/C133_Neeland_batch6.ambient_removed.SCE.rds`
An object of class Seurat 
36591 features across 31275 samples within 1 assay 
Active assay: RNA (36591 features, 0 variable features)</code></pre>
</div>
</div>
<div id="azimuth-annotate" class="section level1">
<h1>Azimuth annotate</h1>
<p>Annotate cells in each batch using the Human Lung Cell Atlas v2.0
reference.</p>
<pre class="r"><code>options(timeout = max(1000000, getOption(&quot;timeout&quot;)))
batches &lt;- str_extract(names(seuLst), &quot;batch[0-6]&quot;)

seuLst &lt;- sapply(1:length(seuLst), function(i){
  out &lt;-  here(&quot;data&quot;,
               paste0(&quot;C133_Neeland_&quot;, batches[i]),
               &quot;data&quot;, 
               &quot;SCEs&quot;, 
               glue(&quot;C133_Neeland_{batches[i]}.azimuth_annotated.SEU.rds&quot;))
  
  if(!file.exists(out)){
    tmp &lt;- RunAzimuth(seuLst[[i]], reference = &quot;lungref&quot;)
    seu &lt;- seuLst[[i]]
    seu@meta.data &lt;- tmp@meta.data
    seu@reductions$ref.umap &lt;- tmp@reductions$ref.umap
    
    if(!is_empty(altExpNames(sceLst[[i]]))){
      seu[[&quot;ADTs&quot;]] &lt;- CreateAssayObject(counts = counts(altExp(sceLst[[i]], 
                                                                &quot;ADT&quot;)))
      
    }
    
    saveRDS(seu, out)
    fs::file_chmod(out, &quot;664&quot;)
    if(any(str_detect(fs::group_ids()$group_name, 
                      &quot;oshlack_lab&quot;))) fs::file_chown(out, 
                                                      group_id = &quot;oshlack_lab&quot;)
    
  } else {
    seu &lt;- readRDS(out)
    
  }
  seu
  
})

seuLst</code></pre>
<pre><code>[[1]]
An object of class Seurat 
33514 features across 25635 samples within 1 assay 
Active assay: RNA (33514 features, 0 variable features)
 1 dimensional reduction calculated: ref.umap

[[2]]
An object of class Seurat 
36754 features across 19522 samples within 2 assays 
Active assay: RNA (36591 features, 0 variable features)
 1 other assay present: ADTs
 1 dimensional reduction calculated: ref.umap

[[3]]
An object of class Seurat 
36754 features across 22386 samples within 2 assays 
Active assay: RNA (36591 features, 0 variable features)
 1 other assay present: ADTs
 1 dimensional reduction calculated: ref.umap

[[4]]
An object of class Seurat 
36754 features across 46052 samples within 2 assays 
Active assay: RNA (36591 features, 0 variable features)
 1 other assay present: ADTs
 1 dimensional reduction calculated: ref.umap

[[5]]
An object of class Seurat 
36754 features across 18858 samples within 2 assays 
Active assay: RNA (36591 features, 0 variable features)
 1 other assay present: ADTs
 1 dimensional reduction calculated: ref.umap

[[6]]
An object of class Seurat 
36754 features across 32959 samples within 2 assays 
Active assay: RNA (36591 features, 0 variable features)
 1 other assay present: ADTs
 1 dimensional reduction calculated: ref.umap

[[7]]
An object of class Seurat 
36754 features across 31275 samples within 2 assays 
Active assay: RNA (36591 features, 0 variable features)
 1 other assay present: ADTs
 1 dimensional reduction calculated: ref.umap</code></pre>
<pre class="r"><code>p &lt;- lapply(seuLst, function(seu){
  DimPlot(seu, reduction = &quot;ref.umap&quot;, group.by = &quot;predicted.ann_level_3&quot;,
          label = TRUE, label.size = 2.75) + 
    NoLegend() +
    paletteer::scale_color_paletteer_d(&quot;miscpalettes::pastel&quot;) -&gt; p1
  DimPlot(seu, reduction = &quot;ref.umap&quot;, group.by = &quot;predicted.ann_level_4&quot;,
          label = TRUE, label.size = 2.5) + NoLegend() -&gt; p2
  p1 / p2
})

p</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/06.0_azimuth_annotation.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
7bf7f42
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/06.0_azimuth_annotation.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
7bf7f42
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/06.0_azimuth_annotation.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-3">
Past versions of unnamed-chunk-6-3.png
</button>
</p>
<div id="fig-unnamed-chunk-6-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
7bf7f42
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/06.0_azimuth_annotation.Rmd/unnamed-chunk-6-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-4">
Past versions of unnamed-chunk-6-4.png
</button>
</p>
<div id="fig-unnamed-chunk-6-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
7bf7f42
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/06.0_azimuth_annotation.Rmd/unnamed-chunk-6-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-5">
Past versions of unnamed-chunk-6-5.png
</button>
</p>
<div id="fig-unnamed-chunk-6-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
7bf7f42
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[6]]</code></pre>
<p><img src="figure/06.0_azimuth_annotation.Rmd/unnamed-chunk-6-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-6">
Past versions of unnamed-chunk-6-6.png
</button>
</p>
<div id="fig-unnamed-chunk-6-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
7bf7f42
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[7]]</code></pre>
<p><img src="figure/06.0_azimuth_annotation.Rmd/unnamed-chunk-6-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-7">
Past versions of unnamed-chunk-6-7.png
</button>
</p>
<div id="fig-unnamed-chunk-6-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
7bf7f42
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-02-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="merge-batches-and-save-object" class="section level1">
<h1>Merge batches and save object</h1>
<p>Create a single <code>Seurat</code> object and clean-up the
metadata.</p>
<pre class="r"><code>out &lt;- here(&quot;data/C133_Neeland_full_clean.SEU.rds&quot;)

if(!file.exists(out)){
  seu &lt;- merge(seuLst[[1]], 
               y = c(seuLst[2:7]))
  rm(seuLst)
  gc()
  
  meta_regex &lt;- &quot;altexps|vireo|hashedDrops|refAssay|batch|Sample|sum|^detected|^dmmHTO|^Capture.|^total|mito_|percent.mt|orig|_call|_score|^scDblFinder|subsets&quot;
  keep &lt;- !str_detect(colnames(seu@meta.data), meta_regex)
  seu@meta.data &lt;- seu@meta.data[, keep]
  
  # extract RNA counts
  rna &lt;- seu[[&quot;RNA&quot;]]@counts
  # remove genes that are not expressed in any cells
  rna &lt;- rna[rowSums(rna) != 0, ]
  # remove genes that are not expressed in &gt;=50 cells across the full dataset
  rna &lt;- rna[nexprs(rna, byrow = TRUE) &gt;= 50,]
  
  seu[[&quot;RNA&quot;]] &lt;- CreateAssayObject(counts = rna)
  
  saveRDS(seu, out)
  fs::file_chmod(out, &quot;664&quot;)
  if(any(str_detect(fs::group_ids()$group_name, 
                    &quot;oshlack_lab&quot;))) fs::file_chown(out, 
                                                    group_id = &quot;oshlack_lab&quot;)
  
} else {
  rm(seuLst)
  gc()
  seu &lt;- readRDS(out)
  
}

seu</code></pre>
<pre><code>An object of class Seurat 
21731 features across 196687 samples within 2 assays 
Active assay: RNA (21568 features, 0 variable features)
 1 other assay present: ADTs</code></pre>
</div>
<div id="save-cell-type-group-objects" class="section level1">
<h1>Save cell type group objects</h1>
<p>Subset cell type groups.</p>
<pre class="r"><code>cellsLst &lt;- list(macrophages = seu[, seu$predicted.ann_level_3 == &quot;Macrophages&quot;],
                 t_cells = seu[, seu$predicted.ann_level_3 %in% c(&quot;T cell lineage&quot;, 
                                                                  &quot;Innate lymphoid cell NK&quot;)],
                 other_cells = seu[, !seu$predicted.ann_level_3 %in% c(&quot;Macrophages&quot;, 
                                                                       &quot;T cell lineage&quot;, 
                                                                       &quot;Innate lymphoid cell NK&quot;)])

cellsLst</code></pre>
<pre><code>$macrophages
An object of class Seurat 
21731 features across 165553 samples within 2 assays 
Active assay: RNA (21568 features, 0 variable features)
 1 other assay present: ADTs

$t_cells
An object of class Seurat 
21731 features across 15447 samples within 2 assays 
Active assay: RNA (21568 features, 0 variable features)
 1 other assay present: ADTs

$other_cells
An object of class Seurat 
21731 features across 15687 samples within 2 assays 
Active assay: RNA (21568 features, 0 variable features)
 1 other assay present: ADTs</code></pre>
<p>Save objects.</p>
<pre class="r"><code>sapply(1:length(cellsLst), function(i){
  out &lt;- here(&quot;data&quot;,
              glue(&quot;C133_Neeland_full_clean_{names(cellsLst[i])}.SEU.rds&quot;))
  
  if(!file.exists(out)) saveRDS(cellsLst[[i]], out)
  fs::file_chmod(out, &quot;664&quot;)
  if(any(str_detect(fs::group_ids()$group_name, 
                    &quot;oshlack_lab&quot;))) fs::file_chown(out, 
                                                    group_id = &quot;oshlack_lab&quot;)
})</code></pre>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL</code></pre>
</div>
<div id="session-info" class="section level1 appendix">
<h1 class="appendix">Session info</h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Melbourne
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] Azimuth_0.4.6               shinyBS_0.61.1             
 [3] lungref.SeuratData_2.0.0    SeuratData_0.2.2           
 [5] SeuratObject_4.1.4          Seurat_4.4.0               
 [7] ggupset_0.3.0               scMerge_1.18.0             
 [9] scater_1.30.1               scran_1.30.2               
[11] scuttle_1.12.0              SingleCellExperiment_1.24.0
[13] SummarizedExperiment_1.32.0 Biobase_2.62.0             
[15] GenomicRanges_1.54.1        GenomeInfoDb_1.38.6        
[17] IRanges_2.36.0              S4Vectors_0.40.2           
[19] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      
[21] matrixStats_1.2.0           patchwork_1.2.0            
[23] glue_1.7.0                  here_1.0.1                 
[25] lubridate_1.9.3             forcats_1.0.0              
[27] stringr_1.5.1               dplyr_1.1.4                
[29] purrr_1.0.2                 readr_2.1.5                
[31] tidyr_1.3.1                 tibble_3.2.1               
[33] ggplot2_3.5.0               tidyverse_2.0.0            
[35] BiocStyle_2.30.0            workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] dichromat_2.0-0.1         progress_1.2.3           
  [3] nnet_7.3-19               goftest_1.2-3            
  [5] DT_0.32                   Biostrings_2.70.2        
  [7] rstan_2.32.5              vctrs_0.6.5              
  [9] spatstat.random_3.2-2     digest_0.6.34            
 [11] png_0.1-8                 git2r_0.33.0             
 [13] ggrepel_0.9.5             deldir_2.0-2             
 [15] parallelly_1.37.0         batchelor_1.18.1         
 [17] renv_1.0.3                MASS_7.3-60.0.1          
 [19] reshape2_1.4.4            httpuv_1.6.14            
 [21] withr_3.0.0               xfun_0.42                
 [23] ellipsis_0.3.2            survival_3.5-8           
 [25] memoise_2.0.1             proxyC_0.3.4             
 [27] ggbeeswarm_0.7.2          zoo_1.8-12               
 [29] gtools_3.9.5              pbapply_1.7-2            
 [31] DEoptimR_1.1-3            prettyunits_1.2.0        
 [33] Formula_1.2-5             rematch2_2.1.2           
 [35] KEGGREST_1.42.0           promises_1.2.1           
 [37] httr_1.4.7                restfulr_0.0.15          
 [39] globals_0.16.2            fitdistrplus_1.1-11      
 [41] cvTools_0.3.2             ps_1.7.6                 
 [43] rstudioapi_0.15.0         miniUI_0.1.1.1           
 [45] generics_0.1.3            base64enc_0.1-3          
 [47] processx_3.8.3            curl_5.2.0               
 [49] zlibbioc_1.48.0           sfsmisc_1.1-17           
 [51] ScaledMatrix_1.10.0       polyclip_1.10-6          
 [53] GenomeInfoDbData_1.2.11   SparseArray_1.2.4        
 [55] CAGEfightR_1.22.0         xtable_1.8-4             
 [57] evaluate_0.23             S4Arrays_1.2.0           
 [59] BiocFileCache_2.10.1      hms_1.1.3                
 [61] irlba_2.3.5.1             filelock_1.0.3           
 [63] colorspace_2.1-0          hdf5r_1.3.9              
 [65] ROCR_1.0-11               reticulate_1.35.0        
 [67] spatstat.data_3.0-4       magrittr_2.0.3           
 [69] lmtest_0.9-40             later_1.3.2              
 [71] viridis_0.6.5             lattice_0.22-5           
 [73] spatstat.geom_3.2-8       future.apply_1.11.1      
 [75] robustbase_0.99-2         getPass_0.2-4            
 [77] XML_3.99-0.16.1           scattermore_1.2          
 [79] cowplot_1.1.3             RcppAnnoy_0.0.22         
 [81] Hmisc_5.1-1               pillar_1.9.0             
 [83] StanHeaders_2.32.5        nlme_3.1-164             
 [85] caTools_1.18.2            compiler_4.3.2           
 [87] beachmat_2.18.1           stringi_1.8.3            
 [89] tensor_1.5                GenomicAlignments_1.38.2 
 [91] plyr_1.8.9                BiocIO_1.12.0            
 [93] crayon_1.5.2              abind_1.4-5              
 [95] googledrive_2.1.1         locfit_1.5-9.8           
 [97] sp_2.1-3                  bit_4.0.5                
 [99] whisker_0.4.1             codetools_0.2-19         
[101] BiocSingular_1.18.0       bslib_0.6.1              
[103] QuickJSR_1.1.3            paletteer_1.6.0          
[105] biovizBase_1.50.0         plotly_4.10.4            
[107] mime_0.12                 splines_4.3.2            
[109] Rcpp_1.0.12               dbplyr_2.4.0             
[111] sparseMatrixStats_1.14.0  prismatic_1.1.1          
[113] cellranger_1.1.0          interp_1.1-6             
[115] knitr_1.45                blob_1.2.4               
[117] utf8_1.2.4                reldist_1.7-2            
[119] AnnotationFilter_1.26.0   fs_1.6.3                 
[121] listenv_0.9.1             checkmate_2.3.1          
[123] DelayedMatrixStats_1.24.0 pkgbuild_1.4.3           
[125] Gviz_1.46.1               Matrix_1.6-5             
[127] callr_3.7.3               statmod_1.5.0            
[129] tzdb_0.4.0                startupmsg_0.9.6.1       
[131] pkgconfig_2.0.3           tools_4.3.2              
[133] cachem_1.0.8              RSQLite_2.3.5            
[135] viridisLite_0.4.2         DBI_1.2.1                
[137] numDeriv_2016.8-1.1       fastmap_1.1.1            
[139] rmarkdown_2.25            scales_1.3.0             
[141] grid_4.3.2                ica_1.0-3                
[143] shinydashboard_0.7.2      Rsamtools_2.18.0         
[145] sass_0.4.8                BiocManager_1.30.22      
[147] VariantAnnotation_1.48.1  dotCall64_1.1-1          
[149] RANN_2.6.1                rpart_4.1.23             
[151] farver_2.1.1              mgcv_1.9-1               
[153] yaml_2.3.8                latticeExtra_0.6-30      
[155] foreign_0.8-86            rtracklayer_1.62.0       
[157] cli_3.6.2                 leiden_0.4.3.1           
[159] lifecycle_1.0.4           uwot_0.1.16              
[161] M3Drop_1.28.0             mvtnorm_1.2-4            
[163] presto_1.0.0              bluster_1.12.0           
[165] backports_1.4.1           BiocParallel_1.36.0      
[167] distr_2.9.3               rjson_0.2.21             
[169] timechange_0.3.0          gtable_0.3.4             
[171] ggridges_0.5.6            densEstBayes_1.0-2.2     
[173] progressr_0.14.0          parallel_4.3.2           
[175] limma_3.58.1              jsonlite_1.8.8           
[177] edgeR_4.0.15              bitops_1.0-7             
[179] assertthat_0.2.1          bit64_4.0.5              
[181] Rtsne_0.17                spatstat.utils_3.0-4     
[183] BiocNeighbors_1.20.2      RcppParallel_5.1.7       
[185] bdsmatrix_1.3-6           highr_0.10               
[187] jquerylib_0.1.4           metapod_1.10.1           
[189] dqrng_0.3.2               loo_2.6.0                
[191] shinyjs_2.1.0             SeuratDisk_0.0.0.9021    
[193] lazyeval_0.2.2            shiny_1.8.0              
[195] ruv_0.9.7.1               htmltools_0.5.7          
[197] sctransform_0.4.1         rappdirs_0.3.3           
[199] ensembldb_2.26.0          spam_2.10-0              
[201] googlesheets4_1.1.1       ResidualMatrix_1.12.0    
[203] XVector_0.42.0            RCurl_1.98-1.14          
[205] rprojroot_2.0.4           BSgenome_1.70.2          
[207] jpeg_0.1-10               gridExtra_2.3            
[209] igraph_2.0.1.1            R6_2.5.1                 
[211] gplots_3.1.3.1            labeling_0.4.3           
[213] GenomicFeatures_1.54.3    cluster_2.1.6            
[215] bbmle_1.0.25.1            gargle_1.5.2             
[217] ProtGenerics_1.34.0       rstantools_2.4.0         
[219] DelayedArray_0.28.0       tidyselect_1.2.0         
[221] vipor_0.4.7               htmlTable_2.4.2          
[223] xml2_1.3.6                inline_0.3.19            
[225] AnnotationDbi_1.64.1      future_1.33.1            
[227] rsvd_1.0.5                munsell_0.5.0            
[229] KernSmooth_2.23-22        data.table_1.15.0        
[231] htmlwidgets_1.6.4         RColorBrewer_1.1-3       
[233] biomaRt_2.58.2            rlang_1.1.3              
[235] spatstat.sparse_3.0-3     spatstat.explore_3.2-6   
[237] fansi_1.0.6               beeswarm_0.4.0           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
