<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2024-05-20" />

<title>Annotate Macrophage clusters</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-inflammation-CITEseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Annotate Macrophage clusters</h1>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">May 20, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-05-20
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>paed-inflammation-CITEseq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmospaedinflammationciteseqtree5fef90559f27ffeec91cadfab85c409a2febd303targetblank5fef905a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/tree/5fef90559f27ffeec91cadfab85c409a2febd303" target="_blank">5fef905</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmospaedinflammationciteseqtree5fef90559f27ffeec91cadfab85c409a2febd303targetblank5fef905a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/tree/5fef90559f27ffeec91cadfab85c409a2febd303" target="_blank">5fef905</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/C133_Neeland_batch1/
    Ignored:    data/C133_Neeland_merged/
    Ignored:    renv/library/
    Ignored:    renv/staging/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/10.0_manual_annotations_macro_cells_decontx.Rmd</code>)
and HTML
(<code>docs/10.0_manual_annotations_macro_cells_decontx.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/blob/5fef90559f27ffeec91cadfab85c409a2febd303/analysis/10.0_manual_annotations_macro_cells_decontx.Rmd" target="_blank">5fef905</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-05-20
</td>
<td>
wflow_publish("analysis/10.0_manual_annotations_macro_cells_decontx.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/blob/9873b3d690703a3fa63d539877540e21d4c68996/analysis/10.0_manual_annotations_macro_cells_decontx.Rmd" target="_blank">9873b3d</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-05-20
</td>
<td>
Add macrophage cluster annotations and annotation analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
</div>
<div id="load-data" class="section level1">
<h1>Load Data</h1>
<pre class="r"><code>ambient &lt;- &quot;_decontx&quot;
out &lt;- here(&quot;data&quot;,
            &quot;C133_Neeland_merged&quot;,
            glue(&quot;C133_Neeland_full_clean{ambient}_integrated_clustered_macrophages.SEU.rds&quot;))
seuInt &lt;- readRDS(file = out)

seuInt</code></pre>
<pre><code>An object of class Seurat 
41688 features across 141121 samples within 5 assays 
Active assay: integrated (3000 features, 3000 variable features)
 4 other assays present: RNA, ADT, ADT.dsb, SCT
 2 dimensional reductions calculated: pca, umap</code></pre>
<div id="update-group-labels" class="section level2">
<h2>Update group labels</h2>
<pre class="r"><code>seuInt@meta.data %&gt;%
  data.frame %&gt;%
  mutate(Status = ifelse(str_detect(Treatment, &quot;ivacaftor&quot;),
                         &quot;CF ivacaftor&quot;,
                         ifelse(str_detect(Treatment, &quot;orkambi&quot;),
                                &quot;CF lumacaftor-ivacaftor&quot;,
                                ifelse(Treatment == &quot;untreated&quot;,
                                       &quot;CF no-modulator&quot;,
                                       &quot;non-CF control&quot;))),
         Status_sub = ifelse(str_detect(Treatment, &quot;ivacaftor&quot;),
                         &quot;CF.IVA&quot;,
                         ifelse(str_detect(Treatment, &quot;orkambi&quot;),
                                &quot;CF.LUMA_IVA&quot;,
                                ifelse(Treatment == &quot;untreated&quot;,
                                       &quot;CF.NO_MOD&quot;,
                                       &quot;NON_CF.CTRL&quot;))),
         Group = ifelse(!Status_sub %in% &quot;NON_CF.CTRL&quot;, 
                        paste(Status_sub, 
                              toupper(substr(Severity, 1, 1)),
                              sep = &quot;.&quot;), 
                        Status_sub),
         Severity = tolower(Severity),
         Participant = strsplit2(sample.id, &quot;.&quot;, fixed = TRUE)[,1]) -&gt; seuInt@meta.data</code></pre>
</div>
</div>
<div id="sub-cluster-labelling" class="section level1">
<h1>Sub-cluster labelling</h1>
<div id="load-manual-annotations" class="section level2">
<h2>Load manual annotations</h2>
<pre class="r"><code>labels &lt;- read_excel(here(&quot;data&quot;,
                          &quot;cluster_annotations&quot;,
                          &quot;macrophages_ambientRNAremoval_01.05.24.xlsx&quot;))

# set selected cluster resolution
grp &lt;- &quot;integrated_snn_res.0.6&quot;
seuInt@meta.data %&gt;%
  rownames_to_column(var = &quot;cell&quot;) %&gt;%
  left_join(labels %&gt;%
              mutate(Cluster = as.factor(Cluster),
                     Annotation = as.factor(Annotation),
                     Broad = as.factor(Broad)),
            by = c(&quot;integrated_snn_res.0.6&quot; = &quot;Cluster&quot;)) %&gt;%
  column_to_rownames(var = &quot;cell&quot;) -&gt; seuInt@meta.data

seuInt$Annotation &lt;- fct_drop(seuInt$Annotation)
seuInt$Broad &lt;- fct_drop(seuInt$Broad)
seuInt</code></pre>
<pre><code>An object of class Seurat 
41688 features across 141121 samples within 5 assays 
Active assay: integrated (3000 features, 3000 variable features)
 4 other assays present: RNA, ADT, ADT.dsb, SCT
 2 dimensional reductions calculated: pca, umap</code></pre>
</div>
<div id="visualise-annotations" class="section level2">
<h2>Visualise annotations</h2>
<pre class="r"><code>options(ggrepel.max.overlaps = Inf)
DimPlot(seuInt, reduction = &#39;umap&#39;, label = TRUE, repel = TRUE, 
        label.size = 3, group.by = &quot;integrated_snn_res.0.6&quot;) + 
  NoLegend() -&gt; p1

DimPlot(seuInt, reduction = &#39;umap&#39;, label = FALSE, group.by = &quot;Annotation&quot;) + 
  scale_color_paletteer_d(&quot;miscpalettes::pastel&quot;) + 
  theme(text = element_text(size = 9),
        axis.text = element_blank(),
        axis.ticks = element_blank()) +
  NoLegend() -&gt; p2

DimPlot(seuInt, reduction = &#39;umap&#39;, label = FALSE, group.by = &quot;Broad&quot;) + 
  NoLegend() +
  scale_color_paletteer_d(&quot;miscpalettes::pastel&quot;) + 
  theme(text = element_text(size = 9),
        axis.text = element_blank(),
        axis.ticks = element_blank()) -&gt; p3

p1</code></pre>
<p><img src="figure/10.0_manual_annotations_macro_cells_decontx.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>LabelClusters(p2, id = &quot;Annotation&quot;,  fontface = &quot;bold&quot;, repel = TRUE, size = 3.5)</code></pre>
<p><img src="figure/10.0_manual_annotations_macro_cells_decontx.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>LabelClusters(p3, id = &quot;Broad&quot;,  fontface = &quot;bold&quot;, repel = TRUE, size = 3.5)</code></pre>
<p><img src="figure/10.0_manual_annotations_macro_cells_decontx.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<div id="no.-cells-per-cluster" class="section level3">
<h3>No. cells per cluster</h3>
<pre class="r"><code>seuInt@meta.data %&gt;%
  ggplot(aes(x = Annotation, fill = Annotation)) +
  geom_bar() +
  geom_text(aes(label = after_stat(count)), stat = &quot;count&quot;,
            vjust = -0.5, colour = &quot;black&quot;, size = 2) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  NoLegend() +
  scale_fill_paletteer_d(&quot;miscpalettes::pastel&quot;)</code></pre>
<p><img src="figure/10.0_manual_annotations_macro_cells_decontx.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="no.-cells-per-cluster-by-disease" class="section level3">
<h3>No. cells per cluster by disease</h3>
<pre class="r"><code>seuInt@meta.data %&gt;%
  ggplot(aes(x = Annotation, fill = Status_sub)) +
  geom_bar(position = &quot;dodge&quot;) +
  geom_text(aes(label = ..count..), stat = &quot;count&quot;,
            vjust = -0.5, colour = &quot;black&quot;, size = 2, 
            position=position_dodge(width=0.9)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1),
        legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/10.0_manual_annotations_macro_cells_decontx.Rmd/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>seuInt@meta.data %&gt;% 
  count(Annotation) %&gt;% 
  mutate(perc = round(n/sum(n)*100, 1)) %&gt;%
  dplyr::rename(`Cell Label` = &quot;Annotation&quot;, 
                `No. Cells` = n,
                `% Cells` = perc) %&gt;%
  knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Cell Label</th>
<th align="right">No. Cells</th>
<th align="right">% Cells</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">macro-alveolar</td>
<td align="right">88915</td>
<td align="right">63.0</td>
</tr>
<tr class="even">
<td align="left">macro-CCL</td>
<td align="right">6912</td>
<td align="right">4.9</td>
</tr>
<tr class="odd">
<td align="left">macro-interstitial</td>
<td align="right">2337</td>
<td align="right">1.7</td>
</tr>
<tr class="even">
<td align="left">macro-lipid</td>
<td align="right">12380</td>
<td align="right">8.8</td>
</tr>
<tr class="odd">
<td align="left">macro-monocyte-derived</td>
<td align="right">9695</td>
<td align="right">6.9</td>
</tr>
<tr class="even">
<td align="left">macro-MT</td>
<td align="right">3224</td>
<td align="right">2.3</td>
</tr>
<tr class="odd">
<td align="left">macro-NFKB</td>
<td align="right">6598</td>
<td align="right">4.7</td>
</tr>
<tr class="even">
<td align="left">macro-proliferating</td>
<td align="right">3467</td>
<td align="right">2.5</td>
</tr>
<tr class="odd">
<td align="left">macro-regulation</td>
<td align="right">73</td>
<td align="right">0.1</td>
</tr>
<tr class="even">
<td align="left">macro-vesicle</td>
<td align="right">5273</td>
<td align="right">3.7</td>
</tr>
<tr class="odd">
<td align="left">macro-viral</td>
<td align="right">2247</td>
<td align="right">1.6</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="rna-marker-gene-analysis" class="section level1">
<h1>RNA marker gene analysis</h1>
<p>Adapted from Dr. Belinda Phipson’s <a
href="https://bphipson.github.io/Human_Development_snRNAseq/14-MarkerAnalysisBCT.html">work</a>
for <span class="citation">[@Sim2021-cg]</span>.</p>
<div id="test-for-marker-genes-using-limma" class="section level2">
<h2>Test for marker genes using <code>limma</code></h2>
<pre class="r"><code># limma-trend for DE
Idents(seuInt) &lt;- &quot;Annotation&quot;

logcounts &lt;- normCounts(DGEList(as.matrix(seuInt[[&quot;RNA&quot;]]@counts)),
                        log = TRUE, prior.count = 0.5)
entrez &lt;- AnnotationDbi::mapIds(org.Hs.eg.db,
                                keys = rownames(logcounts),
                                column = c(&quot;ENTREZID&quot;),
                                keytype = &quot;SYMBOL&quot;,
                                multiVals = &quot;first&quot;)
# remove genes without entrez IDs as these are difficult to interpret biologically
logcounts &lt;- logcounts[!is.na(entrez),]
# remove confounding genes from counts table e.g. mitochondrial, ribosomal etc.
# remove HLA, immunoglobulin, RNA, MT, and RP genes from marker gene analysis
var_regex = &#39;^HLA-|^IG[HJKL]|^RNA|^MT-|^RP&#39; 
logcounts &lt;- logcounts[!str_detect(rownames(logcounts), var_regex),]

maxclust &lt;- length(levels(Idents(seuInt))) - 1

clustgrp &lt;- seuInt$Annotation
clustgrp &lt;- factor(clustgrp)
donor &lt;- factor(seuInt$sample.id)
batch &lt;- factor(seuInt$Batch)

design &lt;- model.matrix(~ 0 + clustgrp + donor)
colnames(design)[1:(length(levels(clustgrp)))] &lt;- levels(clustgrp)

# Create contrast matrix
mycont &lt;- matrix(NA, ncol = length(levels(clustgrp)),
                 nrow = length(levels(clustgrp)))
rownames(mycont) &lt;- colnames(mycont) &lt;- levels(clustgrp)
diag(mycont) &lt;- 1
mycont[upper.tri(mycont)] &lt;- -1/(length(levels(factor(clustgrp))) - 1)
mycont[lower.tri(mycont)] &lt;- -1/(length(levels(factor(clustgrp))) - 1)

# Fill out remaining rows with 0s
zero.rows &lt;- matrix(0, ncol = length(levels(clustgrp)),
                    nrow = (ncol(design) - length(levels(clustgrp))))
fullcont &lt;- rbind(mycont, zero.rows)
rownames(fullcont) &lt;- colnames(design)

fit &lt;- lmFit(logcounts, design)

fit.cont &lt;- contrasts.fit(fit, contrasts = fullcont)
fit.cont &lt;- eBayes(fit.cont, trend = TRUE, robust = TRUE)

summary(decideTests(fit.cont))</code></pre>
<pre><code>       macro-alveolar macro-CCL macro-interstitial macro-lipid
Down             5429      5266               5365        5631
NotSig           6722      8237               6586        7614
Up               3471      2119               3671        2377
       macro-monocyte-derived macro-MT macro-NFKB macro-proliferating
Down                     4538     4128       5052                3151
NotSig                   7296     8553       9659                6165
Up                       3788     2941        911                6306
       macro-regulation macro-vesicle macro-viral
Down               3209          5778        2760
NotSig             9098          8273        8989
Up                 3315          1571        3873</code></pre>
<p>Test relative to a threshold (TREAT).</p>
<pre class="r"><code>tr &lt;- treat(fit.cont, lfc = 0.5)
dt &lt;- decideTests(tr)
summary(dt)</code></pre>
<pre><code>       macro-alveolar macro-CCL macro-interstitial macro-lipid
Down                6        13                167          14
NotSig          15539     15540              15301       15567
Up                 77        69                154          41
       macro-monocyte-derived macro-MT macro-NFKB macro-proliferating
Down                       34        4          1                  17
NotSig                  15551    15539      15596               15307
Up                         37       79         25                 298
       macro-regulation macro-vesicle macro-viral
Down                732             8           2
NotSig            14694         15569       15508
Up                  196            45         112</code></pre>
<p>Mean-difference (MD) plots per cluster.</p>
<pre class="r"><code>par(mfrow=c(4,3))
par(mar=c(2,3,1,2))

for(i in 1:ncol(mycont)){
  plotMD(tr, coef = i, status = dt[,i], hl.cex = 0.5)
  abline(h = 0, col = &quot;lightgrey&quot;)
  lines(lowess(tr$Amean, tr$coefficients[,i]), lwd = 1.5, col = 4)
}</code></pre>
<p><img src="figure/10.0_manual_annotations_macro_cells_decontx.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="limma-marker-gene-dotplot" class="section level2">
<h2><code>limma</code> marker gene dotplot</h2>
<pre class="r"><code>DefaultAssay(seuInt) &lt;- &quot;RNA&quot;
contnames &lt;- colnames(mycont)
top_markers &lt;- NULL
n_markers &lt;- 5

for(i in 1:ncol(mycont)){
  top &lt;- topTreat(tr, coef = i, n = Inf)
  top &lt;- top[top$logFC &gt; 0, ]
  top_markers &lt;- c(top_markers, 
                   setNames(rownames(top)[1:n_markers], 
                            rep(contnames[i], n_markers)))
}

top_markers &lt;- top_markers[!is.na(top_markers)]
top_markers &lt;- top_markers[!duplicated(top_markers)]
cols &lt;- paletteer_d(&quot;miscpalettes::pastel&quot;)[factor(names(top_markers))]

DotPlot(seuInt,    
        features = unname(top_markers),
        group.by = &quot;Annotation&quot;,
        cols = c(&quot;azure1&quot;, &quot;blueviolet&quot;),
        dot.scale = 2.5, 
        assay = &quot;SCT&quot;) +
    RotatedAxis() +
    FontSize(x.text = 8, y.text = 8) +
    labs(y = element_blank(), x = element_blank()) +
    theme(axis.text.x = element_text(color = cols,
                                   angle = 90,
                                   hjust = 1,
                                   vjust = 0.5,
                                   face = &quot;bold&quot;),
        legend.text = element_text(size = 8),
        legend.title = element_text(size = 10))</code></pre>
<p><img src="figure/10.0_manual_annotations_macro_cells_decontx.Rmd/unnamed-chunk-13-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="test-for-marker-genes-using-seurat" class="section level2">
<h2>Test for marker genes using <code>Seurat</code></h2>
<pre class="r"><code>DefaultAssay(seuInt) &lt;- &quot;RNA&quot;
Idents(seuInt) &lt;- &quot;Annotation&quot;

out &lt;- here(&quot;data/cluster_annotations/seurat_markers_macrophages.rds&quot;)

if(!file.exists(out)){
  # restrict genes to same set as for limma analysis
  markers &lt;- FindAllMarkers(seuInt, only.pos = TRUE, 
                            features = rownames(logcounts))
  saveRDS(markers, file = out)

} else {
  markers &lt;- readRDS(out)
  
}

head(markers) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="10%" />
<col width="8%" />
<col width="16%" />
<col width="8%" />
<col width="8%" />
<col width="14%" />
<col width="22%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">p_val</th>
<th align="right">avg_log2FC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
<th align="left">cluster</th>
<th align="left">gene</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">MCEMP1</td>
<td align="right">0</td>
<td align="right">0.7374759</td>
<td align="right">0.975</td>
<td align="right">0.668</td>
<td align="right">0</td>
<td align="left">macro-alveolar</td>
<td align="left">MCEMP1</td>
</tr>
<tr class="even">
<td align="left">FBP1</td>
<td align="right">0</td>
<td align="right">0.6248554</td>
<td align="right">0.977</td>
<td align="right">0.814</td>
<td align="right">0</td>
<td align="left">macro-alveolar</td>
<td align="left">FBP1</td>
</tr>
<tr class="odd">
<td align="left">MRC1</td>
<td align="right">0</td>
<td align="right">0.5883707</td>
<td align="right">0.962</td>
<td align="right">0.716</td>
<td align="right">0</td>
<td align="left">macro-alveolar</td>
<td align="left">MRC1</td>
</tr>
<tr class="even">
<td align="left">CD52</td>
<td align="right">0</td>
<td align="right">0.5708416</td>
<td align="right">0.992</td>
<td align="right">0.863</td>
<td align="right">0</td>
<td align="left">macro-alveolar</td>
<td align="left">CD52</td>
</tr>
<tr class="odd">
<td align="left">CRIP1</td>
<td align="right">0</td>
<td align="right">0.5541694</td>
<td align="right">0.985</td>
<td align="right">0.958</td>
<td align="right">0</td>
<td align="left">macro-alveolar</td>
<td align="left">CRIP1</td>
</tr>
<tr class="even">
<td align="left">IGF1</td>
<td align="right">0</td>
<td align="right">0.5127465</td>
<td align="right">0.201</td>
<td align="right">0.075</td>
<td align="right">0</td>
<td align="left">macro-alveolar</td>
<td align="left">IGF1</td>
</tr>
</tbody>
</table>
</div>
<div id="seurat-marker-gene-dotplot" class="section level2">
<h2><code>Seurat</code> marker gene dotplot</h2>
<pre class="r"><code>DefaultAssay(seuInt) &lt;- &quot;RNA&quot;

maxGenes &lt;- 5
markers %&gt;%
    group_by(cluster) %&gt;%
    top_n(n = maxGenes, wt = avg_log2FC) -&gt; top5

sig &lt;- top5$gene
geneCols &lt;- paletteer_d(&quot;miscpalettes::pastel&quot;)[top5$cluster][!duplicated(sig)]

pal &lt;- paletteer::paletteer_d(&quot;vapoRwave::cool&quot;)
DotPlot(seuInt,
        features = sig[!duplicated(sig)],
        group.by = &quot;Annotation&quot;,
        cols = c(&quot;azure1&quot;, &quot;blueviolet&quot;),
        dot.scale = 2.5,
        assay = &quot;SCT&quot;) +
    FontSize(x.text = 8, y.text = 8) +
    labs(y = element_blank(), x = element_blank()) +
    theme(axis.text.x = element_text(color = cols,
                                   angle = 90,
                                   hjust = 1,
                                   vjust = 0.5,
                                   face = &quot;bold&quot;),
        legend.text = element_text(size = 8),
        legend.title = element_text(size = 10))</code></pre>
<p><img src="figure/10.0_manual_annotations_macro_cells_decontx.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
<!-- ## ADT Expression -->
<!-- Setup ADT annotations -->
<!-- ```{r} -->
<!-- prots_all <- read.csv(file = here("data", -->
<!--                      "C133_Neeland_batch1", -->
<!--                      "data", -->
<!--                      "sample_sheets", -->
<!--                      "ADT_features.csv"))  -->
<!-- head(prots_all) %>% knitr::kable() -->
<!-- ``` -->
<!-- Load the DSB normalised ADT data. -->
<!-- ```{r} -->
<!-- norm.adt <- readRDS(file = here("data/RDS/C133_Neeland_T_cells.DSB.SEU.rds")) -->
<!-- ``` -->
<!-- Put marker names in to the counts. Otherwise we just get the BioLegend codes which aren't very informative. -->
<!-- ```{r} -->
<!-- # Regular expression pattern to remove the substrings -->
<!-- pattern <- "anti-human/mouse |anti-human/mouse/rat |anti-mouse/human " -->
<!-- # Remove the substrings using gsub() -->
<!-- prots_all$name <- gsub(pattern, "", prots_all$name) -->
<!-- rownames(norm.adt) <- prots_all$name -->
<!-- ``` -->
<!-- Subset to only samples with ADT data. -->
<!-- ```{r} -->
<!-- seuSub <- seuInt[, seuInt$Batch != 0] -->
<!-- norm.adt <- norm.adt[, colnames(norm.adt) %in% colnames(seuSub)] -->
<!-- seuSub[["ADT.dsb"]] <- CreateAssayObject(counts = norm.adt) -->
<!-- ``` -->
<!-- ### CD45 expression by cluster -->
<!-- ```{r} -->
<!-- cbind(seuSub@meta.data %>% -->
<!--         dplyr::select(wsnn_res.0.7, Annotation), -->
<!--       as.data.frame(t(seuSub@assays$ADT.dsb@data))) %>% -->
<!--   rownames_to_column(var = "cell") %>% -->
<!--   pivot_longer(c(-cell, -Annotation, -wsnn_res.0.7),  -->
<!--                names_to = "ADT", -->
<!--                values_to = "expression") %>% -->
<!--   dplyr::filter(ADT == "CD45") %>% -->
<!--   ggplot(aes(x = wsnn_res.0.7, y = expression,  -->
<!--              fill = Annotation)) + -->
<!--   geom_boxplot() +  -->
<!--   ggtitle("CD45 ADT") + -->
<!--   scale_fill_paletteer_d("miscpalettes::pastel") -->
<!-- ``` -->
<!-- ### CD45 expression by manual label -->
<!-- ```{r, fig.asp=1} -->
<!-- cbind(seuSub@meta.data %>% -->
<!--         dplyr::select(wsnn_res.0.7, Annotation), -->
<!--       as.data.frame(t(seuSub@assays$ADT.dsb@data))) %>% -->
<!--   rownames_to_column(var = "cell") %>% -->
<!--   pivot_longer(c(-cell, -Annotation, -wsnn_res.0.7),  -->
<!--                names_to = "ADT", -->
<!--                values_to = "expression") %>% -->
<!--   dplyr::filter(ADT == "CD45") %>% -->
<!--   ggplot(aes(x = Annotation, y = expression,  -->
<!--              fill = Annotation)) + -->
<!--   geom_boxplot() +  -->
<!--   NoLegend() + -->
<!--   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) + -->
<!--   ggtitle("CD45 ADT") + -->
<!--   scale_fill_paletteer_d("miscpalettes::pastel") -->
<!-- ``` -->
<!-- ### PTPRC expression by manual labels (all samples) -->
<!-- ```{r, fig.asp=1} -->
<!-- cbind(seuInt@meta.data %>% -->
<!--         dplyr::select(wsnn_res.0.7, Annotation), -->
<!--       as.data.frame(t(seuInt@assays$RNA@data[rownames(seuInt@assays$RNA@data) == "PTPRC", , drop = FALSE]))) %>% -->
<!--   rownames_to_column(var = "cell") %>% -->
<!--   pivot_longer(c(-cell, -Annotation, -wsnn_res.0.7),  -->
<!--                names_to = "RNA", -->
<!--                values_to = "expression") %>% -->
<!--   ggplot(aes(x = Annotation, y = expression,  -->
<!--              fill = Annotation)) + -->
<!--   geom_boxplot() +  -->
<!--   NoLegend() + -->
<!--   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) + -->
<!--   ggtitle("PTPRC RNA") + -->
<!--   scale_fill_paletteer_d("miscpalettes::pastel") -->
<!-- ``` -->
<!-- ## Visualise ADTs -->
<!-- ```{r} -->
<!-- cbind(seuSub@meta.data %>% -->
<!--         dplyr::select(wsnn_res.0.7, Annotation), -->
<!--       as.data.frame(t(seuSub@assays$ADT.dsb@data))) %>% -->
<!--   pivot_longer(c(-Annotation, -wsnn_res.0.7),  -->
<!--                names_to = "ADT", -->
<!--                values_to = "expression") %>% -->
<!--   dplyr::group_by(Annotation, ADT) %>% -->
<!--   dplyr::summarize(Expression = mean(expression)) %>% -->
<!--   ungroup() -> dat -->
<!-- head(dat) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- plot(density(dat$Expression)) -->
<!-- topMax <- 1.8 -->
<!-- abline(v = topMax, lty = 2, col = "grey") -->
<!-- ``` -->
<!-- ### Expression of all ADTs -->
<!-- ```{r, fig.asp=2.5} -->
<!--   dat |> heatmap( -->
<!--     .column = Annotation, -->
<!--     .row = ADT, -->
<!--     .value = Expression, -->
<!--     scale = "none", -->
<!--     rect_gp = grid::gpar(col = "white", lwd = 1), -->
<!--     show_row_names = TRUE, -->
<!--     column_names_gp = grid::gpar(fontsize = 10), -->
<!--     column_title_gp = grid::gpar(fontsize = 12), -->
<!--     row_names_gp = grid::gpar(fontsize = 8), -->
<!--     row_title_gp = grid::gpar(fontsize = 12), -->
<!--     column_title_side = "top", -->
<!--   palette_value = circlize::colorRamp2(seq(-0.5, 2, length.out = 256), -->
<!--                                         viridis::magma(256)), -->
<!--     heatmap_legend_param = list(direction = "vertical")) -->
<!-- ``` -->
<!-- ### Expression of most relevant ADTs -->
<!-- ```{r, fig.asp=1} -->
<!-- ADTs <- str_replace_all(labels$`Relevant marker ADTs`, "HLA", "HLA-") -->
<!-- ADTs <- ADTs[!is.na(ADTs)] -->
<!-- ADTs <- as.vector(t(strsplit2(str_remove_all(ADTs, " "), ","))) -->
<!-- ADTs <- unique(ADTs[ADTs != ""]) -->
<!-- dat |>  -->
<!--   dplyr::filter(rowSums(sapply(glue("\\b{ADTs}\\b\\s*(\\(.*\\))?\\b"),  -->
<!--                                function(x) str_detect(ADT, x))) > 0) |> -->
<!--   heatmap( -->
<!--     .column = Annotation, -->
<!--     .row = ADT, -->
<!--     .value = Expression, -->
<!--     scale = "none", -->
<!--     rect_gp = grid::gpar(col = "white", lwd = 1), -->
<!--     show_row_names = TRUE, -->
<!--     column_names_gp = grid::gpar(fontsize = 10), -->
<!--     column_title_gp = grid::gpar(fontsize = 12), -->
<!--     row_names_gp = grid::gpar(fontsize = 8), -->
<!--     row_title_gp = grid::gpar(fontsize = 12), -->
<!--     column_title_side = "top", -->
<!--     palette_value = circlize::colorRamp2(seq(-0.5, topMax, length.out = 256), -->
<!--                                          viridis::magma(256)), -->
<!--     heatmap_legend_param = list(direction = "vertical")) |>  -->
<!--   add_tile(Annotation, show_legend = FALSE, -->
<!--            show_annotation_name = FALSE,  -->
<!--            palette = paletteer_d("miscpalettes::pastel",  -->
<!--                                  length(levels(top5$cluster))))  -->
<!-- ``` -->
<!-- <!-- wrap_heatmap(f2d) -->
<p>–&gt; <!-- <!-- ``` --> –&gt;</p>
</div>
</div>
<div id="save-data" class="section level1">
<h1>Save data</h1>
<pre class="r"><code>out &lt;- here(&quot;data&quot;,
            &quot;C133_Neeland_merged&quot;,
            glue(&quot;C133_Neeland_full_clean{ambient}_macrophages_annotated_diet.SEU.rds&quot;))
if(!file.exists(out)){
  DefaultAssay(seuInt) &lt;- &quot;RNA&quot;
  saveRDS(DietSeurat(seuInt, assays = &quot;RNA&quot;), out)
}

out &lt;- here(&quot;data&quot;,
            &quot;C133_Neeland_merged&quot;,
            glue(&quot;C133_Neeland_full_clean{ambient}_macrophages_annotated_full.SEU.rds&quot;))
if(!file.exists(out)){
  DefaultAssay(seuInt) &lt;- &quot;RNA&quot;
  saveRDS(seuInt, out)
}</code></pre>
<!-- # Panel figures -->
<!-- ```{r, fig.width=10, fig.asp=1.3} -->
<!-- layout = "AAAABB -->
<!--           AAAABB -->
<!--           AAAABB -->
<!--           CCCCCC -->
<!--           CCCCCC -->
<!--           DDDDDD -->
<!--           DDDDDD -->
<!--           DDDDDD" -->
<!-- ((f2a + ggtitle("")) +  -->
<!--    f2b +  -->
<!--    f2c +  -->
<!--    wrap_heatmap(f2d)) +  -->
<!--   plot_layout(design = layout) + -->
<!--   plot_annotation(tag_levels = "A") & -->
<!--   theme(plot.tag = element_text(size = 14, face = "bold")) -->
<!-- ``` -->
</div>
<div id="session-info" class="section level1 appendix">
<h1 class="appendix">Session info</h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.3 (2024-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] speckle_1.2.0               org.Hs.eg.db_3.18.0        
 [3] AnnotationDbi_1.64.1        readxl_1.4.3               
 [5] tidyHeatmap_1.8.1           paletteer_1.6.0            
 [7] patchwork_1.2.0             glue_1.7.0                 
 [9] here_1.0.1                  dittoSeq_1.14.2            
[11] SeuratObject_4.1.4          Seurat_4.4.0               
[13] lubridate_1.9.3             forcats_1.0.0              
[15] stringr_1.5.1               dplyr_1.1.4                
[17] purrr_1.0.2                 readr_2.1.5                
[19] tidyr_1.3.1                 tibble_3.2.1               
[21] ggplot2_3.5.0               tidyverse_2.0.0            
[23] edgeR_4.0.15                limma_3.58.1               
[25] SingleCellExperiment_1.24.0 SummarizedExperiment_1.32.0
[27] Biobase_2.62.0              GenomicRanges_1.54.1       
[29] GenomeInfoDb_1.38.6         IRanges_2.36.0             
[31] S4Vectors_0.40.2            BiocGenerics_0.48.1        
[33] MatrixGenerics_1.14.0       matrixStats_1.2.0          
[35] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22        splines_4.3.3           later_1.3.2            
  [4] prismatic_1.1.1         bitops_1.0-7            cellranger_1.1.0       
  [7] polyclip_1.10-6         lifecycle_1.0.4         doParallel_1.0.17      
 [10] rprojroot_2.0.4         globals_0.16.2          processx_3.8.3         
 [13] lattice_0.22-5          MASS_7.3-60.0.1         dendextend_1.17.1      
 [16] magrittr_2.0.3          plotly_4.10.4           sass_0.4.8             
 [19] rmarkdown_2.25          jquerylib_0.1.4         yaml_2.3.8             
 [22] httpuv_1.6.14           sctransform_0.4.1       sp_2.1-3               
 [25] spatstat.sparse_3.0-3   reticulate_1.35.0       DBI_1.2.1              
 [28] cowplot_1.1.3           pbapply_1.7-2           RColorBrewer_1.1-3     
 [31] abind_1.4-5             zlibbioc_1.48.0         Rtsne_0.17             
 [34] RCurl_1.98-1.14         git2r_0.33.0            circlize_0.4.15        
 [37] GenomeInfoDbData_1.2.11 ggrepel_0.9.5           irlba_2.3.5.1          
 [40] listenv_0.9.1           spatstat.utils_3.0-4    pheatmap_1.0.12        
 [43] goftest_1.2-3           spatstat.random_3.2-2   fitdistrplus_1.1-11    
 [46] parallelly_1.37.0       leiden_0.4.3.1          codetools_0.2-19       
 [49] DelayedArray_0.28.0     shape_1.4.6             tidyselect_1.2.0       
 [52] farver_2.1.1            viridis_0.6.5           spatstat.explore_3.2-6 
 [55] jsonlite_1.8.8          GetoptLong_1.0.5        ellipsis_0.3.2         
 [58] progressr_0.14.0        iterators_1.0.14        ggridges_0.5.6         
 [61] survival_3.5-8          foreach_1.5.2           tools_4.3.3            
 [64] ica_1.0-3               Rcpp_1.0.12             gridExtra_2.3          
 [67] SparseArray_1.2.4       xfun_0.42               withr_3.0.0            
 [70] BiocManager_1.30.22     fastmap_1.1.1           fansi_1.0.6            
 [73] callr_3.7.3             digest_0.6.34           timechange_0.3.0       
 [76] R6_2.5.1                mime_0.12               colorspace_2.1-0       
 [79] scattermore_1.2         tensor_1.5              RSQLite_2.3.5          
 [82] spatstat.data_3.0-4     utf8_1.2.4              generics_0.1.3         
 [85] renv_1.0.3              data.table_1.15.0       httr_1.4.7             
 [88] htmlwidgets_1.6.4       S4Arrays_1.2.0          whisker_0.4.1          
 [91] uwot_0.1.16             pkgconfig_2.0.3         gtable_0.3.4           
 [94] blob_1.2.4              ComplexHeatmap_2.18.0   lmtest_0.9-40          
 [97] XVector_0.42.0          htmltools_0.5.7         clue_0.3-65            
[100] scales_1.3.0            png_0.1-8               knitr_1.45             
[103] rstudioapi_0.15.0       rjson_0.2.21            tzdb_0.4.0             
[106] reshape2_1.4.4          nlme_3.1-164            GlobalOptions_0.1.2    
[109] cachem_1.0.8            zoo_1.8-12              KernSmooth_2.23-22     
[112] parallel_4.3.3          miniUI_0.1.1.1          pillar_1.9.0           
[115] grid_4.3.3              vctrs_0.6.5             RANN_2.6.1             
[118] promises_1.2.1          xtable_1.8-4            cluster_2.1.6          
[121] evaluate_0.23           cli_3.6.2               locfit_1.5-9.8         
[124] compiler_4.3.3          rlang_1.1.3             crayon_1.5.2           
[127] future.apply_1.11.1     labeling_0.4.3          rematch2_2.1.2         
[130] ps_1.7.6                getPass_0.2-4           plyr_1.8.9             
[133] fs_1.6.3                stringi_1.8.3           viridisLite_0.4.2      
[136] deldir_2.0-2            Biostrings_2.70.2       munsell_0.5.0          
[139] lazyeval_0.2.2          spatstat.geom_3.2-8     Matrix_1.6-5           
[142] hms_1.1.3               bit64_4.0.5             future_1.33.1          
[145] KEGGREST_1.42.0         statmod_1.5.0           shiny_1.8.0            
[148] highr_0.10              ROCR_1.0-11             memoise_2.0.1          
[151] igraph_2.0.1.1          bslib_0.6.1             bit_4.0.5              </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
