<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2024-08-09" />

<title>Inflammation of Paediatric Pulmonary Diseases</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-inflammation-CITEseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Inflammation of Paediatric Pulmonary
Diseases</h1>
<h3 class="subtitle">Cell type proportions analysis - broad labels</h3>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">August 09, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>paed-inflammation-CITEseq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmospaedinflammationciteseqtree01f0f43d543c87aca12c50ae95538cda6f7ddc0ctargetblank01f0f43a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/tree/01f0f43d543c87aca12c50ae95538cda6f7ddc0c" target="_blank">01f0f43</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmospaedinflammationciteseqtree01f0f43d543c87aca12c50ae95538cda6f7ddc0ctargetblank01f0f43a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/tree/01f0f43d543c87aca12c50ae95538cda6f7ddc0c" target="_blank">01f0f43</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/voomByGroup/
    Ignored:    data/.DS_Store
    Ignored:    data/C133_Neeland/
    Ignored:    data/C133_Neeland_batch0/
    Ignored:    data/C133_Neeland_batch1/
    Ignored:    data/C133_Neeland_batch2/
    Ignored:    data/C133_Neeland_batch3/
    Ignored:    data/C133_Neeland_batch4/
    Ignored:    data/C133_Neeland_batch5/
    Ignored:    data/C133_Neeland_batch6/
    Ignored:    data/C133_Neeland_merged/
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  analysis/13.1_DGE_analysis_macro-alveolar_cells_decontx.Rmd
    Untracked:  code/cellbender.sh
    Untracked:  code/move_files.R
    Untracked:  data/heart10k_raw_feature_bc_matrix.h5
    Untracked:  data/oshlack_lab/
    Untracked:  data/output.log
    Untracked:  data/tiny_output.log
    Untracked:  data/tiny_raw_feature_bc_matrix.h5ad

Unstaged changes:
    Modified:   .DS_Store
    Modified:   analysis/06.1_azimuth_annotation_decontx.Rmd
    Modified:   code/run_cellbender.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/14.0_proportions_analysis_broad.Rmd</code>) and HTML
(<code>docs/14.0_proportions_analysis_broad.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/blob/01f0f43d543c87aca12c50ae95538cda6f7ddc0c/analysis/14.0_proportions_analysis_broad.Rmd" target="_blank">01f0f43</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-08-09
</td>
<td>
wflow_publish("analysis/14.*")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>suppressPackageStartupMessages({
  library(SingleCellExperiment)
  library(edgeR)
  library(tidyverse)
  library(ggplot2)
  library(Seurat)
  library(glmGamPoi)
  library(dittoSeq)
  library(clustree)
  library(AnnotationDbi)
  library(org.Hs.eg.db)
  library(glue)
  library(speckle)
  library(patchwork)
  library(paletteer)
  library(tidyHeatmap)
  library(here)
})

set.seed(42)
options(scipen=999)
options(future.globals.maxSize = 6500 * 1024^2)</code></pre>
</div>
<div id="load-data" class="section level1">
<h1>Load Data</h1>
<pre class="r"><code>files &lt;- list.files(here(&quot;data/C133_Neeland_merged&quot;),
                    pattern = &quot;C133_Neeland_full_clean.*(macrophages|t_cells|other_cells)_annotated_diet.SEU.rds&quot;,
                    full.names = TRUE)

seuLst &lt;- lapply(files[2:4], function(f) readRDS(f))

seu &lt;- merge(seuLst[[1]], 
             y = c(seuLst[[2]], 
                   seuLst[[3]]))
seu</code></pre>
<pre><code>An object of class Seurat 
21568 features across 191521 samples within 1 assay 
Active assay: RNA (21568 features, 0 variable features)</code></pre>
<pre><code>             used    (Mb) gc trigger    (Mb) limit (Mb)   max used    (Mb)
Ncells   10360801   553.4   18225151   973.4         NA   13595624   726.1
Vcells 1328307424 10134.2 3629547812 27691.3      65536 3489906798 26625.9</code></pre>
</div>
<div id="analyse-cell-type-proportions" class="section level1">
<h1>Analyse Cell type proportions</h1>
<pre class="r"><code># Differences in cell type proportions
props &lt;- getTransformedProps(clusters = seu$Broad,
                             sample = seu$sample.id, transform=&quot;asin&quot;)
props$Proportions %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="4%" />
<col width="1%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="1%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="1%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="1%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">sample_1.1</th>
<th align="right">sample_15.1</th>
<th align="right">sample_16.1</th>
<th align="right">sample_17.1</th>
<th align="right">sample_18.1</th>
<th align="right">sample_19.1</th>
<th align="right">sample_2.1</th>
<th align="right">sample_20.1</th>
<th align="right">sample_21.1</th>
<th align="right">sample_22.1</th>
<th align="right">sample_23.1</th>
<th align="right">sample_24.1</th>
<th align="right">sample_25.1</th>
<th align="right">sample_26.1</th>
<th align="right">sample_27.1</th>
<th align="right">sample_28.1</th>
<th align="right">sample_29.1</th>
<th align="right">sample_3.1</th>
<th align="right">sample_30.1</th>
<th align="right">sample_31.1</th>
<th align="right">sample_32.1</th>
<th align="right">sample_33.1</th>
<th align="right">sample_34.1</th>
<th align="right">sample_34.2</th>
<th align="right">sample_34.3</th>
<th align="right">sample_35.1</th>
<th align="right">sample_35.2</th>
<th align="right">sample_36.1</th>
<th align="right">sample_36.2</th>
<th align="right">sample_37.1</th>
<th align="right">sample_37.2</th>
<th align="right">sample_37.3</th>
<th align="right">sample_38.1</th>
<th align="right">sample_38.2</th>
<th align="right">sample_38.3</th>
<th align="right">sample_39.1</th>
<th align="right">sample_39.2</th>
<th align="right">sample_4.1</th>
<th align="right">sample_40.1</th>
<th align="right">sample_41.1</th>
<th align="right">sample_42.1</th>
<th align="right">sample_43.1</th>
<th align="right">sample_5.1</th>
<th align="right">sample_6.1</th>
<th align="right">sample_7.1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">B cells</td>
<td align="right">0.0223723</td>
<td align="right">0.0071704</td>
<td align="right">0.0018883</td>
<td align="right">0.0428725</td>
<td align="right">0.0015235</td>
<td align="right">0.0023866</td>
<td align="right">0.0950689</td>
<td align="right">0.0017933</td>
<td align="right">0.0095134</td>
<td align="right">0.0023447</td>
<td align="right">0.2436149</td>
<td align="right">0.0057254</td>
<td align="right">0.0027506</td>
<td align="right">0.0122549</td>
<td align="right">0.0020268</td>
<td align="right">0.0006984</td>
<td align="right">0.0918367</td>
<td align="right">0.0119505</td>
<td align="right">0.0064935</td>
<td align="right">0.0041667</td>
<td align="right">0.0027000</td>
<td align="right">0.0607761</td>
<td align="right">0.0026738</td>
<td align="right">0.0000000</td>
<td align="right">0.0016584</td>
<td align="right">0.0181452</td>
<td align="right">0.0081239</td>
<td align="right">0.0304653</td>
<td align="right">0.0502624</td>
<td align="right">0.0069589</td>
<td align="right">0.0124824</td>
<td align="right">0.0105605</td>
<td align="right">0.0048374</td>
<td align="right">0.0058787</td>
<td align="right">0.0356175</td>
<td align="right">0.0434431</td>
<td align="right">0.0275087</td>
<td align="right">0.0096690</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0181922</td>
<td align="right">0.0046205</td>
<td align="right">0.0282528</td>
</tr>
<tr class="even">
<td align="left">CD4 T cells</td>
<td align="right">0.0092866</td>
<td align="right">0.0275786</td>
<td align="right">0.0129485</td>
<td align="right">0.0176849</td>
<td align="right">0.0082267</td>
<td align="right">0.0023866</td>
<td align="right">0.0364283</td>
<td align="right">0.0087101</td>
<td align="right">0.0192097</td>
<td align="right">0.0082063</td>
<td align="right">0.1650295</td>
<td align="right">0.0262504</td>
<td align="right">0.0174205</td>
<td align="right">0.0323529</td>
<td align="right">0.0133766</td>
<td align="right">0.0060064</td>
<td align="right">0.0299745</td>
<td align="right">0.0061886</td>
<td align="right">0.1911977</td>
<td align="right">0.0722222</td>
<td align="right">0.0155248</td>
<td align="right">0.1547452</td>
<td align="right">0.0113636</td>
<td align="right">0.0051207</td>
<td align="right">0.0035537</td>
<td align="right">0.0173387</td>
<td align="right">0.0164368</td>
<td align="right">0.0113827</td>
<td align="right">0.0425297</td>
<td align="right">0.0153097</td>
<td align="right">0.0163076</td>
<td align="right">0.0268075</td>
<td align="right">0.0056436</td>
<td align="right">0.0102104</td>
<td align="right">0.0274874</td>
<td align="right">0.0250944</td>
<td align="right">0.0585045</td>
<td align="right">0.0106607</td>
<td align="right">0.1434307</td>
<td align="right">0.0584656</td>
<td align="right">0.0374161</td>
<td align="right">0.0462648</td>
<td align="right">0.0801592</td>
<td align="right">0.0137671</td>
<td align="right">0.0381306</td>
</tr>
<tr class="odd">
<td align="left">CD8 T cells</td>
<td align="right">0.0063318</td>
<td align="right">0.0295091</td>
<td align="right">0.0283248</td>
<td align="right">0.0353698</td>
<td align="right">0.0164534</td>
<td align="right">0.0011933</td>
<td align="right">0.0510884</td>
<td align="right">0.0098629</td>
<td align="right">0.0179290</td>
<td align="right">0.0095252</td>
<td align="right">0.1277014</td>
<td align="right">0.0310036</td>
<td align="right">0.0134474</td>
<td align="right">0.0308824</td>
<td align="right">0.0267531</td>
<td align="right">0.0153653</td>
<td align="right">0.0586735</td>
<td align="right">0.0091763</td>
<td align="right">0.0977633</td>
<td align="right">0.0949074</td>
<td align="right">0.0175498</td>
<td align="right">0.1280972</td>
<td align="right">0.0100267</td>
<td align="right">0.0029261</td>
<td align="right">0.0052120</td>
<td align="right">0.0203629</td>
<td align="right">0.0102022</td>
<td align="right">0.0033478</td>
<td align="right">0.0248550</td>
<td align="right">0.0146138</td>
<td align="right">0.0074492</td>
<td align="right">0.0199025</td>
<td align="right">0.0037624</td>
<td align="right">0.0049505</td>
<td align="right">0.0251645</td>
<td align="right">0.0070157</td>
<td align="right">0.0730337</td>
<td align="right">0.0254122</td>
<td align="right">0.1000000</td>
<td align="right">0.0743386</td>
<td align="right">0.0896167</td>
<td align="right">0.2072391</td>
<td align="right">0.0949403</td>
<td align="right">0.0271570</td>
<td align="right">0.0380244</td>
</tr>
<tr class="even">
<td align="left">DC cells</td>
<td align="right">0.0097087</td>
<td align="right">0.0184777</td>
<td align="right">0.0037766</td>
<td align="right">0.0391211</td>
<td align="right">0.0173675</td>
<td align="right">0.0055688</td>
<td align="right">0.0017770</td>
<td align="right">0.0067888</td>
<td align="right">0.0080498</td>
<td align="right">0.0016120</td>
<td align="right">0.0628684</td>
<td align="right">0.0398617</td>
<td align="right">0.0201711</td>
<td align="right">0.0230392</td>
<td align="right">0.0162140</td>
<td align="right">0.0065652</td>
<td align="right">0.0440051</td>
<td align="right">0.0147247</td>
<td align="right">0.0497835</td>
<td align="right">0.0157407</td>
<td align="right">0.0121498</td>
<td align="right">0.0719963</td>
<td align="right">0.0788770</td>
<td align="right">0.0643745</td>
<td align="right">0.0322199</td>
<td align="right">0.0219758</td>
<td align="right">0.0251275</td>
<td align="right">0.0348175</td>
<td align="right">0.0544049</td>
<td align="right">0.0073069</td>
<td align="right">0.0062412</td>
<td align="right">0.0048741</td>
<td align="right">0.0010750</td>
<td align="right">0.0024752</td>
<td align="right">0.0081301</td>
<td align="right">0.0232056</td>
<td align="right">0.0350639</td>
<td align="right">0.0126441</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0204662</td>
<td align="right">0.0045262</td>
<td align="right">0.0059480</td>
</tr>
<tr class="odd">
<td align="left">dividing innate cells</td>
<td align="right">0.0000000</td>
<td align="right">0.0002758</td>
<td align="right">0.0002698</td>
<td align="right">0.0024116</td>
<td align="right">0.0006094</td>
<td align="right">0.0003978</td>
<td align="right">0.0093292</td>
<td align="right">0.0002562</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0108055</td>
<td align="right">0.0012963</td>
<td align="right">0.0006112</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001397</td>
<td align="right">0.0031888</td>
<td align="right">0.0006402</td>
<td align="right">0.0003608</td>
<td align="right">0.0000000</td>
<td align="right">0.0003375</td>
<td align="right">0.0032726</td>
<td align="right">0.0013369</td>
<td align="right">0.0000000</td>
<td align="right">0.0002369</td>
<td align="right">0.0002016</td>
<td align="right">0.0001889</td>
<td align="right">0.0016739</td>
<td align="right">0.0008285</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0004062</td>
<td align="right">0.0000000</td>
<td align="right">0.0003094</td>
<td align="right">0.0011614</td>
<td align="right">0.0035078</td>
<td align="right">0.0017435</td>
<td align="right">0.0006198</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001062</td>
</tr>
<tr class="even">
<td align="left">epithelial cells</td>
<td align="right">0.0426340</td>
<td align="right">0.0132377</td>
<td align="right">0.0005395</td>
<td align="right">0.0032154</td>
<td align="right">0.0053321</td>
<td align="right">0.0019889</td>
<td align="right">0.1537095</td>
<td align="right">0.0043551</td>
<td align="right">0.0104281</td>
<td align="right">0.0038101</td>
<td align="right">0.2092338</td>
<td align="right">0.0034568</td>
<td align="right">0.0006112</td>
<td align="right">0.0034314</td>
<td align="right">0.0275638</td>
<td align="right">0.0006984</td>
<td align="right">0.0446429</td>
<td align="right">0.0017072</td>
<td align="right">0.0010823</td>
<td align="right">0.0018519</td>
<td align="right">0.0000000</td>
<td align="right">0.0219729</td>
<td align="right">0.0093583</td>
<td align="right">0.0087783</td>
<td align="right">0.0037906</td>
<td align="right">0.0094758</td>
<td align="right">0.0094464</td>
<td align="right">0.0077000</td>
<td align="right">0.0038663</td>
<td align="right">0.0076548</td>
<td align="right">0.0020133</td>
<td align="right">0.0052803</td>
<td align="right">0.0016125</td>
<td align="right">0.0018564</td>
<td align="right">0.0154859</td>
<td align="right">0.0029682</td>
<td align="right">0.0069740</td>
<td align="right">0.0032230</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0062536</td>
<td align="right">0.0012258</td>
<td align="right">0.0011683</td>
</tr>
<tr class="odd">
<td align="left">gamma delta T cells</td>
<td align="right">0.0000000</td>
<td align="right">0.0005516</td>
<td align="right">0.0008093</td>
<td align="right">0.0008039</td>
<td align="right">0.0003047</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0003843</td>
<td align="right">0.0007318</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0003241</td>
<td align="right">0.0006112</td>
<td align="right">0.0004902</td>
<td align="right">0.0012161</td>
<td align="right">0.0001397</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0223665</td>
<td align="right">0.0064815</td>
<td align="right">0.0003375</td>
<td align="right">0.0014025</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0002016</td>
<td align="right">0.0000000</td>
<td align="right">0.0003348</td>
<td align="right">0.0000000</td>
<td align="right">0.0010438</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0003094</td>
<td align="right">0.0038715</td>
<td align="right">0.0002698</td>
<td align="right">0.0027121</td>
<td align="right">0.0000000</td>
<td align="right">0.0171533</td>
<td align="right">0.0029101</td>
<td align="right">0.0028432</td>
<td align="right">0.0017689</td>
<td align="right">0.0005685</td>
<td align="right">0.0000000</td>
<td align="right">0.0009559</td>
</tr>
<tr class="even">
<td align="left">innate lymphocyte</td>
<td align="right">0.0029548</td>
<td align="right">0.0353006</td>
<td align="right">0.0045859</td>
<td align="right">0.0077706</td>
<td align="right">0.0039610</td>
<td align="right">0.0023866</td>
<td align="right">0.0075522</td>
<td align="right">0.0019214</td>
<td align="right">0.0012806</td>
<td align="right">0.0038101</td>
<td align="right">0.0245580</td>
<td align="right">0.0046451</td>
<td align="right">0.0058068</td>
<td align="right">0.0083333</td>
<td align="right">0.0044589</td>
<td align="right">0.0006984</td>
<td align="right">0.0082908</td>
<td align="right">0.0014938</td>
<td align="right">0.0339105</td>
<td align="right">0.0129630</td>
<td align="right">0.0006750</td>
<td align="right">0.0144928</td>
<td align="right">0.0006684</td>
<td align="right">0.0007315</td>
<td align="right">0.0016584</td>
<td align="right">0.0036290</td>
<td align="right">0.0039675</td>
<td align="right">0.0006696</td>
<td align="right">0.0138083</td>
<td align="right">0.0013918</td>
<td align="right">0.0018120</td>
<td align="right">0.0020309</td>
<td align="right">0.0002687</td>
<td align="right">0.0003094</td>
<td align="right">0.0046458</td>
<td align="right">0.0010793</td>
<td align="right">0.0048431</td>
<td align="right">0.0047105</td>
<td align="right">0.0229927</td>
<td align="right">0.0177249</td>
<td align="right">0.0378710</td>
<td align="right">0.0168730</td>
<td align="right">0.0136441</td>
<td align="right">0.0071664</td>
<td align="right">0.0023367</td>
</tr>
<tr class="odd">
<td align="left">Macrophages</td>
<td align="right">0.8518362</td>
<td align="right">0.8019857</td>
<td align="right">0.9072026</td>
<td align="right">0.7607181</td>
<td align="right">0.8740098</td>
<td align="right">0.9240255</td>
<td align="right">0.6010662</td>
<td align="right">0.9126425</td>
<td align="right">0.8940724</td>
<td align="right">0.9422626</td>
<td align="right">0.1070727</td>
<td align="right">0.7980987</td>
<td align="right">0.8719438</td>
<td align="right">0.8401961</td>
<td align="right">0.8479935</td>
<td align="right">0.9353262</td>
<td align="right">0.6696429</td>
<td align="right">0.9026889</td>
<td align="right">0.5313853</td>
<td align="right">0.7277778</td>
<td align="right">0.8876139</td>
<td align="right">0.4684432</td>
<td align="right">0.8288770</td>
<td align="right">0.8580834</td>
<td align="right">0.8995499</td>
<td align="right">0.8252016</td>
<td align="right">0.8730399</td>
<td align="right">0.7937730</td>
<td align="right">0.7321182</td>
<td align="right">0.9060543</td>
<td align="right">0.8969197</td>
<td align="right">0.9102356</td>
<td align="right">0.9180328</td>
<td align="right">0.9260520</td>
<td align="right">0.8180410</td>
<td align="right">0.8345926</td>
<td align="right">0.7268501</td>
<td align="right">0.8098426</td>
<td align="right">0.6857664</td>
<td align="right">0.7878307</td>
<td align="right">0.7939270</td>
<td align="right">0.6871683</td>
<td align="right">0.7072200</td>
<td align="right">0.8938237</td>
<td align="right">0.8326075</td>
</tr>
<tr class="even">
<td align="left">mast cells</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0005359</td>
<td align="right">0.0001523</td>
<td align="right">0.0000000</td>
<td align="right">0.0008885</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0098232</td>
<td align="right">0.0001080</td>
<td align="right">0.0012225</td>
<td align="right">0.0014706</td>
<td align="right">0.0008107</td>
<td align="right">0.0000000</td>
<td align="right">0.0063776</td>
<td align="right">0.0000000</td>
<td align="right">0.0007215</td>
<td align="right">0.0023148</td>
<td align="right">0.0000000</td>
<td align="right">0.0112202</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0004738</td>
<td align="right">0.0004032</td>
<td align="right">0.0003779</td>
<td align="right">0.0010044</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0004062</td>
<td align="right">0.0002687</td>
<td align="right">0.0000000</td>
<td align="right">0.0007743</td>
<td align="right">0.0002698</td>
<td align="right">0.0001937</td>
<td align="right">0.0021074</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0005685</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">monocytes</td>
<td align="right">0.0113972</td>
<td align="right">0.0013789</td>
<td align="right">0.0013488</td>
<td align="right">0.0200965</td>
<td align="right">0.0195003</td>
<td align="right">0.0023866</td>
<td align="right">0.0035540</td>
<td align="right">0.0044832</td>
<td align="right">0.0012806</td>
<td align="right">0.0013189</td>
<td align="right">0.0098232</td>
<td align="right">0.0468834</td>
<td align="right">0.0097800</td>
<td align="right">0.0127451</td>
<td align="right">0.0097284</td>
<td align="right">0.0108954</td>
<td align="right">0.0114796</td>
<td align="right">0.0134443</td>
<td align="right">0.0119048</td>
<td align="right">0.0087963</td>
<td align="right">0.0155248</td>
<td align="right">0.0107527</td>
<td align="right">0.0106952</td>
<td align="right">0.0065838</td>
<td align="right">0.0073442</td>
<td align="right">0.0217742</td>
<td align="right">0.0171925</td>
<td align="right">0.0133914</td>
<td align="right">0.0265120</td>
<td align="right">0.0059151</td>
<td align="right">0.0094625</td>
<td align="right">0.0064988</td>
<td align="right">0.0037624</td>
<td align="right">0.0046411</td>
<td align="right">0.0061943</td>
<td align="right">0.0218564</td>
<td align="right">0.0240217</td>
<td align="right">0.1063592</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0187607</td>
<td align="right">0.0086752</td>
<td align="right">0.0065852</td>
</tr>
<tr class="even">
<td align="left">neutrophils</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0056270</td>
<td align="right">0.0012188</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001281</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0108055</td>
<td align="right">0.0006482</td>
<td align="right">0.0003056</td>
<td align="right">0.0014706</td>
<td align="right">0.0000000</td>
<td align="right">0.0004191</td>
<td align="right">0.0063776</td>
<td align="right">0.0004268</td>
<td align="right">0.0010823</td>
<td align="right">0.0013889</td>
<td align="right">0.0006750</td>
<td align="right">0.0229079</td>
<td align="right">0.0033422</td>
<td align="right">0.0051207</td>
<td align="right">0.0021322</td>
<td align="right">0.0022177</td>
<td align="right">0.0005668</td>
<td align="right">0.0549046</td>
<td align="right">0.0035902</td>
<td align="right">0.0003479</td>
<td align="right">0.0004027</td>
<td align="right">0.0012185</td>
<td align="right">0.0000000</td>
<td align="right">0.0003094</td>
<td align="right">0.0000000</td>
<td align="right">0.0089045</td>
<td align="right">0.0092987</td>
<td align="right">0.0004958</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">NK cells</td>
<td align="right">0.0046433</td>
<td align="right">0.0085494</td>
<td align="right">0.0051254</td>
<td align="right">0.0058950</td>
<td align="right">0.0036563</td>
<td align="right">0.0000000</td>
<td align="right">0.0026655</td>
<td align="right">0.0014090</td>
<td align="right">0.0020124</td>
<td align="right">0.0011723</td>
<td align="right">0.0157171</td>
<td align="right">0.0016204</td>
<td align="right">0.0015281</td>
<td align="right">0.0014706</td>
<td align="right">0.0008107</td>
<td align="right">0.0000000</td>
<td align="right">0.0044643</td>
<td align="right">0.0014938</td>
<td align="right">0.0266955</td>
<td align="right">0.0060185</td>
<td align="right">0.0016875</td>
<td align="right">0.0079476</td>
<td align="right">0.0020053</td>
<td align="right">0.0000000</td>
<td align="right">0.0002369</td>
<td align="right">0.0020161</td>
<td align="right">0.0011336</td>
<td align="right">0.0010044</td>
<td align="right">0.0033140</td>
<td align="right">0.0003479</td>
<td align="right">0.0004027</td>
<td align="right">0.0012185</td>
<td align="right">0.0013437</td>
<td align="right">0.0006188</td>
<td align="right">0.0011614</td>
<td align="right">0.0000000</td>
<td align="right">0.0019372</td>
<td align="right">0.0019834</td>
<td align="right">0.0175182</td>
<td align="right">0.0100529</td>
<td align="right">0.0068236</td>
<td align="right">0.0073479</td>
<td align="right">0.0068221</td>
<td align="right">0.0005658</td>
<td align="right">0.0098779</td>
</tr>
<tr class="even">
<td align="left">NK-T cells</td>
<td align="right">0.0000000</td>
<td align="right">0.0041368</td>
<td align="right">0.0002698</td>
<td align="right">0.0005359</td>
<td align="right">0.0001523</td>
<td align="right">0.0000000</td>
<td align="right">0.0013327</td>
<td align="right">0.0002562</td>
<td align="right">0.0005488</td>
<td align="right">0.0001465</td>
<td align="right">0.0019646</td>
<td align="right">0.0000000</td>
<td align="right">0.0003056</td>
<td align="right">0.0014706</td>
<td align="right">0.0000000</td>
<td align="right">0.0001397</td>
<td align="right">0.0012755</td>
<td align="right">0.0002134</td>
<td align="right">0.0014430</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0014025</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0004032</td>
<td align="right">0.0011336</td>
<td align="right">0.0000000</td>
<td align="right">0.0005523</td>
<td align="right">0.0003479</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0007749</td>
<td align="right">0.0000000</td>
<td align="right">0.0025547</td>
<td align="right">0.0002646</td>
<td align="right">0.0004549</td>
<td align="right">0.0010886</td>
<td align="right">0.0005685</td>
<td align="right">0.0000943</td>
<td align="right">0.0009559</td>
</tr>
<tr class="odd">
<td align="left">Proliferating macrophages</td>
<td align="right">0.0367244</td>
<td align="right">0.0493657</td>
<td align="right">0.0321014</td>
<td align="right">0.0560021</td>
<td align="right">0.0467703</td>
<td align="right">0.0572792</td>
<td align="right">0.0310973</td>
<td align="right">0.0462406</td>
<td align="right">0.0332967</td>
<td align="right">0.0247655</td>
<td align="right">0.0000000</td>
<td align="right">0.0392136</td>
<td align="right">0.0531785</td>
<td align="right">0.0303922</td>
<td align="right">0.0490474</td>
<td align="right">0.0222098</td>
<td align="right">0.0197704</td>
<td align="right">0.0354247</td>
<td align="right">0.0151515</td>
<td align="right">0.0444444</td>
<td align="right">0.0445494</td>
<td align="right">0.0187003</td>
<td align="right">0.0407754</td>
<td align="right">0.0475494</td>
<td align="right">0.0414594</td>
<td align="right">0.0556452</td>
<td align="right">0.0317400</td>
<td align="right">0.0451958</td>
<td align="right">0.0425297</td>
<td align="right">0.0320111</td>
<td align="right">0.0457016</td>
<td align="right">0.0097482</td>
<td align="right">0.0588551</td>
<td align="right">0.0411510</td>
<td align="right">0.0514905</td>
<td align="right">0.0275229</td>
<td align="right">0.0236343</td>
<td align="right">0.0115284</td>
<td align="right">0.0083942</td>
<td align="right">0.0473545</td>
<td align="right">0.0268395</td>
<td align="right">0.0312968</td>
<td align="right">0.0278567</td>
<td align="right">0.0375295</td>
<td align="right">0.0333510</td>
</tr>
<tr class="even">
<td align="left">proliferating T/NK</td>
<td align="right">0.0021106</td>
<td align="right">0.0024821</td>
<td align="right">0.0008093</td>
<td align="right">0.0013398</td>
<td align="right">0.0007617</td>
<td align="right">0.0000000</td>
<td align="right">0.0044425</td>
<td align="right">0.0007685</td>
<td align="right">0.0016465</td>
<td align="right">0.0010258</td>
<td align="right">0.0009823</td>
<td align="right">0.0008642</td>
<td align="right">0.0003056</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0006984</td>
<td align="right">0.0000000</td>
<td align="right">0.0004268</td>
<td align="right">0.0086580</td>
<td align="right">0.0009259</td>
<td align="right">0.0006750</td>
<td align="right">0.0018700</td>
<td align="right">0.0000000</td>
<td align="right">0.0007315</td>
<td align="right">0.0004738</td>
<td align="right">0.0010081</td>
<td align="right">0.0013225</td>
<td align="right">0.0003348</td>
<td align="right">0.0008285</td>
<td align="right">0.0006959</td>
<td align="right">0.0008053</td>
<td align="right">0.0008123</td>
<td align="right">0.0005375</td>
<td align="right">0.0009282</td>
<td align="right">0.0007743</td>
<td align="right">0.0002698</td>
<td align="right">0.0029059</td>
<td align="right">0.0007438</td>
<td align="right">0.0021898</td>
<td align="right">0.0010582</td>
<td align="right">0.0042079</td>
<td align="right">0.0009525</td>
<td align="right">0.0039795</td>
<td align="right">0.0008487</td>
<td align="right">0.0016994</td>
</tr>
</tbody>
</table>
<div id="cell-type-proportions-by-sample" class="section level2">
<h2>Cell type proportions by sample</h2>
<pre class="r"><code>props$Proportions %&gt;%
  data.frame %&gt;%
  inner_join(seu@meta.data %&gt;%
               dplyr::select(sample.id, 
                             Disease,
                             Treatment,
                             Status, 
                             Severity,
                             Group,
                             Batch, 
                             Age, 
                             Sex), 
             by = c(&quot;sample&quot; = &quot;sample.id&quot;)) %&gt;%
  distinct()-&gt; dat

ggplot(dat, aes(x = sample, y = Freq, fill = clusters)) +
  geom_bar(stat = &quot;identity&quot;, color = &quot;black&quot;, size = 0.1) +
  theme(axis.text.x = element_text(angle = 90,
                                   vjust = 0.5,
                                   hjust = 1),
        legend.text = element_text(size = 8),
        legend.position = &quot;bottom&quot;) +
  labs(y = &quot;Proportion&quot;, fill = &quot;Cell Label&quot;) +
  scale_fill_paletteer_d(&quot;miscpalettes::pastel&quot;, direction = 1) +
  facet_grid(~Group, scales = &quot;free_x&quot;, space = &quot;free_x&quot;)</code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-6-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="no.-cells-per-sample" class="section level2">
<h2>No. cells per sample</h2>
<pre class="r"><code>props$Counts %&gt;%
  data.frame %&gt;%
  inner_join(seu@meta.data %&gt;%
               dplyr::select(sample.id, 
                             Disease,
                             Treatment,
                             Status, 
                             Severity, 
                             Group,
                             Batch, 
                             Age, 
                             Sex), 
             by = c(&quot;sample&quot; = &quot;sample.id&quot;)) %&gt;%
  distinct() -&gt; dat

ggplot(dat, aes(x = sample, y = Freq, fill = Disease)) +
  geom_bar(stat = &quot;identity&quot;) +
  theme(axis.text.x = element_text(angle = 90,
                                   vjust = 0.5,
                                   hjust = 1),
        legend.text = element_text(size = 8),
        legend.position = &quot;bottom&quot;) +
  labs(y = &quot;No. cells&quot;, fill = &quot;Disease&quot;) +
  facet_grid(~Group, scales = &quot;free_x&quot;, space = &quot;free_x&quot;) +
  geom_hline(yintercept = 1000, linetype = &quot;dashed&quot;)</code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-7-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>Remove sample with too few cells.</p>
<pre class="r"><code>seu &lt;- subset(seu, cells = which(!seu$sample.id %in% c(&quot;sample_23.1&quot;)))

# Differences in cell type proportions
props &lt;- getTransformedProps(clusters = seu$Broad,
                             sample = seu$sample.id, transform=&quot;asin&quot;)
props$Proportions %&gt;% knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="4%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">sample_1.1</th>
<th align="right">sample_15.1</th>
<th align="right">sample_16.1</th>
<th align="right">sample_17.1</th>
<th align="right">sample_18.1</th>
<th align="right">sample_19.1</th>
<th align="right">sample_2.1</th>
<th align="right">sample_20.1</th>
<th align="right">sample_21.1</th>
<th align="right">sample_22.1</th>
<th align="right">sample_24.1</th>
<th align="right">sample_25.1</th>
<th align="right">sample_26.1</th>
<th align="right">sample_27.1</th>
<th align="right">sample_28.1</th>
<th align="right">sample_29.1</th>
<th align="right">sample_3.1</th>
<th align="right">sample_30.1</th>
<th align="right">sample_31.1</th>
<th align="right">sample_32.1</th>
<th align="right">sample_33.1</th>
<th align="right">sample_34.1</th>
<th align="right">sample_34.2</th>
<th align="right">sample_34.3</th>
<th align="right">sample_35.1</th>
<th align="right">sample_35.2</th>
<th align="right">sample_36.1</th>
<th align="right">sample_36.2</th>
<th align="right">sample_37.1</th>
<th align="right">sample_37.2</th>
<th align="right">sample_37.3</th>
<th align="right">sample_38.1</th>
<th align="right">sample_38.2</th>
<th align="right">sample_38.3</th>
<th align="right">sample_39.1</th>
<th align="right">sample_39.2</th>
<th align="right">sample_4.1</th>
<th align="right">sample_40.1</th>
<th align="right">sample_41.1</th>
<th align="right">sample_42.1</th>
<th align="right">sample_43.1</th>
<th align="right">sample_5.1</th>
<th align="right">sample_6.1</th>
<th align="right">sample_7.1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">B cells</td>
<td align="right">0.0223723</td>
<td align="right">0.0071704</td>
<td align="right">0.0018883</td>
<td align="right">0.0428725</td>
<td align="right">0.0015235</td>
<td align="right">0.0023866</td>
<td align="right">0.0950689</td>
<td align="right">0.0017933</td>
<td align="right">0.0095134</td>
<td align="right">0.0023447</td>
<td align="right">0.0057254</td>
<td align="right">0.0027506</td>
<td align="right">0.0122549</td>
<td align="right">0.0020268</td>
<td align="right">0.0006984</td>
<td align="right">0.0918367</td>
<td align="right">0.0119505</td>
<td align="right">0.0064935</td>
<td align="right">0.0041667</td>
<td align="right">0.0027000</td>
<td align="right">0.0607761</td>
<td align="right">0.0026738</td>
<td align="right">0.0000000</td>
<td align="right">0.0016584</td>
<td align="right">0.0181452</td>
<td align="right">0.0081239</td>
<td align="right">0.0304653</td>
<td align="right">0.0502624</td>
<td align="right">0.0069589</td>
<td align="right">0.0124824</td>
<td align="right">0.0105605</td>
<td align="right">0.0048374</td>
<td align="right">0.0058787</td>
<td align="right">0.0356175</td>
<td align="right">0.0434431</td>
<td align="right">0.0275087</td>
<td align="right">0.0096690</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0181922</td>
<td align="right">0.0046205</td>
<td align="right">0.0282528</td>
</tr>
<tr class="even">
<td align="left">CD4 T cells</td>
<td align="right">0.0092866</td>
<td align="right">0.0275786</td>
<td align="right">0.0129485</td>
<td align="right">0.0176849</td>
<td align="right">0.0082267</td>
<td align="right">0.0023866</td>
<td align="right">0.0364283</td>
<td align="right">0.0087101</td>
<td align="right">0.0192097</td>
<td align="right">0.0082063</td>
<td align="right">0.0262504</td>
<td align="right">0.0174205</td>
<td align="right">0.0323529</td>
<td align="right">0.0133766</td>
<td align="right">0.0060064</td>
<td align="right">0.0299745</td>
<td align="right">0.0061886</td>
<td align="right">0.1911977</td>
<td align="right">0.0722222</td>
<td align="right">0.0155248</td>
<td align="right">0.1547452</td>
<td align="right">0.0113636</td>
<td align="right">0.0051207</td>
<td align="right">0.0035537</td>
<td align="right">0.0173387</td>
<td align="right">0.0164368</td>
<td align="right">0.0113827</td>
<td align="right">0.0425297</td>
<td align="right">0.0153097</td>
<td align="right">0.0163076</td>
<td align="right">0.0268075</td>
<td align="right">0.0056436</td>
<td align="right">0.0102104</td>
<td align="right">0.0274874</td>
<td align="right">0.0250944</td>
<td align="right">0.0585045</td>
<td align="right">0.0106607</td>
<td align="right">0.1434307</td>
<td align="right">0.0584656</td>
<td align="right">0.0374161</td>
<td align="right">0.0462648</td>
<td align="right">0.0801592</td>
<td align="right">0.0137671</td>
<td align="right">0.0381306</td>
</tr>
<tr class="odd">
<td align="left">CD8 T cells</td>
<td align="right">0.0063318</td>
<td align="right">0.0295091</td>
<td align="right">0.0283248</td>
<td align="right">0.0353698</td>
<td align="right">0.0164534</td>
<td align="right">0.0011933</td>
<td align="right">0.0510884</td>
<td align="right">0.0098629</td>
<td align="right">0.0179290</td>
<td align="right">0.0095252</td>
<td align="right">0.0310036</td>
<td align="right">0.0134474</td>
<td align="right">0.0308824</td>
<td align="right">0.0267531</td>
<td align="right">0.0153653</td>
<td align="right">0.0586735</td>
<td align="right">0.0091763</td>
<td align="right">0.0977633</td>
<td align="right">0.0949074</td>
<td align="right">0.0175498</td>
<td align="right">0.1280972</td>
<td align="right">0.0100267</td>
<td align="right">0.0029261</td>
<td align="right">0.0052120</td>
<td align="right">0.0203629</td>
<td align="right">0.0102022</td>
<td align="right">0.0033478</td>
<td align="right">0.0248550</td>
<td align="right">0.0146138</td>
<td align="right">0.0074492</td>
<td align="right">0.0199025</td>
<td align="right">0.0037624</td>
<td align="right">0.0049505</td>
<td align="right">0.0251645</td>
<td align="right">0.0070157</td>
<td align="right">0.0730337</td>
<td align="right">0.0254122</td>
<td align="right">0.1000000</td>
<td align="right">0.0743386</td>
<td align="right">0.0896167</td>
<td align="right">0.2072391</td>
<td align="right">0.0949403</td>
<td align="right">0.0271570</td>
<td align="right">0.0380244</td>
</tr>
<tr class="even">
<td align="left">DC cells</td>
<td align="right">0.0097087</td>
<td align="right">0.0184777</td>
<td align="right">0.0037766</td>
<td align="right">0.0391211</td>
<td align="right">0.0173675</td>
<td align="right">0.0055688</td>
<td align="right">0.0017770</td>
<td align="right">0.0067888</td>
<td align="right">0.0080498</td>
<td align="right">0.0016120</td>
<td align="right">0.0398617</td>
<td align="right">0.0201711</td>
<td align="right">0.0230392</td>
<td align="right">0.0162140</td>
<td align="right">0.0065652</td>
<td align="right">0.0440051</td>
<td align="right">0.0147247</td>
<td align="right">0.0497835</td>
<td align="right">0.0157407</td>
<td align="right">0.0121498</td>
<td align="right">0.0719963</td>
<td align="right">0.0788770</td>
<td align="right">0.0643745</td>
<td align="right">0.0322199</td>
<td align="right">0.0219758</td>
<td align="right">0.0251275</td>
<td align="right">0.0348175</td>
<td align="right">0.0544049</td>
<td align="right">0.0073069</td>
<td align="right">0.0062412</td>
<td align="right">0.0048741</td>
<td align="right">0.0010750</td>
<td align="right">0.0024752</td>
<td align="right">0.0081301</td>
<td align="right">0.0232056</td>
<td align="right">0.0350639</td>
<td align="right">0.0126441</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0204662</td>
<td align="right">0.0045262</td>
<td align="right">0.0059480</td>
</tr>
<tr class="odd">
<td align="left">dividing innate cells</td>
<td align="right">0.0000000</td>
<td align="right">0.0002758</td>
<td align="right">0.0002698</td>
<td align="right">0.0024116</td>
<td align="right">0.0006094</td>
<td align="right">0.0003978</td>
<td align="right">0.0093292</td>
<td align="right">0.0002562</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0012963</td>
<td align="right">0.0006112</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001397</td>
<td align="right">0.0031888</td>
<td align="right">0.0006402</td>
<td align="right">0.0003608</td>
<td align="right">0.0000000</td>
<td align="right">0.0003375</td>
<td align="right">0.0032726</td>
<td align="right">0.0013369</td>
<td align="right">0.0000000</td>
<td align="right">0.0002369</td>
<td align="right">0.0002016</td>
<td align="right">0.0001889</td>
<td align="right">0.0016739</td>
<td align="right">0.0008285</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0004062</td>
<td align="right">0.0000000</td>
<td align="right">0.0003094</td>
<td align="right">0.0011614</td>
<td align="right">0.0035078</td>
<td align="right">0.0017435</td>
<td align="right">0.0006198</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001062</td>
</tr>
<tr class="even">
<td align="left">epithelial cells</td>
<td align="right">0.0426340</td>
<td align="right">0.0132377</td>
<td align="right">0.0005395</td>
<td align="right">0.0032154</td>
<td align="right">0.0053321</td>
<td align="right">0.0019889</td>
<td align="right">0.1537095</td>
<td align="right">0.0043551</td>
<td align="right">0.0104281</td>
<td align="right">0.0038101</td>
<td align="right">0.0034568</td>
<td align="right">0.0006112</td>
<td align="right">0.0034314</td>
<td align="right">0.0275638</td>
<td align="right">0.0006984</td>
<td align="right">0.0446429</td>
<td align="right">0.0017072</td>
<td align="right">0.0010823</td>
<td align="right">0.0018519</td>
<td align="right">0.0000000</td>
<td align="right">0.0219729</td>
<td align="right">0.0093583</td>
<td align="right">0.0087783</td>
<td align="right">0.0037906</td>
<td align="right">0.0094758</td>
<td align="right">0.0094464</td>
<td align="right">0.0077000</td>
<td align="right">0.0038663</td>
<td align="right">0.0076548</td>
<td align="right">0.0020133</td>
<td align="right">0.0052803</td>
<td align="right">0.0016125</td>
<td align="right">0.0018564</td>
<td align="right">0.0154859</td>
<td align="right">0.0029682</td>
<td align="right">0.0069740</td>
<td align="right">0.0032230</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0062536</td>
<td align="right">0.0012258</td>
<td align="right">0.0011683</td>
</tr>
<tr class="odd">
<td align="left">gamma delta T cells</td>
<td align="right">0.0000000</td>
<td align="right">0.0005516</td>
<td align="right">0.0008093</td>
<td align="right">0.0008039</td>
<td align="right">0.0003047</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0003843</td>
<td align="right">0.0007318</td>
<td align="right">0.0000000</td>
<td align="right">0.0003241</td>
<td align="right">0.0006112</td>
<td align="right">0.0004902</td>
<td align="right">0.0012161</td>
<td align="right">0.0001397</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0223665</td>
<td align="right">0.0064815</td>
<td align="right">0.0003375</td>
<td align="right">0.0014025</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0002016</td>
<td align="right">0.0000000</td>
<td align="right">0.0003348</td>
<td align="right">0.0000000</td>
<td align="right">0.0010438</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0003094</td>
<td align="right">0.0038715</td>
<td align="right">0.0002698</td>
<td align="right">0.0027121</td>
<td align="right">0.0000000</td>
<td align="right">0.0171533</td>
<td align="right">0.0029101</td>
<td align="right">0.0028432</td>
<td align="right">0.0017689</td>
<td align="right">0.0005685</td>
<td align="right">0.0000000</td>
<td align="right">0.0009559</td>
</tr>
<tr class="even">
<td align="left">innate lymphocyte</td>
<td align="right">0.0029548</td>
<td align="right">0.0353006</td>
<td align="right">0.0045859</td>
<td align="right">0.0077706</td>
<td align="right">0.0039610</td>
<td align="right">0.0023866</td>
<td align="right">0.0075522</td>
<td align="right">0.0019214</td>
<td align="right">0.0012806</td>
<td align="right">0.0038101</td>
<td align="right">0.0046451</td>
<td align="right">0.0058068</td>
<td align="right">0.0083333</td>
<td align="right">0.0044589</td>
<td align="right">0.0006984</td>
<td align="right">0.0082908</td>
<td align="right">0.0014938</td>
<td align="right">0.0339105</td>
<td align="right">0.0129630</td>
<td align="right">0.0006750</td>
<td align="right">0.0144928</td>
<td align="right">0.0006684</td>
<td align="right">0.0007315</td>
<td align="right">0.0016584</td>
<td align="right">0.0036290</td>
<td align="right">0.0039675</td>
<td align="right">0.0006696</td>
<td align="right">0.0138083</td>
<td align="right">0.0013918</td>
<td align="right">0.0018120</td>
<td align="right">0.0020309</td>
<td align="right">0.0002687</td>
<td align="right">0.0003094</td>
<td align="right">0.0046458</td>
<td align="right">0.0010793</td>
<td align="right">0.0048431</td>
<td align="right">0.0047105</td>
<td align="right">0.0229927</td>
<td align="right">0.0177249</td>
<td align="right">0.0378710</td>
<td align="right">0.0168730</td>
<td align="right">0.0136441</td>
<td align="right">0.0071664</td>
<td align="right">0.0023367</td>
</tr>
<tr class="odd">
<td align="left">Macrophages</td>
<td align="right">0.8518362</td>
<td align="right">0.8019857</td>
<td align="right">0.9072026</td>
<td align="right">0.7607181</td>
<td align="right">0.8740098</td>
<td align="right">0.9240255</td>
<td align="right">0.6010662</td>
<td align="right">0.9126425</td>
<td align="right">0.8940724</td>
<td align="right">0.9422626</td>
<td align="right">0.7980987</td>
<td align="right">0.8719438</td>
<td align="right">0.8401961</td>
<td align="right">0.8479935</td>
<td align="right">0.9353262</td>
<td align="right">0.6696429</td>
<td align="right">0.9026889</td>
<td align="right">0.5313853</td>
<td align="right">0.7277778</td>
<td align="right">0.8876139</td>
<td align="right">0.4684432</td>
<td align="right">0.8288770</td>
<td align="right">0.8580834</td>
<td align="right">0.8995499</td>
<td align="right">0.8252016</td>
<td align="right">0.8730399</td>
<td align="right">0.7937730</td>
<td align="right">0.7321182</td>
<td align="right">0.9060543</td>
<td align="right">0.8969197</td>
<td align="right">0.9102356</td>
<td align="right">0.9180328</td>
<td align="right">0.9260520</td>
<td align="right">0.8180410</td>
<td align="right">0.8345926</td>
<td align="right">0.7268501</td>
<td align="right">0.8098426</td>
<td align="right">0.6857664</td>
<td align="right">0.7878307</td>
<td align="right">0.7939270</td>
<td align="right">0.6871683</td>
<td align="right">0.7072200</td>
<td align="right">0.8938237</td>
<td align="right">0.8326075</td>
</tr>
<tr class="even">
<td align="left">mast cells</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0005359</td>
<td align="right">0.0001523</td>
<td align="right">0.0000000</td>
<td align="right">0.0008885</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001080</td>
<td align="right">0.0012225</td>
<td align="right">0.0014706</td>
<td align="right">0.0008107</td>
<td align="right">0.0000000</td>
<td align="right">0.0063776</td>
<td align="right">0.0000000</td>
<td align="right">0.0007215</td>
<td align="right">0.0023148</td>
<td align="right">0.0000000</td>
<td align="right">0.0112202</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0004738</td>
<td align="right">0.0004032</td>
<td align="right">0.0003779</td>
<td align="right">0.0010044</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0004062</td>
<td align="right">0.0002687</td>
<td align="right">0.0000000</td>
<td align="right">0.0007743</td>
<td align="right">0.0002698</td>
<td align="right">0.0001937</td>
<td align="right">0.0021074</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0005685</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">monocytes</td>
<td align="right">0.0113972</td>
<td align="right">0.0013789</td>
<td align="right">0.0013488</td>
<td align="right">0.0200965</td>
<td align="right">0.0195003</td>
<td align="right">0.0023866</td>
<td align="right">0.0035540</td>
<td align="right">0.0044832</td>
<td align="right">0.0012806</td>
<td align="right">0.0013189</td>
<td align="right">0.0468834</td>
<td align="right">0.0097800</td>
<td align="right">0.0127451</td>
<td align="right">0.0097284</td>
<td align="right">0.0108954</td>
<td align="right">0.0114796</td>
<td align="right">0.0134443</td>
<td align="right">0.0119048</td>
<td align="right">0.0087963</td>
<td align="right">0.0155248</td>
<td align="right">0.0107527</td>
<td align="right">0.0106952</td>
<td align="right">0.0065838</td>
<td align="right">0.0073442</td>
<td align="right">0.0217742</td>
<td align="right">0.0171925</td>
<td align="right">0.0133914</td>
<td align="right">0.0265120</td>
<td align="right">0.0059151</td>
<td align="right">0.0094625</td>
<td align="right">0.0064988</td>
<td align="right">0.0037624</td>
<td align="right">0.0046411</td>
<td align="right">0.0061943</td>
<td align="right">0.0218564</td>
<td align="right">0.0240217</td>
<td align="right">0.1063592</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0187607</td>
<td align="right">0.0086752</td>
<td align="right">0.0065852</td>
</tr>
<tr class="even">
<td align="left">neutrophils</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0056270</td>
<td align="right">0.0012188</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0001281</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0006482</td>
<td align="right">0.0003056</td>
<td align="right">0.0014706</td>
<td align="right">0.0000000</td>
<td align="right">0.0004191</td>
<td align="right">0.0063776</td>
<td align="right">0.0004268</td>
<td align="right">0.0010823</td>
<td align="right">0.0013889</td>
<td align="right">0.0006750</td>
<td align="right">0.0229079</td>
<td align="right">0.0033422</td>
<td align="right">0.0051207</td>
<td align="right">0.0021322</td>
<td align="right">0.0022177</td>
<td align="right">0.0005668</td>
<td align="right">0.0549046</td>
<td align="right">0.0035902</td>
<td align="right">0.0003479</td>
<td align="right">0.0004027</td>
<td align="right">0.0012185</td>
<td align="right">0.0000000</td>
<td align="right">0.0003094</td>
<td align="right">0.0000000</td>
<td align="right">0.0089045</td>
<td align="right">0.0092987</td>
<td align="right">0.0004958</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">NK cells</td>
<td align="right">0.0046433</td>
<td align="right">0.0085494</td>
<td align="right">0.0051254</td>
<td align="right">0.0058950</td>
<td align="right">0.0036563</td>
<td align="right">0.0000000</td>
<td align="right">0.0026655</td>
<td align="right">0.0014090</td>
<td align="right">0.0020124</td>
<td align="right">0.0011723</td>
<td align="right">0.0016204</td>
<td align="right">0.0015281</td>
<td align="right">0.0014706</td>
<td align="right">0.0008107</td>
<td align="right">0.0000000</td>
<td align="right">0.0044643</td>
<td align="right">0.0014938</td>
<td align="right">0.0266955</td>
<td align="right">0.0060185</td>
<td align="right">0.0016875</td>
<td align="right">0.0079476</td>
<td align="right">0.0020053</td>
<td align="right">0.0000000</td>
<td align="right">0.0002369</td>
<td align="right">0.0020161</td>
<td align="right">0.0011336</td>
<td align="right">0.0010044</td>
<td align="right">0.0033140</td>
<td align="right">0.0003479</td>
<td align="right">0.0004027</td>
<td align="right">0.0012185</td>
<td align="right">0.0013437</td>
<td align="right">0.0006188</td>
<td align="right">0.0011614</td>
<td align="right">0.0000000</td>
<td align="right">0.0019372</td>
<td align="right">0.0019834</td>
<td align="right">0.0175182</td>
<td align="right">0.0100529</td>
<td align="right">0.0068236</td>
<td align="right">0.0073479</td>
<td align="right">0.0068221</td>
<td align="right">0.0005658</td>
<td align="right">0.0098779</td>
</tr>
<tr class="even">
<td align="left">NK-T cells</td>
<td align="right">0.0000000</td>
<td align="right">0.0041368</td>
<td align="right">0.0002698</td>
<td align="right">0.0005359</td>
<td align="right">0.0001523</td>
<td align="right">0.0000000</td>
<td align="right">0.0013327</td>
<td align="right">0.0002562</td>
<td align="right">0.0005488</td>
<td align="right">0.0001465</td>
<td align="right">0.0000000</td>
<td align="right">0.0003056</td>
<td align="right">0.0014706</td>
<td align="right">0.0000000</td>
<td align="right">0.0001397</td>
<td align="right">0.0012755</td>
<td align="right">0.0002134</td>
<td align="right">0.0014430</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0014025</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0004032</td>
<td align="right">0.0011336</td>
<td align="right">0.0000000</td>
<td align="right">0.0005523</td>
<td align="right">0.0003479</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0007749</td>
<td align="right">0.0000000</td>
<td align="right">0.0025547</td>
<td align="right">0.0002646</td>
<td align="right">0.0004549</td>
<td align="right">0.0010886</td>
<td align="right">0.0005685</td>
<td align="right">0.0000943</td>
<td align="right">0.0009559</td>
</tr>
<tr class="odd">
<td align="left">Proliferating macrophages</td>
<td align="right">0.0367244</td>
<td align="right">0.0493657</td>
<td align="right">0.0321014</td>
<td align="right">0.0560021</td>
<td align="right">0.0467703</td>
<td align="right">0.0572792</td>
<td align="right">0.0310973</td>
<td align="right">0.0462406</td>
<td align="right">0.0332967</td>
<td align="right">0.0247655</td>
<td align="right">0.0392136</td>
<td align="right">0.0531785</td>
<td align="right">0.0303922</td>
<td align="right">0.0490474</td>
<td align="right">0.0222098</td>
<td align="right">0.0197704</td>
<td align="right">0.0354247</td>
<td align="right">0.0151515</td>
<td align="right">0.0444444</td>
<td align="right">0.0445494</td>
<td align="right">0.0187003</td>
<td align="right">0.0407754</td>
<td align="right">0.0475494</td>
<td align="right">0.0414594</td>
<td align="right">0.0556452</td>
<td align="right">0.0317400</td>
<td align="right">0.0451958</td>
<td align="right">0.0425297</td>
<td align="right">0.0320111</td>
<td align="right">0.0457016</td>
<td align="right">0.0097482</td>
<td align="right">0.0588551</td>
<td align="right">0.0411510</td>
<td align="right">0.0514905</td>
<td align="right">0.0275229</td>
<td align="right">0.0236343</td>
<td align="right">0.0115284</td>
<td align="right">0.0083942</td>
<td align="right">0.0473545</td>
<td align="right">0.0268395</td>
<td align="right">0.0312968</td>
<td align="right">0.0278567</td>
<td align="right">0.0375295</td>
<td align="right">0.0333510</td>
</tr>
<tr class="even">
<td align="left">proliferating T/NK</td>
<td align="right">0.0021106</td>
<td align="right">0.0024821</td>
<td align="right">0.0008093</td>
<td align="right">0.0013398</td>
<td align="right">0.0007617</td>
<td align="right">0.0000000</td>
<td align="right">0.0044425</td>
<td align="right">0.0007685</td>
<td align="right">0.0016465</td>
<td align="right">0.0010258</td>
<td align="right">0.0008642</td>
<td align="right">0.0003056</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0006984</td>
<td align="right">0.0000000</td>
<td align="right">0.0004268</td>
<td align="right">0.0086580</td>
<td align="right">0.0009259</td>
<td align="right">0.0006750</td>
<td align="right">0.0018700</td>
<td align="right">0.0000000</td>
<td align="right">0.0007315</td>
<td align="right">0.0004738</td>
<td align="right">0.0010081</td>
<td align="right">0.0013225</td>
<td align="right">0.0003348</td>
<td align="right">0.0008285</td>
<td align="right">0.0006959</td>
<td align="right">0.0008053</td>
<td align="right">0.0008123</td>
<td align="right">0.0005375</td>
<td align="right">0.0009282</td>
<td align="right">0.0007743</td>
<td align="right">0.0002698</td>
<td align="right">0.0029059</td>
<td align="right">0.0007438</td>
<td align="right">0.0021898</td>
<td align="right">0.0010582</td>
<td align="right">0.0042079</td>
<td align="right">0.0009525</td>
<td align="right">0.0039795</td>
<td align="right">0.0008487</td>
<td align="right">0.0016994</td>
</tr>
</tbody>
</table>
</div>
<div id="cell-proportions-by-sample" class="section level2">
<h2>Cell proportions by sample</h2>
<pre class="r"><code>props$Proportions %&gt;%
  data.frame %&gt;%
  inner_join(seu@meta.data %&gt;%
               dplyr::select(sample.id, 
                             Disease,
                             Treatment,
                             Status, 
                             Severity, 
                             Group,
                             Batch, 
                             Age, 
                             Sex), 
             by = c(&quot;sample&quot; = &quot;sample.id&quot;)) %&gt;%
  distinct()-&gt; dat

ggplot(dat, aes(x = sample, y = Freq, fill = clusters)) +
  geom_bar(stat = &quot;identity&quot;, color = &quot;black&quot;, size = 0.1) +
  theme(axis.text.x = element_text(angle = 90,
                                   vjust = 0.5,
                                   hjust = 1),
        legend.text = element_text(size = 8),
        legend.position = &quot;bottom&quot;) +
  labs(y = &quot;Proportion&quot;, fill = &quot;Cell Label&quot;) +
  scale_fill_paletteer_d(&quot;miscpalettes::pastel&quot;, direction = 1) +
  facet_grid(~Group, scales = &quot;free_x&quot;, space = &quot;free_x&quot;)</code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-9-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-proportions-by-cell-type" class="section level2">
<h2>Cell proportions by cell type</h2>
<pre class="r"><code>props$Proportions %&gt;%
  data.frame %&gt;%
  inner_join(seu@meta.data %&gt;%
               dplyr::select(sample.id,
                             Annotation,
                             Broad,
                             Disease,
                             Treatment,
                             Status, 
                             Severity, 
                             Batch, 
                             Age, 
                             Sex), 
             by = c(&quot;sample&quot; = &quot;sample.id&quot;, &quot;clusters&quot; = &quot;Broad&quot;)) %&gt;%
  distinct()-&gt; dat

ggplot(dat, 
       aes(x = clusters, y = Freq, fill = clusters)) +
  geom_boxplot(outlier.size = 0.1, size = 0.25) +
  theme(axis.text.x = element_text(angle = 45,
                                   vjust = 1,
                                   hjust = 1),
        legend.text = element_text(size = 8)) +
  labs(y = &quot;Proportion&quot;) +
  scale_fill_paletteer_d(&quot;miscpalettes::pastel&quot;, direction = 1) +
  NoLegend()</code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="explore-sources-of-variation" class="section level1">
<h1>Explore sources of variation</h1>
<div id="cell-count-data" class="section level2">
<h2>Cell count data</h2>
<p>Look at the sources of variation in the raw cell count level
data.</p>
<pre class="r"><code>dims &lt;- list(c(1,2), c(2:3), c(3,4), c(4,5))
p &lt;- vector(&quot;list&quot;, length(dims))

for(i in 1:length(dims)){
  mds &lt;- plotMDS(props$Counts,
    gene.selection = &quot;common&quot;,
    plot = FALSE, dim.plot = dims[[i]])
  
  data.frame(x = mds$x, 
             y = mds$y,
             sample = rownames(mds$distance.matrix.squared)) %&gt;%
    left_join(seu@meta.data %&gt;%
               dplyr::select(sample.id, 
                             Disease,
                             Treatment,
                             Status, 
                             Severity, 
                             Batch, 
                             Age, 
                             Sex), 
             by = c(&quot;sample&quot; = &quot;sample.id&quot;)) %&gt;%
    distinct() -&gt; dat
  
  p[[i]] &lt;- ggplot(dat, aes(x = x, y = y, 
                            shape = as.factor(Disease),
                            color = as.factor(Batch))) +
    geom_point(size = 3) +
    labs(x = glue(&quot;Principal Component {dims[[i]][1]}&quot;),
         y = glue(&quot;Principal Component {dims[[i]][2]}&quot;),
         colour = &quot;Batch&quot;,
         shape = &quot;Disease&quot;) +
    theme(legend.direction = &quot;horizontal&quot;,
          legend.text = element_text(size = 8),
          legend.title = element_text(size = 9),
          axis.text = element_text(size = 8),
          axis.title = element_text(size = 9)) 
}

wrap_plots(p, cols = 2) + plot_layout(guides = &quot;collect&quot;) &amp;
  theme(legend.position = &quot;bottom&quot;) </code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dims &lt;- list(c(1,2), c(2:3), c(3,4), c(4,5))
p &lt;- vector(&quot;list&quot;, length(dims))

for(i in 1:length(dims)){
  mds &lt;- plotMDS(props$Counts, 
                 gene.selection = &quot;common&quot;,
                 plot = FALSE, dim.plot = dims[[i]])
  
   data.frame(x = mds$x, 
             y = mds$y,
             sample = rownames(mds$distance.matrix.squared)) %&gt;%
    left_join(seu@meta.data %&gt;%
               dplyr::select(sample.id, 
                             Disease,
                             Treatment,
                             Status, 
                             Severity,
                             Participant,
                             Batch, 
                             Age, 
                             Sex), 
             by = c(&quot;sample&quot; = &quot;sample.id&quot;)) %&gt;%
    group_by(sample) %&gt;%
    mutate(ncells = n()) %&gt;%
    ungroup() %&gt;%
    distinct() -&gt; dat
  
  p[[i]] &lt;- ggplot(dat, aes(x = x, y = y, 
                            colour = log2(ncells)))+
    geom_text(aes(label = str_remove_all(sample, &quot;sample_&quot;)), size = 2.5) +
    labs(x = glue(&quot;Principal Component {dims[[i]][1]}&quot;),
         y = glue(&quot;Principal Component {dims[[i]][2]}&quot;),
         colour = &quot;Log2 No. Cells&quot;) +
    theme(legend.direction = &quot;horizontal&quot;,
          legend.text = element_text(size = 8),
          legend.title = element_text(size = 9),
          axis.text = element_text(size = 8),
          axis.title = element_text(size = 9)) +
    scale_colour_viridis_c(option = &quot;magma&quot;)
}

wrap_plots(p, cols = 2) + plot_layout(guides = &quot;collect&quot;) &amp;
  theme(legend.position = &quot;bottom&quot;) </code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-proportion-data" class="section level2">
<h2>Cell proportion data</h2>
<p>Look at the sources of variation in the cell proportions data.</p>
<pre class="r"><code>dims &lt;- list(c(1,2), c(2:3), c(3,4), c(4,5))
p &lt;- vector(&quot;list&quot;, length(dims))

for(i in 1:length(dims)){
  mds &lt;- plotMDS(props$TransformedProps,
    gene.selection = &quot;common&quot;,
    plot = FALSE, dim.plot = dims[[i]])
  
  data.frame(x = mds$x, 
             y = mds$y,
             sample = rownames(mds$distance.matrix.squared)) %&gt;%
    left_join(seu@meta.data %&gt;%
               dplyr::select(sample.id, 
                             Disease,
                             Treatment,
                             Status, 
                             Severity, 
                             Batch, 
                             Age, 
                             Sex), 
             by = c(&quot;sample&quot; = &quot;sample.id&quot;)) %&gt;%
    distinct() -&gt; dat
  
  p[[i]] &lt;- ggplot(dat, aes(x = x, y = y, 
                            shape = as.factor(Disease),
                            color = as.factor(Batch)))+
    geom_point(size = 3) +
    labs(x = glue(&quot;Principal Component {dims[[i]][1]}&quot;),
         y = glue(&quot;Principal Component {dims[[i]][2]}&quot;),
         colour = &quot;Batch&quot;,
         shape = &quot;Disease&quot;) +
    theme(legend.direction = &quot;horizontal&quot;,
          legend.text = element_text(size = 8),
          legend.title = element_text(size = 9),
          axis.text = element_text(size = 8),
          axis.title = element_text(size = 9))
}

wrap_plots(p, cols = 2) + plot_layout(guides = &quot;collect&quot;) &amp;
  theme(legend.position = &quot;bottom&quot;) </code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dims &lt;- list(c(1,2), c(2:3), c(3,4), c(4,5))
p &lt;- vector(&quot;list&quot;, length(dims))

for(i in 1:length(dims)){
  mds &lt;- plotMDS(props$TransformedProps,
    gene.selection = &quot;common&quot;,
    plot = FALSE, dim.plot = dims[[i]])
  
  data.frame(x = mds$x, 
             y = mds$y,
             sample = rownames(mds$distance.matrix.squared)) %&gt;%
    left_join(seu@meta.data %&gt;%
               dplyr::select(sample.id, 
                             Disease,
                             Treatment,
                             Status, 
                             Severity, 
                             Batch, 
                             Age, 
                             Sex), 
             by = c(&quot;sample&quot; = &quot;sample.id&quot;)) %&gt;%
    distinct() -&gt; dat
  
  p[[i]] &lt;- ggplot(dat, aes(x = x, y = y, 
                            shape = as.factor(Disease),
                            color = Sex))+
    geom_point(size = 3) +
    labs(x = glue(&quot;Principal Component {dims[[i]][1]}&quot;),
         y = glue(&quot;Principal Component {dims[[i]][2]}&quot;),
         colour = &quot;Sex&quot;,
         shape = &quot;Disease&quot;) +
    theme(legend.direction = &quot;horizontal&quot;,
          legend.text = element_text(size = 8),
          legend.title = element_text(size = 9),
          axis.text = element_text(size = 8),
          axis.title = element_text(size = 9))
}

wrap_plots(p, cols = 2) + plot_layout(guides = &quot;collect&quot;) &amp;
  theme(legend.position = &quot;bottom&quot;) </code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dims &lt;- list(c(1,2), c(2:3), c(3,4), c(4,5))
p &lt;- vector(&quot;list&quot;, length(dims))

for(i in 1:length(dims)){
  mds &lt;- plotMDS(props$TransformedProps, 
                 gene.selection = &quot;common&quot;,
                 plot = FALSE, dim.plot = dims[[i]])
  
   data.frame(x = mds$x, 
             y = mds$y,
             sample = rownames(mds$distance.matrix.squared)) %&gt;%
    left_join(seu@meta.data %&gt;%
               dplyr::select(sample.id, 
                             Disease,
                             Treatment,
                             Status, 
                             Severity,
                             Participant,
                             Batch, 
                             Age, 
                             Sex), 
             by = c(&quot;sample&quot; = &quot;sample.id&quot;)) %&gt;%
    group_by(sample) %&gt;%
    mutate(ncells = n()) %&gt;%
    ungroup() %&gt;%
    distinct() -&gt; dat
  
  p[[i]] &lt;- ggplot(dat, aes(x = x, y = y, 
                            colour = log2(Age)))+
    geom_text(aes(label = str_remove_all(sample, &quot;sample_&quot;)), size = 2.5) +
    labs(x = glue(&quot;Principal Component {dims[[i]][1]}&quot;),
         y = glue(&quot;Principal Component {dims[[i]][2]}&quot;),
         colour = &quot;Log2 Age&quot;) +
    theme(legend.direction = &quot;horizontal&quot;,
          legend.text = element_text(size = 8),
          legend.title = element_text(size = 9),
          axis.text = element_text(size = 8),
          axis.title = element_text(size = 9)) +
    scale_colour_viridis_c(option = &quot;magma&quot;)
}

wrap_plots(p, cols = 2) + plot_layout(guides = &quot;collect&quot;) &amp;
  theme(legend.position = &quot;bottom&quot;) </code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dims &lt;- list(c(1,2), c(2:3), c(3,4), c(4,5))
p &lt;- vector(&quot;list&quot;, length(dims))

for(i in 1:length(dims)){
  mds &lt;- plotMDS(props$TransformedProps, 
                 gene.selection = &quot;common&quot;,
                 plot = FALSE, dim.plot = dims[[i]])
  
   data.frame(x = mds$x, 
             y = mds$y,
             sample = rownames(mds$distance.matrix.squared)) %&gt;%
    left_join(seu@meta.data %&gt;%
               dplyr::select(sample.id, 
                             Disease,
                             Treatment,
                             Status, 
                             Severity,
                             Participant,
                             Batch, 
                             Age, 
                             Sex), 
             by = c(&quot;sample&quot; = &quot;sample.id&quot;)) %&gt;%
    group_by(sample) %&gt;%
    mutate(ncells = n()) %&gt;%
    ungroup() %&gt;%
    distinct() -&gt; dat
  
  p[[i]] &lt;- ggplot(dat, aes(x = x, y = y, 
                            colour = log2(ncells)))+
    geom_text(aes(label = str_remove_all(sample, &quot;sample_&quot;)), size = 2.5) +
    labs(x = glue(&quot;Principal Component {dims[[i]][1]}&quot;),
         y = glue(&quot;Principal Component {dims[[i]][2]}&quot;),
         colour = &quot;Log2 No. Cells&quot;) +
    theme(legend.direction = &quot;horizontal&quot;,
          legend.text = element_text(size = 8),
          legend.title = element_text(size = 9),
          axis.text = element_text(size = 8),
          axis.title = element_text(size = 9)) +
    scale_colour_viridis_c(option = &quot;magma&quot;)
}

wrap_plots(p, cols = 2) + plot_layout(guides = &quot;collect&quot;) &amp;
  theme(legend.position = &quot;bottom&quot;) </code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="principal-components-versus-traits" class="section level1">
<h1>Principal components versus traits</h1>
<p>Principal components analysis (PCA) allows us to mathematically
determine the sources of variation in the data. We can then investigate
whether these correlate with any of the specifed covariates. First, we
calculate the principal components. The scree plot belows shows us that
most of the variation in this data is captured by the top 7 principal
components.</p>
<pre class="r"><code>PCs &lt;- prcomp(t(props$TransformedProps), center = TRUE, 
              scale = TRUE, retx = TRUE)
loadings = PCs$x # pc loadings
plot(PCs, type=&quot;lines&quot;) # scree plot</code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Collect all of the known sample traits.</p>
<pre class="r"><code>nGenes = nrow(props$TransformedProps)
nSamples = ncol(props$TransformedProps)

info &lt;- seu@meta.data %&gt;%
  dplyr::select(sample.id, 
                Disease,
                Treatment,
                Status, 
                Severity,
                Participant,
                Group,
                Batch, 
                Age, 
                Sex) %&gt;%
  group_by(sample.id) %&gt;%
  mutate(ncells = n()) %&gt;%
  ungroup() %&gt;%
  distinct()

m &lt;- match(colnames(props$TransformedProps), info$sample.id)
info &lt;- info[m,]

datTraits &lt;- info %&gt;% dplyr::select(Participant, Batch, Disease, Status,
                                    Group, Severity, Age, Sex, ncells) %&gt;%
  mutate(Age = log2(Age),
         ncells = log2(ncells),
    Donor = factor(Participant),
         Batch = factor(Batch),
         Disease = factor(Disease, 
                          labels = 1:length(unique(Disease))),
         Group = factor(Group, 
                        labels = 1:length(unique(Group))),
         Treatment = factor(Status, 
                            labels = 1:length(unique(Status))),
         Sex = factor(Sex, labels = length(unique(Sex))),
         Severity = factor(Severity, labels = length(unique(Severity)))) %&gt;%
  mutate(across(everything(), as.numeric)) %&gt;%
  dplyr::select(-Participant, -Status)

datTraits %&gt;% 
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="8%" />
<col width="11%" />
<col width="8%" />
<col width="13%" />
<col width="15%" />
<col width="5%" />
<col width="13%" />
<col width="8%" />
<col width="14%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">Batch</th>
<th align="right">Disease</th>
<th align="right">Group</th>
<th align="right">Severity</th>
<th align="right">Age</th>
<th align="right">Sex</th>
<th align="right">ncells</th>
<th align="right">Donor</th>
<th align="right">Treatment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">4</td>
<td align="right">2</td>
<td align="right">7</td>
<td align="right">1</td>
<td align="right">-0.2590872</td>
<td align="right">2</td>
<td align="right">11.21006</td>
<td align="right">1</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="right">4</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">-0.0939001</td>
<td align="right">2</td>
<td align="right">11.82416</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">4</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">-0.1151479</td>
<td align="right">1</td>
<td align="right">11.85604</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="right">5</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">-0.0441471</td>
<td align="right">1</td>
<td align="right">11.86573</td>
<td align="right">4</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">5</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">0.1428834</td>
<td align="right">2</td>
<td align="right">12.68036</td>
<td align="right">5</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="right">6</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">-0.0729608</td>
<td align="right">1</td>
<td align="right">11.29577</td>
<td align="right">6</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">4</td>
<td align="right">2</td>
<td align="right">7</td>
<td align="right">1</td>
<td align="right">0.1464588</td>
<td align="right">2</td>
<td align="right">11.13635</td>
<td align="right">7</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="right">6</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">3</td>
<td align="right">0.5597097</td>
<td align="right">2</td>
<td align="right">12.93055</td>
<td align="right">8</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">4</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">3</td>
<td align="right">1.5743836</td>
<td align="right">1</td>
<td align="right">12.41627</td>
<td align="right">9</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="right">4</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1.5993830</td>
<td align="right">2</td>
<td align="right">12.73640</td>
<td align="right">10</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="right">6</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2.3883594</td>
<td align="right">2</td>
<td align="right">13.17633</td>
<td align="right">11</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">3</td>
<td align="right">2.2957230</td>
<td align="right">1</td>
<td align="right">11.67596</td>
<td align="right">12</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">2.3360877</td>
<td align="right">2</td>
<td align="right">10.99435</td>
<td align="right">13</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2.2980155</td>
<td align="right">2</td>
<td align="right">11.26854</td>
<td align="right">14</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="right">6</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">2.5790214</td>
<td align="right">1</td>
<td align="right">12.80554</td>
<td align="right">15</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">3</td>
<td align="right">2.5823250</td>
<td align="right">1</td>
<td align="right">10.61471</td>
<td align="right">16</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">6</td>
<td align="right">2</td>
<td align="right">7</td>
<td align="right">1</td>
<td align="right">0.1321035</td>
<td align="right">2</td>
<td align="right">12.19414</td>
<td align="right">17</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">3</td>
<td align="right">2.5889097</td>
<td align="right">1</td>
<td align="right">11.43671</td>
<td align="right">18</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">2.5583683</td>
<td align="right">1</td>
<td align="right">11.07682</td>
<td align="right">19</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">2.5670653</td>
<td align="right">1</td>
<td align="right">11.53284</td>
<td align="right">20</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
<td align="right">2.5730557</td>
<td align="right">2</td>
<td align="right">11.06272</td>
<td align="right">21</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="right">7</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">-0.9343238</td>
<td align="right">1</td>
<td align="right">10.54689</td>
<td align="right">22</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">7</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">0.0918737</td>
<td align="right">1</td>
<td align="right">10.41680</td>
<td align="right">22</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="right">7</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">1.0409164</td>
<td align="right">1</td>
<td align="right">12.04337</td>
<td align="right">22</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">7</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">0.0807044</td>
<td align="right">2</td>
<td align="right">12.27612</td>
<td align="right">23</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="right">7</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">0.9940589</td>
<td align="right">2</td>
<td align="right">12.36987</td>
<td align="right">23</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">7</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">3</td>
<td align="right">-0.0564254</td>
<td align="right">1</td>
<td align="right">11.54448</td>
<td align="right">24</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="right">7</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">3</td>
<td align="right">1.1764977</td>
<td align="right">1</td>
<td align="right">11.82217</td>
<td align="right">24</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">1.5597097</td>
<td align="right">1</td>
<td align="right">11.48884</td>
<td align="right">25</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">2</td>
<td align="right">2.1930156</td>
<td align="right">1</td>
<td align="right">12.27816</td>
<td align="right">25</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">2</td>
<td align="right">2.2980155</td>
<td align="right">1</td>
<td align="right">11.26562</td>
<td align="right">25</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1.5703964</td>
<td align="right">2</td>
<td align="right">11.86147</td>
<td align="right">26</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2.0206033</td>
<td align="right">2</td>
<td align="right">11.65821</td>
<td align="right">26</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2.3485584</td>
<td align="right">2</td>
<td align="right">11.33483</td>
<td align="right">26</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="right">5</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">1.9730702</td>
<td align="right">1</td>
<td align="right">11.85565</td>
<td align="right">27</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="right">5</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">2</td>
<td align="right">2.6297159</td>
<td align="right">1</td>
<td align="right">12.33371</td>
<td align="right">27</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="right">7</td>
<td align="right">2</td>
<td align="right">7</td>
<td align="right">1</td>
<td align="right">0.2923784</td>
<td align="right">2</td>
<td align="right">12.97782</td>
<td align="right">28</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">3</td>
<td align="right">1.5801455</td>
<td align="right">2</td>
<td align="right">11.41996</td>
<td align="right">29</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">1.5801455</td>
<td align="right">2</td>
<td align="right">11.88417</td>
<td align="right">30</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
<td align="right">1.5993178</td>
<td align="right">2</td>
<td align="right">13.10214</td>
<td align="right">31</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">2</td>
<td align="right">7</td>
<td align="right">1</td>
<td align="right">1.5849625</td>
<td align="right">2</td>
<td align="right">12.84333</td>
<td align="right">32</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">2</td>
<td align="right">7</td>
<td align="right">1</td>
<td align="right">3.0699187</td>
<td align="right">1</td>
<td align="right">10.78054</td>
<td align="right">33</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="right">4</td>
<td align="right">2</td>
<td align="right">7</td>
<td align="right">1</td>
<td align="right">2.4204621</td>
<td align="right">2</td>
<td align="right">13.37246</td>
<td align="right">34</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="right">4</td>
<td align="right">2</td>
<td align="right">7</td>
<td align="right">1</td>
<td align="right">2.2356012</td>
<td align="right">1</td>
<td align="right">13.20075</td>
<td align="right">35</td>
<td align="right">4</td>
</tr>
</tbody>
</table>
<p>Correlate known sample traits with the top 10 principal components.
This can help us determine which traits are potentially contributing to
the main sources of variation in the data and should thus be included in
our statistical analysis.</p>
<pre class="r"><code>moduleTraitCor &lt;- suppressWarnings(cor(loadings[, 1:10], datTraits, use = &quot;p&quot;))
moduleTraitPvalue &lt;- WGCNA::corPvalueStudent(moduleTraitCor, (nSamples - 2))

textMatrix &lt;- paste(signif(moduleTraitCor, 2), &quot;\n(&quot;, 
                    signif(moduleTraitPvalue, 1), &quot;)&quot;, sep = &quot;&quot;)
dim(textMatrix) &lt;- dim(moduleTraitCor)

## Display the correlation values within a heatmap plot
par(cex=0.75, mar = c(6, 8.5, 3, 3))
WGCNA::labeledHeatmap(Matrix = t(moduleTraitCor),
                      xLabels = colnames(loadings)[1:10],
                      yLabels = names(datTraits),
                      colorLabels = FALSE,
                      colors = WGCNA::blueWhiteRed(6),
                      textMatrix = t(textMatrix),
                      setStdMargins = FALSE,
                      cex.text = 1,
                      zlim = c(-1,1),
                      main = paste(&quot;PCA-trait relationships: Top 10 PCs&quot;))</code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="statistical-analysis-using-the-propeller-and-limma-approach"
class="section level1">
<h1>Statistical analysis using the <code>propeller</code> and
<code>limma</code> approach</h1>
<p>Create the design matrix.</p>
<pre class="r"><code>group &lt;- factor(info$Group)
participant &lt;- factor(info$Participant)
age &lt;- log2(info$Age)
batch &lt;- factor(info$Batch)
sex &lt;- factor(info$Sex)

design &lt;- model.matrix(~ 0 + group + batch)
colnames(design)[1:7] &lt;- levels(group) 
design</code></pre>
<pre><code>   CF.IVA.M CF.IVA.S CF.LUMA_IVA.M CF.LUMA_IVA.S CF.NO_MOD.M CF.NO_MOD.S
1         0        0             0             0           0           0
2         0        0             0             0           1           0
3         0        0             0             0           1           0
4         0        0             0             0           1           0
5         0        0             0             0           1           0
6         0        0             0             0           1           0
7         0        0             0             0           0           0
8         0        0             0             0           0           1
9         0        0             0             0           0           1
10        1        0             0             0           0           0
11        1        0             0             0           0           0
12        0        0             0             0           0           1
13        0        0             0             0           1           0
14        1        0             0             0           0           0
15        0        0             0             0           1           0
16        0        0             0             0           0           1
17        0        0             0             0           0           0
18        0        0             0             0           0           1
19        0        0             0             0           1           0
20        0        0             0             0           1           0
21        0        1             0             0           0           0
22        0        0             0             0           1           0
23        0        0             0             0           1           0
24        0        0             0             0           1           0
25        0        0             0             0           1           0
26        0        0             0             0           1           0
27        0        0             0             0           0           1
28        0        0             0             1           0           0
29        0        0             0             0           1           0
30        0        0             1             0           0           0
31        0        0             1             0           0           0
32        1        0             0             0           0           0
33        1        0             0             0           0           0
34        1        0             0             0           0           0
35        0        0             0             0           1           0
36        0        0             1             0           0           0
37        0        0             0             0           0           0
38        0        0             0             0           0           1
39        0        0             0             0           1           0
40        0        1             0             0           0           0
41        0        0             0             0           0           0
42        0        0             0             0           0           0
43        0        0             0             0           0           0
44        0        0             0             0           0           0
   NON_CF.CTRL batch1 batch2 batch3 batch4 batch5 batch6
1            1      0      0      1      0      0      0
2            0      0      0      1      0      0      0
3            0      0      0      1      0      0      0
4            0      0      0      0      1      0      0
5            0      0      0      0      1      0      0
6            0      0      0      0      0      1      0
7            1      0      0      1      0      0      0
8            0      0      0      0      0      1      0
9            0      0      0      1      0      0      0
10           0      0      0      1      0      0      0
11           0      0      0      0      0      1      0
12           0      1      0      0      0      0      0
13           0      1      0      0      0      0      0
14           0      1      0      0      0      0      0
15           0      0      0      0      0      1      0
16           0      1      0      0      0      0      0
17           1      0      0      0      0      1      0
18           0      1      0      0      0      0      0
19           0      1      0      0      0      0      0
20           0      1      0      0      0      0      0
21           0      1      0      0      0      0      0
22           0      0      0      0      0      0      1
23           0      0      0      0      0      0      1
24           0      0      0      0      0      0      1
25           0      0      0      0      0      0      1
26           0      0      0      0      0      0      1
27           0      0      0      0      0      0      1
28           0      0      0      0      0      0      1
29           0      0      1      0      0      0      0
30           0      0      1      0      0      0      0
31           0      0      1      0      0      0      0
32           0      0      1      0      0      0      0
33           0      0      1      0      0      0      0
34           0      0      1      0      0      0      0
35           0      0      0      0      1      0      0
36           0      0      0      0      1      0      0
37           1      0      0      0      0      0      1
38           0      0      0      0      0      0      0
39           0      0      0      0      0      0      0
40           0      0      0      0      0      0      0
41           1      0      0      0      0      0      0
42           1      0      1      0      0      0      0
43           1      0      0      1      0      0      0
44           1      0      0      1      0      0      0
attr(,&quot;assign&quot;)
 [1] 1 1 1 1 1 1 1 2 2 2 2 2 2
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$group
[1] &quot;contr.treatment&quot;

attr(,&quot;contrasts&quot;)$batch
[1] &quot;contr.treatment&quot;</code></pre>
<p>Create the contrast matrix.</p>
<pre class="r"><code>contr &lt;- makeContrasts(CF.NO_MODvCF.IVA = 0.5*(CF.NO_MOD.M + CF.NO_MOD.S) - 0.5*(CF.IVA.M + CF.IVA.S),
                       CF.NO_MODvCF.LUMA_IVA = 0.5*(CF.NO_MOD.M + CF.NO_MOD.S) - 0.5*(CF.LUMA_IVA.M + CF.LUMA_IVA.S),
                       CF.NO_MOD.MvCF.NO_MOD.S = CF.NO_MOD.M - CF.NO_MOD.S,
                       CF.NO_MODvNON_CF.CTRL = 0.5*(CF.NO_MOD.M + CF.NO_MOD.S) - NON_CF.CTRL,
                       CF.IVAvNON_CF.CTRL = 0.5*(CF.IVA.M + CF.IVA.S) - NON_CF.CTRL,
                       CF.LUMA_IVAvNON_CF.CTRL = 0.5*(CF.LUMA_IVA.M + CF.LUMA_IVA.S) - NON_CF.CTRL,
                       levels = design)
contr</code></pre>
<pre><code>               Contrasts
Levels          CF.NO_MODvCF.IVA CF.NO_MODvCF.LUMA_IVA CF.NO_MOD.MvCF.NO_MOD.S
  CF.IVA.M                  -0.5                   0.0                       0
  CF.IVA.S                  -0.5                   0.0                       0
  CF.LUMA_IVA.M              0.0                  -0.5                       0
  CF.LUMA_IVA.S              0.0                  -0.5                       0
  CF.NO_MOD.M                0.5                   0.5                       1
  CF.NO_MOD.S                0.5                   0.5                      -1
  NON_CF.CTRL                0.0                   0.0                       0
  batch1                     0.0                   0.0                       0
  batch2                     0.0                   0.0                       0
  batch3                     0.0                   0.0                       0
  batch4                     0.0                   0.0                       0
  batch5                     0.0                   0.0                       0
  batch6                     0.0                   0.0                       0
               Contrasts
Levels          CF.NO_MODvNON_CF.CTRL CF.IVAvNON_CF.CTRL
  CF.IVA.M                        0.0                0.5
  CF.IVA.S                        0.0                0.5
  CF.LUMA_IVA.M                   0.0                0.0
  CF.LUMA_IVA.S                   0.0                0.0
  CF.NO_MOD.M                     0.5                0.0
  CF.NO_MOD.S                     0.5                0.0
  NON_CF.CTRL                    -1.0               -1.0
  batch1                          0.0                0.0
  batch2                          0.0                0.0
  batch3                          0.0                0.0
  batch4                          0.0                0.0
  batch5                          0.0                0.0
  batch6                          0.0                0.0
               Contrasts
Levels          CF.LUMA_IVAvNON_CF.CTRL
  CF.IVA.M                          0.0
  CF.IVA.S                          0.0
  CF.LUMA_IVA.M                     0.5
  CF.LUMA_IVA.S                     0.5
  CF.NO_MOD.M                       0.0
  CF.NO_MOD.S                       0.0
  NON_CF.CTRL                      -1.0
  batch1                            0.0
  batch2                            0.0
  batch3                            0.0
  batch4                            0.0
  batch5                            0.0
  batch6                            0.0</code></pre>
<p>Add random effect for samples from the same individual.</p>
<pre class="r"><code>dupcor &lt;- duplicateCorrelation(props$TransformedProps, design=design,
                                block=participant)
dupcor</code></pre>
<pre><code>$consensus.correlation
[1] 0.6475815

$cor
[1] 0.6475815

$atanh.correlations
 [1]  0.7046178  1.2192721  0.3225704  0.5734219  0.3836834  1.2106267
 [7]  0.1105819  1.0992546  0.8331586  0.4470538  1.7268879  1.5298445
[13]  0.9810777  0.6771234 -0.5360603  0.8016052</code></pre>
<p>Fit the model.</p>
<pre class="r"><code>fit &lt;- lmFit(props$TransformedProps, design=design, block=participant, 
                correlation=dupcor$consensus)
fit2 &lt;- contrasts.fit(fit, contr)
fit2 &lt;- eBayes(fit2, robust=TRUE, trend=FALSE)
pvalue &lt;- 0.05
summary(decideTests(fit2, p.value = pvalue))</code></pre>
<pre><code>       CF.NO_MODvCF.IVA CF.NO_MODvCF.LUMA_IVA CF.NO_MOD.MvCF.NO_MOD.S
Down                  0                     0                       0
NotSig               16                    15                      16
Up                    0                     1                       0
       CF.NO_MODvNON_CF.CTRL CF.IVAvNON_CF.CTRL CF.LUMA_IVAvNON_CF.CTRL
Down                       1                  0                       0
NotSig                    15                 16                      16
Up                         0                  0                       0</code></pre>
<div id="results" class="section level2 tabset">
<h2 class="tabset">Results</h2>
<div id="anova" class="section level3">
<h3>ANOVA</h3>
<pre class="r"><code>topTable(fit2)</code></pre>
<pre><code>                    CF.NO_MODvCF.IVA CF.NO_MODvCF.LUMA_IVA
neutrophils             -0.019722250            0.06035614
monocytes               -0.024786560           -0.02592302
mast cells              -0.019323905            0.01383322
CD8 T cells             -0.031004443           -0.07544362
Macrophages              0.055336009            0.06147213
innate lymphocyte       -0.012832123           -0.04659819
B cells                 -0.028030487           -0.03594194
CD4 T cells             -0.007798471           -0.06683024
NK cells                 0.009068285           -0.01615113
gamma delta T cells      0.016074104            0.01262772
                    CF.NO_MOD.MvCF.NO_MOD.S CF.NO_MODvNON_CF.CTRL
neutrophils                   -0.0354748540          0.0189455379
monocytes                      0.0003329593         -0.0716913715
mast cells                    -0.0109423278         -0.0091787516
CD8 T cells                   -0.0067822197         -0.0774654928
Macrophages                    0.0865178054          0.0897824139
innate lymphocyte             -0.0035924063         -0.0009096075
B cells                       -0.0525320831         -0.0469602254
CD4 T cells                   -0.0546268729         -0.0081952353
NK cells                      -0.0225312262         -0.0036286783
gamma delta T cells           -0.0232651224          0.0197439955
                    CF.IVAvNON_CF.CTRL CF.LUMA_IVAvNON_CF.CTRL    AveExpr
neutrophils                0.038667788            -0.041410598 0.03225974
monocytes                 -0.046904811            -0.045768351 0.09640016
mast cells                 0.010145153            -0.023011974 0.01581934
CD8 T cells               -0.046461050            -0.002021871 0.16947960
Macrophages                0.034446405             0.028310282 1.14015663
innate lymphocyte          0.011922515             0.045688580 0.07422051
B cells                   -0.018929738            -0.011018281 0.10119310
CD4 T cells               -0.000396764             0.058635003 0.16060800
NK cells                  -0.012696963             0.012522454 0.05126451
gamma delta T cells        0.003669892             0.007116278 0.02480051
                           F     P.Value adj.P.Val
neutrophils         4.781092 0.003701551 0.0316720
monocytes           4.723029 0.003959000 0.0316720
mast cells          2.368124 0.072577232 0.3061273
CD8 T cells         2.329604 0.076531821 0.3061273
Macrophages         1.997970 0.117789623 0.3662978
innate lymphocyte   1.878489 0.137361662 0.3662978
B cells             1.732906 0.166309247 0.3760054
CD4 T cells         1.638400 0.188002704 0.3760054
NK cells            1.105613 0.370196928 0.6157466
gamma delta T cells 1.074179 0.384841613 0.6157466</code></pre>
<pre class="r"><code>p &lt;- vector(&quot;list&quot;, ncol(contr))

for(i in 1:ncol(contr)){
props$Proportions %&gt;% data.frame %&gt;%
  left_join(info,
            by = c(&quot;sample&quot; = &quot;sample.id&quot;)) %&gt;%
  dplyr::filter(Group %in%  names(contr[, i])[abs(contr[, i]) &gt; 0]) -&gt; dat
  
  if(length(unique(dat$Group)) &gt; 2) dat$Group &lt;- str_remove(dat$Group, &quot;.(M|S)$&quot;)
ggplot(dat, aes(x = Group,
                y = Freq,
                colour = Group,
                group = Group)) +
  geom_boxplot(outlier.shape = NA, colour = &quot;grey&quot;) +
  geom_jitter(stat = &quot;identity&quot;,
              width = 0.15,
              size = 2) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90,
                                   hjust = 1,
                                   vjust = 0.5),
        legend.position = &quot;bottom&quot;,
        legend.direction = &quot;horizontal&quot;) +
  labs(x = &quot;Group&quot;, y = &quot;Proportion&quot;,
       colour = &quot;Condition&quot;) +
  facet_wrap(~clusters, scales = &quot;free_y&quot;) -&gt; p[[i]]
}

p</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-25-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-25-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-25-3.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-25-4.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-25-5.png" width="768" style="display: block; margin: auto;" /></p>
<pre><code>
[[6]]</code></pre>
<p><img src="figure/14.0_proportions_analysis_broad.Rmd/unnamed-chunk-25-6.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="session-info" class="section level1 appendix">
<h1 class="appendix">Session info</h1>
<summary>
The analysis and this document were prepared using the following
software (click triangle to expand)
</summary>
<details>
<pre class="r"><code>sessioninfo::session_info()</code></pre>
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.3.2 (2023-10-31)
 os       macOS Sonoma 14.5
 system   aarch64, darwin20
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       Australia/Melbourne
 date     2024-08-09
 pandoc   3.1.11 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 ! package              * version    date (UTC) lib source
 P abind                  1.4-5      2016-07-21 [?] RSPM (R 4.3.0)
 P AnnotationDbi        * 1.64.1     2023-11-02 [?] Bioconductor
 P backports              1.4.1      2021-12-13 [?] RSPM (R 4.3.0)
 P base64enc              0.1-3      2015-07-28 [?] RSPM (R 4.3.0)
 P Biobase              * 2.62.0     2023-10-26 [?] Bioconductor
 P BiocGenerics         * 0.48.1     2023-11-02 [?] Bioconductor
 P BiocManager            1.30.22    2023-08-08 [?] RSPM (R 4.3.0)
 P Biostrings             2.70.2     2024-01-30 [?] Bioconductor 3.18 (R 4.3.2)
 P bit                    4.0.5      2022-11-15 [?] RSPM (R 4.3.0)
 P bit64                  4.0.5      2020-08-30 [?] RSPM (R 4.3.0)
 P bitops                 1.0-7      2021-04-24 [?] RSPM (R 4.3.0)
 P blob                   1.2.4      2023-03-17 [?] RSPM (R 4.3.0)
 P bslib                  0.6.1      2023-11-28 [?] RSPM (R 4.3.0)
 P cachem                 1.0.8      2023-05-01 [?] RSPM (R 4.3.0)
 P callr                  3.7.3      2022-11-02 [?] RSPM (R 4.3.0)
 P checkmate              2.3.1      2023-12-04 [?] RSPM (R 4.3.0)
 P circlize               0.4.15     2022-05-10 [?] RSPM (R 4.3.0)
 P cli                    3.6.2      2023-12-11 [?] RSPM (R 4.3.0)
 P clue                   0.3-65     2023-09-23 [?] RSPM (R 4.3.0)
 P cluster                2.1.6      2023-12-01 [?] CRAN (R 4.3.1)
 P clustree             * 0.5.1      2023-11-05 [?] RSPM (R 4.3.0)
 P codetools              0.2-20     2024-03-31 [?] CRAN (R 4.3.1)
 P colorspace             2.1-0      2023-01-23 [?] RSPM (R 4.3.0)
 P ComplexHeatmap         2.18.0     2023-10-26 [?] Bioconductor
 P cowplot                1.1.3      2024-01-22 [?] RSPM (R 4.3.0)
 P crayon                 1.5.2      2022-09-29 [?] RSPM (R 4.3.0)
 P data.table             1.15.0     2024-01-30 [?] RSPM (R 4.3.0)
 P DBI                    1.2.1      2024-01-12 [?] RSPM (R 4.3.0)
 P DelayedArray           0.28.0     2023-11-06 [?] Bioconductor
 P deldir                 2.0-2      2023-11-23 [?] RSPM (R 4.3.0)
 P dendextend             1.17.1     2023-03-25 [?] RSPM (R 4.3.0)
 P digest                 0.6.34     2024-01-11 [?] RSPM (R 4.3.0)
 P dittoSeq             * 1.14.2     2024-02-10 [?] Bioconductor 3.18 (R 4.3.2)
 P doParallel             1.0.17     2022-02-07 [?] RSPM (R 4.3.0)
 P dplyr                * 1.1.4      2023-11-17 [?] RSPM (R 4.3.0)
 P dynamicTreeCut         1.63-1     2016-03-11 [?] RSPM (R 4.3.0)
 P edgeR                * 4.0.15     2024-02-10 [?] Bioconductor 3.18 (R 4.3.2)
 P ellipsis               0.3.2      2021-04-29 [?] RSPM (R 4.3.0)
 P evaluate               0.23       2023-11-01 [?] RSPM (R 4.3.0)
 P fansi                  1.0.6      2023-12-08 [?] RSPM (R 4.3.0)
 P farver                 2.1.1      2022-07-06 [?] RSPM (R 4.3.0)
 P fastcluster            1.2.6      2024-01-12 [?] RSPM (R 4.3.0)
 P fastmap                1.1.1      2023-02-24 [?] RSPM (R 4.3.0)
 P fitdistrplus           1.1-11     2023-04-25 [?] RSPM (R 4.3.0)
 P forcats              * 1.0.0      2023-01-29 [?] RSPM (R 4.3.0)
 P foreach                1.5.2      2022-02-02 [?] RSPM (R 4.3.0)
 P foreign                0.8-86     2023-11-28 [?] CRAN (R 4.3.1)
   Formula                1.2-5      2023-02-24 [1] RSPM (R 4.3.0)
 P fs                     1.6.3      2023-07-20 [?] RSPM (R 4.3.0)
 P future                 1.33.1     2023-12-22 [?] RSPM (R 4.3.0)
 P future.apply           1.11.1     2023-12-21 [?] RSPM (R 4.3.0)
 P generics               0.1.3      2022-07-05 [?] RSPM (R 4.3.0)
 P GenomeInfoDb         * 1.38.6     2024-02-10 [?] Bioconductor 3.18 (R 4.3.2)
 P GenomeInfoDbData       1.2.11     2024-02-16 [?] Bioconductor
 P GenomicRanges        * 1.54.1     2023-10-30 [?] Bioconductor
 P GetoptLong             1.0.5      2020-12-15 [?] RSPM (R 4.3.0)
 P getPass                0.2-4      2023-12-10 [?] RSPM (R 4.3.0)
 P ggforce                0.4.2      2024-02-19 [?] RSPM (R 4.3.0)
 P ggplot2              * 3.5.0      2024-02-23 [?] RSPM (R 4.3.0)
 P ggraph               * 2.2.0      2024-02-27 [?] RSPM (R 4.3.0)
 P ggrepel                0.9.5      2024-01-10 [?] RSPM (R 4.3.0)
 P ggridges               0.5.6      2024-01-23 [?] RSPM (R 4.3.0)
 P git2r                  0.33.0     2023-11-26 [?] RSPM (R 4.3.0)
   glmGamPoi            * 1.14.3     2024-02-10 [1] Bioconductor 3.18 (R 4.3.2)
 P GlobalOptions          0.1.2      2020-06-10 [?] RSPM (R 4.3.0)
 P globals                0.16.2     2022-11-21 [?] RSPM (R 4.3.0)
 P glue                 * 1.7.0      2024-01-09 [?] RSPM (R 4.3.0)
 P GO.db                  3.18.0     2024-02-21 [?] Bioconductor
 P goftest                1.2-3      2021-10-07 [?] RSPM (R 4.3.0)
 P graphlayouts           1.1.0      2024-01-19 [?] RSPM (R 4.3.0)
 P gridExtra              2.3        2017-09-09 [?] RSPM (R 4.3.0)
 P gtable                 0.3.4      2023-08-21 [?] RSPM (R 4.3.0)
 P here                 * 1.0.1      2020-12-13 [?] RSPM (R 4.3.0)
 P highr                  0.10       2022-12-22 [?] RSPM (R 4.3.0)
   Hmisc                  5.1-1      2023-09-12 [1] RSPM (R 4.3.0)
 P hms                    1.1.3      2023-03-21 [?] RSPM (R 4.3.0)
   htmlTable              2.4.2      2023-10-29 [1] RSPM (R 4.3.0)
 P htmltools              0.5.7      2023-11-03 [?] RSPM (R 4.3.0)
 P htmlwidgets            1.6.4      2023-12-06 [?] RSPM (R 4.3.0)
 P httpuv                 1.6.14     2024-01-26 [?] RSPM (R 4.3.0)
 P httr                   1.4.7      2023-08-15 [?] RSPM (R 4.3.0)
 P ica                    1.0-3      2022-07-08 [?] RSPM (R 4.3.0)
 P igraph                 2.0.1.1    2024-01-30 [?] RSPM (R 4.3.0)
 P impute                 1.76.0     2023-10-26 [?] Bioconductor
 P IRanges              * 2.36.0     2023-10-26 [?] Bioconductor
 P irlba                  2.3.5.1    2022-10-03 [?] RSPM (R 4.3.0)
 P iterators              1.0.14     2022-02-05 [?] RSPM (R 4.3.0)
 P jquerylib              0.1.4      2021-04-26 [?] RSPM (R 4.3.0)
 P jsonlite               1.8.8      2023-12-04 [?] RSPM (R 4.3.0)
 P KEGGREST               1.42.0     2023-10-26 [?] Bioconductor
 P KernSmooth             2.23-22    2023-07-10 [?] CRAN (R 4.3.2)
 P knitr                  1.45       2023-10-30 [?] RSPM (R 4.3.0)
 P labeling               0.4.3      2023-08-29 [?] RSPM (R 4.3.0)
 P later                  1.3.2      2023-12-06 [?] RSPM (R 4.3.0)
 P lattice                0.22-6     2024-03-20 [?] CRAN (R 4.3.1)
 P lazyeval               0.2.2      2019-03-15 [?] RSPM (R 4.3.0)
 P leiden                 0.4.3.1    2023-11-17 [?] RSPM (R 4.3.0)
 P lifecycle              1.0.4      2023-11-07 [?] RSPM (R 4.3.0)
 P limma                * 3.58.1     2023-11-02 [?] Bioconductor
 P listenv                0.9.1      2024-01-29 [?] RSPM (R 4.3.0)
 P lmtest                 0.9-40     2022-03-21 [?] RSPM (R 4.3.0)
 P locfit                 1.5-9.8    2023-06-11 [?] RSPM (R 4.3.0)
 P lubridate            * 1.9.3      2023-09-27 [?] RSPM (R 4.3.0)
 P magrittr               2.0.3      2022-03-30 [?] RSPM (R 4.3.0)
 P MASS                   7.3-60.0.1 2024-01-13 [?] CRAN (R 4.3.1)
 P Matrix                 1.6-5      2024-01-11 [?] CRAN (R 4.3.1)
 P MatrixGenerics       * 1.14.0     2023-10-26 [?] Bioconductor
 P matrixStats          * 1.2.0      2023-12-11 [?] RSPM (R 4.3.0)
 P memoise                2.0.1      2021-11-26 [?] RSPM (R 4.3.0)
 P mime                   0.12       2021-09-28 [?] RSPM (R 4.3.0)
 P miniUI                 0.1.1.1    2018-05-18 [?] RSPM (R 4.3.0)
 P munsell                0.5.0      2018-06-12 [?] RSPM (R 4.3.0)
 P nlme                   3.1-164    2023-11-27 [?] CRAN (R 4.3.1)
 P nnet                   7.3-19     2023-05-03 [?] CRAN (R 4.3.2)
 P org.Hs.eg.db         * 3.18.0     2024-02-21 [?] Bioconductor
 P paletteer            * 1.6.0      2024-01-21 [?] RSPM (R 4.3.0)
 P parallelly             1.37.0     2024-02-14 [?] RSPM (R 4.3.0)
 P patchwork            * 1.2.0      2024-01-08 [?] RSPM (R 4.3.0)
 P pbapply                1.7-2      2023-06-27 [?] RSPM (R 4.3.0)
 P pheatmap               1.0.12     2019-01-04 [?] RSPM (R 4.3.0)
 P pillar                 1.9.0      2023-03-22 [?] RSPM (R 4.3.0)
 P pkgconfig              2.0.3      2019-09-22 [?] RSPM (R 4.3.0)
 P plotly                 4.10.4     2024-01-13 [?] RSPM (R 4.3.0)
 P plyr                   1.8.9      2023-10-02 [?] RSPM (R 4.3.0)
 P png                    0.1-8      2022-11-29 [?] RSPM (R 4.3.0)
 P polyclip               1.10-6     2023-09-27 [?] RSPM (R 4.3.0)
 P preprocessCore         1.64.0     2023-10-26 [?] Bioconductor
 P prismatic              1.1.1      2022-08-15 [?] RSPM (R 4.3.0)
 P processx               3.8.3      2023-12-10 [?] RSPM (R 4.3.0)
 P progressr              0.14.0     2023-08-10 [?] RSPM (R 4.3.0)
 P promises               1.2.1      2023-08-10 [?] RSPM (R 4.3.0)
 P ps                     1.7.6      2024-01-18 [?] RSPM (R 4.3.0)
 P purrr                * 1.0.2      2023-08-10 [?] RSPM (R 4.3.0)
 P R6                     2.5.1      2021-08-19 [?] RSPM (R 4.3.0)
 P RANN                   2.6.1      2019-01-08 [?] RSPM (R 4.3.0)
 P RColorBrewer           1.1-3      2022-04-03 [?] RSPM (R 4.3.0)
 P Rcpp                   1.0.12     2024-01-09 [?] RSPM (R 4.3.0)
 P RcppAnnoy              0.0.22     2024-01-23 [?] RSPM (R 4.3.0)
 P RCurl                  1.98-1.14  2024-01-09 [?] RSPM (R 4.3.0)
 P readr                * 2.1.5      2024-01-10 [?] RSPM (R 4.3.0)
 P rematch2               2.1.2      2020-05-01 [?] RSPM (R 4.3.0)
   renv                   1.0.3      2023-09-19 [1] CRAN (R 4.3.1)
 P reshape2               1.4.4      2020-04-09 [?] RSPM (R 4.3.0)
 P reticulate             1.35.0     2024-01-31 [?] RSPM (R 4.3.0)
 P rjson                  0.2.21     2022-01-09 [?] RSPM (R 4.3.0)
 P rlang                  1.1.3      2024-01-10 [?] RSPM (R 4.3.0)
 P rmarkdown              2.25       2023-09-18 [?] RSPM (R 4.3.0)
 P ROCR                   1.0-11     2020-05-02 [?] RSPM (R 4.3.0)
 P rpart                  4.1.23     2023-12-05 [?] CRAN (R 4.3.1)
 P rprojroot              2.0.4      2023-11-05 [?] RSPM (R 4.3.0)
 P RSQLite                2.3.5      2024-01-21 [?] RSPM (R 4.3.0)
 P rstudioapi             0.15.0     2023-07-07 [?] RSPM (R 4.3.0)
 P Rtsne                  0.17       2023-12-07 [?] RSPM (R 4.3.0)
 P S4Arrays               1.2.0      2023-10-26 [?] Bioconductor
 P S4Vectors            * 0.40.2     2023-11-25 [?] Bioconductor 3.18 (R 4.3.2)
 P sass                   0.4.8      2023-12-06 [?] RSPM (R 4.3.0)
 P scales                 1.3.0      2023-11-28 [?] RSPM (R 4.3.0)
 P scattermore            1.2        2023-06-12 [?] RSPM (R 4.3.0)
 P sctransform            0.4.1      2023-10-19 [?] RSPM (R 4.3.0)
 P sessioninfo            1.2.2      2021-12-06 [?] RSPM (R 4.3.0)
 P Seurat               * 4.4.0      2023-09-28 [?] RSPM (R 4.3.2)
 P SeuratObject         * 4.1.4      2023-09-26 [?] RSPM (R 4.3.2)
 P shape                  1.4.6      2021-05-19 [?] RSPM (R 4.3.0)
 P shiny                  1.8.0      2023-11-17 [?] RSPM (R 4.3.0)
 P SingleCellExperiment * 1.24.0     2023-11-06 [?] Bioconductor
 P sp                     2.1-3      2024-01-30 [?] RSPM (R 4.3.0)
 P SparseArray            1.2.4      2024-02-10 [?] Bioconductor 3.18 (R 4.3.2)
 P spatstat.data          3.0-4      2024-01-15 [?] RSPM (R 4.3.0)
 P spatstat.explore       3.2-6      2024-02-01 [?] RSPM (R 4.3.0)
 P spatstat.geom          3.2-8      2024-01-26 [?] RSPM (R 4.3.0)
 P spatstat.random        3.2-2      2023-11-29 [?] RSPM (R 4.3.0)
 P spatstat.sparse        3.0-3      2023-10-24 [?] RSPM (R 4.3.0)
 P spatstat.utils         3.0-4      2023-10-24 [?] RSPM (R 4.3.0)
 P speckle              * 1.2.0      2023-10-26 [?] Bioconductor
 P statmod                1.5.0      2023-01-06 [?] RSPM (R 4.3.0)
 P stringi                1.8.3      2023-12-11 [?] RSPM (R 4.3.0)
 P stringr              * 1.5.1      2023-11-14 [?] RSPM (R 4.3.0)
 P SummarizedExperiment * 1.32.0     2023-11-06 [?] Bioconductor
 P survival               3.5-8      2024-02-14 [?] CRAN (R 4.3.1)
 P tensor                 1.5        2012-05-05 [?] RSPM (R 4.3.0)
 P tibble               * 3.2.1      2023-03-20 [?] RSPM (R 4.3.0)
 P tidygraph              1.3.1      2024-01-30 [?] RSPM (R 4.3.0)
 P tidyHeatmap          * 1.8.1      2022-05-20 [?] RSPM (R 4.3.2)
 P tidyr                * 1.3.1      2024-01-24 [?] RSPM (R 4.3.0)
 P tidyselect             1.2.0      2022-10-10 [?] RSPM (R 4.3.0)
 P tidyverse            * 2.0.0      2023-02-22 [?] RSPM (R 4.3.0)
 P timechange             0.3.0      2024-01-18 [?] RSPM (R 4.3.0)
 P tweenr                 2.0.3      2024-02-26 [?] RSPM (R 4.3.0)
 P tzdb                   0.4.0      2023-05-12 [?] RSPM (R 4.3.0)
 P utf8                   1.2.4      2023-10-22 [?] RSPM (R 4.3.0)
 P uwot                   0.1.16     2023-06-29 [?] RSPM (R 4.3.0)
 P vctrs                  0.6.5      2023-12-01 [?] RSPM (R 4.3.0)
 P viridis                0.6.5      2024-01-29 [?] RSPM (R 4.3.0)
 P viridisLite            0.4.2      2023-05-02 [?] RSPM (R 4.3.0)
 P WGCNA                  1.72-5     2023-12-07 [?] RSPM (R 4.3.2)
 P whisker                0.4.1      2022-12-05 [?] RSPM (R 4.3.0)
 P withr                  3.0.0      2024-01-16 [?] RSPM (R 4.3.0)
 P workflowr            * 1.7.1      2023-08-23 [?] RSPM (R 4.3.2)
 P xfun                   0.42       2024-02-08 [?] RSPM (R 4.3.0)
 P xtable                 1.8-4      2019-04-21 [?] RSPM (R 4.3.0)
 P XVector                0.42.0     2023-10-26 [?] Bioconductor
 P yaml                   2.3.8      2023-12-11 [?] RSPM (R 4.3.0)
 P zlibbioc               1.48.0     2023-10-26 [?] Bioconductor
 P zoo                    1.8-12     2023-04-13 [?] RSPM (R 4.3.0)

 [1] /Users/maksimovicjovana/Work/Projects/MCRI/melanie.neeland/paed-inflammation-CITEseq/renv/library/R-4.3/aarch64-apple-darwin20
 [2] /Users/maksimovicjovana/Library/Caches/org.R-project.R/R/renv/sandbox/R-4.3/aarch64-apple-darwin20/ac5c2659

 P ── Loaded and on-disk path mismatch.

──────────────────────────────────────────────────────────────────────────────</code></pre>
</details>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Melbourne
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] here_1.0.1                  tidyHeatmap_1.8.1          
 [3] paletteer_1.6.0             patchwork_1.2.0            
 [5] speckle_1.2.0               glue_1.7.0                 
 [7] org.Hs.eg.db_3.18.0         AnnotationDbi_1.64.1       
 [9] clustree_0.5.1              ggraph_2.2.0               
[11] dittoSeq_1.14.2             glmGamPoi_1.14.3           
[13] SeuratObject_4.1.4          Seurat_4.4.0               
[15] lubridate_1.9.3             forcats_1.0.0              
[17] stringr_1.5.1               dplyr_1.1.4                
[19] purrr_1.0.2                 readr_2.1.5                
[21] tidyr_1.3.1                 tibble_3.2.1               
[23] ggplot2_3.5.0               tidyverse_2.0.0            
[25] edgeR_4.0.15                limma_3.58.1               
[27] SingleCellExperiment_1.24.0 SummarizedExperiment_1.32.0
[29] Biobase_2.62.0              GenomicRanges_1.54.1       
[31] GenomeInfoDb_1.38.6         IRanges_2.36.0             
[33] S4Vectors_0.40.2            BiocGenerics_0.48.1        
[35] MatrixGenerics_1.14.0       matrixStats_1.2.0          
[37] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] fs_1.6.3                spatstat.sparse_3.0-3   bitops_1.0-7           
  [4] httr_1.4.7              RColorBrewer_1.1-3      doParallel_1.0.17      
  [7] dynamicTreeCut_1.63-1   backports_1.4.1         tools_4.3.2            
 [10] sctransform_0.4.1       utf8_1.2.4              R6_2.5.1               
 [13] lazyeval_0.2.2          uwot_0.1.16             GetoptLong_1.0.5       
 [16] withr_3.0.0             sp_2.1-3                gridExtra_2.3          
 [19] preprocessCore_1.64.0   progressr_0.14.0        WGCNA_1.72-5           
 [22] cli_3.6.2               spatstat.explore_3.2-6  labeling_0.4.3         
 [25] sass_0.4.8              prismatic_1.1.1         spatstat.data_3.0-4    
 [28] ggridges_0.5.6          pbapply_1.7-2           foreign_0.8-86         
 [31] sessioninfo_1.2.2       parallelly_1.37.0       impute_1.76.0          
 [34] rstudioapi_0.15.0       RSQLite_2.3.5           generics_0.1.3         
 [37] shape_1.4.6             ica_1.0-3               spatstat.random_3.2-2  
 [40] dendextend_1.17.1       GO.db_3.18.0            Matrix_1.6-5           
 [43] fansi_1.0.6             abind_1.4-5             lifecycle_1.0.4        
 [46] whisker_0.4.1           yaml_2.3.8              SparseArray_1.2.4      
 [49] Rtsne_0.17              grid_4.3.2              blob_1.2.4             
 [52] promises_1.2.1          crayon_1.5.2            miniUI_0.1.1.1         
 [55] lattice_0.22-6          cowplot_1.1.3           KEGGREST_1.42.0        
 [58] pillar_1.9.0            knitr_1.45              ComplexHeatmap_2.18.0  
 [61] rjson_0.2.21            future.apply_1.11.1     codetools_0.2-20       
 [64] leiden_0.4.3.1          getPass_0.2-4           data.table_1.15.0      
 [67] vctrs_0.6.5             png_0.1-8               gtable_0.3.4           
 [70] rematch2_2.1.2          cachem_1.0.8            xfun_0.42              
 [73] S4Arrays_1.2.0          mime_0.12               tidygraph_1.3.1        
 [76] survival_3.5-8          pheatmap_1.0.12         iterators_1.0.14       
 [79] statmod_1.5.0           ellipsis_0.3.2          fitdistrplus_1.1-11    
 [82] ROCR_1.0-11             nlme_3.1-164            bit64_4.0.5            
 [85] RcppAnnoy_0.0.22        rprojroot_2.0.4         bslib_0.6.1            
 [88] irlba_2.3.5.1           rpart_4.1.23            KernSmooth_2.23-22     
 [91] Hmisc_5.1-1             colorspace_2.1-0        DBI_1.2.1              
 [94] nnet_7.3-19             tidyselect_1.2.0        processx_3.8.3         
 [97] bit_4.0.5               compiler_4.3.2          git2r_0.33.0           
[100] htmlTable_2.4.2         DelayedArray_0.28.0     plotly_4.10.4          
[103] checkmate_2.3.1         scales_1.3.0            lmtest_0.9-40          
[106] callr_3.7.3             digest_0.6.34           goftest_1.2-3          
[109] spatstat.utils_3.0-4    rmarkdown_2.25          XVector_0.42.0         
[112] base64enc_0.1-3         htmltools_0.5.7         pkgconfig_2.0.3        
[115] highr_0.10              fastmap_1.1.1           rlang_1.1.3            
[118] GlobalOptions_0.1.2     htmlwidgets_1.6.4       shiny_1.8.0            
[121] farver_2.1.1            jquerylib_0.1.4         zoo_1.8-12             
[124] jsonlite_1.8.8          RCurl_1.98-1.14         magrittr_2.0.3         
[127] Formula_1.2-5           GenomeInfoDbData_1.2.11 munsell_0.5.0          
[130] Rcpp_1.0.12             viridis_0.6.5           reticulate_1.35.0      
[133] stringi_1.8.3           zlibbioc_1.48.0         MASS_7.3-60.0.1        
[136] plyr_1.8.9              parallel_4.3.2          listenv_0.9.1          
[139] ggrepel_0.9.5           deldir_2.0-2            Biostrings_2.70.2      
[142] graphlayouts_1.1.0      splines_4.3.2           tensor_1.5             
[145] hms_1.1.3               circlize_0.4.15         locfit_1.5-9.8         
[148] ps_1.7.6                fastcluster_1.2.6       igraph_2.0.1.1         
[151] spatstat.geom_3.2-8     reshape2_1.4.4          evaluate_0.23          
[154] renv_1.0.3              BiocManager_1.30.22     tzdb_0.4.0             
[157] foreach_1.5.2           tweenr_2.0.3            httpuv_1.6.14          
[160] RANN_2.6.1              polyclip_1.10-6         future_1.33.1          
[163] clue_0.3-65             scattermore_1.2         ggforce_0.4.2          
[166] xtable_1.8-4            later_1.3.2             viridisLite_0.4.2      
[169] memoise_2.0.1           cluster_2.1.6           timechange_0.3.0       
[172] globals_0.16.2         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
