<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2025-01-06" />

<title>Figure 4</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-inflammation-CITEseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Figure 4</h1>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">January 06, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>paed-inflammation-CITEseq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmospaedinflammationciteseqtree0a567aa96f82b8738f8e04a33782941bb49840e5targetblank0a567aaa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/tree/0a567aa96f82b8738f8e04a33782941bb49840e5" target="_blank">0a567aa</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmospaedinflammationciteseqtree0a567aa96f82b8738f8e04a33782941bb49840e5targetblank0a567aaa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/tree/0a567aa96f82b8738f8e04a33782941bb49840e5" target="_blank">0a567aa</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    analysis/obsolete/
    Ignored:    data/C133_Neeland_batch1/
    Ignored:    data/C133_Neeland_merged/
    Ignored:    output/dge_analysis/obsolete/
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  analysis/14.0_DGE_analysis_T-cells.Rmd
    Untracked:  analysis/99.0_Figure_1.Rmd
    Untracked:  analysis/99.0_Figure_2.Rmd
    Untracked:  analysis/99.0_Figure_3.Rmd
    Untracked:  analysis/99.0_Figure_5.Rmd
    Untracked:  analysis/99.0_Supplementary_Figure_ADTs.Rmd
    Untracked:  broad_markers_seurat.csv
    Untracked:  code/background_job.R
    Untracked:  code/reverse_modifier_severity_comparisons.sh
    Untracked:  data/intermediate_objects/CD4 T cells.CF_samples.fit.rds
    Untracked:  data/intermediate_objects/CD4 T cells.all_samples.fit.rds
    Untracked:  data/intermediate_objects/CD8 T cells.CF_samples.fit.rds
    Untracked:  data/intermediate_objects/CD8 T cells.all_samples.fit.rds
    Untracked:  data/intermediate_objects/DC cells.CF_samples.fit.rds
    Untracked:  data/intermediate_objects/DC cells.all_samples.fit.rds
    Untracked:  data/intermediate_objects/T cells.CF_samples.fit.rds
    Untracked:  data/intermediate_objects/T cells.all_samples.fit.rds
    Untracked:  output/dge_analysis/T cells/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/06.0_azimuth_annotation.Rmd
    Modified:   analysis/09.0_integrate_cluster_macro_cells.Rmd
    Modified:   analysis/13.1_DGE_analysis_macro-alveolar.Rmd
    Modified:   analysis/13.7_DGE_analysis_macro-proliferating.Rmd
    Deleted:    analysis/14.0_proportions_analysis_ann_level_1.Rmd
    Deleted:    analysis/14.1_proportions_analysis_ann_level_3_non-macrophages.Rmd
    Deleted:    analysis/14.2_proportions_analysis_ann_level_3_macrophages.Rmd
    Deleted:    analysis/15.0_Figure_1.Rmd
    Deleted:    analysis/16.0_Figure_2.Rmd
    Deleted:    analysis/17.0_Supplementary_Figure_ADTs.Rmd
    Modified:   analysis/index.Rmd
    Modified:   code/utility.R
    Modified:   data/cluster_annotations/marker_proteins_TNK_supp.xlsx
    Modified:   data/cluster_annotations/marker_proteins_macrophages_supp.xlsx
    Modified:   data/cluster_annotations/marker_proteins_other_supp.xlsx
    Modified:   data/cluster_annotations/seurat_markers_all_cells.rds
    Modified:   data/intermediate_objects/macro-APOC2+.CF_samples.fit.rds
    Modified:   data/intermediate_objects/macro-APOC2+.all_samples.fit.rds
    Modified:   data/intermediate_objects/macro-CCL.CF_samples.fit.rds
    Modified:   data/intermediate_objects/macro-CCL.all_samples.fit.rds
    Modified:   data/intermediate_objects/macro-IFI27.CF_samples.fit.rds
    Modified:   data/intermediate_objects/macro-IFI27.all_samples.fit.rds
    Modified:   data/intermediate_objects/macro-alveolar.CF_samples.fit.rds
    Modified:   data/intermediate_objects/macro-alveolar.all_samples.fit.rds
    Modified:   data/intermediate_objects/macro-lipid.CF_samples.fit.rds
    Modified:   data/intermediate_objects/macro-lipid.all_samples.fit.rds
    Modified:   data/intermediate_objects/macro-monocyte-derived.CF_samples.fit.rds
    Modified:   data/intermediate_objects/macro-monocyte-derived.all_samples.fit.rds
    Modified:   data/intermediate_objects/macro-proliferating.CF_samples.fit.rds
    Modified:   data/intermediate_objects/macro-proliferating.all_samples.fit.rds
    Modified:   data/intermediate_objects/macrophages.CF_samples.fit.rds
    Modified:   data/intermediate_objects/macrophages.all_samples.fit.rds
    Modified:   output/dge_analysis/macro-APOC2+/ORA.FIBROSIS.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.FIBROSIS.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.FIBROSIS.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.GO.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.GO.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.GO.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.HALLMARK.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.HALLMARK.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.HALLMARK.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.REACTOME.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.REACTOME.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.REACTOME.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.WP.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.WP.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-APOC2+/ORA.WP.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.FIBROSIS.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.FIBROSIS.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.FIBROSIS.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.FIBROSIS.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.GO.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.GO.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.GO.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.GO.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.HALLMARK.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.HALLMARK.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.HALLMARK.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.HALLMARK.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.REACTOME.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.REACTOME.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.REACTOME.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.REACTOME.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.WP.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.WP.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.WP.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-CCL/ORA.WP.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.FIBROSIS.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.FIBROSIS.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.FIBROSIS.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.FIBROSIS.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.GO.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.GO.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.GO.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.GO.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.HALLMARK.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.HALLMARK.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.HALLMARK.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.HALLMARK.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.REACTOME.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.REACTOME.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.REACTOME.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.REACTOME.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.WP.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.WP.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.WP.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-IFI27/ORA.WP.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.FIBROSIS.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.FIBROSIS.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.FIBROSIS.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.FIBROSIS.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.GO.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.GO.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.GO.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.GO.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.HALLMARK.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.HALLMARK.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.HALLMARK.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.HALLMARK.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.REACTOME.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.REACTOME.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.REACTOME.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.REACTOME.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.WP.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.WP.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.WP.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-alveolar/ORA.WP.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.FIBROSIS.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.FIBROSIS.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.FIBROSIS.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.FIBROSIS.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.GO.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.GO.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.GO.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.GO.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.HALLMARK.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.HALLMARK.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.HALLMARK.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.HALLMARK.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.REACTOME.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.REACTOME.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.REACTOME.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.REACTOME.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.WP.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.WP.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.WP.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-lipid/ORA.WP.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.FIBROSIS.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.FIBROSIS.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.FIBROSIS.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.FIBROSIS.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.GO.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.GO.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.GO.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.GO.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.HALLMARK.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.HALLMARK.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.HALLMARK.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.HALLMARK.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.REACTOME.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.REACTOME.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.REACTOME.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.REACTOME.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.WP.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.WP.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.WP.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-monocyte-derived/ORA.WP.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.FIBROSIS.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.FIBROSIS.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.FIBROSIS.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.GO.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.GO.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.GO.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.HALLMARK.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.HALLMARK.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.HALLMARK.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.REACTOME.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.REACTOME.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.REACTOME.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.WP.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.WP.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macro-proliferating/ORA.WP.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macrophages/ORA.FIBROSIS.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macrophages/ORA.FIBROSIS.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macrophages/ORA.FIBROSIS.CF.LUMA_IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macrophages/ORA.FIBROSIS.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macrophages/ORA.FIBROSIS.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macrophages/ORA.GO.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macrophages/ORA.GO.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macrophages/ORA.GO.CF.LUMA_IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macrophages/ORA.GO.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macrophages/ORA.GO.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macrophages/ORA.HALLMARK.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macrophages/ORA.HALLMARK.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macrophages/ORA.HALLMARK.CF.LUMA_IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macrophages/ORA.HALLMARK.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macrophages/ORA.HALLMARK.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macrophages/ORA.REACTOME.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macrophages/ORA.REACTOME.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macrophages/ORA.REACTOME.CF.LUMA_IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macrophages/ORA.REACTOME.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macrophages/ORA.REACTOME.CF.NO_MODvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macrophages/ORA.WP.CF.IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macrophages/ORA.WP.CF.IVAvNON_CF.CTRL.csv
    Modified:   output/dge_analysis/macrophages/ORA.WP.CF.LUMA_IVAvCF.NO_MOD.csv
    Modified:   output/dge_analysis/macrophages/ORA.WP.CF.NO_MOD.SvCF.NO_MOD.M.csv
    Modified:   output/dge_analysis/macrophages/ORA.WP.CF.NO_MODvNON_CF.CTRL.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/99.0_Figure_4.Rmd</code>) and
HTML (<code>docs/99.0_Figure_4.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/blob/0a567aa96f82b8738f8e04a33782941bb49840e5/analysis/99.0_Figure_4.Rmd" target="_blank">0a567aa</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2025-01-06
</td>
<td>
wflow_publish("analysis/99.0_Figure_4.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Load libraries.</p>
<pre class="r"><code>suppressPackageStartupMessages({
 library(SingleCellExperiment)
 library(edgeR)
 library(tidyverse)
 library(ggplot2)
 library(Seurat)
 library(glmGamPoi)
 library(dittoSeq)
 library(here)
 library(clustree)
 library(patchwork)
 library(AnnotationDbi)
 library(org.Hs.eg.db)
 library(glue)
 library(speckle)
 library(tidyHeatmap)
 library(paletteer)
 library(dsb)
 library(ggh4x)
  library(readxl)
})

source(here(&quot;code/utility.R&quot;))</code></pre>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<pre class="r"><code>files &lt;- list.files(here(&quot;data/C133_Neeland_merged&quot;),
                    pattern = &quot;C133_Neeland_full_clean.*(macrophages|t_cells|other_cells)_annotated_diet.SEU.rds&quot;,
                    full.names = TRUE)

seuLst &lt;- lapply(files[2:4], function(f) readRDS(f))

seu &lt;- merge(seuLst[[1]], 
             y = c(seuLst[[2]], 
                   seuLst[[3]]))
seu</code></pre>
<pre><code>An object of class Seurat 
21568 features across 194407 samples within 1 assay 
Active assay: RNA (21568 features, 0 variable features)</code></pre>
<pre><code>             used    (Mb) gc trigger    (Mb)   max used    (Mb)
Ncells   12100924   646.3   20376418  1088.3   14090832   752.6
Vcells 1354185555 10331.7 3693781618 28181.4 3551519297 27096.0</code></pre>
<p>Create sample meta data table.</p>
<pre class="r"><code>props &lt;- getTransformedProps(clusters = seu$ann_level_3,
                             sample = seu$sample.id, transform=&quot;asin&quot;)

seu@meta.data %&gt;%
  dplyr::select(sample.id,
                Participant,
                Disease,
                Treatment,
                Severity,
                Group,
                Group_severity,
                Batch, 
                Age, 
                Sex) %&gt;%
     left_join(props$Counts %&gt;% 
                 data.frame %&gt;%
                 group_by(sample) %&gt;%
                 summarise(ncells = sum(Freq)),
               by = c(&quot;sample.id&quot; = &quot;sample&quot;)) %&gt;%
    distinct() -&gt; info

head(info) %&gt;% knitr::kable()</code></pre>
<table>
<colgroup>
<col width="10%" />
<col width="10%" />
<col width="7%" />
<col width="17%" />
<col width="8%" />
<col width="8%" />
<col width="13%" />
<col width="5%" />
<col width="8%" />
<col width="3%" />
<col width="6%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">sample.id</th>
<th align="left">Participant</th>
<th align="left">Disease</th>
<th align="left">Treatment</th>
<th align="left">Severity</th>
<th align="left">Group</th>
<th align="left">Group_severity</th>
<th align="right">Batch</th>
<th align="right">Age</th>
<th align="left">Sex</th>
<th align="right">ncells</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">sample_33.1</td>
<td align="left">sample_33</td>
<td align="left">CF</td>
<td align="left">treated (ivacaftor)</td>
<td align="left">severe</td>
<td align="left">CF.IVA</td>
<td align="left">CF.IVA.S</td>
<td align="right">1</td>
<td align="right">5.950685</td>
<td align="left">M</td>
<td align="right">2139</td>
</tr>
<tr class="even">
<td align="left">sample_25.1</td>
<td align="left">sample_25</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">severe</td>
<td align="left">CF.NO_MOD</td>
<td align="left">CF.NO_MOD.S</td>
<td align="right">1</td>
<td align="right">4.910000</td>
<td align="left">F</td>
<td align="right">3272</td>
</tr>
<tr class="odd">
<td align="left">sample_29.1</td>
<td align="left">sample_29</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">severe</td>
<td align="left">CF.NO_MOD</td>
<td align="left">CF.NO_MOD.S</td>
<td align="right">1</td>
<td align="right">5.989041</td>
<td align="left">F</td>
<td align="right">1568</td>
</tr>
<tr class="even">
<td align="left">sample_27.1</td>
<td align="left">sample_27</td>
<td align="left">CF</td>
<td align="left">treated (ivacaftor)</td>
<td align="left">mild</td>
<td align="left">CF.IVA</td>
<td align="left">CF.IVA.M</td>
<td align="right">1</td>
<td align="right">4.917808</td>
<td align="left">M</td>
<td align="right">2467</td>
</tr>
<tr class="odd">
<td align="left">sample_32.1</td>
<td align="left">sample_32</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="left">CF.NO_MOD</td>
<td align="left">CF.NO_MOD.M</td>
<td align="right">1</td>
<td align="right">5.926027</td>
<td align="left">F</td>
<td align="right">2963</td>
</tr>
<tr class="even">
<td align="left">sample_26.1</td>
<td align="left">sample_26</td>
<td align="left">CF</td>
<td align="left">untreated</td>
<td align="left">mild</td>
<td align="left">CF.NO_MOD</td>
<td align="left">CF.NO_MOD.M</td>
<td align="right">1</td>
<td align="right">5.049315</td>
<td align="left">M</td>
<td align="right">2040</td>
</tr>
</tbody>
</table>
</div>
<div id="prepare-figure-panels" class="section level1">
<h1>Prepare figure panels</h1>
<pre class="r"><code># HSP+ B, CD4 T-IFN

props &lt;- getTransformedProps(clusters = seu$ann_level_3[!str_detect(seu$ann_level_3, &quot;macro&quot;)],
                             sample = seu$sample.id[!str_detect(seu$ann_level_3, &quot;macro&quot;)], transform=&quot;asin&quot;)

props$Proportions %&gt;% data.frame %&gt;%
  left_join(info,
            by = c(&quot;sample&quot; = &quot;sample.id&quot;)) %&gt;%
  dplyr::filter(Group_severity %in% c(&quot;CF.NO_MOD.S&quot;, &quot;CF.NO_MOD.M&quot;),
                clusters %in% c(&quot;CD4 T-IFN&quot;, 
                                &quot;HSP+ B cells&quot;)) -&gt; dat

sig_names &lt;- as_labeller(
     c(&quot;CD4 T-IFN&quot; = &quot;CD4 T-IFN&quot;,
       &quot;HSP+ B cells&quot; = &quot;HSP+ B cells&quot;))

pal &lt;- RColorBrewer::brewer.pal(8, &quot;Accent&quot;)[1:2]
names(pal) &lt;- c(&quot;CF.NO_MOD.S&quot;, &quot;CF.NO_MOD.M&quot;)
  
dat %&gt;%
ggplot(aes(x = Group_severity,
                y = Freq,
                colour = Group_severity)) +
  geom_jitter(stat = &quot;identity&quot;,
              width = 0.15,
              size = 1) +
  theme_classic() +
  theme(axis.text.x = element_blank(),
          axis.ticks.x = element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(7),
          legend.position = &quot;bottom&quot;,
          legend.direction = &quot;horizontal&quot;,
          strip.text = element_text(size = 8)) +
  labs(x = &quot;Group&quot;, y = &quot;Proportion&quot;,
       colour = &quot;Group&quot;) +
  facet_wrap(~clusters, scales = &quot;free_y&quot;, ncol = 4,
             labeller = sig_names) +
  stat_summary(
    geom = &quot;point&quot;,
    fun.y = &quot;mean&quot;,
    col = &quot;black&quot;,
    shape = &quot;_&quot;,
    size = 10) +
  scale_color_manual(values = pal) -&gt; p1

p1</code></pre>
<p><img src="figure/99.0_Figure_4.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>seu@meta.data %&gt;% 
    data.frame %&gt;% 
    dplyr::select(ann_level_2) %&gt;%
    dplyr::filter(str_detect(ann_level_2, &quot;macro&quot;)) %&gt;%
    group_by(ann_level_2) %&gt;% 
    count() %&gt;%
    janitor::adorn_totals(name = &quot;macrophages&quot;) %&gt;%
    arrange(-n) %&gt;%
    dplyr::rename(cell = ann_level_2) -&gt; cell_freq

cell_freq</code></pre>
<pre><code>                   cell      n
            macrophages 165209
         macro-alveolar  52563
            macro-IFI27  24864
              macro-CCL  21246
 macro-monocyte-derived  13461
           macro-APOC2+  13354
            macro-lipid  12452
             macro-IGF1   8229
    macro-proliferating   6821
               macro-MT   4037
     macro-interstitial   3412
                macro-T   2722
              macro-IFN   2048</code></pre>
<pre class="r"><code>files &lt;- list.files(here(&quot;data/intermediate_objects&quot;),
            pattern = &quot;macro.*CF_samples&quot;, 
            full.names = TRUE) 
 
cutoff &lt;- 0.05 
cont_name &lt;- &quot;CF.NO_MOD.SvCF.NO_MOD.M&quot;  
lfc_cutoff &lt;- 0
suffix &lt;- &quot;.CF_samples.fit.rds&quot;
  
get_deg_data(files, cont_name, cell_freq, treat_lfc = lfc_cutoff,
             suffix = suffix) -&gt; dat</code></pre>
<pre><code>Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. </code></pre>
<pre class="r"><code>genes &lt;- c(&quot;ARHGEF5&quot;,
           &quot;GPD1&quot;,
           &quot;ITGA1&quot;,
           &quot;SIRPB1&quot;,
           &quot;TLCD4&quot;,
           &quot;GBP3&quot;,
           &quot;CLIC2&quot;,
           &quot;LILRB2&quot;,
           &quot;MMP14&quot;,
           &quot;SLC28A3&quot;,
           &quot;SLC46A1&quot;,
           &quot;ARPIN&quot;,
           &quot;CTSK&quot;,
           &quot;VAMP5&quot;)

dat %&gt;%
  dplyr::select(gene, cell, logFC) %&gt;%
  distinct() %&gt;%
  dplyr::filter(gene %in% sort(genes)) %&gt;%
  pivot_wider(
    names_from = cell,        # Column whose values become new column names
    values_from = logFC,
    values_fill = list(logFC = NA)) %&gt;%
   arrange(across(all_of(cell_freq$cell[cell_freq$cell %in% .$cell]))) %&gt;%
  column_to_rownames(var = &quot;gene&quot;) -&gt; dat_lfc</code></pre>
<pre class="r"><code>pal_dt &lt;- c(paletteer::paletteer_d(&quot;RColorBrewer::Set1&quot;)[2:1], &quot;grey&quot;) 

col_fun &lt;- circlize::colorRamp2(seq(0, 100000, length.out = 9), 
                                (RColorBrewer::brewer.pal(9, &quot;PiYG&quot;)))
col_split &lt;- c(rep(&quot;aggregate&quot;, 1), rep(&quot;sub-type&quot;, ncol(dat_lfc) - 1))

col_lfc_fun &lt;- circlize::colorRamp2(seq(-2, 2, length.out = 3), 
                                    c(pal_dt[1], &quot;white&quot;, pal_dt[2]))

ComplexHeatmap::HeatmapAnnotation(df = cell_freq %&gt;%
                                    dplyr::filter(cell %in% colnames(dat_lfc)) %&gt;%
                                    column_to_rownames(var = &quot;cell&quot;) %&gt;%
                                    dplyr::rename(`No. cells` = n),
                                  which = &quot;column&quot;,
                                  show_annotation_name = FALSE,
                                  col = list(`No. cells` = col_fun),
                                  annotation_legend_param = list(
                                    `No. cells` = list(direction = &quot;vertical&quot;,
                                                       labels_gp = grid::gpar(fontsize = 7)))) -&gt; col_ann

ComplexHeatmap::Heatmap(dat_lfc[order(rowSums(dat_lfc, na.rm = TRUE),
                                      decreasing = TRUE),], 
                        name = &quot;logFC&quot;,
                        column_split = col_split,
                        column_title = NULL,
                        cluster_rows = FALSE,
                        cluster_columns = FALSE,
                        rect_gp = grid::gpar(col = &quot;white&quot;, lwd = 1),
                        row_names_gp = grid::gpar(fontsize = 7),
                        column_names_gp = grid::gpar(fontsize = 8),
                        column_names_rot = 90,
                        top_annotation = col_ann,
                        col = col_lfc_fun,
                        #right_annotation = row_ann,
                        heatmap_legend_param = list(direction = &quot;vertical&quot;,
                                                    labels_gp = grid::gpar(fontsize = 7))) -&gt; plot_lfc

ComplexHeatmap::draw(as(list(plot_lfc), &quot;HeatmapList&quot;), 
                     heatmap_legend_side = &quot;right&quot;, 
                     annotation_legend_side = &quot;right&quot;,
                     merge_legends = TRUE) -&gt; plot_lfc</code></pre>
<p><img src="figure/99.0_Figure_4.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_lfc</code></pre>
<p><img src="figure/99.0_Figure_4.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>bind_rows(lapply(files, function(f){
  deg_results &lt;- readRDS(f)
  lrt &lt;- glmLRT(deg_results$fit, 
                contrast = deg_results$contr[,cont_name])
  tmp &lt;- cbind(summary(decideTests(lrt, p.value = 0.05)) %&gt;% data.frame,
                    cell = unlist(str_split(str_remove(f, suffix), &quot;/&quot;))[8])
  tmp
})) -&gt; dat_deg</code></pre>
<pre class="r"><code>bind_rows(lapply(files, function(f){
  deg_results &lt;- readRDS(f)
  lrt &lt;- glmLRT(deg_results$fit, 
                contrast = deg_results$contr[,cont_name])
  tmp &lt;- cbind(summary(decideTests(lrt, p.value = cutoff)) %&gt;% data.frame,
                    cell = unlist(str_split(str_remove(f, suffix), &quot;/&quot;))[8])
  tmp
})) -&gt; dat_deg</code></pre>
<pre class="r"><code>dat_deg %&gt;% 
  left_join(cell_freq) -&gt; dat_deg

dat_deg %&gt;%
  dplyr::filter(Var1 != &quot;NotSig&quot;) %&gt;%
  ggplot(aes(x = fct_reorder(cell, -n), y = Freq, fill = Var1)) +
  geom_col(position = &quot;dodge&quot;) +
  scale_fill_manual(values = pal_dt) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 30,
                                   hjust = 1,
                                   vjust = 1,
                                   size = 7),
        legend.position = &quot;top&quot;) +
  geom_text(aes(label = Freq), 
            position = position_dodge(width = 0.9),
            vjust = -0.5,
            size = 2.5) +
  labs(x = &quot;Cell Type&quot;,
       y = &quot;No. DEG (FDR &lt; 0.05)&quot;,
       fill = &quot;Direction&quot;) -&gt; deg_barplot

deg_barplot</code></pre>
<p><img src="figure/99.0_Figure_4.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-4" class="section level1">
<h1>Figure 4</h1>
<pre class="r"><code>layout &lt;- &quot;
ABB
CCC
&quot;

wrap_elements(p1 + theme(legend.direction = &quot;vertical&quot;)) +
(wrap_elements(deg_barplot + theme(axis.title.x = element_blank(),
                                   legend.text = element_text(size = 8)))) +
  #wrap_elements(volc_plot + theme(strip.text = element_text(size = 7))) +
  wrap_elements(grid::grid.grabExpr(ComplexHeatmap::draw(plot_lfc))) +
  plot_layout(design = layout) +
  plot_annotation(tag_levels = &quot;A&quot;)  &amp;
  theme(plot.tag = element_text(size = 16,
                                face = &quot;bold&quot;,
                                family = &quot;arial&quot;))</code></pre>
<p><img src="figure/99.0_Figure_4.Rmd/unnamed-chunk-13-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="supplementary-figure-x" class="section level1">
<h1>Supplementary Figure X</h1>
<pre class="r"><code>dat %&gt;%
  dplyr::select(gene, cell, logFC) %&gt;%
  distinct() %&gt;%
  pivot_wider(
    names_from = cell,        # Column whose values become new column names
    values_from = logFC,
    values_fill = list(logFC = NA)) %&gt;%
   arrange(across(all_of(cell_freq$cell[cell_freq$cell %in% .$cell]))) %&gt;%
  column_to_rownames(var = &quot;gene&quot;) -&gt; dat_lfc_supp</code></pre>
<pre class="r"><code>col_fun &lt;- circlize::colorRamp2(seq(0, 100000, length.out = 9), 
                                (RColorBrewer::brewer.pal(9, &quot;PiYG&quot;)))
col_split &lt;- c(rep(&quot;aggregate&quot;, 1), rep(&quot;sub-type&quot;, ncol(dat_lfc_supp) - 1))

col_lfc_fun &lt;- circlize::colorRamp2(seq(-2, 2, length.out = 3), 
                                    c(pal_dt[1], &quot;white&quot;, pal_dt[2]))

ComplexHeatmap::HeatmapAnnotation(df = cell_freq %&gt;%
                                    dplyr::filter(cell %in% colnames(dat_lfc_supp)) %&gt;%
                                    column_to_rownames(var = &quot;cell&quot;) %&gt;%
                                    dplyr::rename(`No. cells` = n),
                                  which = &quot;column&quot;,
                                  show_annotation_name = FALSE,
                                  col = list(`No. cells` = col_fun),
                                  annotation_legend_param = list(
                                    `No. cells` = list(direction = &quot;vertical&quot;,
                                                       labels_gp = grid::gpar(fontsize = 7)))) -&gt; col_ann

ComplexHeatmap::HeatmapAnnotation(df = data.frame(multiple = (rowSums(!is.na(dat_lfc_supp)) &gt; 1)),
                                  which = &quot;row&quot;,
                                  col = list(multiple = c(&quot;FALSE&quot; = &quot;#fdcce5&quot;,&quot;TRUE&quot; = &quot;#8bd3c7&quot;)),
                                  annotation_legend_param = list(
                                    multiple = list(direction = &quot;vertical&quot;,
                                                    ncol = 1))) -&gt; row_ann

ComplexHeatmap::Heatmap(dat_lfc_supp, 
                        name = &quot;logFC&quot;,
                        column_split = col_split,
                        column_title = NULL,
                        cluster_rows = FALSE,
                        cluster_columns = FALSE,
                        rect_gp = grid::gpar(col = &quot;white&quot;, lwd = 1),
                        row_names_gp = grid::gpar(fontsize = 7),
                        column_names_gp = grid::gpar(fontsize = 8),
                        column_names_rot = 90,
                        top_annotation = col_ann,
                        col = col_lfc_fun,
                        right_annotation = row_ann,
                        heatmap_legend_param = list(direction = &quot;vertical&quot;,
                                                    labels_gp = grid::gpar(fontsize = 7))) -&gt; plot_lfc

ComplexHeatmap::draw(as(list(plot_lfc), &quot;HeatmapList&quot;), 
                     heatmap_legend_side = &quot;right&quot;, 
                     annotation_legend_side = &quot;right&quot;,
                     merge_legends = TRUE) -&gt; plot_lfc_supp</code></pre>
<p><img src="figure/99.0_Figure_4.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_lfc_supp</code></pre>
<p><img src="figure/99.0_Figure_4.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>get_deg_data(files, cont_name, cell_freq, treat_lfc = lfc_cutoff,
             suffix = suffix, cutoff = 1) -&gt; dat_all</code></pre>
<pre><code>Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. 
Zero log2-FC threshold detected. Switch to glmLRT() instead. </code></pre>
<pre class="r"><code>dat_all %&gt;% 
  left_join(cell_freq) %&gt;%
  mutate(Direction = as.factor(ifelse(sig == -1, &quot;Down&quot;,
                                      ifelse(sig == 1, &quot;Up&quot;, &quot;N.S.&quot;))),
         cell = fct_reorder(cell, -n)) -&gt; dat_all

ggplot(dat_all, aes(x = logFC, y = -log10(FDR), colour = Direction)) + 
  geom_point(size = 0.5) + 
  facet_wrap(~cell, ncol = 4) + 
  theme_classic() +
  scale_color_manual(values = pal_dt[c(1,3,2)]) +
  ggrepel::geom_text_repel(data = dat_all[dat_all$sig != 0,], 
                           aes(label = gene), size = 2) -&gt; volc_plot

volc_plot</code></pre>
<p><img src="figure/99.0_Figure_4.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<!-- ```{r, fig.asp=1} -->
<!-- volc_plot <- draw_treat_volcano_plot(cell = "macrophages", -->
<!--                                      suffix = suffix, -->
<!--                                      cutoff = cutoff, -->
<!--                                      lfc_cutoff = lfc_cutoff) -->
<!-- volc_plot -->
<!-- ``` -->
<!-- ```{r, fig.asp=1, fig.width=8} -->
<!-- #genes <- c("LILRB2", "CLIC2", "MMP14", "SLC46A1", "CTSK", "VAMP5", -->
<!-- #          "SLC28A3", "ITGA1", "TLCD4", "GBP3", "GPD1", "ARHGEF5", "SIRPB1") -->
<!-- genes <- c("LILRB2", "CLIC2", "MMP14", "CTSK", "ITGA1", "SIRPB1") -->
<!-- bind_rows(lapply(files[c(8,1:7)], function(f){ -->
<!--   deg_results <- readRDS(f) -->
<!--   deg_results <- readRDS(f) -->
<!--   norm_counts <- cpm(deg_results$adj$normalizedCounts, log = TRUE) %>%  -->
<!--     data.frame %>% -->
<!--     rownames_to_column(var = "gene") %>% -->
<!--     pivot_longer(-gene,  -->
<!--                  names_to = "sample",  -->
<!--                  values_to = "logCPM") %>% -->
<!--     left_join(deg_results$fit$samples[,"group", drop = FALSE] %>% -->
<!--                 data.frame %>% -->
<!--                 rownames_to_column(var = "sample")) %>% -->
<!--     mutate(cell = unlist(str_split(str_remove(f, suffix), "/"))[8]) %>% -->
<!--   dplyr::filter(gene %in% genes) %>% -->
<!--   dplyr::filter(group %in% str_split(cont_name, "v")[[1]]) -->
<!-- })) -> dat -->
<!-- lapply(unique(dat$gene), function(gene){ -->
<!--   ggplot(dat[dat$gene == gene,],  -->
<!--          aes(x = group, y = logCPM, colour = group)) + -->
<!--     # geom_violin() + -->
<!--     geom_jitter(width = 0.15, -->
<!--                 size = 1) + -->
<!--     stat_summary(geom = "point", -->
<!--                  fun.y = "mean", -->
<!--                  col = "black", -->
<!--                  shape = "_", -->
<!--                  size = 10) + -->
<!--     facet_wrap(vars(cell),   -->
<!--                ncol = length(unique(dat$cell))) + -->
<!--     theme_classic() + -->
<!--     theme(axis.text.x = element_blank(), -->
<!--           axis.ticks.x = element_blank(), -->
<!--           axis.title.x = element_blank(), -->
<!--           axis.text.y = element_text(7), -->
<!--           legend.position = "bottom", -->
<!--           legend.direction = "horizontal", -->
<!--           strip.text = element_text(size = 8)) + -->
<!--     labs(colour = "Group", -->
<!--          y = gene) + -->
<!--     scale_color_manual(values = pal) -->
<!-- }) -> gene_plot -->
<!-- wrap_plots(gene_plot, nrow = length(unique(dat$gene)), byrow = TRUE) + -->
<!--   plot_layout(guides = "collect") & -->
<!--   theme(legend.position = "bottom") -->
<!-- ``` -->
<!-- ```{r, fig.asp=2, fig.width=5} -->
<!-- ggplot(dat,  -->
<!--          aes(x = gene, y = logCPM, fill = group)) + -->
<!--      geom_boxplot(position="dodge") + -->
<!--     facet_wrap(vars(cell),   -->
<!--                nrow = length(unique(dat$cell))) + -->
<!--     theme_classic() + -->
<!--     theme(axis.text = element_text(7), -->
<!--           legend.position = "right", -->
<!--           legend.direction = "vertical", -->
<!--           strip.text = element_text(size = 8)) + -->
<!--     scale_fill_brewer(palette = "Accent") + -->
<!--   labs(fill = "Group", -->
<!--        x = "Gene", -->
<!--        y = "log2 CPM") -> top_gene_plot -->
<!-- top_gene_plot -->
<!-- ``` -->
<!-- ```{r, fig.width=8, fig.asp=0.6} -->
<!-- col_fun <- circlize::colorRamp2(seq(-4, 4, length.out = 11), -->
<!--                                 rev(RColorBrewer::brewer.pal(11, "PiYG"))) -->
<!-- # col_fun_list <- setNames(replicate(length(uniq_genes), col_fun, simplify = FALSE), -->
<!-- #                          names(uniq_genes)) -->
<!-- sample_ann <- dat %>%  -->
<!--   dplyr::select(sample, group) %>%  -->
<!--   distinct %>%  -->
<!--   data.frame() %>% -->
<!--   arrange(group) %>% -->
<!--   column_to_rownames(var = "sample")  -->
<!-- lapply(unique(dat$cell), function(cell_type){ -->
<!--   dat %>% -->
<!--     dplyr::filter(cell == cell_type) %>% -->
<!--     dplyr::select(-group, -cell) %>% -->
<!--     pivot_wider(names_from = sample,        # Column whose values become new column names -->
<!--     values_from = logCPM) %>% -->
<!--     column_to_rownames(var = "gene") -->
<!-- }) -> logCPM -->
<!-- names(logCPM) <- unique(dat$cell) -->
<!-- col_ann <- ComplexHeatmap::HeatmapAnnotation(df = sample_ann, -->
<!--                                         #col = col_fun_list, -->
<!--                                         #show_legend = c(TRUE, rep(FALSE, 4)), -->
<!--                                         which = "column", -->
<!--                                         # annotation_legend_param = list( -->
<!--                                         #   macrophages = list( -->
<!--                                         #     title = "log2(FC)" -->
<!--                                         #   )), -->
<!--                                         annotation_name_gp = grid::gpar(fontsize = 8) -->
<!--                                         #annotation_name_side = "top" -->
<!--                                         ) -->
<!-- col_split <- sample_ann$group -->
<!-- # row_split <- unlist(sapply(seq_along(uniq_genes), function(i){ -->
<!-- #   rep(names(uniq_genes)[i], length(uniq_genes[[i]])) -->
<!-- # })) -->
<!-- col_fun_main <- circlize::colorRamp2(seq(-4, 4, length.out = 11), -->
<!--                                      rev(RColorBrewer::brewer.pal(11, "RdBu"))) -->
<!-- lapply(seq_along(logCPM), function(i){ -->
<!--   m <- match(rownames(sample_ann), colnames(logCPM[[i]])) -->
<!--   ComplexHeatmap::Heatmap(t(scale(t(logCPM[[i]][, m[!is.na(m)]]))), -->
<!--                           name = "scaled exp.", -->
<!--                           cluster_rows = TRUE, -->
<!--                           cluster_columns = TRUE, -->
<!--                           rect_gp = grid::gpar(col = "white", lwd = 1), -->
<!--                           row_names_gp = grid::gpar(fontsize = 8), -->
<!--                           column_names_gp = grid::gpar(fontsize = 8), -->
<!-- #                          top_annotation = col_ann[!is.na(m)], -->
<!--                           column_split = col_split[!is.na(m)], -->
<!--                           cluster_column_slices = FALSE, -->
<!--                           #row_split = row_split, -->
<!--                           row_title_gp = grid::gpar(fontsize = 8), -->
<!--                           col = col_fun_main, -->
<!--                           row_title = NULL, -->
<!--                          # column_title = names(logCPM)[i] -->
<!--                           ) -->
<!-- }) -> foo -->
<!-- #as(foo, "HeatmapList") -->
<!-- library(ComplexHeatmap) -->
<!-- ht_list = foo[[1]] %v% foo[[2]] %v% foo[[4]] %v% foo[[5]]  -->
<!-- ht_list -->
<!-- ``` -->
<!-- ```{r} -->
<!-- Hs.c2.all <- convert_gmt_to_list(here("data/c2.all.v2024.1.Hs.entrez.gmt")) -->
<!-- Hs.h.all <- convert_gmt_to_list(here("data/h.all.v2024.1.Hs.entrez.gmt")) -->
<!-- Hs.c5.all <- convert_gmt_to_list(here("data/c5.all.v2024.1.Hs.entrez.gmt")) -->
<!-- fibrosis <- create_custom_gene_lists_from_file(here("data/fibrosis_gene_sets.csv")) -->
<!-- # add fibrosis sets from REACTOME and WIKIPATHWAYS -->
<!-- fibrosis <- c(lapply(fibrosis, function(l) l[!is.na(l)]), -->
<!--               Hs.c2.all[str_detect(names(Hs.c2.all), "FIBROSIS")]) -->
<!-- gene_sets_list <- list(HALLMARK = Hs.h.all, -->
<!--                        GO = Hs.c5.all, -->
<!--                        REACTOME = Hs.c2.all[str_detect(names(Hs.c2.all), "REACTOME")], -->
<!--                        WP = Hs.c2.all[str_detect(names(Hs.c2.all), "^WP")], -->
<!--                        FIBROSIS = fibrosis) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #library(gggsea) -->
<!-- f <- files[8] -->
<!-- deg_results <- readRDS(f) -->
<!-- lrt <- glmLRT(deg_results$fit, -->
<!--               contrast = deg_results$contr[, cont_name]) -->
<!-- gns <- AnnotationDbi::mapIds(org.Hs.eg.db, -->
<!--                              keys = rownames(lrt), -->
<!--                              column = c("ENTREZID"), -->
<!--                              keytype = "SYMBOL", -->
<!--                              multiVals = "first") -->
<!-- statistic <- sign(lrt$table$logFC) * sqrt(lrt$table$LR) -->
<!-- names(statistic) <- unname(gns[rownames(lrt)]) -->
<!-- barcodeplot(statistic, -->
<!--             ids2indices(gene_sets_list$FIBROSIS["fibrosis_transcriptional_signature"], -->
<!--             names(statistic))[[1]]) -->
<!-- ``` -->
</div>
<div id="session-info" class="section level1">
<h1>Session info</h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.3 (2024-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] readxl_1.4.3                ggh4x_0.2.8                
 [3] dsb_1.0.3                   paletteer_1.6.0            
 [5] tidyHeatmap_1.8.1           speckle_1.2.0              
 [7] glue_1.7.0                  org.Hs.eg.db_3.18.0        
 [9] AnnotationDbi_1.64.1        patchwork_1.2.0            
[11] clustree_0.5.1              ggraph_2.2.0               
[13] here_1.0.1                  dittoSeq_1.14.2            
[15] glmGamPoi_1.14.3            SeuratObject_4.1.4         
[17] Seurat_4.4.0                lubridate_1.9.3            
[19] forcats_1.0.0               stringr_1.5.1              
[21] dplyr_1.1.4                 purrr_1.0.2                
[23] readr_2.1.5                 tidyr_1.3.1                
[25] tibble_3.2.1                ggplot2_3.5.0              
[27] tidyverse_2.0.0             edgeR_4.0.15               
[29] limma_3.58.1                SingleCellExperiment_1.24.0
[31] SummarizedExperiment_1.32.0 Biobase_2.62.0             
[33] GenomicRanges_1.54.1        GenomeInfoDb_1.38.6        
[35] IRanges_2.36.0              S4Vectors_0.40.2           
[37] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      
[39] matrixStats_1.2.0           workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] fs_1.6.3                spatstat.sparse_3.0-3   bitops_1.0-7           
  [4] httr_1.4.7              RColorBrewer_1.1-3      doParallel_1.0.17      
  [7] tools_4.3.3             sctransform_0.4.1       utf8_1.2.4             
 [10] R6_2.5.1                lazyeval_0.2.2          uwot_0.1.16            
 [13] GetoptLong_1.0.5        withr_3.0.0             sp_2.1-3               
 [16] gridExtra_2.3           progressr_0.14.0        cli_3.6.2              
 [19] Cairo_1.6-2             spatstat.explore_3.2-6  prismatic_1.1.1        
 [22] labeling_0.4.3          sass_0.4.8              spatstat.data_3.0-4    
 [25] ggridges_0.5.6          pbapply_1.7-2           parallelly_1.37.0      
 [28] rstudioapi_0.15.0       RSQLite_2.3.5           generics_0.1.3         
 [31] shape_1.4.6             ica_1.0-3               spatstat.random_3.2-2  
 [34] dendextend_1.17.1       Matrix_1.6-5            fansi_1.0.6            
 [37] abind_1.4-5             lifecycle_1.0.4         whisker_0.4.1          
 [40] yaml_2.3.8              snakecase_0.11.1        SparseArray_1.2.4      
 [43] Rtsne_0.17              grid_4.3.3              blob_1.2.4             
 [46] promises_1.2.1          crayon_1.5.2            miniUI_0.1.1.1         
 [49] lattice_0.22-5          cowplot_1.1.3           KEGGREST_1.42.0        
 [52] pillar_1.9.0            knitr_1.45              ComplexHeatmap_2.18.0  
 [55] rjson_0.2.21            future.apply_1.11.1     codetools_0.2-19       
 [58] leiden_0.4.3.1          getPass_0.2-4           data.table_1.15.0      
 [61] vctrs_0.6.5             png_0.1-8               cellranger_1.1.0       
 [64] gtable_0.3.4            rematch2_2.1.2          cachem_1.0.8           
 [67] xfun_0.42               S4Arrays_1.2.0          mime_0.12              
 [70] tidygraph_1.3.1         survival_3.7-0          pheatmap_1.0.12        
 [73] iterators_1.0.14        statmod_1.5.0           ellipsis_0.3.2         
 [76] fitdistrplus_1.1-11     ROCR_1.0-11             nlme_3.1-164           
 [79] bit64_4.0.5             RcppAnnoy_0.0.22        rprojroot_2.0.4        
 [82] bslib_0.6.1             irlba_2.3.5.1           KernSmooth_2.23-24     
 [85] colorspace_2.1-0        DBI_1.2.1               tidyselect_1.2.0       
 [88] processx_3.8.3          bit_4.0.5               compiler_4.3.3         
 [91] git2r_0.33.0            DelayedArray_0.28.0     plotly_4.10.4          
 [94] scales_1.3.0            lmtest_0.9-40           callr_3.7.3            
 [97] digest_0.6.34           goftest_1.2-3           spatstat.utils_3.0-4   
[100] rmarkdown_2.25          XVector_0.42.0          htmltools_0.5.7        
[103] pkgconfig_2.0.3         highr_0.10              fastmap_1.1.1          
[106] rlang_1.1.3             GlobalOptions_0.1.2     htmlwidgets_1.6.4      
[109] shiny_1.8.0             farver_2.1.1            jquerylib_0.1.4        
[112] zoo_1.8-12              jsonlite_1.8.8          mclust_6.1             
[115] RCurl_1.98-1.14         magrittr_2.0.3          GenomeInfoDbData_1.2.11
[118] munsell_0.5.0           Rcpp_1.0.12             viridis_0.6.5          
[121] reticulate_1.35.0       stringi_1.8.3           zlibbioc_1.48.0        
[124] MASS_7.3-60.0.1         plyr_1.8.9              parallel_4.3.3         
[127] listenv_0.9.1           ggrepel_0.9.5           deldir_2.0-2           
[130] Biostrings_2.70.2       graphlayouts_1.1.0      splines_4.3.3          
[133] tensor_1.5              hms_1.1.3               circlize_0.4.15        
[136] locfit_1.5-9.8          ps_1.7.6                igraph_2.0.1.1         
[139] spatstat.geom_3.2-8     reshape2_1.4.4          evaluate_0.23          
[142] renv_1.0.3              BiocManager_1.30.22     tzdb_0.4.0             
[145] foreach_1.5.2           tweenr_2.0.3            httpuv_1.6.14          
[148] RANN_2.6.1              polyclip_1.10-6         future_1.33.1          
[151] clue_0.3-65             scattermore_1.2         ggforce_0.4.2          
[154] janitor_2.2.0           xtable_1.8-4            later_1.3.2            
[157] viridisLite_0.4.2       memoise_2.0.1           cluster_2.1.6          
[160] timechange_0.3.0        globals_0.16.2         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
