<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2024-04-09" />

<title>Inflammation of Paediatric Pulmonary Diseases</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-inflammation-CITEseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Inflammation of Paediatric Pulmonary
Diseases</h1>
<h3 class="subtitle">Differential expression analysis of T cells</h3>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">April 09, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-04-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>paed-inflammation-CITEseq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmospaedinflammationciteseqtree652af576e96f14417ac35b1454b69644e71602c3targetblank652af57a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/tree/652af576e96f14417ac35b1454b69644e71602c3" target="_blank">652af57</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmospaedinflammationciteseqtree652af576e96f14417ac35b1454b69644e71602c3targetblank652af57a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/tree/652af576e96f14417ac35b1454b69644e71602c3" target="_blank">652af57</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/C133_Neeland_batch0/
    Ignored:    data/C133_Neeland_batch1/
    Ignored:    data/C133_Neeland_batch2/
    Ignored:    data/C133_Neeland_batch3/
    Ignored:    data/C133_Neeland_batch4/
    Ignored:    data/C133_Neeland_batch5/
    Ignored:    data/C133_Neeland_batch6/
    Ignored:    data/C133_Neeland_merged/
    Ignored:    data/cluster_annotations/
    Ignored:    renv/library/
    Ignored:    renv/staging/

Untracked files:
    Untracked:  analysis/11.0_manual_annotations_t_cells_decontx.Rmd
    Untracked:  analysis/scar.Rmd
    Untracked:  analysis/scar.nb.html
    Untracked:  code/voomByGroup/
    Untracked:  dataC133_Neeland_mergedC133_Neeland_full_clean_decontx_annotated_diet.SEU.rds
    Untracked:  dataC133_Neeland_mergedC133_Neeland_full_clean_decontx_annotated_full.SEU.rds

Unstaged changes:
    Modified:   .DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/TNK-DGE.Rmd</code>) and HTML
(<code>docs/TNK-DGE.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-inflammation-citeseq/blob/652af576e96f14417ac35b1454b69644e71602c3/analysis/TNK-DGE.Rmd" target="_blank">652af57</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2024-04-09
</td>
<td>
wflow_publish("analysis/TNK-DGE.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>suppressPackageStartupMessages({
  library(BiocStyle)
  library(tidyverse)
  library(here)
  library(glue)
  library(Seurat)
  library(patchwork)
  library(paletteer)
  library(limma)
  library(edgeR)
  library(RUVSeq)
  library(scMerge)
  library(SingleCellExperiment)
  library(scater)
  library(tidyHeatmap)
  library(EGSEAdata)
  library(org.Hs.eg.db)
  library(TxDb.Hsapiens.UCSC.hg38.knownGene)
  library(missMethyl)
})</code></pre>
</div>
<div id="load-data" class="section level1">
<h1>Load Data</h1>
<pre class="r"><code>ambient &lt;- &quot;_decontx&quot;
file &lt;- here(glue(&quot;data&quot;,
            &quot;C133_Neeland_merged&quot;,
            &quot;C133_Neeland_full_clean{ambient}_annotated_diet.SEU.rds&quot;))

seu &lt;- readRDS(file)
seu</code></pre>
<pre><code>An object of class Seurat 
19973 features across 15511 samples within 1 assay 
Active assay: RNA (19973 features, 0 variable features)</code></pre>
</div>
<div id="create-pseudobulk-samples-by-cell-type-broad---modified"
class="section level1">
<h1>Create pseudobulk samples by cell type (Broad - modified)</h1>
<p>Use <em>cell type</em> and <em>sample</em> as our two factors; each
column of the output corresponds to one unique combination of these two
factors. Exclude the <em>macro-T</em> “cells” from the macrophages
pseudobulks because we know they are doublets. In this modified
pseudobulk CD8 and CD4 T-cells will be aggegated to form an overall
T-cell pseudobulk.</p>
<pre class="r"><code>out &lt;- here(&quot;data/C133_Neeland_all_annotated.PBCB-mod.SCE.rds&quot;)

seu@meta.data %&gt;%
  data.frame %&gt;%
  mutate(Status = ifelse(str_detect(Treatment, &quot;^treated&quot;), 
                         &quot;treated&quot;, 
                         Treatment),
         Group = ifelse(Status %in% c(&quot;treated&quot;, &quot;untreated&quot;), 
                        paste(Disease, 
                              toupper(substr(Status, 1, 1)),
                              toupper(substr(Severity, 1, 1)),
                              sep = &quot;.&quot;), 
                        Disease),
         Severity = tolower(Severity),
         Participant = strsplit2(sample.id, &quot;.&quot;, fixed = TRUE)[,1]) -&gt; seu@meta.data

sce &lt;- SingleCellExperiment(list(counts = seu[[&quot;RNA&quot;]]@counts),
                            colData = seu@meta.data)
sce$Broad_mod &lt;- ifelse(sce$Broad %in% c(&quot;CD4 T cells&quot;, &quot;CD8 T cells&quot;),
                        &quot;T-cell&quot;, as.character(sce$Broad))

if(!file.exists(out)){
  broadSummed &lt;- aggregateAcrossCells(sce, 
                                 id = colData(sce)[, c(&quot;Broad_mod&quot;, &quot;sample.id&quot;)])
  saveRDS(broadSummed, file = out)
  
} else {
  broadSummed &lt;- readRDS(file = out)
  
}

# pbCells &lt;- counts(broadSummed)
broadSummed</code></pre>
<pre><code>class: SingleCellExperiment 
dim: 19973 227 
metadata(0):
assays(1): counts
rownames(19973): A1BG A1BG-AS1 ... ZNRD2 ZRANB2-AS2
rowData names(0):
colnames: NULL
colData names(75): nCount_RNA nFeature_RNA ... sample.id ncells
reducedDimNames(0):
mainExpName: NULL
altExpNames(0):</code></pre>
<div id="no.-cells-per-cell-type" class="section level2">
<h2>No. cells per cell type</h2>
<pre class="r"><code>#yPB$samples %&gt;% 
colData(broadSummed) %&gt;%
  data.frame %&gt;%
  group_by(Broad_mod) %&gt;%
  summarise(total = sum(ncells)) %&gt;%
  ggplot(aes(x = fct_reorder(Broad_mod, total), 
             y = total, fill = Broad_mod)) +
  geom_col() + 
  geom_text(aes(label = total), vjust = -0.5, colour = &quot;black&quot;, size = 2.5) +
  scale_y_log10() +
  labs(x = &quot;Cell label&quot;,
       y = &quot;Log 10 No. cells&quot;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        legend.position = &quot;bottom&quot;) +
  geom_hline(yintercept = 1000, linetype = &quot;dashed&quot;) +
    NoLegend()</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="no.-cells-per-cell-type-per-sample" class="section level2">
<h2>No. cells per cell type per sample</h2>
<p>Apart from macrophages, most cell types do no have &gt;50 cells for
many samples, making DGE between groups difficult. CD4 and CD8 T-cells
have been aggregated into a single category to boost numbers for
analysis.</p>
<pre class="r"><code>#yPB$samples %&gt;% 
 colData(broadSummed) %&gt;%
  data.frame %&gt;%
  arrange(Group) %&gt;%
  ggplot(aes(x = fct_inorder(sample.id), 
             y = ncells, fill = Group)) +
  geom_col() + 
  scale_fill_brewer(palette = &quot;Set2&quot;) +
  scale_y_log10() +
  facet_wrap(~Broad_mod, ncol = 2) + 
  labs(x = &quot;Cell label&quot;,
       y = &quot;Log10 No. cells&quot;) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5,
                                   size = 8),
        legend.position = &quot;bottom&quot;) +
  geom_hline(yintercept = 50, linetype = &quot;dashed&quot;) +
  geom_hline(yintercept = 25, linetype = &quot;dotted&quot;)</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-6-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="data-preparation" class="section level1 tabset">
<h1 class="tabset">Data preparation</h1>
<div id="extract-cell-type" class="section level2">
<h2>Extract cell type</h2>
<p>Make a <code>DGElist</code> object.</p>
<pre class="r"><code>yPB &lt;- DGEList(counts = counts(broadSummed),
               samples = colData(broadSummed) %&gt;% data.frame)
dim(yPB)</code></pre>
<pre><code>[1] 19973   227</code></pre>
<p>Remove genes with zero counts in all samples.</p>
<pre class="r"><code>keep &lt;- rowSums(yPB$counts) &gt; 0 
yFlt &lt;- yPB[keep, ]
dim(yFlt)</code></pre>
<pre><code>[1] 18969   227</code></pre>
<p>Extract only the <em>T-cell</em> cells.</p>
<pre class="r"><code>cell &lt;- &quot;T-cell&quot;
ySub &lt;- yFlt[, yFlt$samples$Broad_mod == cell]
ySub$samples$Broad &lt;- ySub$samples$Broad_mod
dim(ySub)</code></pre>
<pre><code>[1] 18969    45</code></pre>
</div>
<div id="filter-samples-genes" class="section level2">
<h2>Filter samples &amp; genes</h2>
<p>Examine MDS plot for outlier samples.</p>
<pre class="r"><code>dims &lt;- list(c(1,2), c(2:3), c(3,4), c(4,5))
p &lt;- vector(&quot;list&quot;, length(dims))

for(i in 1:length(dims)){
  
  mds &lt;- limma::plotMDS(edgeR::cpm(ySub, 
                                   log = TRUE), 
                        gene.selection = &quot;common&quot;,
                        plot = FALSE, dim.plot = dims[[i]])
  
  data.frame(x = mds$x, 
             y = mds$y,
             sample = rownames(mds$distance.matrix.squared)) %&gt;%
    left_join(rownames_to_column(ySub$samples, var = &quot;sample&quot;)) %&gt;%
    mutate(Batch = as.factor(Batch)) -&gt; dat
  
  p[[i]] &lt;- ggplot(dat, aes(x = x, y = y, 
                            colour = Batch,
                            shape = Group))+
    geom_point(size = 3) +
    ggrepel::geom_text_repel(aes(label = sample.id),
                             size = 2) +
    labs(x = glue(&quot;Principal Component {dims[[i]][1]}&quot;),
         y = glue(&quot;Principal Component {dims[[i]][2]}&quot;),
         shape = &quot;Group&quot;,
         colour = &quot;Batch&quot;) +
    theme(legend.direction = &quot;horizontal&quot;,
          legend.text = element_text(size = 8),
          legend.title = element_text(size = 9),
          axis.text = element_text(size = 8),
          axis.title = element_text(size = 9))  +
    scale_colour_brewer(palette = &quot;Set1&quot;) -&gt; p[[i]]
}

wrap_plots(p, ncol = 2) + 
  plot_layout(guides = &quot;collect&quot;) &amp;
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Examine number of cells per sample. Identify outliers and
cross-reference with MDS plot. Determine a threshold for minimum number
of cells per sample.</p>
<pre class="r"><code>minCells &lt;- 50

ySub$samples %&gt;%
  ggplot(aes(x = sample.id, y = ncells, fill = ncells &gt; minCells)) +
  geom_col() +
  labs(fill = &quot;Keep&quot;) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  geom_hline(yintercept = minCells, linetype = &quot;dashed&quot;)</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Filter out samples with less than previously determined minimum
number of cells.</p>
<pre class="r"><code>ySub &lt;- ySub[, ySub$samples$ncells &gt; minCells]
dim(ySub)</code></pre>
<pre><code>[1] 18969    37</code></pre>
<p>Re-examine MDS plot.</p>
<pre class="r"><code>dims &lt;- list(c(1,2), c(2:3), c(3,4), c(4,5))
p &lt;- vector(&quot;list&quot;, length(dims))

for(i in 1:length(dims)){
  
  mds &lt;- limma::plotMDS(edgeR::cpm(ySub, 
                                   log = TRUE), 
                        gene.selection = &quot;common&quot;,
                        plot = FALSE, dim.plot = dims[[i]])
  
  data.frame(x = mds$x, 
             y = mds$y,
             sample = rownames(mds$distance.matrix.squared)) %&gt;%
    left_join(rownames_to_column(ySub$samples, var = &quot;sample&quot;)) %&gt;%
    mutate(Batch = as.factor(Batch)) -&gt; dat
  
  p[[i]] &lt;- ggplot(dat, aes(x = x, y = y, 
                            colour = Batch,
                            shape = Group))+
    geom_point(size = 3) +
    ggrepel::geom_text_repel(aes(label = sample.id),
                             size = 2) +
    labs(x = glue(&quot;Principal Component {dims[[i]][1]}&quot;),
         y = glue(&quot;Principal Component {dims[[i]][2]}&quot;),
         shape = &quot;Group&quot;,
         colour = &quot;Batch&quot;) +
    theme(legend.direction = &quot;horizontal&quot;,
          legend.text = element_text(size = 8),
          legend.title = element_text(size = 9),
          axis.text = element_text(size = 8),
          axis.title = element_text(size = 9))  +
    scale_colour_brewer(palette = &quot;Set1&quot;) -&gt; p[[i]]
}

wrap_plots(p, ncol = 2) + 
  plot_layout(guides = &quot;collect&quot;) &amp;
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Filter out genes with no ENTREZ IDs and very low expression.</p>
<p>Filter out genes with no ENTREZ IDs and very low expression.</p>
<pre class="r"><code>gns &lt;- AnnotationDbi::mapIds(org.Hs.eg.db,
                             keys = rownames(ySub),
                             column = c(&quot;ENTREZID&quot;),
                             keytype = &quot;SYMBOL&quot;,
                             multiVals = &quot;first&quot;)
keep &lt;- !is.na(gns)
ySub &lt;- ySub[keep,]

thresh &lt;- 2.5
m &lt;- rowMedians(edgeR::cpm(ySub$counts, log = TRUE))
plot(density(m))
abline(v = thresh, lty = 2)</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># filter out genes with low median expression
keep &lt;- m &gt; thresh
table(keep)</code></pre>
<pre><code>keep
FALSE  TRUE 
 5442  9725 </code></pre>
<pre class="r"><code>ySub &lt;- ySub[keep, ]
dim(ySub)</code></pre>
<pre><code>[1] 9725   37</code></pre>
</div>
<div id="examine-covariates" class="section level2">
<h2>Examine covariates</h2>
<p>Principal components analysis (PCA) allows us to mathematically
determine the sources of variation in the data. We can then investigate
whether these correlate with any of the specifed covariates.</p>
<p>Prepare the data.</p>
<pre class="r"><code>PCs &lt;- prcomp(t(edgeR::cpm(ySub$counts, log = TRUE)), 
              center = TRUE, retx = TRUE)
loadings = PCs$x # pc loadings


nGenes = nrow(ySub)
nSamples = ncol(ySub)

datTraits &lt;- ySub$samples %&gt;% dplyr::select(Batch, Disease, Treatment, 
                                            Severity, Age, Sex, ncells) %&gt;%
  mutate(Batch = factor(Batch),
         Disease = factor(Disease, 
                          labels = 1:length(unique(Disease))),
         Treatment = factor(Treatment, 
                            labels = 1:length(unique(Treatment))),
         Sex = factor(Sex, labels = length(unique(Sex))),
         Severity = factor(Severity, labels = length(unique(Severity)))) %&gt;%
  mutate(across(everything(), as.numeric))

moduleTraitCor &lt;- suppressWarnings(cor(loadings[, 1:min(10, nSamples)], 
                                       datTraits, use = &quot;p&quot;))
moduleTraitPvalue &lt;- WGCNA::corPvalueStudent(moduleTraitCor, (nSamples-2))

textMatrix &lt;- paste(signif(moduleTraitCor, 2), &quot;\n(&quot;, 
                    signif(moduleTraitPvalue, 1), &quot;)&quot;, sep = &quot;&quot;)
dim(textMatrix) &lt;- dim(moduleTraitCor)</code></pre>
<p>Output results.</p>
<pre class="r"><code>par(mfrow = c(2, 1))
plot(PCs, type=&quot;lines&quot;, main = cell) # scree plot

## Display the correlation values within a heatmap plot
par(cex=0.75, mar = c(3, 5, 2, 1))
WGCNA::labeledHeatmap(Matrix = t(moduleTraitCor),
                            xLabels = colnames(loadings)[1:min(10, nSamples)],
                            yLabels = names(datTraits),
                            colorLabels = FALSE,
                            colors = WGCNA::blueWhiteRed(6),
                            textMatrix = t(textMatrix),
                            setStdMargins = FALSE,
                            cex.text = 1,
                            zlim = c(-1,1),
                            main = paste0(&quot;PCA-trait relationships: Top &quot;, 
                                          min(10, nSamples), 
                                          &quot; PCs&quot;))</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="examine-data-properties" class="section level2">
<h2>Examine data properties</h2>
<p>Use <code>limma</code> with <code>voomByGroup</code> to assess the
mean-variance relationships within the different sample groups.</p>
<pre class="r"><code>source(here(&quot;code/voomByGroup/voomByGroup.R&quot;))

group &lt;- factor(ySub$samples$Group)
batch &lt;- factor(ySub$samples$Batch)
age &lt;- log2(ySub$samples$Age)
sex &lt;- factor(ySub$samples$Sex)

# model sample groups, taking into account batch, age and sex
design &lt;- model.matrix(~ 0 + group + batch + age + sex)
colnames(design)[1:length(levels(group))] &lt;- levels(group)

contr &lt;- makeContrasts(HvCF.U = Healthy - 0.5*(CF.U.M + CF.U.S),
                       HvCF = Healthy - 0.25*(CF.U.M + CF.U.S + CF.T.M + CF.T.S),
                       CF.TvCF.U = 0.5*(CF.T.M + CF.T.S) - 0.5*(CF.U.M + CF.U.S),
                       CF.U.MvCF.U.S = CF.U.M - CF.U.S,
                       CF.MvCF.S = 0.5*(CF.T.M + CF.U.M) - 0.5*(CF.T.M + CF.U.S),
                       levels = design)

y &lt;- DGEList(counts = ySub$counts)
y &lt;- calcNormFactors(y)

vbg &lt;- voomByGroup(y, design = design, group = group, plot = &quot;all&quot;)</code></pre>
<pre><code>Group:
1 CF.T.M 
2 CF.T.S 
3 CF.U.M 
4 CF.U.S 
5 Healthy </code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-18-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-18-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-18-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-18-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-18-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>Use <code>voomWithQualityWeights</code> to account for differences in
sample quality. Use the <code>duplicateCorrelation</code> function to
calculate the consensus correlation between replicate samples from the
same individuals.</p>
<pre class="r"><code>participant &lt;- factor(ySub$samples$Participant)

vqw &lt;- voomWithQualityWeights(y, design = design, plot = TRUE)</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corfit &lt;- duplicateCorrelation(vqw, design, block = participant)
corfit$consensus.correlation</code></pre>
<pre><code>[1] 0.107848</code></pre>
</div>
<div id="view-limma-results" class="section level2">
<h2>View <em>limma</em> results</h2>
<p>As the consensus correlation is high, include
<code>participant</code> as a random effect in the model to account for
the correlation between replicate samples from the same individual.</p>
<pre class="r"><code>vqw &lt;- voomWithQualityWeights(y, design, block = participant, 
                              correlation = corfit$consensus, plot = TRUE)</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fit &lt;- lmFit(vqw, design, block = participant, correlation = corfit$consensus)
fit2 &lt;- contrasts.fit(fit, contr)
fit2 &lt;- eBayes(fit2, robust = TRUE, trend = FALSE)

cutoff &lt;- 0.05
dt &lt;- decideTests(fit2, p.value = cutoff)
s &lt;- summary(dt)

pal &lt;- c(paletteer::paletteer_d(&quot;RColorBrewer::Set1&quot;)[1:2], &quot;grey&quot;) 
s[-2,] %&gt;% 
  data.frame %&gt;%
  dplyr::rename(Direction = Var1, name = Var2, value = Freq) %&gt;% 
  ggplot(aes(x = name, y = value, fill = Direction)) +
  geom_col(position = &quot;dodge&quot;) +
  geom_text(aes(label = value), 
            position = position_dodge(width = 0.9),
            vjust = -0.5,
            size = 3) +
  labs(y = glue(&quot;No. DGE (FDR &lt; {cutoff})&quot;),
       x = &quot;Contrast&quot;) +
  theme(axis.text.x = element_text(angle = 45,
                                   hjust = 1,
                                   vjust = 1)) +
      scale_fill_manual(values = pal)</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-20-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>if(!file.exists(here(&quot;data/Hs.c5.all.v7.1.entrez.rds&quot;)))
  download.file(&quot;https://bioinf.wehi.edu.au/MSigDB/v7.1/Hs.c5.all.v7.1.entrez.rds&quot;,
                here(&quot;data/Hs.c5.all.v7.1.entrez.rds&quot;))
Hs.c5.all &lt;- readRDS(here(&quot;data/Hs.c5.all.v7.1.entrez.rds&quot;))

if(!file.exists(here(&quot;data/Hs.c2.cp.reactome.v7.1.entrez.rds&quot;)))
  download.file(&quot;https://bioinf.wehi.edu.au/MSigDB/v7.1/Hs.c2.cp.reactome.v7.1.entrez.rds&quot;,
                here(&quot;data/Hs.c2.cp.reactome.v7.1.entrez.rds&quot;))
Hs.c2.reactome &lt;- readRDS(here(&quot;data/Hs.c2.cp.reactome.v7.1.entrez.rds&quot;))

if(!file.exists(here(&quot;data/Hs.h.all.v7.1.entrez.rds&quot;)))
  download.file(&quot;https://bioinf.wehi.edu.au/MSigDB/v7.1/Hs.h.all.v7.1.entrez.rds&quot;,
                here(&quot;data/Hs.h.all.v7.1.entrez.rds&quot;))
Hs.h.all &lt;- readRDS(here(&quot;data/Hs.h.all.v7.1.entrez.rds&quot;))

# get gene indices for all gene set genes
h.id &lt;- ids2indices(Hs.h.all, unname(gns[rownames(fit2)]))
c2.id &lt;- ids2indices(Hs.c2.reactome, unname(gns[rownames(fit2)]))
c5.id &lt;- ids2indices(Hs.c5.all, unname(gns[rownames(fit2)]))</code></pre>
<pre class="r"><code># translate gene SYMBOLS to entrez gene IDs 
gns &lt;- AnnotationDbi::mapIds(org.Hs.eg.db, 
                             keys = rownames(fit2), 
                             column = c(&quot;ENTREZID&quot;),
                             keytype = &quot;SYMBOL&quot;,
                             multiVals = &quot;first&quot;)

# get transcript lengths
txdb &lt;- TxDb.Hsapiens.UCSC.hg38.knownGene
txlen &lt;- transcriptLengths(txdb)
txlen %&gt;%
  dplyr::select(gene_id, tx_len) %&gt;%
  group_by(gene_id) %&gt;%
  summarise(max_len = max(tx_len)) %&gt;%
  dplyr::filter(gene_id %in% gns) -&gt; supertxlen

m &lt;- match(supertxlen$gene_id, gns)
all(supertxlen$gene_id == gns[m])</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>#library(missMethyl)

# outDir &lt;- here(&quot;output&quot;,&quot;dge_analysis&quot;)
# if(!dir.exists(outDir)) dir.create(outDir)
# cellDir &lt;- file.path(outDir, cell)
# if(!dir.exists(cellDir)) dir.create(cellDir)

p &lt;- vector(&quot;list&quot;, ncol(contr))

for(i in 1:(ncol(contr))){
  # lrt &lt;- glmLRT(fit, contrast = contr[,i])
  # topTags(lrt, n = Inf) %&gt;%
  #   data.frame -&gt; top
  top &lt;- topTable(fit2, coef = i, number = Inf)
  
  if(sum(top$adj.P.Val &lt; cutoff) &gt; 0){
    # top 500 DGE results
    # write.table(top[1:500, ], 
    #             file = file.path(cellDir, glue(&quot;{colnames(contr)[i]}.csv&quot;)),
    #             sep = &quot;,&quot;, quote = F, col.names = NA)
    
    # test DGEs for over-representation of fibrosis gene sets
    # read in fibrosis gene sets
    read.csv2(file = here(&quot;data/fibrosis_gene_sets.csv&quot;),
              header = TRUE, sep = &quot;,&quot;) %&gt;% 
      pivot_longer(cols = everything()) %&gt;%
      mutate(value = str_trim(value),
             entrez = unname(unlist(AnnotationDbi::mapIds(org.Hs.eg.db, 
                                                          keys = value, 
                                                          column = c(&quot;ENTREZID&quot;),
                                                          keytype = &quot;SYMBOL&quot;,
                                                          multiVals = &quot;first&quot;))[value])) %&gt;%
      dplyr::rename(&quot;symbol&quot; = &quot;value&quot;,
                    &quot;source&quot; = &quot;name&quot;) -&gt; fibrosis_table
      
      fibrosis_list &lt;- split(fibrosis_table$entrez, fibrosis_table$source)
    
    deg &lt;- rownames(top)[top$adj.P.Val &lt; cutoff]
    gsa &lt;- topGSA(gsaseq(unname(gns[deg]),
                         universe = unname(gns),
                         collection = fibrosis_list,
                         plot.bias = FALSE,
                         gene.length = supertxlen$max_len[match(unname(gns), 
                                                                supertxlen$gene_id)]))
    
    # get gene indices for all fibrosis gene set genes
    f.id &lt;- ids2indices(fibrosis_list, unname(gns[rownames(fit2)]))
    
    # run camera competitive gene set test  
    # use signed likelihood ratio test statistic as recommended by GS here: https://support.bioconductor.org/p/112937/
    statistic &lt;- fit2$t[,i]
    f.cam &lt;- cameraPR(statistic, f.id)
    
    # write fibrosis over-representation results to file
    # write.table(gsa %&gt;%
    #               data.frame %&gt;%
    #               rownames_to_column(var = &quot;Set&quot;), 
    #             file = file.path(cellDir, glue(&quot;fibrosis.ORA.{colnames(contr)[i]}.csv&quot;)),
    #             sep = &quot;,&quot;, quote = F, col.names = NA)
    # 
    # # write fibrosis GSEA results to file
    # write.table(f.cam, 
    #             file = file.path(cellDir, glue(&quot;fibrosis.GSEA.{colnames(contr)[i]}.csv&quot;)),
    #             sep = &quot;,&quot;, quote = F, col.names = NA)
    
    # run camera competitive gene set test  
    # use signed likelihood ratio test statistic as recommended by GS here: https://support.bioconductor.org/p/112937/
    h.cam &lt;- cameraPR(statistic, h.id)
    c2.cam &lt;- cameraPR(statistic, c2.id)
    c5.cam &lt;- cameraPR(statistic, c5.id)
    
    # write top 50 GSEA results to file
    # write.table(h.cam[1:50,], 
    #             file = file.path(cellDir, glue(&quot;h.{colnames(contr)[i]}.csv&quot;)),
    #             sep = &quot;,&quot;, quote = F, col.names = NA)
    # write.table(c2.cam[1:50,], 
    #             file = file.path(cellDir, glue(&quot;c2.{colnames(contr)[i]}.csv&quot;)),
    #             sep = &quot;,&quot;, quote = F, col.names = NA)
    # write.table(c5.cam[1:50,], 
    #             file = file.path(cellDir, glue(&quot;c5.{colnames(contr)[i]}.csv&quot;)),
    #             sep = &quot;,&quot;, quote = F, col.names = NA)
    
    top %&gt;% 
      mutate(sig = ifelse(adj.P.Val &lt;= cutoff, glue(&quot;&lt;= {cutoff}&quot;), 
                          glue(&quot;&gt; {cutoff}&quot;))) %&gt;%
      rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
      left_join(dt[,i] %&gt;% 
                  data.frame %&gt;%
                  rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
                  dplyr::rename(status = 2)) %&gt;%
      mutate(status = ifelse(status == 1, &quot;Up&quot;,
                             ifelse(status == -1, &quot;Down&quot;,
                                    &quot;NotSig&quot;))) %&gt;%
      mutate(status = as.factor(status)) %&gt;%
      mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) %&gt;%
      ggplot(aes(x = logFC, y = -log10(P.Value), color = status)) +
      geom_point(alpha = 0.75) +
      ggrepel::geom_text_repel(data = function(x) subset(x, adj.P.Val &lt; cutoff), 
                               aes(x = logFC, y = -log10(P.Value), 
                                   label = SYMBOL), 
                               size = 2, colour = &quot;black&quot;, max.overlaps = 15) +
      labs(x = expression(~log[2]~&quot;(Fold Change)&quot;), 
           y = expression(~-log[10]~&quot;(P-value)&quot;),
           colour = glue(&quot;FDR &lt; {cutoff}&quot;)) +
      scale_colour_manual(values = pal) +
      theme_classic() +
      theme(legend.position = &quot;bottom&quot;) -&gt; p1
    
    # plot up to top 9 DGE
    grps &lt;- strsplit2(colnames(contr)[i], &quot;v&quot;)[1,]
    
    edgeR::cpm(ySub$counts, log = TRUE) %&gt;% 
      data.frame %&gt;%
      rownames_to_column(var = &quot;gene&quot;) %&gt;%
      pivot_longer(-gene, 
                   names_to = &quot;sample&quot;, 
                   values_to = &quot;raw&quot;) %&gt;%
      left_join(rownames_to_column(ySub$samples, 
                                   var = &quot;sample&quot;)) %&gt;%
      mutate(Batch = as.factor(Batch)) %&gt;%
      dplyr::filter(Group %in% names(contr[,i])[abs(contr[,i]) &gt; 0],
                    gene %in% rownames(top)[1:min(9, max(which(top$adj.P.Val &lt; cutoff)))]) %&gt;%
      mutate(Group = ifelse(str_detect(Group, str_remove(grps[1], &quot;CF.&quot;)),
                            grps[1], 
                            grps[2])) %&gt;%
      ggplot(aes(x = Group,
                 y = raw,
                 colour = Group)) +
      geom_boxplot(outlier.shape = NA, colour = &quot;grey&quot;) +
      geom_jitter(stat = &quot;identity&quot;,
                  width = 0.15,
                  size = 1.25) +
      theme_classic() +
      theme(axis.text.x = element_text(angle = 90,
                                       hjust = 1,
                                       vjust = 0.5),
            legend.position = &quot;bottom&quot;,
            legend.direction = &quot;horizontal&quot;,
            strip.text = element_text(size = 7),
            axis.text.y = element_text(size = 6)) +
      labs(x = &quot;Group&quot;, y = &quot;log2 CPM&quot;) +
      facet_wrap(~gene, scales = &quot;free_y&quot;) + 
      scale_color_brewer(palette = &quot;Set2&quot;) +
      ggtitle(colnames(contr)[i]) -&gt; p2
    
    # plot fibrosis gene set over-representation results
    gsa %&gt;% 
      data.frame %&gt;%
      rownames_to_column(var = &quot;Set&quot;) %&gt;%
      mutate(Rank = 1:n()) %&gt;%
      ggplot(aes(x = -log10(FDR), y = fct_reorder(Set, -Rank), colour = DE/N*100)) +
      geom_point(aes(size = N)) +
      geom_vline(xintercept = -log10(0.05),
                 linetype = &quot;dashed&quot;)  +
      scale_colour_viridis_c(option = &quot;plasma&quot;) +
      labs(y = &quot;Gene set&quot;,
           colour = &quot;% DEGs&quot;,
           size = &quot;Set size&quot;) +
      ggtitle(&quot;Fibrosis over-represenation&quot;) +
      theme_classic(base_size = 8) +
      theme(legend.key.height = unit(0.3, &quot;cm&quot;),
            legend.key.width = unit(0.3, &quot;cm&quot;),
            legend.text = element_text(size = 7),
            title = element_text(size = 8),
            legend.position = &quot;bottom&quot;,
            legend.direction = &quot;vertical&quot;) -&gt; p3
    
    # plot fibrosis GSEA results
    f.cam %&gt;% 
      data.frame %&gt;%
      rownames_to_column(var = &quot;Set&quot;) %&gt;%
      mutate(Rank = 1:n()) %&gt;%
      ggplot(aes(x = -log10(FDR), y = fct_reorder(Set, -Rank),
                 colour = Direction)) +
      geom_point(aes(size = NGenes)) +
      geom_vline(xintercept = -log10(0.05),
                 linetype = &quot;dashed&quot;)  +
      scale_colour_manual(values = pal) +
      labs(y = &quot;Gene set&quot;) +
      ggtitle(&quot;Fibrosis GSEA&quot;) +
      theme_classic(base_size = 8) +
      theme(legend.key.height = unit(0.3, &quot;cm&quot;),
            legend.key.width = unit(0.3, &quot;cm&quot;),
            legend.text = element_text(size = 7),
            title = element_text(size = 8),
            legend.position = &quot;bottom&quot;,
            legend.direction = &quot;vertical&quot;) -&gt; p4
    
    # GSEA top 10 plots
    h.cam %&gt;%
      dplyr::slice(1:10) %&gt;%
      rownames_to_column(var = &quot;Set&quot;) %&gt;%
      mutate(Type = &quot;HALLMARK&quot;,
             Rank = 1:10) %&gt;%
      bind_rows(c2.cam %&gt;%
                  dplyr::slice(1:10) %&gt;%
                  rownames_to_column(var = &quot;Set&quot;) %&gt;%
                  mutate(Type = &quot;REACTOME&quot;,
                         Rank = 1:10)) %&gt;%
      bind_rows(c5.cam %&gt;%
                  dplyr::slice(1:10) %&gt;%
                  rownames_to_column(var = &quot;Set&quot;) %&gt;%
                  mutate(Type = &quot;GO&quot;,
                         Rank = 1:10)) %&gt;%
      mutate(Set = str_wrap(str_replace_all(Set, &quot;_&quot;, &quot; &quot;), width = 75),
             Set = str_remove_all(Set, &quot;GO |REACTOME |HALLMARK &quot;)) %&gt;%
      ggplot(aes(x = -log10(FDR), y = fct_reorder(Set, -Rank),
                 colour = Direction)) +
      geom_point(aes(size = NGenes)) +
      facet_wrap(~Type, ncol = 1, scales = &quot;free_y&quot;) +
      geom_vline(xintercept = -log10(0.05),
                 linetype = &quot;dashed&quot;)  +
      scale_colour_manual(values = pal) +
      labs(y = &quot;Gene set&quot;) +
      theme_classic(base_size = 10) -&gt; p5

    layout &lt;- &quot;
      AAAA
      AAAA
      AAAA
      BBBB
      BBBB
      EEEE
      EEEE
      EEEE
      EEEE&quot;
    
    p[[i]] &lt;- wrap_elements(p1 + p2) + 
      wrap_elements(p3 + p4) + 
      wrap_elements(p5) +
      plot_layout(design = layout)
  }
}

p</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-23-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-23-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]
NULL

[[4]]
NULL

[[5]]
NULL</code></pre>
</div>
</div>
<div id="section" class="section level1 unnumbered">
<h1 class="unnumbered"></h1>
</div>
<div id="statistical-analysis" class="section level1 tabset">
<h1 class="tabset">Statistical analysis</h1>
<p>Use <code>RUVseq</code> and <code>edgeR</code> for differential
expression analysis between sample groups.</p>
<div id="define-negative-control-genes" class="section level2">
<h2>Define negative control genes</h2>
<p>Use house-keeping genes (HKG) identified from human single-cell
RNAseq experiments.</p>
<pre class="r"><code>data(&quot;segList&quot;, package = &quot;scMerge&quot;)

ctl &lt;- rownames(ySub) %in% segList$human$human_scSEG
table(ctl)</code></pre>
<pre><code>ctl
FALSE  TRUE 
 8750   975 </code></pre>
<p>Plot HKG expression profiles across all the samples.</p>
<pre class="r"><code>edgeR::cpm(ySub$counts, log = TRUE) %&gt;% 
  data.frame %&gt;%
  rownames_to_column(var = &quot;gene&quot;) %&gt;%
  pivot_longer(-gene, names_to = &quot;sample&quot;) %&gt;%
  left_join(rownames_to_column(ySub$samples, 
                               var = &quot;sample&quot;)) %&gt;%
  dplyr::filter(gene %in% segList$human$human_scSEG) %&gt;%
  dplyr::filter(Broad == cell) %&gt;%
  mutate(Batch = as.factor(Batch),
         Treatment = as.factor(Treatment),
         Severity = as.factor(Severity)) -&gt; dat

dat %&gt;%
  heatmap(gene, sample, value,
          scale = &quot;row&quot;,
          show_row_names = FALSE,
          show_column_names = FALSE) %&gt;%
  add_tile(Disease) %&gt;%
  add_tile(Treatment) %&gt;%
  add_tile(Severity) %&gt;%
  add_tile(Batch) %&gt;%
  add_tile(Participant) %&gt;%
  add_tile(Age) %&gt;%
  add_tile(Sex)</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-25-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="select-k" class="section level2">
<h2>Select <code>k</code></h2>
<p>Select the value for the <code>k</code> parameter i.e. the number of
columns of the <code>W</code> matrix that will be included in the
modelling.</p>
<pre class="r"><code>x1 &lt;- as.factor(ySub$samples$Batch)
cols1 &lt;- RColorBrewer::brewer.pal(7, &quot;Set2&quot;)

par(mfrow = c(1,3))
EDASeq::plotRLE(edgeR::cpm(ySub$counts), 
                col = cols1[x1], ylim = c(-0.5, 0.5),
                main = &quot;Raw&quot;)
legend(&quot;topleft&quot;, fill = cols1, legend = paste0(&quot;Batch &quot;, 0:6))
EDASeq::plotPCA(edgeR::cpm(ySub$counts), 
                col = cols1[x1], labels = FALSE,
                pch = 19)

x2 &lt;- as.factor(ySub$samples$Disease)
cols2 &lt;- RColorBrewer::brewer.pal(4, &quot;Set1&quot;)
EDASeq::plotPCA(edgeR::cpm(ySub$counts), 
                col = cols2[x2], labels = FALSE,
                pch = 19)</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for(k in 1:10){
  adj &lt;- RUVg(ySub$counts, ctl, k = k)
  W &lt;- adj$W
  
  # create the design matrix
  design &lt;- model.matrix(~ 0 + group + W + age + sex)
  colnames(design)[1:length(levels(group))] &lt;- levels(group)
  
  # add the factors for the replicate samples
  dups &lt;- unique(ySub$samples$Participant[duplicated(ySub$samples$Participant)])
  dups &lt;- sapply(dups, function(d){
    ifelse(ySub$samples$Participant == d, 1, 0)  
  }, USE.NAMES = TRUE)
  
  contr &lt;- makeContrasts(HvCF.U = Healthy - 0.5*(CF.U.M + CF.U.S),
                         CF.TvCF.U = 0.5*(CF.T.M + CF.T.S) - 0.5*(CF.U.M + CF.U.S),
                         CF.U.MvCF.U.S = CF.U.M - CF.U.S,
                         levels = design)
  
  y &lt;- DGEList(counts = ySub$counts)
  y &lt;- calcNormFactors(y)
  y &lt;- estimateGLMCommonDisp(y, design)
  y &lt;- estimateGLMTagwiseDisp(y, design)
  fit &lt;- glmFit(y, design)
  
  x1 &lt;- as.factor(ySub$samples$Batch)
  cols1 &lt;- RColorBrewer::brewer.pal(7, &quot;Set2&quot;)
  
  par(mfrow = c(1,3))
  EDASeq::plotRLE(edgeR::cpm(adj$normalizedCounts), 
                  col = cols1[x1], ylim = c(-0.5, 0.5),
                  main = paste0(&quot;K = &quot;, k))
  legend(&quot;topleft&quot;, fill = cols1, legend = paste0(&quot;Batch &quot;, 0:6))
  EDASeq::plotPCA(edgeR::cpm(adj$normalizedCounts), 
                  col = cols1[x1], labels = FALSE,
                  pch = 19)
  
  x2 &lt;- as.factor(ySub$samples$Group)
  cols2 &lt;- RColorBrewer::brewer.pal(5, &quot;Set1&quot;)
  EDASeq::plotPCA(edgeR::cpm(adj$normalizedCounts), 
                  col = cols2[x2], labels = FALSE,
                  pch = 19)
  legend(&quot;topleft&quot;, fill = cols2, legend = levels(factor(ySub$samples$Group)))
  
  lrt &lt;- glmLRT(fit, contrast = contr[, 1])
  hist(lrt$table$PValue, main = paste0(&quot;K = &quot;, k, &quot; &quot;, colnames(contr)[1]))
  lrt &lt;- glmLRT(fit, contrast = contr[, 2])
  hist(lrt$table$PValue, main = paste0(&quot;K = &quot;, k, &quot; &quot;, colnames(contr)[2]))
  lrt &lt;- glmLRT(fit, contrast = contr[, 3])
  hist(lrt$table$PValue, main = paste0(&quot;K = &quot;, k, &quot; &quot;, colnames(contr)[3]))
}</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-2.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-3.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-4.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-5.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-6.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-7.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-8.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-9.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-10.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-11.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-12.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-13.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-14.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-15.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-16.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-17.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-18.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-19.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-20.png" width="864" style="display: block; margin: auto;" /><img src="figure/TNK-DGE.Rmd/unnamed-chunk-26-21.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="test-for-dge" class="section level2">
<h2>Test for DGE</h2>
<p>Test for DGE using <code>RUVSeq</code> and <code>edgeR</code>. First,
create design matrix to model the sample groups and take into account
the unwanted variation, age, sex and replicate samples from the same
individual.</p>
<pre class="r"><code># use RUVSeq to identify the factors of unwanted variation
adj &lt;- RUVg(ySub$counts, ctl, k = 3)
W &lt;- adj$W

# define the sample groups
group &lt;- factor(ySub$samples$Group)

# create the design matrix
design &lt;- model.matrix(~ 0 + group + W + age + sex)
colnames(design)[1:length(levels(group))] &lt;- levels(group)

# add the factors for the replicate samples
dups &lt;- unique(ySub$samples$Participant[duplicated(ySub$samples$Participant)])
dups &lt;- sapply(dups, function(d){
  ifelse(ySub$samples$Participant == d, 1, 0)  
}, USE.NAMES = TRUE)

design &lt;- cbind(design, dups)
design %&gt;% knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="6%" />
<col width="6%" />
<col width="6%" />
<col width="6%" />
<col width="6%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="4%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">CF.T.M</th>
<th align="right">CF.T.S</th>
<th align="right">CF.U.M</th>
<th align="right">CF.U.S</th>
<th align="right">Healthy</th>
<th align="right">WW_1</th>
<th align="right">WW_2</th>
<th align="right">WW_3</th>
<th align="right">age</th>
<th align="right">sexM</th>
<th align="right">sample_35</th>
<th align="right">sample_37</th>
<th align="right">sample_39</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.1048065</td>
<td align="right">0.0054928</td>
<td align="right">-0.0302837</td>
<td align="right">-0.0939001</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.1573904</td>
<td align="right">-0.0388280</td>
<td align="right">0.0908887</td>
<td align="right">-0.1151479</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0596822</td>
<td align="right">0.0175639</td>
<td align="right">-0.0453488</td>
<td align="right">-0.0441471</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0957410</td>
<td align="right">-0.0199102</td>
<td align="right">-0.0466733</td>
<td align="right">0.1428834</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">-0.0835825</td>
<td align="right">-0.0147553</td>
<td align="right">-0.0482503</td>
<td align="right">0.1464588</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">-0.1198077</td>
<td align="right">-0.1050663</td>
<td align="right">-0.0456284</td>
<td align="right">0.5597097</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">-0.1004050</td>
<td align="right">0.0239409</td>
<td align="right">0.0075121</td>
<td align="right">1.5743836</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.1799767</td>
<td align="right">-0.0524649</td>
<td align="right">0.0917676</td>
<td align="right">1.5993830</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">-0.0160048</td>
<td align="right">-0.0914790</td>
<td align="right">-0.2165898</td>
<td align="right">1.9720631</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.0702306</td>
<td align="right">-0.0145163</td>
<td align="right">-0.2934049</td>
<td align="right">2.3883594</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">-0.0282879</td>
<td align="right">0.2870413</td>
<td align="right">0.2219840</td>
<td align="right">2.2957230</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0110668</td>
<td align="right">0.2295060</td>
<td align="right">0.1594423</td>
<td align="right">2.3360877</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0474976</td>
<td align="right">0.2508945</td>
<td align="right">0.2711534</td>
<td align="right">2.2980155</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.1244666</td>
<td align="right">-0.1117880</td>
<td align="right">-0.0225701</td>
<td align="right">2.5790214</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.0120309</td>
<td align="right">0.2360823</td>
<td align="right">0.1595635</td>
<td align="right">2.5823250</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">-0.2055862</td>
<td align="right">-0.2649939</td>
<td align="right">0.0654772</td>
<td align="right">0.1321035</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.3115546</td>
<td align="right">0.3131937</td>
<td align="right">-0.1496071</td>
<td align="right">2.5889097</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.1555949</td>
<td align="right">0.2880240</td>
<td align="right">0.0131611</td>
<td align="right">2.5583683</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0308695</td>
<td align="right">0.2422836</td>
<td align="right">0.2489584</td>
<td align="right">2.5670653</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.2226012</td>
<td align="right">0.2723742</td>
<td align="right">-0.1024958</td>
<td align="right">2.5730557</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0706894</td>
<td align="right">-0.0707720</td>
<td align="right">-0.1468500</td>
<td align="right">0.0807044</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0875375</td>
<td align="right">-0.1049505</td>
<td align="right">-0.0784646</td>
<td align="right">0.9940589</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0352946</td>
<td align="right">-0.0865651</td>
<td align="right">-0.1673927</td>
<td align="right">1.1764977</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.1828618</td>
<td align="right">-0.0998068</td>
<td align="right">0.1813889</td>
<td align="right">1.5597097</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.1546375</td>
<td align="right">-0.0129977</td>
<td align="right">0.1269464</td>
<td align="right">2.1930156</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.1624324</td>
<td align="right">-0.0848724</td>
<td align="right">0.0952999</td>
<td align="right">2.2980155</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.1506883</td>
<td align="right">-0.0952173</td>
<td align="right">0.1026107</td>
<td align="right">2.3485584</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">-0.0876473</td>
<td align="right">0.0439731</td>
<td align="right">0.0205071</td>
<td align="right">1.9730702</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.1376469</td>
<td align="right">0.0537279</td>
<td align="right">-0.2849637</td>
<td align="right">2.6297159</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">-0.0375185</td>
<td align="right">-0.0677095</td>
<td align="right">-0.2528950</td>
<td align="right">0.2923784</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.3435247</td>
<td align="right">-0.3185943</td>
<td align="right">0.1169517</td>
<td align="right">1.5801455</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.2708175</td>
<td align="right">-0.2258089</td>
<td align="right">0.2528744</td>
<td align="right">1.5801455</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.3623125</td>
<td align="right">-0.2414722</td>
<td align="right">0.1812328</td>
<td align="right">1.5993178</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0.4123004</td>
<td align="right">-0.2208887</td>
<td align="right">0.1720484</td>
<td align="right">1.5849625</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">-0.0445885</td>
<td align="right">-0.0181922</td>
<td align="right">-0.1295199</td>
<td align="right">3.0699187</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">-0.0011924</td>
<td align="right">0.0390738</td>
<td align="right">-0.2055982</td>
<td align="right">2.4204621</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0.0816450</td>
<td align="right">0.0584778</td>
<td align="right">-0.3132323</td>
<td align="right">2.2356012</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>Create the contrast matrix for the sample group comparisons.</p>
<pre class="r"><code>contr &lt;- makeContrasts(HvCF.U = Healthy - 0.5*(CF.U.M + CF.U.S),
                       HvCF = Healthy - 0.25*(CF.U.M + CF.U.S + CF.T.M + CF.T.S),
                       CF.TvCF.U = 0.5*(CF.T.M + CF.T.S) - 0.5*(CF.U.M + CF.U.S),
                       CF.U.MvCF.U.S = CF.U.M - CF.U.S,
                       CF.MvCF.S = 0.5*(CF.T.M + CF.U.M) - 0.5*(CF.T.M + CF.U.S),
                       levels = design)
contr %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">HvCF.U</th>
<th align="right">HvCF</th>
<th align="right">CF.TvCF.U</th>
<th align="right">CF.U.MvCF.U.S</th>
<th align="right">CF.MvCF.S</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CF.T.M</td>
<td align="right">0.0</td>
<td align="right">-0.25</td>
<td align="right">0.5</td>
<td align="right">0</td>
<td align="right">0.0</td>
</tr>
<tr class="even">
<td align="left">CF.T.S</td>
<td align="right">0.0</td>
<td align="right">-0.25</td>
<td align="right">0.5</td>
<td align="right">0</td>
<td align="right">0.0</td>
</tr>
<tr class="odd">
<td align="left">CF.U.M</td>
<td align="right">-0.5</td>
<td align="right">-0.25</td>
<td align="right">-0.5</td>
<td align="right">1</td>
<td align="right">0.5</td>
</tr>
<tr class="even">
<td align="left">CF.U.S</td>
<td align="right">-0.5</td>
<td align="right">-0.25</td>
<td align="right">-0.5</td>
<td align="right">-1</td>
<td align="right">-0.5</td>
</tr>
<tr class="odd">
<td align="left">Healthy</td>
<td align="right">1.0</td>
<td align="right">1.00</td>
<td align="right">0.0</td>
<td align="right">0</td>
<td align="right">0.0</td>
</tr>
<tr class="even">
<td align="left">WW_1</td>
<td align="right">0.0</td>
<td align="right">0.00</td>
<td align="right">0.0</td>
<td align="right">0</td>
<td align="right">0.0</td>
</tr>
<tr class="odd">
<td align="left">WW_2</td>
<td align="right">0.0</td>
<td align="right">0.00</td>
<td align="right">0.0</td>
<td align="right">0</td>
<td align="right">0.0</td>
</tr>
<tr class="even">
<td align="left">WW_3</td>
<td align="right">0.0</td>
<td align="right">0.00</td>
<td align="right">0.0</td>
<td align="right">0</td>
<td align="right">0.0</td>
</tr>
<tr class="odd">
<td align="left">age</td>
<td align="right">0.0</td>
<td align="right">0.00</td>
<td align="right">0.0</td>
<td align="right">0</td>
<td align="right">0.0</td>
</tr>
<tr class="even">
<td align="left">sexM</td>
<td align="right">0.0</td>
<td align="right">0.00</td>
<td align="right">0.0</td>
<td align="right">0</td>
<td align="right">0.0</td>
</tr>
<tr class="odd">
<td align="left">sample_35</td>
<td align="right">0.0</td>
<td align="right">0.00</td>
<td align="right">0.0</td>
<td align="right">0</td>
<td align="right">0.0</td>
</tr>
<tr class="even">
<td align="left">sample_37</td>
<td align="right">0.0</td>
<td align="right">0.00</td>
<td align="right">0.0</td>
<td align="right">0</td>
<td align="right">0.0</td>
</tr>
<tr class="odd">
<td align="left">sample_39</td>
<td align="right">0.0</td>
<td align="right">0.00</td>
<td align="right">0.0</td>
<td align="right">0</td>
<td align="right">0.0</td>
</tr>
</tbody>
</table>
<p>Fit the model.</p>
<pre class="r"><code>y &lt;- DGEList(counts = ySub$counts)
y &lt;- calcNormFactors(y)
y &lt;- estimateGLMCommonDisp(y, design)
y &lt;- estimateGLMTagwiseDisp(y, design)
fit &lt;- glmFit(y, design)</code></pre>
</div>
<div id="results-summary" class="section level2">
<h2>Results summary</h2>
<p>Summarise the results.</p>
<pre class="r"><code>cutoff &lt;- 0.05
  
dt &lt;- lapply(1:ncol(contr), function(i){
  decideTests(glmLRT(fit, contrast = contr[,i]),
                            p.value = cutoff)
})

s &lt;- sapply(dt, function(d){
  summary(d)
})
colnames(s) &lt;- colnames(contr)
rownames(s) &lt;- c(&quot;Down&quot;, &quot;NotSig&quot;, &quot;Up&quot;)

pal &lt;- c(paletteer::paletteer_d(&quot;RColorBrewer::Set1&quot;)[1:2], &quot;grey&quot;) 

s[-2,] %&gt;% 
  data.frame %&gt;%
  rownames_to_column(var = &quot;Direction&quot;) %&gt;%
  pivot_longer(-Direction) %&gt;%
  ggplot(aes(x = name, y = value, fill = Direction)) +
  geom_col(position = &quot;dodge&quot;) +
  geom_text(aes(label = value), 
            position = position_dodge(width = 0.9),
            vjust = -0.5,
            size = 3) +
  labs(y = glue(&quot;No. DGE (FDR &lt; {cutoff})&quot;),
       x = &quot;Contrast&quot;) +
      scale_fill_manual(values = pal)</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Test for over-representation of fibrosis signature genes our sets of
DGEs, adjusting for gene length bias. First, setup the transcript
length.</p>
<pre class="r"><code># translate gene SYMBOLS to entrez gene IDs 
gns &lt;- AnnotationDbi::mapIds(org.Hs.eg.db, 
                             keys = rownames(fit), 
                             column = c(&quot;ENTREZID&quot;),
                             keytype = &quot;SYMBOL&quot;,
                             multiVals = &quot;first&quot;)

# get transcript lengths
txdb &lt;- TxDb.Hsapiens.UCSC.hg38.knownGene
txlen &lt;- transcriptLengths(txdb)
txlen %&gt;%
  dplyr::select(gene_id, tx_len) %&gt;%
  group_by(gene_id) %&gt;%
  summarise(max_len = max(tx_len)) %&gt;%
  dplyr::filter(gene_id %in% gns) -&gt; supertxlen

m &lt;- match(supertxlen$gene_id, gns)
all(supertxlen$gene_id == gns[m])</code></pre>
<pre><code>[1] TRUE</code></pre>
<p>Perform gene set enrichment analysis (GSEA) using the
<code>cameraPR</code> method. <code>cameraPR</code> tests whether a set
of genes is highly ranked relative to other genes in terms of
differential expression, accounting for inter-gene correlation. Prepare
the Broad MSigDB Hallmark gene sets, Reactome pathways and GO terms.</p>
<pre class="r"><code>if(!file.exists(here(&quot;data/Hs.c5.all.v7.1.entrez.rds&quot;)))
  download.file(&quot;https://bioinf.wehi.edu.au/MSigDB/v7.1/Hs.c5.all.v7.1.entrez.rds&quot;,
                here(&quot;data/Hs.c5.all.v7.1.entrez.rds&quot;))
Hs.c5.all &lt;- readRDS(here(&quot;data/Hs.c5.all.v7.1.entrez.rds&quot;))

if(!file.exists(here(&quot;data/Hs.c2.cp.reactome.v7.1.entrez.rds&quot;)))
  download.file(&quot;https://bioinf.wehi.edu.au/MSigDB/v7.1/Hs.c2.cp.reactome.v7.1.entrez.rds&quot;,
                here(&quot;data/Hs.c2.cp.reactome.v7.1.entrez.rds&quot;))
Hs.c2.reactome &lt;- readRDS(here(&quot;data/Hs.c2.cp.reactome.v7.1.entrez.rds&quot;))

if(!file.exists(here(&quot;data/Hs.h.all.v7.1.entrez.rds&quot;)))
  download.file(&quot;https://bioinf.wehi.edu.au/MSigDB/v7.1/Hs.h.all.v7.1.entrez.rds&quot;,
                here(&quot;data/Hs.h.all.v7.1.entrez.rds&quot;))
Hs.h.all &lt;- readRDS(here(&quot;data/Hs.h.all.v7.1.entrez.rds&quot;))</code></pre>
<p>For each contrast, display the significant DGE and GSEA results,
along with volcano plots and strip charts for top 9 DGE.</p>
<pre class="r"><code>#library(missMethyl)

outDir &lt;- here(&quot;output&quot;,&quot;dge_analysis&quot;)
if(!dir.exists(outDir)) dir.create(outDir)
cellDir &lt;- file.path(outDir, cell)
if(!dir.exists(cellDir)) dir.create(cellDir)

p &lt;- vector(&quot;list&quot;, ncol(contr))

for(i in 1:(ncol(contr))){
  lrt &lt;- glmLRT(fit, contrast = contr[,i])
  topTags(lrt, n = Inf) %&gt;%
    data.frame -&gt; top
  
  if(sum(top$FDR &lt; cutoff) &gt; 0){
    # top 500 DGE results
    write.table(top[1:500, ], 
                file = file.path(cellDir, glue(&quot;{colnames(contr)[i]}.csv&quot;)),
                sep = &quot;,&quot;, quote = F, col.names = NA)
    
    # test DGEs for over-representation of fibrosis gene sets
    # read in fibrosis gene sets
    read.csv2(file = here(&quot;data/fibrosis_gene_sets.csv&quot;),
              header = TRUE, sep = &quot;,&quot;) %&gt;% 
      pivot_longer(cols = everything()) %&gt;%
      mutate(value = str_trim(value),
             entrez = unname(unlist(AnnotationDbi::mapIds(org.Hs.eg.db, 
                                                          keys = value, 
                                                          column = c(&quot;ENTREZID&quot;),
                                                          keytype = &quot;SYMBOL&quot;,
                                                          multiVals = &quot;first&quot;))[value])) %&gt;%
      dplyr::rename(&quot;symbol&quot; = &quot;value&quot;,
                    &quot;source&quot; = &quot;name&quot;) -&gt; fibrosis_table
      
      fibrosis_list &lt;- split(fibrosis_table$entrez, fibrosis_table$source)
    
    deg &lt;- rownames(top)[top$FDR &lt; cutoff]
    gsa &lt;- topGSA(gsaseq(unname(gns[deg]),
                         universe = unname(gns),
                         collection = fibrosis_list,
                         plot.bias = FALSE,
                         gene.length = supertxlen$max_len[match(unname(gns), 
                                                                supertxlen$gene_id)]))
    
    # get gene indices for all fibrosis gene set genes
    f.id &lt;- ids2indices(fibrosis_list, unname(gns[rownames(lrt)]))
    
    # run camera competitive gene set test  
    # use signed likelihood ratio test statistic as recommended by GS here: https://support.bioconductor.org/p/112937/
    statistic &lt;- sign(lrt$table$logFC) * sqrt(lrt$table$LR)
    f.cam &lt;- cameraPR(statistic, f.id)
    
    # write fibrosis over-representation results to file
    write.table(gsa %&gt;%
                  data.frame %&gt;%
                  rownames_to_column(var = &quot;Set&quot;), 
                file = file.path(cellDir, glue(&quot;fibrosis.ORA.{colnames(contr)[i]}.csv&quot;)),
                sep = &quot;,&quot;, quote = F, col.names = NA)
    
    # write fibrosis GSEA results to file
    write.table(f.cam, 
                file = file.path(cellDir, glue(&quot;fibrosis.GSEA.{colnames(contr)[i]}.csv&quot;)),
                sep = &quot;,&quot;, quote = F, col.names = NA)
    
    # get gene indices for all gene set genes
    h.id &lt;- ids2indices(Hs.h.all, unname(gns[rownames(lrt)]))
    c2.id &lt;- ids2indices(Hs.c2.reactome, unname(gns[rownames(lrt)]))
    c5.id &lt;- ids2indices(Hs.c5.all, unname(gns[rownames(lrt)]))
    
    # run camera competitive gene set test  
    # use signed likelihood ratio test statistic as recommended by GS here: https://support.bioconductor.org/p/112937/
    h.cam &lt;- cameraPR(statistic, h.id)
    c2.cam &lt;- cameraPR(statistic, c2.id)
    c5.cam &lt;- cameraPR(statistic, c5.id)
    
    # write top 50 GSEA results to file
    write.table(h.cam[1:50,], 
                file = file.path(cellDir, glue(&quot;h.{colnames(contr)[i]}.csv&quot;)),
                sep = &quot;,&quot;, quote = F, col.names = NA)
    write.table(c2.cam[1:50,], 
                file = file.path(cellDir, glue(&quot;c2.{colnames(contr)[i]}.csv&quot;)),
                sep = &quot;,&quot;, quote = F, col.names = NA)
    write.table(c5.cam[1:50,], 
                file = file.path(cellDir, glue(&quot;c5.{colnames(contr)[i]}.csv&quot;)),
                sep = &quot;,&quot;, quote = F, col.names = NA)
    
    top %&gt;% 
      mutate(sig = ifelse(FDR &lt;= cutoff, glue(&quot;&lt;= {cutoff}&quot;), 
                          glue(&quot;&gt; {cutoff}&quot;))) %&gt;%
      rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
      left_join(dt[[i]][,1] %&gt;% 
                  data.frame %&gt;%
                  rownames_to_column(var = &quot;SYMBOL&quot;) %&gt;%
                  dplyr::rename(status = 2)) %&gt;%
      mutate(status = ifelse(status == 1, &quot;Up&quot;,
                             ifelse(status == -1, &quot;Down&quot;,
                                    &quot;NotSig&quot;))) %&gt;%
      mutate(status = as.factor(status)) %&gt;%
      mutate(status = fct_relevel(status, &quot;NotSig&quot;, after = Inf)) %&gt;%
      ggplot(aes(x = logFC, y = -log10(PValue), color = status)) +
      geom_point(alpha = 0.75) +
      ggrepel::geom_text_repel(data = function(x) subset(x, FDR &lt; cutoff), 
                               aes(x = logFC, y = -log10(PValue), 
                                   label = SYMBOL), 
                               size = 2, colour = &quot;black&quot;, max.overlaps = 15) +
      labs(x = expression(~log[2]~&quot;(Fold Change)&quot;), 
           y = expression(~-log[10]~&quot;(P-value)&quot;),
           colour = glue(&quot;FDR &lt; {cutoff}&quot;)) +
      scale_colour_manual(values = pal) +
      theme_classic() +
      theme(legend.position = &quot;bottom&quot;) -&gt; p1
    
    # plot up to top 9 DGE
    grps &lt;- strsplit2(colnames(contr)[i], &quot;v&quot;)[1,]
    
    edgeR::cpm(ySub$counts, log = TRUE) %&gt;% 
      data.frame %&gt;%
      rownames_to_column(var = &quot;gene&quot;) %&gt;%
      pivot_longer(-gene, 
                   names_to = &quot;sample&quot;, 
                   values_to = &quot;raw&quot;) %&gt;%
      inner_join(edgeR::cpm(adj$normalizedCounts, log = TRUE) %&gt;% 
                   data.frame %&gt;%
                   rownames_to_column(var = &quot;gene&quot;) %&gt;%
                   pivot_longer(-gene, 
                                names_to = &quot;sample&quot;, 
                                values_to = &quot;norm&quot;)) %&gt;%
      left_join(rownames_to_column(ySub$samples, 
                                   var = &quot;sample&quot;)) %&gt;%
      mutate(Batch = as.factor(Batch)) %&gt;%
      dplyr::filter(Group %in% names(contr[,i])[abs(contr[,i]) &gt; 0],
                    gene %in% rownames(top)[1:min(9, max(which(top$FDR &lt; cutoff)))]) %&gt;%
      mutate(Group = ifelse(str_detect(Group, str_remove(grps[1], &quot;CF.&quot;)),
                            grps[1], 
                            grps[2])) %&gt;%
      ggplot(aes(x = Group,
                 y = norm,
                 colour = Group)) +
      geom_boxplot(outlier.shape = NA, colour = &quot;grey&quot;) +
      geom_jitter(stat = &quot;identity&quot;,
                  width = 0.15,
                  size = 1.25) +
      geom_jitter(aes(x = Group,
                      y = raw), stat = &quot;identity&quot;,
                  width = 0.15,
                  size = 1.25, 
                  alpha = 0.2) +
      theme_classic() +
      theme(axis.text.x = element_text(angle = 90,
                                       hjust = 1,
                                       vjust = 0.5),
            legend.position = &quot;bottom&quot;,
            legend.direction = &quot;horizontal&quot;,
            strip.text = element_text(size = 7),
            axis.text.y = element_text(size = 6)) +
      labs(x = &quot;Group&quot;, y = &quot;log2 CPM&quot;) +
      facet_wrap(~gene, scales = &quot;free_y&quot;) + 
      scale_color_brewer(palette = &quot;Set2&quot;) +
      ggtitle(colnames(contr)[i]) -&gt; p2
    
    # plot fibrosis gene set over-representation results
    gsa %&gt;% 
      data.frame %&gt;%
      rownames_to_column(var = &quot;Set&quot;) %&gt;%
      mutate(Rank = 1:n()) %&gt;%
      ggplot(aes(x = -log10(FDR), y = fct_reorder(Set, -Rank), colour = DE/N*100)) +
      geom_point(aes(size = N)) +
      geom_vline(xintercept = -log10(0.05),
                 linetype = &quot;dashed&quot;)  +
      scale_colour_viridis_c(option = &quot;plasma&quot;) +
      labs(y = &quot;Gene set&quot;,
           colour = &quot;% DEGs&quot;,
           size = &quot;Set size&quot;) +
      ggtitle(&quot;Fibrosis over-represenation&quot;) +
      theme_classic(base_size = 8) +
      theme(legend.key.height = unit(0.3, &quot;cm&quot;),
            legend.key.width = unit(0.3, &quot;cm&quot;),
            legend.text = element_text(size = 7),
            title = element_text(size = 8),
            legend.position = &quot;bottom&quot;,
            legend.direction = &quot;vertical&quot;) -&gt; p3
    
    # plot fibrosis GSEA results
    f.cam %&gt;% 
      data.frame %&gt;%
      rownames_to_column(var = &quot;Set&quot;) %&gt;%
      mutate(Rank = 1:n()) %&gt;%
      ggplot(aes(x = -log10(FDR), y = fct_reorder(Set, -Rank),
                 colour = Direction)) +
      geom_point(aes(size = NGenes)) +
      geom_vline(xintercept = -log10(0.05),
                 linetype = &quot;dashed&quot;)  +
      scale_colour_manual(values = pal) +
      labs(y = &quot;Gene set&quot;) +
      ggtitle(&quot;Fibrosis GSEA&quot;) +
      theme_classic(base_size = 8) +
      theme(legend.key.height = unit(0.3, &quot;cm&quot;),
            legend.key.width = unit(0.3, &quot;cm&quot;),
            legend.text = element_text(size = 7),
            title = element_text(size = 8),
            legend.position = &quot;bottom&quot;,
            legend.direction = &quot;vertical&quot;) -&gt; p4
    
    # GSEA top 10 plots
    h.cam %&gt;%
      dplyr::slice(1:10) %&gt;%
      rownames_to_column(var = &quot;Set&quot;) %&gt;%
      mutate(Type = &quot;HALLMARK&quot;,
             Rank = 1:10) %&gt;%
      bind_rows(c2.cam %&gt;%
                  dplyr::slice(1:10) %&gt;%
                  rownames_to_column(var = &quot;Set&quot;) %&gt;%
                  mutate(Type = &quot;REACTOME&quot;,
                         Rank = 1:10)) %&gt;%
      bind_rows(c5.cam %&gt;%
                  dplyr::slice(1:10) %&gt;%
                  rownames_to_column(var = &quot;Set&quot;) %&gt;%
                  mutate(Type = &quot;GO&quot;,
                         Rank = 1:10)) %&gt;%
      mutate(Set = str_wrap(str_replace_all(Set, &quot;_&quot;, &quot; &quot;), width = 75),
             Set = str_remove_all(Set, &quot;GO |REACTOME |HALLMARK &quot;)) %&gt;%
      ggplot(aes(x = -log10(FDR), y = fct_reorder(Set, -Rank),
                 colour = Direction)) +
      geom_point(aes(size = NGenes)) +
      facet_wrap(~Type, ncol = 1, scales = &quot;free_y&quot;) +
      geom_vline(xintercept = -log10(0.05),
                 linetype = &quot;dashed&quot;)  +
      scale_colour_manual(values = pal) +
      labs(y = &quot;Gene set&quot;) +
      theme_classic(base_size = 10) -&gt; p5

    layout &lt;- &quot;
      AAAA
      AAAA
      AAAA
      BBBB
      BBBB
      EEEE
      EEEE
      EEEE
      EEEE&quot;
    
    p[[i]] &lt;- wrap_elements(p1 + p2) + 
      wrap_elements(p3 + p4) + 
      wrap_elements(p5) +
      plot_layout(design = layout)
  }
}

p</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-33-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/TNK-DGE.Rmd/unnamed-chunk-33-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]
NULL

[[4]]
NULL

[[5]]
NULL</code></pre>
</div>
<div id="section-1" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
</div>
</div>
<div id="session-info" class="section level1 appendix">
<h1 class="appendix">Session info</h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Melbourne
tzcode source: internal

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices datasets  utils    
[8] methods   base     

other attached packages:
 [1] missMethyl_1.36.0                                  
 [2] IlluminaHumanMethylationEPICanno.ilm10b4.hg19_0.6.0
 [3] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.1 
 [4] minfi_1.48.0                                       
 [5] bumphunter_1.44.0                                  
 [6] locfit_1.5-9.8                                     
 [7] iterators_1.0.14                                   
 [8] foreach_1.5.2                                      
 [9] TxDb.Hsapiens.UCSC.hg38.knownGene_3.18.0           
[10] GenomicFeatures_1.54.3                             
[11] org.Hs.eg.db_3.18.0                                
[12] AnnotationDbi_1.64.1                               
[13] EGSEAdata_1.30.0                                   
[14] tidyHeatmap_1.8.1                                  
[15] scater_1.30.1                                      
[16] scuttle_1.12.0                                     
[17] SingleCellExperiment_1.24.0                        
[18] scMerge_1.18.0                                     
[19] RUVSeq_1.36.0                                      
[20] EDASeq_2.36.0                                      
[21] ShortRead_1.60.0                                   
[22] GenomicAlignments_1.38.2                           
[23] SummarizedExperiment_1.32.0                        
[24] MatrixGenerics_1.14.0                              
[25] matrixStats_1.2.0                                  
[26] Rsamtools_2.18.0                                   
[27] GenomicRanges_1.54.1                               
[28] Biostrings_2.70.2                                  
[29] GenomeInfoDb_1.38.6                                
[30] XVector_0.42.0                                     
[31] IRanges_2.36.0                                     
[32] S4Vectors_0.40.2                                   
[33] BiocParallel_1.36.0                                
[34] Biobase_2.62.0                                     
[35] BiocGenerics_0.48.1                                
[36] edgeR_4.0.15                                       
[37] limma_3.58.1                                       
[38] paletteer_1.6.0                                    
[39] patchwork_1.2.0                                    
[40] SeuratObject_4.1.4                                 
[41] Seurat_4.4.0                                       
[42] glue_1.7.0                                         
[43] here_1.0.1                                         
[44] lubridate_1.9.3                                    
[45] forcats_1.0.0                                      
[46] stringr_1.5.1                                      
[47] dplyr_1.1.4                                        
[48] purrr_1.0.2                                        
[49] readr_2.1.5                                        
[50] tidyr_1.3.1                                        
[51] tibble_3.2.1                                       
[52] ggplot2_3.5.0                                      
[53] tidyverse_2.0.0                                    
[54] BiocStyle_2.30.0                                   
[55] workflowr_1.7.1                                    

loaded via a namespace (and not attached):
  [1] igraph_2.0.1.1            ica_1.0-3                
  [3] plotly_4.10.4             Formula_1.2-5            
  [5] rematch2_2.1.2            zlibbioc_1.48.0          
  [7] tidyselect_1.2.0          bit_4.0.5                
  [9] doParallel_1.0.17         clue_0.3-65              
 [11] lattice_0.22-5            rjson_0.2.21             
 [13] nor1mix_1.3-2             M3Drop_1.28.0            
 [15] blob_1.2.4                rngtools_1.5.2           
 [17] S4Arrays_1.2.0            base64_2.0.1             
 [19] scrime_1.3.5              png_0.1-8                
 [21] ResidualMatrix_1.12.0     cli_3.6.2                
 [23] askpass_1.2.0             openssl_2.1.1            
 [25] multtest_2.58.0           goftest_1.2-3            
 [27] BiocIO_1.12.0             bluster_1.12.0           
 [29] BiocNeighbors_1.20.2      densEstBayes_1.0-2.2     
 [31] uwot_0.1.16               dendextend_1.17.1        
 [33] curl_5.2.0                mime_0.12                
 [35] evaluate_0.23             leiden_0.4.3.1           
 [37] ComplexHeatmap_2.18.0     stringi_1.8.3            
 [39] backports_1.4.1           XML_3.99-0.16.1          
 [41] httpuv_1.6.14             magrittr_2.0.3           
 [43] rappdirs_0.3.3            splines_4.3.2            
 [45] mclust_6.1                BiasedUrn_2.0.11         
 [47] jpeg_0.1-10               doRNG_1.8.6              
 [49] sctransform_0.4.1         ggbeeswarm_0.7.2         
 [51] DBI_1.2.1                 HDF5Array_1.30.0         
 [53] genefilter_1.84.0         jquerylib_0.1.4          
 [55] withr_3.0.0               git2r_0.33.0             
 [57] rprojroot_2.0.4           lmtest_0.9-40            
 [59] bdsmatrix_1.3-6           rtracklayer_1.62.0       
 [61] BiocManager_1.30.22       htmlwidgets_1.6.4        
 [63] fs_1.6.3                  biomaRt_2.58.2           
 [65] ggrepel_0.9.5             labeling_0.4.3           
 [67] SparseArray_1.2.4         DEoptimR_1.1-3           
 [69] annotate_1.80.0           reticulate_1.35.0        
 [71] zoo_1.8-12                knitr_1.45               
 [73] beanplot_1.3.1            timechange_0.3.0         
 [75] fansi_1.0.6               caTools_1.18.2           
 [77] grid_4.3.2                data.table_1.15.0        
 [79] rhdf5_2.46.1              ruv_0.9.7.1              
 [81] R.oo_1.26.0               irlba_2.3.5.1            
 [83] ellipsis_0.3.2            aroma.light_3.32.0       
 [85] lazyeval_0.2.2            yaml_2.3.8               
 [87] survival_3.5-8            scattermore_1.2          
 [89] crayon_1.5.2              RcppAnnoy_0.0.22         
 [91] RColorBrewer_1.1-3        progressr_0.14.0         
 [93] later_1.3.2               ggridges_0.5.6           
 [95] codetools_0.2-19          base64enc_0.1-3          
 [97] GlobalOptions_0.1.2       KEGGREST_1.42.0          
 [99] bbmle_1.0.25.1            Rtsne_0.17               
[101] shape_1.4.6               startupmsg_0.9.6.1       
[103] filelock_1.0.3            foreign_0.8-86           
[105] pkgconfig_2.0.3           xml2_1.3.6               
[107] getPass_0.2-4             sfsmisc_1.1-17           
[109] spatstat.sparse_3.0-3     viridisLite_0.4.2        
[111] xtable_1.8-4              interp_1.1-6             
[113] fastcluster_1.2.6         highr_0.10               
[115] hwriter_1.3.2.1           plyr_1.8.9               
[117] httr_1.4.7                tools_4.3.2              
[119] globals_0.16.2            pkgbuild_1.4.3           
[121] beeswarm_0.4.0            htmlTable_2.4.2          
[123] checkmate_2.3.1           nlme_3.1-164             
[125] loo_2.6.0                 dbplyr_2.4.0             
[127] digest_0.6.34             numDeriv_2016.8-1.1      
[129] Matrix_1.6-5              farver_2.1.1             
[131] tzdb_0.4.0                reshape2_1.4.4           
[133] viridis_0.6.5             cvTools_0.3.2            
[135] rpart_4.1.23              cachem_1.0.8             
[137] BiocFileCache_2.10.1      polyclip_1.10-6          
[139] WGCNA_1.72-5              Hmisc_5.1-1              
[141] generics_0.1.3            proxyC_0.3.4             
[143] dynamicTreeCut_1.63-1     mvtnorm_1.2-4            
[145] parallelly_1.37.0         statmod_1.5.0            
[147] impute_1.76.0             ScaledMatrix_1.10.0      
[149] GEOquery_2.70.0           pbapply_1.7-2            
[151] dqrng_0.3.2               utf8_1.2.4               
[153] siggenes_1.76.0           StanHeaders_2.32.5       
[155] gtools_3.9.5              preprocessCore_1.64.0    
[157] gridExtra_2.3             shiny_1.8.0              
[159] GenomeInfoDbData_1.2.11   R.utils_2.12.3           
[161] rhdf5filters_1.14.1       RCurl_1.98-1.14          
[163] memoise_2.0.1             rmarkdown_2.25           
[165] scales_1.3.0              R.methodsS3_1.8.2        
[167] future_1.33.1             reshape_0.8.9            
[169] RANN_2.6.1                renv_1.0.3               
[171] Cairo_1.6-2               illuminaio_0.44.0        
[173] spatstat.data_3.0-4       rstudioapi_0.15.0        
[175] cluster_2.1.6             QuickJSR_1.1.3           
[177] whisker_0.4.1             rstantools_2.4.0         
[179] spatstat.utils_3.0-4      hms_1.1.3                
[181] fitdistrplus_1.1-11       munsell_0.5.0            
[183] cowplot_1.1.3             colorspace_2.1-0         
[185] quadprog_1.5-8            rlang_1.1.3              
[187] DelayedMatrixStats_1.24.0 sparseMatrixStats_1.14.0 
[189] circlize_0.4.15           mgcv_1.9-1               
[191] xfun_0.42                 reldist_1.7-2            
[193] abind_1.4-5               rstan_2.32.5             
[195] Rhdf5lib_1.24.2           bitops_1.0-7             
[197] ps_1.7.6                  promises_1.2.1           
[199] inline_0.3.19             RSQLite_2.3.5            
[201] DelayedArray_0.28.0       GO.db_3.18.0             
[203] compiler_4.3.2            prettyunits_1.2.0        
[205] beachmat_2.18.1           listenv_0.9.1            
[207] Rcpp_1.0.12               BiocSingular_1.18.0      
[209] tensor_1.5                MASS_7.3-60.0.1          
[211] progress_1.2.3            spatstat.random_3.2-2    
[213] R6_2.5.1                  fastmap_1.1.1            
[215] vipor_0.4.7               distr_2.9.3              
[217] ROCR_1.0-11               rsvd_1.0.5               
[219] nnet_7.3-19               gtable_0.3.4             
[221] KernSmooth_2.23-22        latticeExtra_0.6-30      
[223] miniUI_0.1.1.1            deldir_2.0-2             
[225] htmltools_0.5.7           RcppParallel_5.1.7       
[227] bit64_4.0.5               spatstat.explore_3.2-6   
[229] lifecycle_1.0.4           processx_3.8.3           
[231] callr_3.7.3               restfulr_0.0.15          
[233] sass_0.4.8                vctrs_0.6.5              
[235] spatstat.geom_3.2-8       robustbase_0.99-2        
[237] scran_1.30.2              sp_2.1-3                 
[239] future.apply_1.11.1       bslib_0.6.1              
[241] pillar_1.9.0              batchelor_1.18.1         
[243] prismatic_1.1.1           gplots_3.1.3.1           
[245] metapod_1.10.1            jsonlite_1.8.8           
[247] GetoptLong_1.0.5         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
