---
title: "Preprocessing the C133_Neeland_batch0 data set"
description: |
author:
  - name: Jovana Maksimovic
date: "`r Sys.Date()`"
editor_options: 
  chunk_output_type: inline
---

```{r, echo = FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE)
```

```{r setup}
suppressPackageStartupMessages({
    library(here)
    library(BiocStyle)
    library(ggplot2)
    library(cowplot)
    library(patchwork)
    library(tidyverse)
    library(SingleCellExperiment)
    library(DropletUtils)
    library(scater)
})
```


# Overview

Bronchoalveolar lavage (BAL) samples were collected from 4 individuals: 1 control sample and 3 cystic fibrosis (CF) samples. The samples were run on teh 10X Chromium and sequenced at the [Garvan-Weizmann Centre for Cellular Genomics (GWCCG)](https://www.garvan.org.au/research/garvan-weizmann). The multiplexed samples were sequenced on an Illumina NovaSeq 6000 (NovaSeq Control Software v1.3.1 / Real Time Analysis v3.3.3) ) using a NovaSeq S1 200 cycle kit (Illumina, 20012864). The `cellranger count` pipeline (version 6.0.2) was used for alignment, filtering, barcode counting, and UMI counting from FASTQ files. The GRCh38 reference was used for the alignment. The number of cells from the pipeline was forced to 10,000. 
View the **CellRanger** capture-specific web summaries: [A](A_web_summary.html), [B](B_web_summary.html), [C](C_web_summary.html), [D](D_web_summary.html).

```{r}
sample_metadata_df <- read_csv(
  here("data/C133_Neeland_batch0/data/sample_sheets/Sample_information.csv"))

knitr::kable(sample_metadata_df)
```

# Set up the data

```{r}
sce <- readRDS(here("data", "C133_Neeland_batch0",
                    "data", "SCEs", "C133_Neeland_batch0.CellRanger.SCE.rds"))
sce$Capture <- factor(sce$Sample)
capture_names <- levels(sce$Capture)
capture_names <- setNames(capture_names, capture_names)
sce$Sample <- NULL
sce
```

# Call cells from empty droplets

```{r rankplot, fig.cap = "Total UMI count for each barcode in the dataset, plotted against its rank (in decreasing order of total counts). The inferred locations of the inflection (dark green dashed lines) and knee points (blue dashed lines) are also shown.", fig.asp = 1, results = "hide"}

par(mfrow = c(2, 2))
lapply(capture_names, function(cn) {
  sce <- sce[, sce$Capture == cn]
  bcrank <- barcodeRanks(counts(sce))
  # Only showing unique points for plotting speed.
  uniq <- !duplicated(bcrank$rank)
  plot(
    x = bcrank$rank[uniq],
    y = bcrank$total[uniq],
    log = "xy",
    xlab = "Rank",
    ylab = "Total UMI count",
    main = cn,
    cex.lab = 1.2,
    xlim = c(1, 500000),
    ylim = c(1, 200000))
  abline(h = metadata(bcrank)$inflection, col = "darkgreen", lty = 2)
  abline(h = metadata(bcrank)$knee, col = "dodgerblue", lty = 2)
})
```

Remove empty droplets.

```{r}
empties <- do.call(rbind, lapply(capture_names, function(cn) {
  message(cn)
  empties <- readRDS(
    here("data",
         "C133_Neeland_batch0",
         "data",
         "emptyDrops", paste0(cn, ".emptyDrops.rds")))
  empties$Capture <- cn
  empties
}))
tapply(
  empties$FDR,
  empties$Capture,
  function(x) sum(x <= 0.001, na.rm = TRUE)) |>
  knitr::kable(
    caption = "Number of non-empty droplets identified using `emptyDrops()` from **DropletUtils**.")

sce <- sce[, which(empties$FDR <= 0.001)]
sce
```

## Add per cell quality control information

```{r}
sce <- scuttle::addPerCellQC(sce)
head(colData(sce)) %>%
  data.frame %>%
  knitr::kable()
```


# Save data

```{r}
saveRDS(
   sce,
   here("data",
        "C133_Neeland_batch0",
        "data",
        "SCEs",
        "C133_Neeland_batch0.preprocessed.SCE.rds"))
```

# Session info {.appendix}

