---
title: "Inflammation of Paediatric Pulmonary Diseases"
subtitle: "DGE analysis of alveolar macrophages: CF modifier and severity"
author: "Jovana Maksimovic"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
site: workflowr::wflow_site
output:
  bookdown::html_document2:
    base_format: workflowr::wflow_html
    toc: yes
    toc_float: yes
    theme: cosmo
    highlight: textmate
    number_sections: true
---

```{r, echo = FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE)
```

# Load libraries

```{r, message=FALSE, warning=FALSE}
suppressPackageStartupMessages({
  library(BiocStyle)
  library(tidyverse)
  library(here)
  library(glue)
  library(Seurat)
  library(patchwork)
  library(paletteer)
  library(limma)
  library(edgeR)
  library(RUVSeq)
  library(scMerge)
  library(SingleCellExperiment)
  library(scater)
  library(tidyHeatmap)
  library(EGSEAdata)
  library(org.Hs.eg.db)
  library(TxDb.Hsapiens.UCSC.hg38.knownGene)
  library(missMethyl)
})
```


# Load Data

```{r}
ambient <- ""
file <- here("data",
            "C133_Neeland_merged",
            glue("C133_Neeland_full_clean{ambient}_macrophages_annotated_diet.SEU.rds"))

seu <- readRDS(file)
seu
```


# Create pseudobulk samples by cell type (Annotation)

Use *cell type* and *sample* as our two factors; each column of the output
corresponds to one unique combination of these two factors. 

```{r}
out <- here("data",
            "C133_Neeland_merged",
            glue("C133_Neeland_full_clean{ambient}_macrophages_pseudobulk.rds"))

sce <- SingleCellExperiment(list(counts = seu[["RNA"]]@counts),
                            colData = seu@meta.data)

if(!file.exists(out)){
  pseudoBulk <- aggregateAcrossCells(sce, 
                                 id = colData(sce)[, c("Annotation", "sample.id")])
  saveRDS(pseudoBulk, file = out)
  
} else {
  pseudoBulk <- readRDS(file = out)
  
}

pseudoBulk
```


## Code micro information

Create a factor that identifies individuals that were infected with the top 4 clinically important pathogens at time of sample collection i.e. *Pseudomonas aeruginosa*, *Staphylococcus aureus*, *Haemophilus influenzae*, and *Aspergillus*.

```{r}
important_micro <- c("Pseudomonas aeruginosa", "Staphylococcus aureus",
                     "Haemophilus influenzae", "Aspergillus", "S. aureus",
                     "Staph Aureus (Methicillin Resistant)", "MRSA")

pseudoBulk$Micro_code <- sapply(strsplit(pseudoBulk$Bacteria_type, ","), function(bacteria){
  any(tolower(str_trim(bacteria)) %in% tolower(important_micro))
})

table(pseudoBulk$Micro_code)
```

## No. cells per cell type

```{r}
colData(pseudoBulk) %>%
  data.frame %>%
  group_by(Annotation) %>%
  summarise(total = sum(ncells)) %>%
  ggplot(aes(x = fct_reorder(Annotation, total), 
             y = total, fill = Annotation)) +
  geom_col() + 
  geom_text(aes(label = total), vjust = -0.5, colour = "black", size = 2.5) +
  scale_y_log10() +
  labs(x = "Cell label",
       y = "Log 10 No. cells") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        legend.position = "bottom") +
  geom_hline(yintercept = 1000, linetype = "dashed") +
    NoLegend()
```

## No. cells per cell type per sample

How many pseudobulk samples are comprised of >50 cells?

```{r, fig.asp = 1.5, fig.width=9}
 colData(pseudoBulk) %>%
  data.frame %>%
  arrange(Group) %>%
  ggplot(aes(x = fct_inorder(sample.id), 
             y = ncells, fill = Group)) +
  geom_col() + 
  scale_fill_brewer(palette = "Set2") +
  scale_y_log10() +
  facet_wrap(~Annotation, ncol = 2) + 
  labs(x = "Sample",
       y = "Log10 No. cells") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5,
                                   size = 8),
        legend.position = "bottom") +
  geom_hline(yintercept = 50, linetype = "dashed") +
  geom_hline(yintercept = 25, linetype = "dotted")
```

# Data preparation {.tabset}
## Extract cell type

Make a `DGElist` object.

```{r}
yPB <- DGEList(counts = counts(pseudoBulk),
               samples = colData(pseudoBulk) %>% data.frame)
dim(yPB)
```

Remove genes with zero counts in all samples.

```{r}
keep <- rowSums(yPB$counts) > 0 
yFlt <- yPB[keep, ]
dim(yFlt)
```

Extract only the *macro-alveolar* cells. 

```{r}
cell <- "macro-alveolar"
ySub <- yFlt[, yFlt$samples$Annotation == cell]
dim(ySub)
```

Extract only the CF samples.

```{r}
ySub <- ySub[, ySub$samples$Disease != "Healthy"]
```

## Filter samples & genes

Examine MDS plot for outlier samples.

```{r, fig.asp=1}
mds_by_factor <- function(data, factor, lab){
  dims <- list(c(1,2), c(2:3), c(3,4), c(4,5))
  p <- vector("list", length(dims))
  
  for(i in 1:length(dims)){
    
    mds <- limma::plotMDS(edgeR::cpm(data, 
                                     log = TRUE), 
                          gene.selection = "common",
                          plot = FALSE, dim.plot = dims[[i]])
    
    data.frame(x = mds$x, 
               y = mds$y,
               sample = rownames(mds$distance.matrix.squared)) %>%
      left_join(rownames_to_column(data$samples, var = "sample")) -> dat
    
    p[[i]] <- ggplot(dat, aes(x = x, y = y, 
                              colour = eval(parse(text=(factor))))) +
      geom_point(size = 3) +
      ggrepel::geom_text_repel(aes(label = sample.id),
                               size = 2) +
      labs(x = glue("Principal Component {dims[[i]][1]}"),
           y = glue("Principal Component {dims[[i]][2]}"),
           colour = lab) +
      theme(legend.direction = "horizontal",
            legend.text = element_text(size = 8),
            legend.title = element_text(size = 9),
            axis.text = element_text(size = 8),
            axis.title = element_text(size = 9)) -> p[[i]]
  }
  
  wrap_plots(p, ncol = 2) + 
    plot_layout(guides = "collect") &
    theme(legend.position = "bottom")
}

mds_by_factor(ySub, "as.factor(Batch)", "Batch") & scale_color_brewer(palette = "Set1")
mds_by_factor(ySub, "as.factor(Sex)", "Sex") & scale_color_brewer(palette = "Set2")
mds_by_factor(ySub, "log2(Age)", "Log2 Age") & scale_colour_viridis_c(option = "magma") 
mds_by_factor(ySub, "as.factor(Status_sub)", "Group") & scale_color_brewer(palette = "Dark2")
mds_by_factor(ySub, "as.factor(Severity)", "Severity") & scale_color_brewer(palette = "Accent")
mds_by_factor(ySub, "as.factor(Micro_code)", "Infection") & scale_color_brewer(palette = "Pastel1")
```


Examine number of cells per sample. Identify outliers and cross-reference with MDS plot. Determine a threshold for minimum number of cells per sample.

```{r}
minCells <- 50

ySub$samples %>%
  ggplot(aes(x = sample.id, y = ncells, fill = Micro_code)) +
  geom_col() +
  labs(fill = "Infection") + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  geom_hline(yintercept = minCells, linetype = "dashed") +
  facet_grid(~Status_sub, space = "free_x", scales = "free_x") +
  scale_fill_brewer(palette = "Pastel1")

ySub$samples %>%
  ggplot(aes(x = sample.id, y = ncells, fill = Severity)) +
  geom_col() +
  labs(fill = "Severity") + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  geom_hline(yintercept = minCells, linetype = "dashed") +
  facet_grid(~Status_sub, space = "free_x", scales = "free_x") +
  scale_fill_brewer(palette = "Accent")
```

Filter out samples with less than previously determined minimum number of cells.

```{r}
ySub <- ySub[, ySub$samples$ncells > minCells]
dim(ySub)
```

Re-examine MDS plots.

```{r, fig.asp=1}
mds_by_factor(ySub, "as.factor(Batch)", "Batch") & scale_color_brewer(palette = "Set1")
mds_by_factor(ySub, "as.factor(Sex)", "Sex") & scale_color_brewer(palette = "Set2")
mds_by_factor(ySub, "log2(Age)", "Log2 Age") & scale_colour_viridis_c(option = "magma") 
mds_by_factor(ySub, "as.factor(Status_sub)", "Group") & scale_color_brewer(palette = "Dark2")
mds_by_factor(ySub, "as.factor(Severity)", "Severity") & scale_color_brewer(palette = "Accent")
mds_by_factor(ySub, "as.factor(Micro_code)", "Infection") & scale_color_brewer(palette = "Pastel1")
```

Filter out genes with no ENTREZ IDs and very low expression.

```{r}
gns <- AnnotationDbi::mapIds(org.Hs.eg.db,
                             keys = rownames(ySub),
                             column = c("ENTREZID"),
                             keytype = "SYMBOL",
                             multiVals = "first")
keep <- !is.na(gns)
ySub <- ySub[keep,]

thresh <- 0
m <- rowMedians(edgeR::cpm(ySub$counts, log = TRUE))
plot(density(m))
abline(v = thresh, lty = 2)
```

```{r}
# filter out genes with low median expression
keep <- m > thresh
table(keep)
ySub <- ySub[keep, ]
dim(ySub)
```

## Examine covariates

Principal components analysis (PCA) allows us to mathematically determine the sources of variation in the data. We can then investigate whether these correlate with any of the specifed covariates.

Prepare the data.

```{r}
PCs <- prcomp(t(edgeR::cpm(ySub$counts, log = TRUE)), 
              center = TRUE, retx = TRUE)
loadings = PCs$x # pc loadings


nGenes = nrow(ySub)
nSamples = ncol(ySub)

datTraits <- ySub$samples %>% dplyr::select(Batch, Treatment, Micro_code,
                                            Severity, Age, Sex, ncells) %>%
  mutate(Batch = factor(Batch),
         Treatment = factor(Treatment, 
                            labels = 1:length(unique(Treatment))),
         Sex = factor(Sex, labels = length(unique(Sex))),
         Severity = factor(Severity, labels = length(unique(Severity)))) %>%
  mutate(across(everything(), as.numeric))

moduleTraitCor <- suppressWarnings(cor(loadings[, 1:min(10, nSamples)], 
                                       datTraits, use = "p"))
moduleTraitPvalue <- WGCNA::corPvalueStudent(moduleTraitCor, (nSamples-2))

textMatrix <- paste(signif(moduleTraitCor, 2), "\n(", 
                    signif(moduleTraitPvalue, 1), ")", sep = "")
dim(textMatrix) <- dim(moduleTraitCor)
```

Output results.

```{r, fig.asp=1.3}
par(mfrow = c(2, 1))
plot(PCs, type="lines", main = cell) # scree plot

## Display the correlation values within a heatmap plot
par(cex=0.75, mar = c(3, 5, 2, 1))
WGCNA::labeledHeatmap(Matrix = t(moduleTraitCor),
                            xLabels = colnames(loadings)[1:min(10, nSamples)],
                            yLabels = names(datTraits),
                            colorLabels = FALSE,
                            colors = WGCNA::blueWhiteRed(6),
                            textMatrix = t(textMatrix),
                            setStdMargins = FALSE,
                            cex.text = 1,
                            zlim = c(-1,1),
                            main = paste0("PCA-trait relationships: Top ", 
                                          min(10, nSamples), 
                                          " PCs"))
```

# {-}

# Statistical analysis with `RUVseq` 

Use `RUVseq` and `edgeR` for differential expression analysis between sample groups.

## Define negative control genes

Use house-keeping genes (HKG) identified from human single-cell RNAseq experiments.

```{r}
data("segList", package = "scMerge")

HKGs <- segList$human$bulkRNAseqHK
ctl <- rownames(ySub) %in% HKGs
table(ctl)
```

Plot HKG expression profiles across all the samples.

```{r, fig.asp=1.5, fig.width=10}
edgeR::cpm(ySub$counts, log = TRUE) %>% 
  data.frame %>%
  rownames_to_column(var = "gene") %>%
  pivot_longer(-gene, names_to = "sample") %>%
  left_join(rownames_to_column(ySub$samples, 
                               var = "sample")) %>%
  dplyr::filter(gene %in% HKGs) %>%
  dplyr::filter(Annotation == cell) %>%
  mutate(Batch = as.factor(Batch)) -> dat

dat %>%
  heatmap(gene, sample, value,
          scale = "row",
          show_row_names = FALSE,
          show_column_names = FALSE) %>%
  add_tile(Status_sub) %>%
  add_tile(Severity) %>%
  add_tile(Batch) %>%
  add_tile(Participant) %>%
  add_tile(Age) %>%
  add_tile(Sex)
```

```{r, fig.asp=1}
mds_by_factor(ySub[rownames(ySub) %in% HKGs,], "as.factor(Batch)", "Batch") & scale_color_brewer(palette = "Set1")
mds_by_factor(ySub[rownames(ySub) %in% HKGs,], "as.factor(Sex)", "Sex") & scale_color_brewer(palette = "Set2")
mds_by_factor(ySub[rownames(ySub) %in% HKGs,], "log2(Age)", "Log2 Age") & scale_colour_viridis_c(option = "magma") 
mds_by_factor(ySub[rownames(ySub) %in% HKGs,], "as.factor(Status_sub)", "Group") & scale_color_brewer(palette = "Dark2")
mds_by_factor(ySub[rownames(ySub) %in% HKGs,], "as.factor(Severity)", "Severity") & 
  scale_color_brewer(palette = "Accent")
mds_by_factor(ySub[rownames(ySub) %in% HKGs,], "as.factor(Micro_code)", "Infection") & scale_color_brewer(palette = "Pastel1")
```

## Examine covariates vs. HKGs

Investigate whether HKG PCAs correlate with any known covariates.
Prepare the data.

```{r}
PCs <- prcomp(t(edgeR::cpm(ySub$counts[ctl, ], log = TRUE)), 
              center = TRUE, retx = TRUE)
loadings = PCs$x # pc loadings


nGenes = nrow(ySub)
nSamples = ncol(ySub)

datTraits <- ySub$samples %>% dplyr::select(Batch, Treatment, 
                                            Severity, Age, Sex, ncells, Micro_code) %>%
  mutate(Batch = factor(Batch),
         Treatment = factor(Treatment, 
                            labels = 1:length(unique(Treatment))),
         Sex = factor(Sex, labels = length(unique(Sex))),
         Severity = factor(Severity, labels = length(unique(Severity)))) %>%
  mutate(across(everything(), as.numeric))

moduleTraitCor <- suppressWarnings(cor(loadings[, 1:min(10, nSamples)], 
                                       datTraits, use = "p"))
moduleTraitPvalue <- WGCNA::corPvalueStudent(moduleTraitCor, (nSamples-2))

textMatrix <- paste(signif(moduleTraitCor, 2), "\n(", 
                    signif(moduleTraitPvalue, 1), ")", sep = "")
dim(textMatrix) <- dim(moduleTraitCor)
```

Output results.

```{r, fig.asp=1.3}
par(mfrow = c(2, 1))
plot(PCs, type="lines", main = cell) # scree plot

## Display the correlation values within a heatmap plot
par(cex=0.75, mar = c(3, 5, 2, 1))
WGCNA::labeledHeatmap(Matrix = t(moduleTraitCor),
                            xLabels = colnames(loadings)[1:min(10, nSamples)],
                            yLabels = names(datTraits),
                            colorLabels = FALSE,
                            colors = WGCNA::blueWhiteRed(6),
                            textMatrix = t(textMatrix),
                            setStdMargins = FALSE,
                            cex.text = 1,
                            zlim = c(-1,1),
                            main = paste0("PCA-trait relationships: Top ", 
                                          min(10, nSamples), 
                                          " PCs"))
```

## Compare groups {.tabset}

### CF modifiers

#### Set up the model

First, we need to select `k` for use with `RUVseq`.
Examine the structure of the raw pseudobulk data.

```{r, fig.asp=0.4}
x1 <- as.factor(ySub$samples$Batch)
cols1 <- RColorBrewer::brewer.pal(7, "Set2")

par(mfrow = c(1,3))
EDASeq::plotRLE(edgeR::cpm(ySub$counts), 
                col = cols1[x1], ylim = c(-0.5, 0.5),
                main = "Raw RLE by batch", las = 2)
EDASeq::plotPCA(edgeR::cpm(ySub$counts), 
                col = cols1[x1], labels = FALSE,
                pch = 19, main = "Raw PCA by batch")
x2 <- as.factor(ySub$samples$Status_sub)
cols2 <- RColorBrewer::brewer.pal(4, "Set1")
EDASeq::plotPCA(edgeR::cpm(ySub$counts), 
                col = cols2[x2], labels = FALSE,
                pch = 19, main = "Raw PCA by disease")
```


Select the value for the `k` parameter i.e. the number of columns of the `W` matrix that will be included in the modelling.

```{r, fig.asp=0.8}
# define the sample groups
group <- factor(ySub$samples$Status_sub)
micro <- factor(ySub$samples$Micro_code)
sex <- factor(ySub$samples$Sex)
severity <- factor(ySub$samples$Severity)
age <- log2(ySub$samples$Age)

for(k in 1:6){
  adj <- RUVg(ySub$counts, ctl, k = k)
  W <- adj$W
  
  # create the design matrix
  design <- model.matrix(~0 + group + W + sex + micro + severity + age)
  colnames(design)[1:length(levels(group))] <- levels(group)
  
  # add the factors for the replicate samples
  dups <- unique(ySub$samples$Participant[duplicated(ySub$samples$Participant)])
  dups <- sapply(dups, function(d){
    ifelse(ySub$samples$Participant == d, 1, 0)  
  }, USE.NAMES = TRUE)
  
  contr <- makeContrasts(CF.NO_MODvCF.IVA = CF.NO_MOD - CF.IVA,
                         CF.NO_MODvCF.LUMA_IVA = CF.NO_MOD - CF.LUMA_IVA,
                         levels = design)
  
  y <- DGEList(counts = ySub$counts)
  y <- calcNormFactors(y)
  y <- estimateGLMCommonDisp(y, design)
  y <- estimateGLMTagwiseDisp(y, design)
  fit <- glmFit(y, design)
  
  x1 <- as.factor(ySub$samples$Batch)
  cols1 <- RColorBrewer::brewer.pal(7, "Set2")
  
  par(mfrow = c(2,3))
  EDASeq::plotRLE(edgeR::cpm(adj$normalizedCounts), 
                  col = cols1[x1], ylim = c(-0.5, 0.5),
                  main = paste0("K = ", k, " RLE by batch"))
  EDASeq::plotPCA(edgeR::cpm(adj$normalizedCounts), 
                  col = cols1[x1], labels = FALSE,
                  pch = 19,
                  main = paste0("K = ", k, " PCA by batch"))
  
  x2 <- as.factor(ySub$samples$Status_sub)
  cols2 <- RColorBrewer::brewer.pal(5, "Set1")
  EDASeq::plotPCA(edgeR::cpm(adj$normalizedCounts), 
                  col = cols2[x2], labels = FALSE,
                  pch = 19,
                  main = paste0("K = ", k, " PCA by disease"))
  
  lrt <- glmLRT(fit, contrast = contr[, 1])
  hist(lrt$table$PValue, main = paste0("K = ", k, " ", colnames(contr)[1]),
       cex.main = 0.8)
  lrt <- glmLRT(fit, contrast = contr[, 2])
  hist(lrt$table$PValue, main = paste0("K = ", k, " ", colnames(contr)[2]),
       cex.main = 0.8)

}
```

Test for DGE using `RUVSeq` and `edgeR`. First, create design matrix to model the sample groups and take into account the unwanted variation, age, sex, severity and replicate samples from the same individual. Also include a factor for presence of top 4 clinically important organisms as we are only comparing CF samples which have *all* been tested for the presence of various microorganisms.

```{r}
# use RUVSeq to identify the factors of unwanted variation
adj <- RUVg(ySub$counts, ctl, k = 4)
W <- adj$W
  
# create the design matrix
design <- model.matrix(~ 0 + group + W + sex + micro + severity + age)
colnames(design)[1:length(levels(group))] <- levels(group)

# add the factors for the replicate samples
dups <- unique(ySub$samples$Participant[duplicated(ySub$samples$Participant)])
dups <- sapply(dups, function(d){
  ifelse(ySub$samples$Participant == d, 1, 0)  
}, USE.NAMES = TRUE)

design <- cbind(design, dups)
design %>% knitr::kable()
```


```{r}
edgeR::cpm(ySub$counts, log = TRUE) %>% 
      data.frame %>%
      rownames_to_column(var = "gene") %>%
      pivot_longer(-gene, 
                   names_to = "sample", 
                   values_to = "raw") %>%
      inner_join(edgeR::cpm(adj$normalizedCounts, log = TRUE) %>% 
                   data.frame %>%
                   rownames_to_column(var = "gene") %>%
                   pivot_longer(-gene, 
                                names_to = "sample", 
                                values_to = "norm")) %>%
      left_join(rownames_to_column(ySub$samples, 
                                   var = "sample")) %>%
      mutate(Batch = as.factor(Batch)) %>%
      dplyr::filter(gene %in% c("ZFY", "EIF1AY", "XIST")) %>%
      ggplot(aes(x = Sex,
                 y = norm,
                 colour = Sex)) +
      geom_boxplot(outlier.shape = NA, colour = "grey") +
      geom_jitter(stat = "identity",
                  width = 0.15,
                  size = 1.25) +
      geom_jitter(aes(x = Sex,
                      y = raw), stat = "identity",
                  width = 0.15,
                  size = 2, 
                  alpha = 0.2,
                  stroke = 0) +
     ggrepel::geom_text_repel(aes(label = sample.id),
                             size = 2) +
      theme_classic() +
      theme(axis.text.x = element_text(angle = 90,
                                       hjust = 1,
                                       vjust = 0.5),
            legend.position = "bottom",
            legend.direction = "horizontal",
            strip.text = element_text(size = 7),
            axis.text.y = element_text(size = 6)) +
      labs(x = "Group", y = "log2 CPM") +
      facet_wrap(~gene, scales = "free_y") + 
      scale_color_brewer(palette = "Set2") +
      ggtitle("Sex gene expression check") -> p2

p2
```

Create the contrast matrix for the sample group comparisons.

```{r}
contr <- makeContrasts(CF.NO_MODvCF.IVA = CF.NO_MOD - CF.IVA,
                CF.NO_MODvCF.LUMA_IVA = CF.NO_MOD - CF.LUMA_IVA,
                levels = design)

contr %>% knitr::kable()
```

Fit the model.

```{r}
y <- DGEList(counts = ySub$counts)
y <- calcNormFactors(y)
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmFit(y, design)
```


#### DGE results summary

```{r}
cutoff <- 0.05
  
dt <- lapply(1:ncol(contr), function(i){
  decideTests(glmLRT(fit, contrast = contr[,i]),
                            p.value = cutoff)
})

s <- sapply(dt, function(d){
  summary(d)
})
colnames(s) <- colnames(contr)
rownames(s) <- c("Down", "NotSig", "Up")

pal <- c(paletteer::paletteer_d("RColorBrewer::Set1")[2:1], "grey") 

s[-2,] %>% 
  data.frame %>%
  rownames_to_column(var = "Direction") %>%
  pivot_longer(-Direction) %>%
  ggplot(aes(x = name, y = value, fill = Direction)) +
  geom_col(position = "dodge") +
  geom_text(aes(label = value), 
            position = position_dodge(width = 0.9),
            vjust = -0.5,
            size = 3) +
  labs(y = glue("No. DGE (FDR < {cutoff})"),
       x = "Contrast") +
      scale_fill_manual(values = pal) +
  theme(axis.text.x = element_text(angle = 45,
                                   hjust = 1,
                                   vjust = 1)) +
      scale_fill_manual(values = pal)
```

Explore results of statistical analysis for each contrast with significant DGEs.
First, setup the output directories.

```{r}
outDir <- here("output","dge_analysis")
if(!dir.exists(outDir)) dir.create(outDir)
cellDir <- file.path(outDir, cell)
if(!dir.exists(cellDir)) dir.create(cellDir)
```


Write significant DGEs for this comparison to file.

```{r}
# set index for current contrast
i = 1

lrt <- glmLRT(fit, contrast = contr[,i])
topTags(lrt, n = Inf) %>%
  data.frame -> top

write.table(top[1:500, ], 
            file = file.path(cellDir, glue("{colnames(contr)[i]}.csv")),
            sep = ",", quote = F, col.names = NA)
```


#### Volcano plot

```{r, fig.asp=0.9}
top %>% 
  mutate(sig = ifelse(FDR <= cutoff, glue("<= {cutoff}"), 
                      glue("> {cutoff}"))) %>%
  rownames_to_column(var = "SYMBOL") %>%
  left_join(dt[[i]][,1] %>% 
              data.frame %>%
              rownames_to_column(var = "SYMBOL") %>%
              dplyr::rename(status = 2)) %>%
  mutate(status = ifelse(status == 1, "Up",
                         ifelse(status == -1, "Down",
                                "NotSig"))) %>%
  mutate(status = as.factor(status)) %>%
  mutate(status = fct_relevel(status, "NotSig", after = Inf)) %>%
  ggplot(aes(x = logFC, y = -log10(PValue), color = status)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = function(x) subset(x, FDR < cutoff), 
                           aes(x = logFC, y = -log10(PValue), 
                               label = SYMBOL), 
                           size = 2, colour = "black", max.overlaps = 15) +
  labs(x = expression(~log[2]~"(Fold Change)"), 
       y = expression(~-log[10]~"(P-value)"),
       colour = glue("FDR < {cutoff}")) +
  scale_colour_manual(values = pal) +
  theme_classic() +
  theme(legend.position = "bottom") -> p1

p1
```

#### Top DGE stripchart

```{r, fig.asp=1}
# plot up to top 9 DGE
grps <- strsplit2(colnames(contr)[i], "v")[1,]

edgeR::cpm(ySub$counts, log = TRUE) %>% 
  data.frame %>%
  rownames_to_column(var = "gene") %>%
  pivot_longer(-gene, 
               names_to = "sample", 
               values_to = "raw") %>%
  inner_join(edgeR::cpm(adj$normalizedCounts, log = TRUE) %>% 
               data.frame %>%
               rownames_to_column(var = "gene") %>%
               pivot_longer(-gene, 
                            names_to = "sample", 
                            values_to = "norm")) %>%
  left_join(rownames_to_column(ySub$samples, 
                               var = "sample")) %>%
  mutate(Batch = as.factor(Batch)) %>%
  dplyr::filter(Status_sub %in% names(contr[,i])[abs(contr[,i]) > 0],
                gene %in% rownames(top)[1:min(9, max(which(top$FDR < cutoff)))]) %>%
  mutate(Group = ifelse(str_detect(Group, str_remove(grps[1], "CF.")),
                        grps[1],
                        grps[2])) %>%
  ggplot(aes(x = Group,
             y = norm,
             colour = Group)) +
  geom_boxplot(outlier.shape = NA, colour = "grey") +
  geom_jitter(stat = "identity",
              width = 0.15,
              size = 1.25) +
  geom_jitter(aes(x = Group,
                  y = raw), stat = "identity",
              width = 0.15,
              size = 1.25, 
              alpha = 0.2) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90,
                                   hjust = 1,
                                   vjust = 0.5),
        legend.position = "bottom",
        legend.direction = "horizontal",
        strip.text = element_text(size = 7),
        axis.text.y = element_text(size = 6)) +
  labs(x = "Group", y = "log2 CPM") +
  facet_wrap(~gene, scales = "free_y") + 
  scale_color_brewer(palette = "Set2") +
  ggtitle(colnames(contr)[i]) -> p2

p2
```

Gene set enrichment analysis.

Perform gene set enrichment analysis (GSEA) using the `cameraPR` method. `cameraPR` tests whether a set of genes is highly ranked relative to other genes in terms of differential expression, accounting for inter-gene correlation. Prepare the Broad MSigDB Hallmark gene sets and Reactome pathways.  

```{r}
if(!file.exists(here("data/Hs.c2.cp.reactome.v7.1.entrez.rds")))
  download.file("https://bioinf.wehi.edu.au/MSigDB/v7.1/Hs.c2.cp.reactome.v7.1.entrez.rds",
                here("data/Hs.c2.cp.reactome.v7.1.entrez.rds"))
Hs.c2.reactome <- readRDS(here("data/Hs.c2.cp.reactome.v7.1.entrez.rds"))

if(!file.exists(here("data/Hs.h.all.v7.1.entrez.rds")))
  download.file("https://bioinf.wehi.edu.au/MSigDB/v7.1/Hs.h.all.v7.1.entrez.rds",
                here("data/Hs.h.all.v7.1.entrez.rds"))
Hs.h.all <- readRDS(here("data/Hs.h.all.v7.1.entrez.rds"))
```

Write the GSEA results to file for this comparison.

```{r}
# get gene indices for all gene set genes
h.id <- ids2indices(Hs.h.all, unname(gns[rownames(lrt)]))
c2.id <- ids2indices(Hs.c2.reactome, unname(gns[rownames(lrt)]))

# run camera competitive gene set test  
# use signed likelihood ratio test statistic as recommended by GS here: https://support.bioconductor.org/p/112937/
statistic <- sign(lrt$table$logFC) * sqrt(lrt$table$LR)
h.cam <- cameraPR(statistic, h.id)
c2.cam <- cameraPR(statistic, c2.id)

# write top 50 GSEA results to file
write.table(h.cam[1:50,], 
            file = file.path(cellDir, glue("h.{colnames(contr)[i]}.csv")),
            sep = ",", quote = F, col.names = NA)
write.table(c2.cam[1:50,], 
            file = file.path(cellDir, glue("c2.{colnames(contr)[i]}.csv")),
            sep = ",", quote = F, col.names = NA)
```

#### Top gene sets

```{r}
# GSEA top 10 plots
h.cam %>%
  dplyr::slice(1:10) %>%
  rownames_to_column(var = "Set") %>%
  mutate(Type = "HALLMARK",
         Rank = 1:10) %>%
  bind_rows(c2.cam %>%
              dplyr::slice(1:10) %>%
              rownames_to_column(var = "Set") %>%
              mutate(Type = "REACTOME",
                     Rank = 1:10)) %>%
  mutate(Set = str_wrap(str_replace_all(Set, "_", " "), width = 75),
         Set = str_remove_all(Set, "REACTOME |HALLMARK ")) %>%
  ggplot(aes(x = -log10(FDR), y = fct_reorder(Set, -Rank),
             colour = Direction)) +
  geom_point(aes(size = NGenes)) +
  facet_wrap(~Type, ncol = 1, scales = "free_y") +
  geom_vline(xintercept = -log10(0.05),
             linetype = "dashed")  +
  scale_colour_manual(values = pal) +
  labs(y = "Gene set") +
  theme_classic(base_size = 10) -> p3

p3
```

Test two fibrosis gene sets using both over-represenationa analysis (ORA) and gene set enrichment analysis (GSEA). 
Adjust for gene length bias in ORA.
First, setup the transcript length.

```{r}
# translate gene SYMBOLS to entrez gene IDs 
gns <- AnnotationDbi::mapIds(org.Hs.eg.db, 
                             keys = rownames(fit), 
                             column = c("ENTREZID"),
                             keytype = "SYMBOL",
                             multiVals = "first")

# get transcript lengths
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
txlen <- transcriptLengths(txdb)
txlen %>%
  dplyr::select(gene_id, tx_len) %>%
  group_by(gene_id) %>%
  summarise(max_len = max(tx_len)) %>%
  dplyr::filter(gene_id %in% gns) -> supertxlen

m <- match(supertxlen$gene_id, gns)
all(supertxlen$gene_id == gns[m])
```

Generate table of results.

```{r}
# test DGEs for over-representation of fibrosis gene sets
# read in fibrosis gene sets
read.csv2(file = here("data/fibrosis_gene_sets.csv"),
          header = TRUE, sep = ",") %>% 
  pivot_longer(cols = everything()) %>%
  mutate(value = str_trim(value),
         entrez = unname(unlist(AnnotationDbi::mapIds(org.Hs.eg.db, 
                                                      keys = value, 
                                                      column = c("ENTREZID"),
                                                      keytype = "SYMBOL",
                                                      multiVals = "first"))[value])) %>%
  dplyr::rename("symbol" = "value",
                "source" = "name") -> fibrosis_table

fibrosis_list <- split(fibrosis_table$entrez, fibrosis_table$source)
    
deg <- rownames(top)[top$FDR < cutoff]
gsa <- topGSA(gsaseq(unname(gns[deg]),
                     universe = unname(gns),
                     collection = fibrosis_list,
                     plot.bias = FALSE,
                     gene.length = supertxlen$max_len[match(unname(gns), 
                                                            supertxlen$gene_id)]))
gsa %>% knitr::kable()

# get gene indices for all fibrosis gene set genes
f.id <- ids2indices(fibrosis_list, unname(gns[rownames(lrt)]))
f.cam <- cameraPR(statistic, f.id)
f.cam %>% knitr::kable()
```

#### Fibrosis gene sets

```{r}    
# plot fibrosis gene set over-representation results
gsa %>% 
  data.frame %>%
  rownames_to_column(var = "Set") %>%
  mutate(Rank = 1:n()) %>%
  ggplot(aes(x = -log10(FDR), y = fct_reorder(Set, -Rank), colour = DE/N*100)) +
  geom_point(aes(size = N)) +
  geom_vline(xintercept = -log10(0.05),
             linetype = "dashed")  +
  scale_colour_viridis_c(option = "plasma") +
  labs(y = "Gene set",
       colour = "% DEGs",
       size = "Set size") +
  ggtitle("Fibrosis over-represenation") +
  theme_classic(base_size = 10) +
  theme(legend.key.height = unit(0.3, "cm"),
        legend.key.width = unit(0.3, "cm")) -> p4 

# plot fibrosis GSEA results
f.cam %>% 
  data.frame %>%
  rownames_to_column(var = "Set") %>%
  mutate(Rank = 1:n()) %>%
  ggplot(aes(x = -log10(FDR), y = fct_reorder(Set, -Rank),
             colour = Direction)) +
  geom_point(aes(size = NGenes)) +
  geom_vline(xintercept = -log10(0.05),
             linetype = "dashed")  +
  scale_colour_manual(values = pal) +
  labs(y = "Gene set") +
  ggtitle("Fibrosis GSEA") +
  theme_classic(base_size = 10) +
  theme(legend.key.height = unit(0.3, "cm"),
        legend.key.width = unit(0.3, "cm")) -> p5

p4 / p5
```


### CF severity

#### Set up the model

First, we need to select `k` for use with `RUVseq`.
Examine the structure of the raw pseudobulk data.

```{r, fig.asp=0.4}
x1 <- as.factor(ySub$samples$Batch)
cols1 <- RColorBrewer::brewer.pal(7, "Set2")

par(mfrow = c(1,3))
EDASeq::plotRLE(edgeR::cpm(ySub$counts), 
                col = cols1[x1], ylim = c(-0.5, 0.5),
                main = "Raw RLE by batch", las = 2)
EDASeq::plotPCA(edgeR::cpm(ySub$counts), 
                col = cols1[x1], labels = FALSE,
                pch = 19, main = "Raw PCA by batch")
x2 <- as.factor(ySub$samples$Status_sub)
cols2 <- RColorBrewer::brewer.pal(4, "Set1")
EDASeq::plotPCA(edgeR::cpm(ySub$counts), 
                col = cols2[x2], labels = FALSE,
                pch = 19, main = "Raw PCA by disease")
```


Select the value for the `k` parameter i.e. the number of columns of the `W` matrix that will be included in the modelling.

```{r, fig.asp=0.8}
# define the sample groups
group <- factor(ySub$samples$Group)
micro <- factor(ySub$samples$Micro_code)
sex <- factor(ySub$samples$Sex)
age <- log2(ySub$samples$Age)

for(k in 1:6){
  adj <- RUVg(ySub$counts, ctl, k = k)
  W <- adj$W
  
  # create the design matrix
  design <- model.matrix(~0 + group + W + sex + micro + age)
  colnames(design)[1:length(levels(group))] <- levels(group)
  
  # add the factors for the replicate samples
  dups <- unique(ySub$samples$Participant[duplicated(ySub$samples$Participant)])
  dups <- sapply(dups, function(d){
    ifelse(ySub$samples$Participant == d, 1, 0)  
  }, USE.NAMES = TRUE)
  
  contr <- makeContrasts(CF.NO_MOD.MvCF.NO_MOD.S = CF.NO_MOD.M - CF.NO_MOD.S,
                         levels = design)
  
  y <- DGEList(counts = ySub$counts)
  y <- calcNormFactors(y)
  y <- estimateGLMCommonDisp(y, design)
  y <- estimateGLMTagwiseDisp(y, design)
  fit <- glmFit(y, design)
  
  x1 <- as.factor(ySub$samples$Batch)
  cols1 <- RColorBrewer::brewer.pal(7, "Set2")
  
  par(mfrow = c(2,3))
  EDASeq::plotRLE(edgeR::cpm(adj$normalizedCounts), 
                  col = cols1[x1], ylim = c(-0.5, 0.5),
                  main = paste0("K = ", k, " RLE by batch"))
  EDASeq::plotPCA(edgeR::cpm(adj$normalizedCounts), 
                  col = cols1[x1], labels = FALSE,
                  pch = 19,
                  main = paste0("K = ", k, " PCA by batch"))
  
  x2 <- as.factor(group)
  cols2 <- RColorBrewer::brewer.pal(5, "Set1")
  EDASeq::plotPCA(edgeR::cpm(adj$normalizedCounts), 
                  col = cols2[x2], labels = FALSE,
                  pch = 19,
                  main = paste0("K = ", k, " PCA by disease"))
  
  lrt <- glmLRT(fit, contrast = contr[, 1])
  hist(lrt$table$PValue, main = paste0("K = ", k, " ", colnames(contr)[1]),
       cex.main = 0.8)
  
}
```

Test for DGE using `RUVSeq` and `edgeR`. First, create design matrix to model the sample groups and take into account the unwanted variation, age, sex, severity and replicate samples from the same individual. Also include a factor for presence of top 4 clinically important organisms as we are only comparing CF samples which have *all* been tested for the presence of various microorganisms.

```{r}
# use RUVSeq to identify the factors of unwanted variation
adj <- RUVg(ySub$counts, ctl, k = 4)
W <- adj$W
  
# create the design matrix
design <- model.matrix(~ 0 + group + W + sex + micro + age)
colnames(design)[1:length(levels(group))] <- levels(group)

# add the factors for the replicate samples
dups <- unique(ySub$samples$Participant[duplicated(ySub$samples$Participant)])
dups <- sapply(dups, function(d){
  ifelse(ySub$samples$Participant == d, 1, 0)  
}, USE.NAMES = TRUE)

design <- cbind(design, dups)
design %>% knitr::kable()
```


```{r}
edgeR::cpm(ySub$counts, log = TRUE) %>% 
      data.frame %>%
      rownames_to_column(var = "gene") %>%
      pivot_longer(-gene, 
                   names_to = "sample", 
                   values_to = "raw") %>%
      inner_join(edgeR::cpm(adj$normalizedCounts, log = TRUE) %>% 
                   data.frame %>%
                   rownames_to_column(var = "gene") %>%
                   pivot_longer(-gene, 
                                names_to = "sample", 
                                values_to = "norm")) %>%
      left_join(rownames_to_column(ySub$samples, 
                                   var = "sample")) %>%
      mutate(Batch = as.factor(Batch)) %>%
      dplyr::filter(gene %in% c("ZFY", "EIF1AY", "XIST")) %>%
      ggplot(aes(x = Sex,
                 y = norm,
                 colour = Sex)) +
      geom_boxplot(outlier.shape = NA, colour = "grey") +
      geom_jitter(stat = "identity",
                  width = 0.15,
                  size = 1.25) +
      geom_jitter(aes(x = Sex,
                      y = raw), stat = "identity",
                  width = 0.15,
                  size = 2, 
                  alpha = 0.2,
                  stroke = 0) +
     ggrepel::geom_text_repel(aes(label = sample.id),
                             size = 2) +
      theme_classic() +
      theme(axis.text.x = element_text(angle = 90,
                                       hjust = 1,
                                       vjust = 0.5),
            legend.position = "bottom",
            legend.direction = "horizontal",
            strip.text = element_text(size = 7),
            axis.text.y = element_text(size = 6)) +
      labs(x = "Group", y = "log2 CPM") +
      facet_wrap(~gene, scales = "free_y") + 
      scale_color_brewer(palette = "Set2") +
      ggtitle("Sex gene expression check") -> p2

p2
```

Create the contrast matrix for the sample group comparisons.

```{r}
contr <- makeContrasts(CF.NO_MOD.MvCF.NO_MOD.S = CF.NO_MOD.M - CF.NO_MOD.S,
                         levels = design)

contr %>% knitr::kable()
```

Fit the model.

```{r}
y <- DGEList(counts = ySub$counts)
y <- calcNormFactors(y)
y <- estimateGLMCommonDisp(y, design)
y <- estimateGLMTagwiseDisp(y, design)
fit <- glmFit(y, design)
```


#### DGE results summary

```{r}
cutoff <- 0.05
  
dt <- lapply(1:ncol(contr), function(i){
  decideTests(glmLRT(fit, contrast = contr[,i]),
                            p.value = cutoff)
})

s <- sapply(dt, function(d){
  summary(d)
})
colnames(s) <- colnames(contr)
rownames(s) <- c("Down", "NotSig", "Up")

pal <- c(paletteer::paletteer_d("RColorBrewer::Set1")[2:1], "grey") 

s[-2,] %>% 
  data.frame %>%
  rownames_to_column(var = "Direction") %>%
  pivot_longer(-Direction) %>%
  ggplot(aes(x = name, y = value, fill = Direction)) +
  geom_col(position = "dodge") +
  geom_text(aes(label = value), 
            position = position_dodge(width = 0.9),
            vjust = -0.5,
            size = 3) +
  labs(y = glue("No. DGE (FDR < {cutoff})"),
       x = "Contrast") +
      scale_fill_manual(values = pal) +
  theme(axis.text.x = element_text(angle = 45,
                                   hjust = 1,
                                   vjust = 1)) +
      scale_fill_manual(values = pal)
```

Explore results of statistical analysis for each contrast with significant DGEs.
First, setup the output directories.

```{r}
outDir <- here("output","dge_analysis")
if(!dir.exists(outDir)) dir.create(outDir)
cellDir <- file.path(outDir, cell)
if(!dir.exists(cellDir)) dir.create(cellDir)
```


Write significant DGEs for this comparison to file.

```{r}
# set index for current contrast
i = 1

lrt <- glmLRT(fit, contrast = contr[,i])
topTags(lrt, n = Inf) %>%
  data.frame -> top

write.table(top[1:500, ], 
            file = file.path(cellDir, glue("{colnames(contr)[i]}.csv")),
            sep = ",", quote = F, col.names = NA)
```


#### Volcano plot

```{r, fig.asp=0.9}
top %>% 
  mutate(sig = ifelse(FDR <= cutoff, glue("<= {cutoff}"), 
                      glue("> {cutoff}"))) %>%
  rownames_to_column(var = "SYMBOL") %>%
  left_join(dt[[i]][,1] %>% 
              data.frame %>%
              rownames_to_column(var = "SYMBOL") %>%
              dplyr::rename(status = 2)) %>%
  mutate(status = ifelse(status == 1, "Up",
                         ifelse(status == -1, "Down",
                                "NotSig"))) %>%
  mutate(status = as.factor(status)) %>%
  mutate(status = fct_relevel(status, "NotSig", after = Inf)) %>%
  ggplot(aes(x = logFC, y = -log10(PValue), color = status)) +
  geom_point(alpha = 0.75) +
  ggrepel::geom_text_repel(data = function(x) subset(x, FDR < cutoff), 
                           aes(x = logFC, y = -log10(PValue), 
                               label = SYMBOL), 
                           size = 2, colour = "black", max.overlaps = 15) +
  labs(x = expression(~log[2]~"(Fold Change)"), 
       y = expression(~-log[10]~"(P-value)"),
       colour = glue("FDR < {cutoff}")) +
  scale_colour_manual(values = pal) +
  theme_classic() +
  theme(legend.position = "bottom") -> p1

p1
```

#### Top DGE stripchart

```{r, fig.asp=1}
# plot up to top 9 DGE
grps <- strsplit2(colnames(contr)[i], "v")[1,]

edgeR::cpm(ySub$counts, log = TRUE) %>% 
  data.frame %>%
  rownames_to_column(var = "gene") %>%
  pivot_longer(-gene, 
               names_to = "sample", 
               values_to = "raw") %>%
  inner_join(edgeR::cpm(adj$normalizedCounts, log = TRUE) %>% 
               data.frame %>%
               rownames_to_column(var = "gene") %>%
               pivot_longer(-gene, 
                            names_to = "sample", 
                            values_to = "norm")) %>%
  left_join(rownames_to_column(ySub$samples, 
                               var = "sample")) %>%
  mutate(Batch = as.factor(Batch)) %>%
  dplyr::filter(Group %in% names(contr[,i])[abs(contr[,i]) > 0],
                gene %in% rownames(top)[1:min(9, max(which(top$FDR < cutoff)))]) %>%
  mutate(Group = ifelse(str_detect(Group, str_remove(grps[1], "CF.")),
                        grps[1],
                        grps[2])) %>%
  ggplot(aes(x = Group,
             y = norm,
             colour = Group)) +
  geom_boxplot(outlier.shape = NA, colour = "grey") +
  geom_jitter(stat = "identity",
              width = 0.15,
              size = 1.25) +
  geom_jitter(aes(x = Group,
                  y = raw), stat = "identity",
              width = 0.15,
              size = 1.25, 
              alpha = 0.2) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90,
                                   hjust = 1,
                                   vjust = 0.5),
        legend.position = "bottom",
        legend.direction = "horizontal",
        strip.text = element_text(size = 7),
        axis.text.y = element_text(size = 6)) +
  labs(x = "Group", y = "log2 CPM") +
  facet_wrap(~gene, scales = "free_y") + 
  scale_color_brewer(palette = "Set2") +
  ggtitle(colnames(contr)[i]) -> p2

p2
```

Perform gene set enrichment analysis (GSEA) using the `cameraPR` method. `cameraPR` tests whether a set of genes is highly ranked relative to other genes in terms of differential expression, accounting for inter-gene correlation. Prepare the Broad MSigDB Hallmark gene sets and Reactome pathways.  

```{r}
if(!file.exists(here("data/Hs.c2.cp.reactome.v7.1.entrez.rds")))
  download.file("https://bioinf.wehi.edu.au/MSigDB/v7.1/Hs.c2.cp.reactome.v7.1.entrez.rds",
                here("data/Hs.c2.cp.reactome.v7.1.entrez.rds"))
Hs.c2.reactome <- readRDS(here("data/Hs.c2.cp.reactome.v7.1.entrez.rds"))

if(!file.exists(here("data/Hs.h.all.v7.1.entrez.rds")))
  download.file("https://bioinf.wehi.edu.au/MSigDB/v7.1/Hs.h.all.v7.1.entrez.rds",
                here("data/Hs.h.all.v7.1.entrez.rds"))
Hs.h.all <- readRDS(here("data/Hs.h.all.v7.1.entrez.rds"))
```

Write the GSEA results to file for this comparison.

```{r}
# get gene indices for all gene set genes
h.id <- ids2indices(Hs.h.all, unname(gns[rownames(lrt)]))
c2.id <- ids2indices(Hs.c2.reactome, unname(gns[rownames(lrt)]))

# run camera competitive gene set test  
# use signed likelihood ratio test statistic as recommended by GS here: https://support.bioconductor.org/p/112937/
statistic <- sign(lrt$table$logFC) * sqrt(lrt$table$LR)
h.cam <- cameraPR(statistic, h.id)
c2.cam <- cameraPR(statistic, c2.id)

# write top 50 GSEA results to file
write.table(h.cam[1:50,], 
            file = file.path(cellDir, glue("h.{colnames(contr)[i]}.csv")),
            sep = ",", quote = F, col.names = NA)
write.table(c2.cam[1:50,], 
            file = file.path(cellDir, glue("c2.{colnames(contr)[i]}.csv")),
            sep = ",", quote = F, col.names = NA)
```

#### Top gene sets

```{r}
# GSEA top 10 plots
h.cam %>%
  dplyr::slice(1:10) %>%
  rownames_to_column(var = "Set") %>%
  mutate(Type = "HALLMARK",
         Rank = 1:10) %>%
  bind_rows(c2.cam %>%
              dplyr::slice(1:10) %>%
              rownames_to_column(var = "Set") %>%
              mutate(Type = "REACTOME",
                     Rank = 1:10)) %>%
  mutate(Set = str_wrap(str_replace_all(Set, "_", " "), width = 75),
         Set = str_remove_all(Set, "REACTOME |HALLMARK ")) %>%
  ggplot(aes(x = -log10(FDR), y = fct_reorder(Set, -Rank),
             colour = Direction)) +
  geom_point(aes(size = NGenes)) +
  facet_wrap(~Type, ncol = 1, scales = "free_y") +
  geom_vline(xintercept = -log10(0.05),
             linetype = "dashed")  +
  scale_colour_manual(values = pal) +
  labs(y = "Gene set") +
  theme_classic(base_size = 10) -> p3

p3
```

Test two fibrosis gene sets using both over-represenationa analysis (ORA) and gene set enrichment analysis (GSEA). 
Adjust for gene length bias in ORA.
First, setup the transcript length.

```{r}
# translate gene SYMBOLS to entrez gene IDs 
gns <- AnnotationDbi::mapIds(org.Hs.eg.db, 
                             keys = rownames(fit), 
                             column = c("ENTREZID"),
                             keytype = "SYMBOL",
                             multiVals = "first")

# get transcript lengths
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
txlen <- transcriptLengths(txdb)
txlen %>%
  dplyr::select(gene_id, tx_len) %>%
  group_by(gene_id) %>%
  summarise(max_len = max(tx_len)) %>%
  dplyr::filter(gene_id %in% gns) -> supertxlen

m <- match(supertxlen$gene_id, gns)
all(supertxlen$gene_id == gns[m])
```

Generate table of results.

```{r}
# test DGEs for over-representation of fibrosis gene sets
# read in fibrosis gene sets
read.csv2(file = here("data/fibrosis_gene_sets.csv"),
          header = TRUE, sep = ",") %>% 
  pivot_longer(cols = everything()) %>%
  mutate(value = str_trim(value),
         entrez = unname(unlist(AnnotationDbi::mapIds(org.Hs.eg.db, 
                                                      keys = value, 
                                                      column = c("ENTREZID"),
                                                      keytype = "SYMBOL",
                                                      multiVals = "first"))[value])) %>%
  dplyr::rename("symbol" = "value",
                "source" = "name") -> fibrosis_table

fibrosis_list <- split(fibrosis_table$entrez, fibrosis_table$source)
    
deg <- rownames(top)[top$FDR < cutoff]
gsa <- topGSA(gsaseq(unname(gns[deg]),
                     universe = unname(gns),
                     collection = fibrosis_list,
                     plot.bias = FALSE,
                     gene.length = supertxlen$max_len[match(unname(gns), 
                                                            supertxlen$gene_id)]))
gsa %>% knitr::kable()

# get gene indices for all fibrosis gene set genes
f.id <- ids2indices(fibrosis_list, unname(gns[rownames(lrt)]))
f.cam <- cameraPR(statistic, f.id)
f.cam %>% knitr::kable()
```

#### Fibrosis gene sets

```{r}    
# plot fibrosis gene set over-representation results
gsa %>% 
  data.frame %>%
  rownames_to_column(var = "Set") %>%
  mutate(Rank = 1:n()) %>%
  ggplot(aes(x = -log10(FDR), y = fct_reorder(Set, -Rank), colour = DE/N*100)) +
  geom_point(aes(size = N)) +
  geom_vline(xintercept = -log10(0.05),
             linetype = "dashed")  +
  scale_colour_viridis_c(option = "plasma") +
  labs(y = "Gene set",
       colour = "% DEGs",
       size = "Set size") +
  ggtitle("Fibrosis over-represenation") +
  theme_classic(base_size = 10) +
  theme(legend.key.height = unit(0.3, "cm"),
        legend.key.width = unit(0.3, "cm")) -> p4 

# plot fibrosis GSEA results
f.cam %>% 
  data.frame %>%
  rownames_to_column(var = "Set") %>%
  mutate(Rank = 1:n()) %>%
  ggplot(aes(x = -log10(FDR), y = fct_reorder(Set, -Rank),
             colour = Direction)) +
  geom_point(aes(size = NGenes)) +
  geom_vline(xintercept = -log10(0.05),
             linetype = "dashed")  +
  scale_colour_manual(values = pal) +
  labs(y = "Gene set") +
  ggtitle("Fibrosis GSEA") +
  theme_classic(base_size = 10) +
  theme(legend.key.height = unit(0.3, "cm"),
        legend.key.width = unit(0.3, "cm")) -> p5

p4 / p5
```



## {-}


# Session info {.appendix}











