---
title: "Single-cell atlas of BAL from preschool cystic fibrosis reveals key inflammatory pathways modified by ivacaftor in the early life lung"
description: |
author:
  - name: Jovana Maksimovic
date: "`r Sys.Date()`"
site: workflowr::wflow_site
output:
  html_document: 
    theme: cosmo
---

```{r, echo = FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE, number_sections = FALSE)
```


# Abstract {-}

Aberrant inflammation and structural lung damage occurs early in life for people with cystic fibrosis (CF). Even in the era of CFTR modulators, anti-inflammatory therapy may still be needed to prevent establishment, and lifelong consequences, of bronchiectasis. In this study, we integrated transcriptome-wide single-cell RNA sequencing data, highly multiplexed surface protein expression, and functional pathway analysis to comprehensively investigate inflammatory cell responses in 37 BAL samples from children with CF and 8 BAL samples from non-CF controls. We show widespread dysregulation of macrophage function in the preschool CF lung, including alterations in pathways associated with SARS-COV-1 and -2 responses, TNF/IFN signalling and cholesterol homeostasis, that were further altered by the early development of bronchiectasis. We showed that the CFTR modulator ivacaftor restores some of these macrophage-related functional deficits and reduces expression of pathways associated with neutrophil infiltration, however the modulator lumacaftor/ivacaftor was not associated with any detectable change in response. This work represents the most comprehensive, multi-omic single-cell analysis of bronchoalveolar lavage from preschool children and the results inform the future development of anti-inflammatory therapy for children with CF.

# Authors {-}
Jovana Maksimovic^1,2,9\*^, Shivanthan Shanthikumar ^2,3,4\*^, George Howitt^1,9^, Gunjan Dixit^1,9^, Peter F Hickey^5,6^, William Ho^5^, Casey J.A. Anttila^5^, Daniel V. Brown^5^, Anne Senabouth^7,8^, Dominik Kaczorowski^7,8^, Daniela Amann-Zalcenstein^5,6^, Joseph E. Powell^7,8^, Sarath C. Ranganathan^2,3,4^, Alicia Oshlack^1,9,10^, Melanie R. Neeland^2,3#^.

^1^ Computational Biology Program, Peter MacCallum Cancer Centre, Parkville, VIC, Australia

^2^ Respiratory Diseases, Murdoch Children’s Research Institute, Parkville, VIC, Australia

^3^ Department of Paediatrics, University of Melbourne, Parkville, VIC, Australia

^4^ Respiratory and Sleep Medicine, Royal Children’s Hospital, Parkville, VIC, Australia

^5^ Advanced Genomics Facility, Advanced Technology and Biology Division, WEHI, Parkville, VIC, Australia

^6^ Department of Medical Biology, University of Melbourne, Parkville, VIC, Australia

^7^ Garvan-Weizmann Centre for Cellular Genomics, Garvan Institute of Medical Research, Sydney, NSW, Australia

^8^ UNSW Cellular Genomics Futures Institute, University of New South Wales, Sydney, NSW, Australia

^9^ Sir Peter MacCallum Department of Oncology, University of Melbourne, Parkville, VIC, Australia

^10^ School of Mathematics and Statistics, University of Melbourne, Parkville, VIC, Australia

^\*^Contributed equally

# Analysis Overview {-}

Follow the links below to view the different parts of the analysis. 

1. Remove empty droplets and demultiplex
    i) [Batch 0](01.0_preprocess_batch0.html)
    ii) [Batch 1](01.1_preprocess_batch1.html)
    iii) [Batch 2](01.2_preprocess_batch2.html)
    iv) [Batch 3](01.3_preprocess_batch3.html)
    v) [Batch 4](01.4_preprocess_batch4.html)
    vi) [Batch 5](01.5_preprocess_batch5.html)
    vii) [Batch 6](01.6_preprocess_batch6.html)

<!-- line break -->
2. [Quality control](02.0_quality_control.html)

3. [Identify doublets](03.0_identify_doublets.html)

4. [Filter doublets](04.0_filter_doublets.html)

5. [Remove ambient contamination](05.0_remove_ambient.html)

6. Annotate using Human Lung Cell Atlas v2.0
    i) [Without ambient removal](06.0_azimuth_annotation.html)
    ii) [With ambient removal](06.1_azimuth_annotation_decontx.html)
    
<!-- line break -->
7. [Cluster T/NK cells](07.1_integrate_cluster_t_cells_decontx.html)

8. [Cluster rare cells](08.1_integrate_cluster_other_cells_decontx.html)

9. [Cluster macrophage cells](09.0_integrate_cluster_macro_cells.html)

10. [Annotate macrophage cells](10.0_manual_annotations_macro_cells.html) 

11. [Annotate T/NK cells](11.0_manual_annotations_t_cells_decontx.html)  

12. [Annotate rare cells](12.0_manual_annotations_other_cells.html) 

<!-- line break -->
13. Differential expression analysis of macrophages
    i) [Macrophages](13.0_DGE_analysis_macrophages.html)
    ii) [Alveolar macrophages](13.1_DGE_analysis_macro-alveolar.html)
    iii) [APOC2+ macrophages](13.2_DGE_analysis_macro-APOC2+.html)
    iv) [CCL macrophages](13.3_DGE_analysis_macro-CCL.html)
    v) [IFI21 macrophages](13.4_DGE_analysis_macro-IFI27.html)
    vi) [Lipid macrophages](13.5_DGE_analysis_macro-lipid.html)
    vii) [Monocyte-derived macrophages](13.6_DGE_analysis_macro-monocyte-derived.html)
    viii) [Proliferating macrophages](13.7_DGE_analysis_macro-proliferating.html)

<!-- line break -->
14. Differential expression analysis of T/NK and rare cells
    i) [CD4 T-cells](14.0_DGE_analysis_CD4-T-cells.html) 
    ii) [CD8 T-cells](14.1_DGE_analysis_CD8-T-cells.html)
    iii) [DC cells](14.2_DGE_analysis_DC-cells.html)
    
<!-- line break -->
15. Analyse cell types proportions
    i) [Annotation level 1](15.0_proportions_analysis_ann_level_1.html) 
    ii) [Annotation level 3, non macro. cells](15.1_proportions_analysis_ann_level_3_non-macrophages.html)
    iii) [Annotation level 3, macro. cells](15.2_proportions_analysis_ann_level_3_macrophages.html)

<!-- line break -->
15. [Figure 1](99.0_Figure_1.html)

<!-- line break -->
16. [Figure 2](99.0_Figure_2.html)

<!-- line break -->
16. [Figure 3](99.0_Figure_3.html)

<!-- line break -->
16. [Figure 4](99.0_Figure_4.html)

<!-- line break -->
16. [Figure 5](99.0_Figure_5.html)

<!-- line break -->
17. [Supplementary Figure ADTs](99.0_Supplementary_Figure_ADTs.html)    

<!-- line break -->
18. [Supplementary Figures 3 and 5](99.0_Supplementary_Figure_3-5.html)    

<!-- line break -->
<!-- To ensure reproducibility, I had to install these versions of the following packages specifically: Seurat_4.4.0,  SeuratObject_4.1.4, SeuratData_0.2.2 (very specifically this commit "satijalab/seurat-data@d6a8ce61ccb21a3b204f194d07009772c822791d", Azimuth_0.4.6. -->

<!-- As per this issue, https://github.com/satijalab/seurat-object/issues/165, if there is an error running sctransform, remove all relevant packages (and purge if using renv) and then install using the following method exactly, to build from source. -->

<!-- if (!requireNamespace("remotes", quietly = TRUE)) { -->
<!--   install.packages("remotes") -->
<!-- } -->
<!-- options(repos = c("https://satijalab.r-universe.dev", getOption("repos"))) -->
<!-- remotes::install_version("SeuratObject", "4.1.4") -->
<!-- remotes::install_version("Seurat", "4.4.0", upgrade = FALSE) # remotes will try to upgrade SeuratObject to v5, we need to tell it no -->


# Licenses {-}

The code in this analysis is covered by the [MIT license][mit] and the written
content on this website is covered by a [Creative Commons CC-BY][cc] license.


# Version Information  {-}

**R version**: `r R.version.string`

**Bioconductor version**: `r BiocManager::version()`


[mit]: https://choosealicense.com/licenses/mit/ "MIT License"
[cc]: https://choosealicense.com/licenses/cc-by-4.0/ "CC-BY License"

